<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.43">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Atlantis Documentation – atlantisuserguide_parti_clean</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Atlantis Documentation</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-user-guides" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">User Guides</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-user-guides">    
        <li>
    <a class="dropdown-item" href="../../user_guides/quarto_site/part1/index.html">
 <span class="dropdown-text">Part I - Atlantis User Guide</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../user_guides/quarto_site/part2/index.qmd">
 <span class="dropdown-text">Part II - [Coming Soon]</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item">
    <a class="nav-link" href="../../confluence/index.html"> 
<span class="menu-text">Confluence Wiki</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../resources/index.html"> 
<span class="menu-text">Resources</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#foreward" id="toc-foreward" class="nav-link active" data-scroll-target="#foreward">0. FOREWARD</a></li>
  <li><a href="#introduction" id="toc-introduction" class="nav-link" data-scroll-target="#introduction">1. INTRODUCTION</a>
  <ul>
  <li><a href="#goals-of-this-manual-and-where-to-go-from-here" id="toc-goals-of-this-manual-and-where-to-go-from-here" class="nav-link" data-scroll-target="#goals-of-this-manual-and-where-to-go-from-here"><strong>1.1. Goals of this manual and where to go from here</strong></a></li>
  <li><a href="#terms-and-conventions-used-in-the-manual" id="toc-terms-and-conventions-used-in-the-manual" class="nav-link" data-scroll-target="#terms-and-conventions-used-in-the-manual"><strong>1.2. Terms and conventions used in the manual</strong></a></li>
  <li><a href="#what-is-atlantis" id="toc-what-is-atlantis" class="nav-link" data-scroll-target="#what-is-atlantis"><strong>1.3. What is Atlantis?</strong></a></li>
  <li><a href="#existing-applications-and-questions-studied" id="toc-existing-applications-and-questions-studied" class="nav-link" data-scroll-target="#existing-applications-and-questions-studied"><strong>1.4. Existing applications and questions studied</strong></a></li>
  <li><a href="#how-does-atlantis-compare-to-other-marine-ecosystem-models" id="toc-how-does-atlantis-compare-to-other-marine-ecosystem-models" class="nav-link" data-scroll-target="#how-does-atlantis-compare-to-other-marine-ecosystem-models"><strong>1.5. How does Atlantis compare to other marine ecosystem models?</strong></a></li>
  <li><a href="#main-assumptions" id="toc-main-assumptions" class="nav-link" data-scroll-target="#main-assumptions"><strong>1.6. Main assumptions</strong></a></li>
  </ul></li>
  <li><a href="#installation-and-running" id="toc-installation-and-running" class="nav-link" data-scroll-target="#installation-and-running">2. INSTALLATION AND RUNNING</a>
  <ul>
  <li><a href="#how-to-check-out-and-install-the-code" id="toc-how-to-check-out-and-install-the-code" class="nav-link" data-scroll-target="#how-to-check-out-and-install-the-code"><strong>2.1 How to check out and install the code</strong></a></li>
  <li><a href="#input-files" id="toc-input-files" class="nav-link" data-scroll-target="#input-files"><strong>2.2. Input files</strong></a></li>
  <li><a href="#overview-of-atlantis-submodels-and-the-main-routine" id="toc-overview-of-atlantis-submodels-and-the-main-routine" class="nav-link" data-scroll-target="#overview-of-atlantis-submodels-and-the-main-routine"><strong>2.3. Overview of Atlantis submodels and the main() routine</strong></a></li>
  <li><a href="#delving-into-the-code-finding-the-files-and-routines-that-describe-the-processes-you-are-interested-in" id="toc-delving-into-the-code-finding-the-files-and-routines-that-describe-the-processes-you-are-interested-in" class="nav-link" data-scroll-target="#delving-into-the-code-finding-the-files-and-routines-that-describe-the-processes-you-are-interested-in"><strong>2.4. Delving into the code: finding the files and routines that describe the processes you are interested in</strong></a></li>
  <li><a href="#how-to-setup-and-start-simulations" id="toc-how-to-setup-and-start-simulations" class="nav-link" data-scroll-target="#how-to-setup-and-start-simulations"><strong>2.5. How to setup and start simulations</strong></a>
  <ul class="collapse">
  <li><a href="#crash-introduction-into-command-line-interfaces" id="toc-crash-introduction-into-command-line-interfaces" class="nav-link" data-scroll-target="#crash-introduction-into-command-line-interfaces"><em>2.5.1. Crash introduction into Command line interfaces</em></a></li>
  <li><a href="#setting-up-the-bat-file" id="toc-setting-up-the-bat-file" class="nav-link" data-scroll-target="#setting-up-the-bat-file"><em>2.5.2. Setting up the BAT file</em></a></li>
  <li><a href="#parameter-files-called-in-the-bat-command-line" id="toc-parameter-files-called-in-the-bat-command-line" class="nav-link" data-scroll-target="#parameter-files-called-in-the-bat-command-line"><em>2.5.3. Parameter files called in the BAT command line</em></a></li>
  <li><a href="#setting-up-batch-simulations" id="toc-setting-up-batch-simulations" class="nav-link" data-scroll-target="#setting-up-batch-simulations"><em>2.5.4. Setting up batch simulations</em></a></li>
  <li><a href="#initialising-simulations" id="toc-initialising-simulations" class="nav-link" data-scroll-target="#initialising-simulations"><em>2.5.5. Initialising simulations</em></a></li>
  </ul></li>
  <li><a href="#simulation-outputs" id="toc-simulation-outputs" class="nav-link" data-scroll-target="#simulation-outputs"><strong>2.6. Simulation outputs</strong></a></li>
  <li><a href="#introduction-to-netcdf-files" id="toc-introduction-to-netcdf-files" class="nav-link" data-scroll-target="#introduction-to-netcdf-files"><strong>2.7. Introduction to NetCDF files</strong></a></li>
  <li><a href="#introduction-to-atlantis-variables" id="toc-introduction-to-atlantis-variables" class="nav-link" data-scroll-target="#introduction-to-atlantis-variables"><strong>2.8. Introduction to Atlantis variables</strong></a></li>
  <li><a href="#tools-and-packages-to-view-and-analyse-atlantis-input-and-output-files" id="toc-tools-and-packages-to-view-and-analyse-atlantis-input-and-output-files" class="nav-link" data-scroll-target="#tools-and-packages-to-view-and-analyse-atlantis-input-and-output-files"><strong>2.9 Tools and packages to view and analyse Atlantis input and output files</strong></a></li>
  <li><a href="#troubleshooting-on-installation-and-running" id="toc-troubleshooting-on-installation-and-running" class="nav-link" data-scroll-target="#troubleshooting-on-installation-and-running"><strong>2.10. Troubleshooting on installation and running</strong></a></li>
  </ul></li>
  <li><a href="#geometry-of-the-model-domain" id="toc-geometry-of-the-model-domain" class="nav-link" data-scroll-target="#geometry-of-the-model-domain">3. GEOMETRY OF THE MODEL DOMAIN</a>
  <ul>
  <li><a href="#boxes-and-layers-defining-the-geometry-of-the-model" id="toc-boxes-and-layers-defining-the-geometry-of-the-model" class="nav-link" data-scroll-target="#boxes-and-layers-defining-the-geometry-of-the-model"><strong>3.1. Boxes and layers defining the geometry of the model</strong></a></li>
  <li><a href="#how-to-define-a-geometry-file-for-a-new-model" id="toc-how-to-define-a-geometry-file-for-a-new-model" class="nav-link" data-scroll-target="#how-to-define-a-geometry-file-for-a-new-model"><strong>3.2. How to define a geometry file for a new model?</strong></a></li>
  <li><a href="#practical-steps-for-building-the-bgm-files-from-gis-shapefiles" id="toc-practical-steps-for-building-the-bgm-files-from-gis-shapefiles" class="nav-link" data-scroll-target="#practical-steps-for-building-the-bgm-files-from-gis-shapefiles"><strong>3.3. Practical steps for building the BGM files from GIS shapefiles</strong></a></li>
  <li><a href="#how-to-view-your-model-geometry" id="toc-how-to-view-your-model-geometry" class="nav-link" data-scroll-target="#how-to-view-your-model-geometry"><strong>3.4. How to view your model geometry?</strong></a></li>
  <li><a href="#using-r-for-bgm-geometry" id="toc-using-r-for-bgm-geometry" class="nav-link" data-scroll-target="#using-r-for-bgm-geometry"><strong>3.5 Using R for BGM geometry</strong></a></li>
  </ul></li>
  <li><a href="#hydrodynamic-salinity-and-temperature-forcing-data" id="toc-hydrodynamic-salinity-and-temperature-forcing-data" class="nav-link" data-scroll-target="#hydrodynamic-salinity-and-temperature-forcing-data">4. HYDRODYNAMIC, SALINITY AND TEMPERATURE FORCING DATA</a>
  <ul>
  <li><a href="#general-introduction-to-forcing-of-atlantis-models" id="toc-general-introduction-to-forcing-of-atlantis-models" class="nav-link" data-scroll-target="#general-introduction-to-forcing-of-atlantis-models"><strong>4.1. General introduction to forcing of Atlantis models</strong></a></li>
  <li><a href="#transforming-oceanographic-model-output-into-atlantis-forcing-files" id="toc-transforming-oceanographic-model-output-into-atlantis-forcing-files" class="nav-link" data-scroll-target="#transforming-oceanographic-model-output-into-atlantis-forcing-files"><strong>4.2. Transforming oceanographic model output into Atlantis forcing files</strong></a></li>
  <li><a href="#things-to-keep-in-mind-about-hydrodynamic-forcing-files" id="toc-things-to-keep-in-mind-about-hydrodynamic-forcing-files" class="nav-link" data-scroll-target="#things-to-keep-in-mind-about-hydrodynamic-forcing-files"><strong>4.3. Things to keep in mind about hydrodynamic forcing files</strong></a></li>
  <li><a href="#numbering-of-vertical-layers-in-the-input-and-output-nc-files" id="toc-numbering-of-vertical-layers-in-the-input-and-output-nc-files" class="nav-link" data-scroll-target="#numbering-of-vertical-layers-in-the-input-and-output-nc-files"><strong>4.4. Numbering of vertical layers in the input and output NC files</strong></a></li>
  <li><a href="#viewing-and-checking-forcing-input-files-with-r" id="toc-viewing-and-checking-forcing-input-files-with-r" class="nav-link" data-scroll-target="#viewing-and-checking-forcing-input-files-with-r"><strong>4.5. Viewing and checking forcing input files with R</strong></a></li>
  <li><a href="#light-and-noise-pollution" id="toc-light-and-noise-pollution" class="nav-link" data-scroll-target="#light-and-noise-pollution"><strong>4.6. Light and noise pollution</strong></a></li>
  </ul></li>
  <li><a href="#the-physics-submodel" id="toc-the-physics-submodel" class="nav-link" data-scroll-target="#the-physics-submodel">5. THE PHYSICS SUBMODEL</a>
  <ul>
  <li><a href="#physical-and-biogeochemical-variables-and-their-characteristics" id="toc-physical-and-biogeochemical-variables-and-their-characteristics" class="nav-link" data-scroll-target="#physical-and-biogeochemical-variables-and-their-characteristics"><strong>5.1. Physical and biogeochemical variables and their characteristics</strong></a></li>
  <li><a href="#the-main-physics-routine" id="toc-the-main-physics-routine" class="nav-link" data-scroll-target="#the-main-physics-routine"><strong>5.2. The main <em>physics()</em> routine</strong></a></li>
  <li><a href="#other-parameters-in-physics.prm-not-included-in-the-physics-routine" id="toc-other-parameters-in-physics.prm-not-included-in-the-physics-routine" class="nav-link" data-scroll-target="#other-parameters-in-physics.prm-not-included-in-the-physics-routine"><strong>5.3. Other parameters in <em>physics.prm</em> not included in the <em>physics()</em> routine</strong></a></li>
  <li><a href="#when-to-use-vertical-and-horizontal-transport-and-mixing" id="toc-when-to-use-vertical-and-horizontal-transport-and-mixing" class="nav-link" data-scroll-target="#when-to-use-vertical-and-horizontal-transport-and-mixing"><strong>5.4. When to use vertical and horizontal transport and mixing?</strong></a></li>
  <li><a href="#latest-developments" id="toc-latest-developments" class="nav-link" data-scroll-target="#latest-developments"><strong>5.5. Latest developments</strong></a>
  <ul class="collapse">
  <li><a href="#land" id="toc-land" class="nav-link" data-scroll-target="#land"><strong><em>5.5.1 Land</em></strong></a></li>
  <li><a href="#ice" id="toc-ice" class="nav-link" data-scroll-target="#ice"><strong><em>5.5.2. Ice</em></strong></a></li>
  <li><a href="#contaminants" id="toc-contaminants" class="nav-link" data-scroll-target="#contaminants"><strong><em>5.5.3. Contaminants</em></strong></a></li>
  </ul></li>
  <li><a href="#outputs-of-physics-tracers-in-the-output.nc-file" id="toc-outputs-of-physics-tracers-in-the-output.nc-file" class="nav-link" data-scroll-target="#outputs-of-physics-tracers-in-the-output.nc-file"><strong>5.6. Outputs of physics tracers in the <em>output.nc</em> file</strong></a></li>
  <li><a href="#importance-of-optional-physics-routines" id="toc-importance-of-optional-physics-routines" class="nav-link" data-scroll-target="#importance-of-optional-physics-routines"><strong>5.7. Importance of optional physics routines</strong></a></li>
  <li><a href="#checking-hydrodynamic-transports" id="toc-checking-hydrodynamic-transports" class="nav-link" data-scroll-target="#checking-hydrodynamic-transports"><strong>5.8. Checking hydrodynamic transports</strong></a></li>
  </ul></li>
  <li><a href="#definitions-of-functional-groups" id="toc-definitions-of-functional-groups" class="nav-link" data-scroll-target="#definitions-of-functional-groups">6. DEFINITIONS OF FUNCTIONAL GROUPS</a>
  <ul>
  <li><a href="#introduction-to-functional-groups" id="toc-introduction-to-functional-groups" class="nav-link" data-scroll-target="#introduction-to-functional-groups"><strong>6.1. Introduction to functional groups</strong></a></li>
  <li><a href="#defining-functional-groups" id="toc-defining-functional-groups" class="nav-link" data-scroll-target="#defining-functional-groups"><strong>6.2. Defining functional groups</strong></a></li>
  <li><a href="#defining-the-grouptype-correctly" id="toc-defining-the-grouptype-correctly" class="nav-link" data-scroll-target="#defining-the-grouptype-correctly"><strong>6.3. Defining the GroupType correctly</strong></a></li>
  <li><a href="#modifying-functional-groups-in-parameterised-models" id="toc-modifying-functional-groups-in-parameterised-models" class="nav-link" data-scroll-target="#modifying-functional-groups-in-parameterised-models"><strong>6.4. Modifying functional groups in parameterised models</strong></a></li>
  </ul></li>
  <li><a href="#run-parameters" id="toc-run-parameters" class="nav-link" data-scroll-target="#run-parameters">7. RUN PARAMETERS</a></li>
  <li><a href="#force-parameters" id="toc-force-parameters" class="nav-link" data-scroll-target="#force-parameters">8. FORCE PARAMETERS</a></li>
  <li><a href="#primary-producer-processes" id="toc-primary-producer-processes" class="nav-link" data-scroll-target="#primary-producer-processes">9. PRIMARY PRODUCER PROCESSES</a>
  <ul>
  <li><a href="#fluxes-in-primary-producers" id="toc-fluxes-in-primary-producers" class="nav-link" data-scroll-target="#fluxes-in-primary-producers"><strong>9.1. Fluxes in primary producers</strong></a></li>
  <li><a href="#primary-producer-growth" id="toc-primary-producer-growth" class="nav-link" data-scroll-target="#primary-producer-growth"><strong>9.2. Primary producer growth</strong></a>
  <ul class="collapse">
  <li><a href="#primary-producer-nutrient-limitation" id="toc-primary-producer-nutrient-limitation" class="nav-link" data-scroll-target="#primary-producer-nutrient-limitation"><strong><em>9.2.1. Primary producer nutrient limitation</em></strong></a></li>
  <li><a href="#primary-producer-light-limitation" id="toc-primary-producer-light-limitation" class="nav-link" data-scroll-target="#primary-producer-light-limitation"><strong><em>9.2.2. Primary producer light limitation</em></strong></a></li>
  <li><a href="#primary-macrophyte-producer-space-limitation" id="toc-primary-macrophyte-producer-space-limitation" class="nav-link" data-scroll-target="#primary-macrophyte-producer-space-limitation"><strong><em>9.2.3. Primary macrophyte producer space limitation</em></strong></a></li>
  <li><a href="#effect-of-eddies-on-primary-production" id="toc-effect-of-eddies-on-primary-production" class="nav-link" data-scroll-target="#effect-of-eddies-on-primary-production"><strong><em>9.2.4 Effect of eddies on primary production</em></strong></a></li>
  </ul></li>
  <li><a href="#primary-producer-mortality" id="toc-primary-producer-mortality" class="nav-link" data-scroll-target="#primary-producer-mortality"><strong>9.3. Primary producer mortality</strong></a></li>
  <li><a href="#encystment-of-primary-producers" id="toc-encystment-of-primary-producers" class="nav-link" data-scroll-target="#encystment-of-primary-producers"><strong>9.4. Encystment of primary producers</strong></a></li>
  <li><a href="#effect-of-temperature-and-salinity-on-primary-production" id="toc-effect-of-temperature-and-salinity-on-primary-production" class="nav-link" data-scroll-target="#effect-of-temperature-and-salinity-on-primary-production"><strong>9.5. Effect of temperature and salinity on primary production</strong></a></li>
  <li><a href="#growth-of-mixotrophic-primary-producers" id="toc-growth-of-mixotrophic-primary-producers" class="nav-link" data-scroll-target="#growth-of-mixotrophic-primary-producers"><strong>9.6. Growth of mixotrophic primary producers</strong></a></li>
  </ul></li>
  <li><a href="#consumer-processes" id="toc-consumer-processes" class="nav-link" data-scroll-target="#consumer-processes">10. CONSUMER PROCESSES</a>
  <ul>
  <li><a href="#fluxes-in-consumer-biomass-pools" id="toc-fluxes-in-consumer-biomass-pools" class="nav-link" data-scroll-target="#fluxes-in-consumer-biomass-pools"><strong>10.1. Fluxes in consumer biomass pools</strong></a></li>
  <li><a href="#consumer-feeding" id="toc-consumer-feeding" class="nav-link" data-scroll-target="#consumer-feeding"><strong>10.2. Consumer feeding</strong></a>
  <ul class="collapse">
  <li><a href="#grazing" id="toc-grazing" class="nav-link" data-scroll-target="#grazing"><em>10.2.1 Grazing</em></a></li>
  <li><a href="#what-are-the-c-and-mum-parameters" id="toc-what-are-the-c-and-mum-parameters" class="nav-link" data-scroll-target="#what-are-the-c-and-mum-parameters"><em>10.2.2 What are the</em> C <em>and</em> mum <em>parameters?</em></a></li>
  <li><a href="#what-determines-prey-biomass-available-for-predation" id="toc-what-determines-prey-biomass-available-for-predation" class="nav-link" data-scroll-target="#what-determines-prey-biomass-available-for-predation"><strong><em>10.2.3. What determines prey biomass available for predation?</em></strong></a></li>
  <li><a href="#habitat-refuge" id="toc-habitat-refuge" class="nav-link" data-scroll-target="#habitat-refuge"><strong><em>10.2.4. Habitat refuge</em></strong></a></li>
  <li><a href="#size-refuge-from-predation" id="toc-size-refuge-from-predation" class="nav-link" data-scroll-target="#size-refuge-from-predation"><strong><em>10.2.5. Size refuge from predation</em></strong></a></li>
  <li><a href="#availability-of-fisheries-catches-to-opportunistic-catch-eaters" id="toc-availability-of-fisheries-catches-to-opportunistic-catch-eaters" class="nav-link" data-scroll-target="#availability-of-fisheries-catches-to-opportunistic-catch-eaters"><strong><em>10.2.6. Availability of fisheries catches to opportunistic catch-eaters</em></strong></a></li>
  <li><a href="#effect-of-temperature-and-salinity-on-feeding-parameters" id="toc-effect-of-temperature-and-salinity-on-feeding-parameters" class="nav-link" data-scroll-target="#effect-of-temperature-and-salinity-on-feeding-parameters"><em>10.2.7. Effect of temperature and salinity on feeding parameters</em></a></li>
  <li><a href="#effect-of-oxygen-limitation-on-epibenthic-invertebrate-feeding-rates" id="toc-effect-of-oxygen-limitation-on-epibenthic-invertebrate-feeding-rates" class="nav-link" data-scroll-target="#effect-of-oxygen-limitation-on-epibenthic-invertebrate-feeding-rates"><em>10.2.8. Effect of oxygen limitation on epibenthic invertebrate feeding rates</em></a></li>
  <li><a href="#effect-of-space-limitation-on-epibenthic-invertebrate-feeding-rates" id="toc-effect-of-space-limitation-on-epibenthic-invertebrate-feeding-rates" class="nav-link" data-scroll-target="#effect-of-space-limitation-on-epibenthic-invertebrate-feeding-rates"><em>10.2.9. Effect of space limitation on epibenthic invertebrate feeding rates</em></a></li>
  <li><a href="#other-factors-affecting-feeding-parameters" id="toc-other-factors-affecting-feeding-parameters" class="nav-link" data-scroll-target="#other-factors-affecting-feeding-parameters"><em>10.2.10. Other factors affecting feeding parameters</em></a></li>
  <li><a href="#other-feeding-functional-responses" id="toc-other-feeding-functional-responses" class="nav-link" data-scroll-target="#other-feeding-functional-responses"><em>10.2.11 Other feeding functional responses</em></a></li>
  </ul></li>
  <li><a href="#food-assimilation-in-consumers" id="toc-food-assimilation-in-consumers" class="nav-link" data-scroll-target="#food-assimilation-in-consumers"><strong>10.3. Food assimilation in consumers</strong></a></li>
  <li><a href="#maintenance-respiration-in-consumers" id="toc-maintenance-respiration-in-consumers" class="nav-link" data-scroll-target="#maintenance-respiration-in-consumers"><strong>10.4. Maintenance (respiration) in consumers</strong></a>
  <ul class="collapse">
  <li><a href="#scaling-of-respiration-costs-depending-on-individuals-condition" id="toc-scaling-of-respiration-costs-depending-on-individuals-condition" class="nav-link" data-scroll-target="#scaling-of-respiration-costs-depending-on-individuals-condition"><strong><em>10.4.1 Scaling of respiration costs depending on individual’s condition</em></strong></a></li>
  </ul></li>
  <li><a href="#growth" id="toc-growth" class="nav-link" data-scroll-target="#growth"><strong>10.5. Growth</strong></a>
  <ul class="collapse">
  <li><a href="#growth-and-energy-allocation-in-age-structured-groups" id="toc-growth-and-energy-allocation-in-age-structured-groups" class="nav-link" data-scroll-target="#growth-and-energy-allocation-in-age-structured-groups"><strong><em>10.5.1. Growth and energy allocation in age-structured groups</em></strong></a></li>
  <li><a href="#when-is-an-age-structured-group-starving-and-what-does-it-mean" id="toc-when-is-an-age-structured-group-starving-and-what-does-it-mean" class="nav-link" data-scroll-target="#when-is-an-age-structured-group-starving-and-what-does-it-mean"><em>10.5.2. When is an age-structured group starving and what does it mean?</em></a></li>
  <li><a href="#growth-in-consumer-biomass-pools" id="toc-growth-in-consumer-biomass-pools" class="nav-link" data-scroll-target="#growth-in-consumer-biomass-pools"><strong><em>10.5.3 Growth in consumer biomass pools</em></strong></a></li>
  <li><a href="#feeding-on-material-outside-the-model" id="toc-feeding-on-material-outside-the-model" class="nav-link" data-scroll-target="#feeding-on-material-outside-the-model"><strong><em>10.5.4 Feeding on material outside the model</em></strong></a></li>
  <li><a href="#effect-of-contaminants-on-the-growth-model" id="toc-effect-of-contaminants-on-the-growth-model" class="nav-link" data-scroll-target="#effect-of-contaminants-on-the-growth-model"><strong><em>10.5.5 Effect of contaminants on the growth model</em></strong></a></li>
  </ul></li>
  <li><a href="#mortality-in-consumers" id="toc-mortality-in-consumers" class="nav-link" data-scroll-target="#mortality-in-consumers"><strong>10.6. Mortality in consumers</strong></a>
  <ul class="collapse">
  <li><a href="#mortality-in-age-structured-groups" id="toc-mortality-in-age-structured-groups" class="nav-link" data-scroll-target="#mortality-in-age-structured-groups"><strong><em>10.6.1. Mortality in age-structured groups</em></strong></a></li>
  <li><a href="#starvation-mortality-in-age-structured-groups" id="toc-starvation-mortality-in-age-structured-groups" class="nav-link" data-scroll-target="#starvation-mortality-in-age-structured-groups"><strong><em>10.6.2. Starvation mortality in age-structured groups</em></strong></a></li>
  <li><a href="#extra-mortality-in-age-structured-groups" id="toc-extra-mortality-in-age-structured-groups" class="nav-link" data-scroll-target="#extra-mortality-in-age-structured-groups"><strong><em>10.6.3. Extra mortality in age-structured groups</em></strong></a></li>
  <li><a href="#mortality-in-biomass-pool-consumers" id="toc-mortality-in-biomass-pool-consumers" class="nav-link" data-scroll-target="#mortality-in-biomass-pool-consumers"><strong><em>10.6.4. Mortality in biomass pool consumers</em></strong></a></li>
  <li><a href="#temperature-and-acidification-effect-on-mortality" id="toc-temperature-and-acidification-effect-on-mortality" class="nav-link" data-scroll-target="#temperature-and-acidification-effect-on-mortality"><strong><em>10.6.5. Temperature and acidification effect on mortality</em></strong></a></li>
  <li><a href="#turbidity-effect-on-mortality" id="toc-turbidity-effect-on-mortality" class="nav-link" data-scroll-target="#turbidity-effect-on-mortality"><strong><em>10.6.6. Turbidity effect on mortality</em></strong></a></li>
  </ul></li>
  <li><a href="#waste-production" id="toc-waste-production" class="nav-link" data-scroll-target="#waste-production"><strong>10.7. Waste production</strong></a></li>
  <li><a href="#spawn" id="toc-spawn" class="nav-link" data-scroll-target="#spawn"><strong>10.8. Spawn</strong></a>
  <ul class="collapse">
  <li><a href="#sexual-maturation-in-age-structured-groups" id="toc-sexual-maturation-in-age-structured-groups" class="nav-link" data-scroll-target="#sexual-maturation-in-age-structured-groups"><em>10.8.1. Sexual maturation in age-structured groups</em></a></li>
  <li><a href="#spawning-in-age-structured-groups" id="toc-spawning-in-age-structured-groups" class="nav-link" data-scroll-target="#spawning-in-age-structured-groups"><strong><em>10.8.2. Spawning in age-structured groups</em></strong></a></li>
  <li><a href="#maturation-and-spawn-in-age-structured-biomass-pools" id="toc-maturation-and-spawn-in-age-structured-biomass-pools" class="nav-link" data-scroll-target="#maturation-and-spawn-in-age-structured-biomass-pools"><strong><em>10.8.3. Maturation and spawn in age-structured biomass pools</em></strong></a></li>
  </ul></li>
  <li><a href="#recruitment-and-ageing" id="toc-recruitment-and-ageing" class="nav-link" data-scroll-target="#recruitment-and-ageing"><strong>10.9. Recruitment and ageing</strong></a>
  <ul class="collapse">
  <li><a href="#routines-used-to-calculate-recruitment-in-age-structured-groups" id="toc-routines-used-to-calculate-recruitment-in-age-structured-groups" class="nav-link" data-scroll-target="#routines-used-to-calculate-recruitment-in-age-structured-groups"><strong><em>10.9.1 Routines used to calculate recruitment in age-structured groups</em></strong></a></li>
  <li><a href="#stock-recruitment-options" id="toc-stock-recruitment-options" class="nav-link" data-scroll-target="#stock-recruitment-options"><strong><em>10.9.2. Stock-recruitment options</em></strong></a></li>
  <li><a href="#environmental-habitat-and-external-recruitment-scalars" id="toc-environmental-habitat-and-external-recruitment-scalars" class="nav-link" data-scroll-target="#environmental-habitat-and-external-recruitment-scalars"><strong><em>10.9.3. Environmental, habitat and external recruitment scalars</em></strong></a></li>
  <li><a href="#spatial-distribution-of-recruits" id="toc-spatial-distribution-of-recruits" class="nav-link" data-scroll-target="#spatial-distribution-of-recruits"><strong><em>10.9.4. Spatial distribution of recruits</em></strong></a></li>
  <li><a href="#factors-defining-temporal-arrival-of-recruits" id="toc-factors-defining-temporal-arrival-of-recruits" class="nav-link" data-scroll-target="#factors-defining-temporal-arrival-of-recruits"><strong><em>10.9.5. Factors defining temporal arrival of recruits</em></strong></a></li>
  <li><a href="#modifying-size-at-recruitment" id="toc-modifying-size-at-recruitment" class="nav-link" data-scroll-target="#modifying-size-at-recruitment"><strong><em>10.9.6. Modifying size at recruitment</em></strong></a></li>
  <li><a href="#stochastic-variation-around-mean-recruitment" id="toc-stochastic-variation-around-mean-recruitment" class="nav-link" data-scroll-target="#stochastic-variation-around-mean-recruitment"><strong><em>10.9.7. Stochastic variation around mean recruitment</em></strong></a></li>
  <li><a href="#ageing-of-individuals" id="toc-ageing-of-individuals" class="nav-link" data-scroll-target="#ageing-of-individuals"><strong><em>10.9.8. Ageing of individuals</em></strong></a></li>
  </ul></li>
  </ul></li>
  <li><a href="#distribution-and-movement" id="toc-distribution-and-movement" class="nav-link" data-scroll-target="#distribution-and-movement">11. DISTRIBUTION AND MOVEMENT</a>
  <ul>
  <li><a href="#spatial-distribution-without-migration-or-density-dependent-movement" id="toc-spatial-distribution-without-migration-or-density-dependent-movement" class="nav-link" data-scroll-target="#spatial-distribution-without-migration-or-density-dependent-movement"><strong>11.1 Spatial distribution without migration or density dependent movement</strong></a></li>
  <li><a href="#spatial-distribution-with-density-dependent-movement" id="toc-spatial-distribution-with-density-dependent-movement" class="nav-link" data-scroll-target="#spatial-distribution-with-density-dependent-movement"><strong>11.2 Spatial distribution with density dependent movement</strong></a></li>
  <li><a href="#forcing-the-model-distributions-with-species-distribution-models" id="toc-forcing-the-model-distributions-with-species-distribution-models" class="nav-link" data-scroll-target="#forcing-the-model-distributions-with-species-distribution-models"><strong>11.3. Forcing the model distributions with species distribution models</strong></a></li>
  <li><a href="#migration-into-and-out-of-the-model-domain" id="toc-migration-into-and-out-of-the-model-domain" class="nav-link" data-scroll-target="#migration-into-and-out-of-the-model-domain"><strong>11.4. Migration into and out of the model domain</strong></a>
  <ul class="collapse">
  <li><a href="#annual-migrations" id="toc-annual-migrations" class="nav-link" data-scroll-target="#annual-migrations"><strong><em>11.4.1. Annual migrations</em></strong></a></li>
  <li><a href="#multi-year-migrations" id="toc-multi-year-migrations" class="nav-link" data-scroll-target="#multi-year-migrations"><strong><em>11.4.2. Multi-year migrations</em></strong></a></li>
  </ul></li>
  <li><a href="#external-population-model" id="toc-external-population-model" class="nav-link" data-scroll-target="#external-population-model"><strong>11.5. External population model</strong></a></li>
  <li><a href="#invading-species" id="toc-invading-species" class="nav-link" data-scroll-target="#invading-species"><strong>11.6. Invading species</strong></a></li>
  </ul></li>
  <li><a href="#processes-in-bacterial-and-inanimate-pools" id="toc-processes-in-bacterial-and-inanimate-pools" class="nav-link" data-scroll-target="#processes-in-bacterial-and-inanimate-pools">12. PROCESSES IN BACTERIAL AND INANIMATE POOLS</a>
  <ul>
  <li><a href="#bacterial-processes" id="toc-bacterial-processes" class="nav-link" data-scroll-target="#bacterial-processes"><strong>12.1. Bacterial processes</strong></a></li>
  <li><a href="#biogeochemical-processes" id="toc-biogeochemical-processes" class="nav-link" data-scroll-target="#biogeochemical-processes"><strong>12.2 Biogeochemical processes</strong></a></li>
  </ul></li>
  <li><a href="#influence-of-environmental-factors-on-ecological-processes" id="toc-influence-of-environmental-factors-on-ecological-processes" class="nav-link" data-scroll-target="#influence-of-environmental-factors-on-ecological-processes">13. INFLUENCE OF ENVIRONMENTAL FACTORS ON ECOLOGICAL PROCESSES</a>
  <ul>
  <li><a href="#temperature-two-methods-to-get-temperature-scalar" id="toc-temperature-two-methods-to-get-temperature-scalar" class="nav-link" data-scroll-target="#temperature-two-methods-to-get-temperature-scalar"><strong>13.1. Temperature: two methods to get temperature scalar</strong></a></li>
  <li><a href="#temperature-effects-on-feeding-parameters-and-assimilation-efficiency" id="toc-temperature-effects-on-feeding-parameters-and-assimilation-efficiency" class="nav-link" data-scroll-target="#temperature-effects-on-feeding-parameters-and-assimilation-efficiency"><strong>13.2. Temperature: effects on feeding parameters and assimilation efficiency</strong></a></li>
  <li><a href="#salinity" id="toc-salinity" class="nav-link" data-scroll-target="#salinity"><strong>13.3. Salinity</strong></a></li>
  <li><a href="#acidification" id="toc-acidification" class="nav-link" data-scroll-target="#acidification"><strong>13.4. Acidification</strong></a></li>
  <li><a href="#oxygen" id="toc-oxygen" class="nav-link" data-scroll-target="#oxygen"><strong>13.5. Oxygen</strong></a></li>
  </ul></li>
  <li><a href="#final-overview-of-ecology-routines" id="toc-final-overview-of-ecology-routines" class="nav-link" data-scroll-target="#final-overview-of-ecology-routines">14. FINAL OVERVIEW OF ECOLOGY ROUTINES</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">REFERENCES</a></li>
  <li><a href="#appendix-1-tips-for-calibrating-biological-model-notes-from-the-2015-atlantis-summit" id="toc-appendix-1-tips-for-calibrating-biological-model-notes-from-the-2015-atlantis-summit" class="nav-link" data-scroll-target="#appendix-1-tips-for-calibrating-biological-model-notes-from-the-2015-atlantis-summit">Appendix 1: TIPS FOR CALIBRATING BIOLOGICAL MODEL: NOTES FROM THE 2015 ATLANTIS SUMMIT</a>
  <ul>
  <li><a href="#tips-for-parameterizing-biomass-pools-invertebrates" id="toc-tips-for-parameterizing-biomass-pools-invertebrates" class="nav-link" data-scroll-target="#tips-for-parameterizing-biomass-pools-invertebrates"><strong>Tips for parameterizing biomass pools (invertebrates)</strong></a></li>
  <li><a href="#tips-for-parameterizing-age-structured-vertebrate-groups" id="toc-tips-for-parameterizing-age-structured-vertebrate-groups" class="nav-link" data-scroll-target="#tips-for-parameterizing-age-structured-vertebrate-groups"><strong>Tips for parameterizing age-structured (vertebrate) groups</strong></a>
  <ul class="collapse">
  <li><a href="#starvation-and-respiration" id="toc-starvation-and-respiration" class="nav-link" data-scroll-target="#starvation-and-respiration">Starvation and respiration</a></li>
  <li><a href="#movement" id="toc-movement" class="nav-link" data-scroll-target="#movement">Movement</a></li>
  <li><a href="#reproduction" id="toc-reproduction" class="nav-link" data-scroll-target="#reproduction">Reproduction</a></li>
  <li><a href="#growth-consumption-and-predator-prey-functional-response" id="toc-growth-consumption-and-predator-prey-functional-response" class="nav-link" data-scroll-target="#growth-consumption-and-predator-prey-functional-response">Growth, consumption, and predator-prey functional response</a></li>
  <li><a href="#other-mortality" id="toc-other-mortality" class="nav-link" data-scroll-target="#other-mortality">Other mortality</a></li>
  </ul></li>
  <li><a href="#tips-on-parameterizing-biogeochemistry" id="toc-tips-on-parameterizing-biogeochemistry" class="nav-link" data-scroll-target="#tips-on-parameterizing-biogeochemistry"><strong>Tips on parameterizing biogeochemistry</strong></a></li>
  <li><a href="#tips-on-handling-ocean-acidification" id="toc-tips-on-handling-ocean-acidification" class="nav-link" data-scroll-target="#tips-on-handling-ocean-acidification"><strong>Tips on handling ocean acidification</strong></a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">




<p><strong>Atlantis User’s Guide</strong></p>
<p><strong>Part I:</strong></p>
<p><strong>General Overview, Physics &amp; Ecology</strong></p>
<p><strong>___________________________________</strong></p>
<p>Asta Audzijonyte, Rebecca Gorton, Isaac Kaplan,</p>
<p>Jacob M. Kasper, Elizabeth A. Fulton</p>
<p>Original Date: April 2017</p>
<p>Current Update: May 2025</p>
<p><img src="markdown_output/media/image1.jpg" style="width:1.86944in;height:1.86944in"></p>
<p>****</p>
<p>TABLE OF CONTENTS</p>
<p><a href="#foreward">FOREWARD [10](#foreward)</a></p>
<p><a href="#introduction">1. INTRODUCTION [11](#introduction)</a></p>
<p><a href="#goals-of-this-manual-and-where-to-go-from-here"><strong>1.1. Goals of this manual and where to go from here</strong> [11](#goals-of-this-manual-and-where-to-go-from-here)</a></p>
<p><a href="#terms-and-conventions-used-in-the-manual"><strong>1.2. Terms and conventions used in the manual</strong> [12](#terms-and-conventions-used-in-the-manual)</a></p>
<p><a href="#what-is-atlantis"><strong>1.3. What is Atlantis?</strong> [13](#what-is-atlantis)</a></p>
<p><a href="#existing-applications-and-questions-studied"><strong>1.4. Existing applications and questions studied</strong> [15](#existing-applications-and-questions-studied)</a></p>
<p><a href="#how-does-atlantis-compare-to-other-marine-ecosystem-models"><strong>1.5. How does Atlantis compare to other marine ecosystem models?</strong> [19](#how-does-atlantis-compare-to-other-marine-ecosystem-models)</a></p>
<p><a href="#main-assumptions"><strong>1.6. Main assumptions</strong> [21](#main-assumptions)</a></p>
<p><a href="#installation-and-running">2. INSTALLATION AND RUNNING [22](#installation-and-running)</a></p>
<p><a href="#how-to-check-out-and-install-the-code"><strong>2.1 How to check out and install the code</strong> [22](#how-to-check-out-and-install-the-code)</a></p>
<p><a href="#input-files"><strong>2.2. Input files</strong> [27](#input-files)</a></p>
<p><a href="#overview-of-atlantis-submodels-and-the-main-routine"><strong>2.3. Overview of Atlantis submodels and the main() routine</strong> [28](#overview-of-atlantis-submodels-and-the-main-routine)</a></p>
<p><a href="#delving-into-the-code-finding-the-files-and-routines-that-describe-the-processes-you-are-interested-in"><strong>2.4. Delving into the code: finding the files and routines that describe the processes you are interested in</strong> [30](#delving-into-the-code-finding-the-files-and-routines-that-describe-the-processes-you-are-interested-in)</a></p>
<p><a href="#how-to-setup-and-start-simulations"><strong>2.5. How to setup and start simulations</strong> [33](#how-to-setup-and-start-simulations)</a></p>
<p><a href="#crash-introduction-into-command-line-interfaces"><em>2.5.1. Crash introduction into Command line interfaces</em> [33](#crash-introduction-into-command-line-interfaces)</a></p>
<p><a href="#setting-up-the-bat-file"><em>2.5.2. Setting up the BAT file</em> [34](#setting-up-the-bat-file)</a></p>
<p><a href="#parameter-files-called-in-the-bat-command-line"><em>2.5.3. Parameter files called in the BAT command line</em> [35](#parameter-files-called-in-the-bat-command-line)</a></p>
<p><a href="#setting-up-batch-simulations"><em>2.5.4. Setting up batch simulations</em> [36](#setting-up-batch-simulations)</a></p>
<p><a href="#initialising-simulations"><em>2.5.5. Initialising simulations</em> [37](#initialising-simulations)</a></p>
<p><a href="#_Toc123075512"><strong>2.6. Simulation outputs</strong> [39](#_Toc123075512)</a></p>
<p><a href="#_Toc123075513"><strong>2.7. Introduction to NetCDF files</strong> [44](#_Toc123075513)</a></p>
<p><a href="#_Toc123075514"><strong>2.8. Introduction to Atlantis variables</strong> [47](#_Toc123075514)</a></p>
<p><a href="#chapter29"><strong>2.9 Tools and packages to view and analyse Atlantis input and output files</strong> [50](#chapter29)</a></p>
<p><a href="#troubleshooting-on-installation-and-running"><strong>2.10. Troubleshooting on installation and running</strong> [51](#troubleshooting-on-installation-and-running)</a></p>
<p><a href="#_Toc123075517">3. GEOMETRY OF THE MODEL DOMAIN [55](#_Toc123075517)</a></p>
<p><a href="#_Toc123075518"><strong>3.1. Boxes and layers defining the geometry of the model</strong> [55](#_Toc123075518)</a></p>
<p><a href="#_Toc123075519"><strong>3.2. How to define a geometry file for a new model?</strong> [57](#_Toc123075519)</a></p>
<p><a href="#practical-steps-for-building-the-bgm-files-from-gis-shapefiles"><strong>3.3. Practical steps for building the BGM files from GIS shapefiles</strong> [62](#practical-steps-for-building-the-bgm-files-from-gis-shapefiles)</a></p>
<p><a href="#_Toc123075521"><strong>3.4. How to view your model geometry?</strong> [63](#_Toc123075521)</a></p>
<p><a href="#_Toc123075522"><strong>3.5 Using R for BGM geometry</strong> [63](#_Toc123075522)</a></p>
<p><a href="#_Toc123075523">4. HYDRODYNAMIC, SALINITY AND TEMPERATURE FORCING DATA [64](#_Toc123075523)</a></p>
<p><a href="#_Toc123075524"><strong>4.1. General introduction to forcing of Atlantis models</strong> [64](#_Toc123075524)</a></p>
<p><a href="#_Toc123075525"><strong>4.2. Transforming oceanographic model output into Atlantis forcing files</strong> [66](#_Toc123075525)</a></p>
<p><a href="#_Toc123075526"><strong>4.3. Things to keep in mind about hydrodynamic forcing files</strong> [67](#_Toc123075526)</a></p>
<p><a href="#_Toc123075527"><strong>4.4. Numbering of vertical layers in the input and output NC files</strong> [69](#_Toc123075527)</a></p>
<p><a href="#_Toc123075528"><strong>4.5. Viewing and checking forcing input files with R</strong> [70](#_Toc123075528)</a></p>
<p><a href="#_Toc123075529"><strong>4.6. Light and noise pollution</strong> [71](#_Toc123075529)</a></p>
<p><a href="#_Toc123075530">5. THE PHYSICS SUBMODEL [71](#_Toc123075530)</a></p>
<p><a href="#physical-and-biogeochemical-variables-and-their-characteristics"><strong>5.1. Physical and biogeochemical variables and their characteristics</strong> [72](#physical-and-biogeochemical-variables-and-their-characteristics)</a></p>
<p><a href="#_Toc123075532"><strong>5.2. The main <em>physics()</em> routine</strong> [74](#_Toc123075532)</a></p>
<p><a href="#_Toc123075533"><strong>5.3. Other parameters in <em>physics.prm</em> not included in the <em>physics()</em> routine</strong> [87](#_Toc123075533)</a></p>
<p><a href="#_Toc123075534"><strong>5.4. When to use vertical and horizontal transport and mixing?</strong> [89](#_Toc123075534)</a></p>
<p><a href="#_Toc123075535"><strong>5.5. Latest developments</strong> [91](#_Toc123075535)</a></p>
<p><a href="#_Toc123075536"><em>5.5.1 Land</em> [91](#_Toc123075536)</a></p>
<p><a href="#_Toc123075537"><em>5.5.2. Ice</em> [91](#_Toc123075537)</a></p>
<p><a href="#_Toc123075538"><em>5.5.3. Contaminants</em> [91](#_Toc123075538)</a></p>
<p><a href="#_Toc123075539"><strong>5.6. Outputs of physics tracers in the <em>output.nc</em> file</strong> [95](#_Toc123075539)</a></p>
<p><a href="#_Toc123075540"><strong>5.7. Importance of optional physics routines</strong> [98](#_Toc123075540)</a></p>
<p><a href="#_Toc123075541"><strong>5.8. Checking hydrodynamic transports</strong> [100](#_Toc123075541)</a></p>
<p><a href="#definitions-of-functional-groups">6. DEFINITIONS OF FUNCTIONAL GROUPS [101](#definitions-of-functional-groups)</a></p>
<p><a href="#introduction-to-functional-groups"><strong>6.1. Introduction to functional groups</strong> [101](#introduction-to-functional-groups)</a></p>
<p><a href="#defining-functional-groups"><strong>6.2. Defining functional groups</strong> [105](#defining-functional-groups)</a></p>
<p><a href="#_Toc123075545"><strong>6.3. Defining the GroupType correctly</strong> [111](#_Toc123075545)</a></p>
<p><a href="#_Toc123075546"><strong>6.4. Modifying functional groups in parameterised models</strong> [112](#_Toc123075546)</a></p>
<p><a href="#run-parameters">7. RUN PARAMETERS [112](#run-parameters)</a></p>
<p><a href="#force-parameters">8. FORCE PARAMETERS [119](#force-parameters)</a></p>
<p><a href="#_Toc123075549">9. PRIMARY PRODUCER PROCESSES [130](#_Toc123075549)</a></p>
<p><a href="#_Toc123075550"><strong>9.1. Fluxes in primary producers</strong> [130](#_Toc123075550)</a></p>
<p><a href="#_Toc123075551"><strong>9.2. Primary producer growth</strong> [132](#_Toc123075551)</a></p>
<p><a href="#_Toc123075552"><em>9.2.1. Primary producer nutrient limitation</em> [132](#_Toc123075552)</a></p>
<p><a href="#_Toc123075553"><em>9.2.2. Primary producer light limitation</em> [133](#_Toc123075553)</a></p>
<p><a href="#_Toc123075554"><em>9.2.3. Primary macrophyte producer space limitation</em> [134](#_Toc123075554)</a></p>
<p><a href="#_Toc123075555"><em>9.2.4 Effect of eddies on primary production</em> [135](#_Toc123075555)</a></p>
<p><a href="#_Toc123075556"><strong>9.3. Primary producer mortality</strong> [136](#_Toc123075556)</a></p>
<p><a href="#_Toc123075557"><strong>9.4. Encystment of primary producers</strong> [137](#_Toc123075557)</a></p>
<p><a href="#_Toc123075558"><strong>9.5. Effect of temperature and salinity on primary production</strong> [137](#_Toc123075558)</a></p>
<p><a href="#growth-of-mixotrophic-primary-producers"><strong>9.6. Growth of mixotrophic primary producers</strong> [138](#growth-of-mixotrophic-primary-producers)</a></p>
<p><a href="#consumer-processes">10. CONSUMER PROCESSES [139](#consumer-processes)</a></p>
<p><a href="#_Toc123075561"><strong>10.1. Fluxes in consumer biomass pools</strong> [140](#_Toc123075561)</a></p>
<p><a href="#consumer-feeding"><strong>10.2. Consumer feeding</strong> [141](#consumer-feeding)</a></p>
<p><a href="#grazing"><em>10.2.1 Grazing</em> [142](#grazing)</a></p>
<p><a href="#what-are-the-c-and-mum-parameters"><em>10.2.2 What are the</em> C <em>and</em> mum <em>parameters?</em> [144](#what-are-the-c-and-mum-parameters)</a></p>
<p><a href="#_Toc123075565"><em>10.2.3. What determines prey biomass available for predation?</em> [148](#_Toc123075565)</a></p>
<p><a href="#_Toc123075566"><em>10.2.5. Size refuge from predation</em> [153](#_Toc123075566)</a></p>
<p><a href="#_Toc123075567"><em>10.2.6. Availability of fisheries catches to opportunistic catch-eaters</em> [157](#_Toc123075567)</a></p>
<p><a href="#effect-of-temperature-and-salinity-on-feeding-parameters"><em>10.2.7. Effect of temperature and salinity on feeding parameters</em> [157](#effect-of-temperature-and-salinity-on-feeding-parameters)</a></p>
<p><a href="#effect-of-oxygen-limitation-on-epibenthic-invertebrate-feeding-rates"><em>10.2.8. Effect of oxygen limitation on epibenthic invertebrate feeding rates</em> [157](#effect-of-oxygen-limitation-on-epibenthic-invertebrate-feeding-rates)</a></p>
<p><a href="#effect-of-space-limitation-on-epibenthic-invertebrate-feeding-rates"><em>10.2.9. Effect of space limitation on epibenthic invertebrate feeding rates</em> [159](#effect-of-space-limitation-on-epibenthic-invertebrate-feeding-rates)</a></p>
<p><a href="#other-factors-affecting-feeding-parameters"><em>10.2.10. Other factors affecting feeding parameters</em> [159](#other-factors-affecting-feeding-parameters)</a></p>
<p><a href="#other-feeding-functional-responses"><em>10.2.11 Other feeding functional responses</em> [160](#other-feeding-functional-responses)</a></p>
<p><a href="#food-assimilation-in-consumers"><strong>10.3. Food assimilation in consumers</strong> [164](#food-assimilation-in-consumers)</a></p>
<p><a href="#maintenance-respiration-in-consumers"><strong>10.4. Maintenance (respiration) in consumers</strong> [165](#maintenance-respiration-in-consumers)</a></p>
<p><a href="#_Toc123075575"><em>10.4.1 Scaling of respiration costs depending on individual’s condition</em> [166](#_Toc123075575)</a></p>
<p><a href="#growth"><strong>10.5. Growth</strong> [167](#growth)</a></p>
<p><a href="#_Toc123075577"><em>10.5.1. Growth and energy allocation in age-structured groups</em> [167](#_Toc123075577)</a></p>
<p><a href="#when-is-an-age-structured-group-starving-and-what-does-it-mean"><em>10.5.2. When is an age-structured group starving and what does it mean?</em> [169](#when-is-an-age-structured-group-starving-and-what-does-it-mean)</a></p>
<p><a href="#_Toc123075579"><em>10.5.3 Growth in consumer biomass pools</em> [169](#_Toc123075579)</a></p>
<p><a href="#_Toc123075580"><em>10.5.4 Feeding on material outside the model</em> [172](#_Toc123075580)</a></p>
<p><a href="#_Toc123075581"><em>10.5.4 Effect of contaminants on the growth model</em> [174](#_Toc123075581)</a></p>
<p><a href="#mortality-in-consumers"><strong>10.6. Mortality in consumers</strong> [176](#mortality-in-consumers)</a></p>
<p><a href="#_Toc123075583"><em>10.6.1. Mortality in age-structured groups</em> [176](#_Toc123075583)</a></p>
<p><a href="#_Toc123075584"><em>10.6.2. Starvation mortality in age-structured groups</em> [177](#_Toc123075584)</a></p>
<p><a href="#_Toc123075585"><em>10.6.3. Extra mortality in age-structured groups</em> [178](#_Toc123075585)</a></p>
<p><a href="#_Toc123075586"><em>10.6.4. Mortality in biomass pool consumers</em> [179](#_Toc123075586)</a></p>
<p><a href="#_Toc123075587"><em>10.6.5. Temperature and acidification effect on mortality</em> [179](#_Toc123075587)</a></p>
<p><a href="#_Toc123075588"><em>10.6.6. Turbidity effect on mortality</em> [179](#_Toc123075588)</a></p>
<p><a href="#waste-production"><strong>10.7. Waste production</strong> [180](#waste-production)</a></p>
<p><a href="#spawn"><strong>10.8. Spawn</strong> [184](#spawn)</a></p>
<p><a href="#sexual-maturation-in-age-structured-groups"><em>10.8.1. Sexual maturation in age-structured groups</em> [184](#sexual-maturation-in-age-structured-groups)</a></p>
<p><a href="#recruitment-and-ageing"><strong>10.9. Recruitment and ageing</strong> [186](#recruitment-and-ageing)</a></p>
<p><a href="#_Toc123075593"><em>10.9.1 Routines used to calculate recruitment in age-structured groups</em> [187](#_Toc123075593)</a></p>
<p><a href="#_Toc123075594"><em>10.9.2. Stock-recruitment options</em> [187](#_Toc123075594)</a></p>
<p><a href="#_Toc123075595"><em>10.9.3. Environmental, habitat and external recruitment scalars</em> [193](#_Toc123075595)</a></p>
<p><a href="#_Toc123075596"><em>10.9.4. Spatial distribution of recruits</em> [195](#_Toc123075596)</a></p>
<p><a href="#_Toc123075597"><em>10.9.5. Factors defining temporal arrival of recruits</em> [196](#_Toc123075597)</a></p>
<p><a href="#_Toc123075598"><em>10.9.6. Modifying size at recruitment</em> [196](#_Toc123075598)</a></p>
<p><a href="#_Toc123075599"><em>10.9.7. Stochastic variation around mean recruitment</em> [197](#_Toc123075599)</a></p>
<p><a href="#_Toc123075600"><em>10.9.8. Ageing of individuals</em> [198](#_Toc123075600)</a></p>
<p><a href="#_Toc123075601">11. DISTRIBUTION AND MOVEMENT [199](#_Toc123075601)</a></p>
<p><a href="#_Toc123075602">11.1 Spatial distribution without migration or density dependent movement [200](#_Toc123075602)</a></p>
<p><a href="#_Toc123075603">11.2 Spatial distribution with density dependent movement [201](#_Toc123075603)</a></p>
<p><a href="#_Toc123075604">11.3. Forcing the model distributions with species distribution models [204](#_Toc123075604)</a></p>
<p><a href="#_Toc123075605">11.4. Migration into and out of the model domain [206](#_Toc123075605)</a></p>
<p><a href="#_Toc123075606">11.5. External population model [209](#_Toc123075606)</a></p>
<p><a href="#_Toc123075607">11.5. Invading species [209](#_Toc123075607)</a></p>
<p><a href="#_Toc123075608">12. PROCESSES IN BACTERIAL AND INANIMATE POOLS [210](#_Toc123075608)</a></p>
<p><a href="#bacterial-processes"><strong>12.1. Bacterial processes</strong> [210](#bacterial-processes)</a></p>
<p><a href="#biogeochemical-processes"><strong>12.2 Biogeochemical processes</strong> [213](#biogeochemical-processes)</a></p>
<p><a href="#influence-of-environmental-factors-on-ecological-processes">13. INFLUENCE OF ENVIRONMENTAL FACTORS ON ECOLOGICAL PROCESSES [213](#influence-of-environmental-factors-on-ecological-processes)</a></p>
<p><a href="#temperature-two-methods-to-get-temperature-scalar"><strong>13.1. Temperature: two methods to get temperature scalar</strong> [213](#temperature-two-methods-to-get-temperature-scalar)</a></p>
<p><a href="#temperature-effects-on-feeding-parameters-and-assimilation-efficiency"><strong>13.2. Temperature: effects on feeding parameters and assimilation efficiency</strong> [215](#temperature-effects-on-feeding-parameters-and-assimilation-efficiency)</a></p>
<p><a href="#salinity"><strong>13.3. Salinity</strong> [216](#salinity)</a></p>
<p><a href="#acidification"><strong>13.4. Acidification</strong> [217](#acidification)</a></p>
<p><a href="#oxygen"><strong>13.5. Oxygen</strong> [218](#oxygen)</a></p>
<p><a href="#final-overview-of-ecology-routines">14. FINAL OVERVIEW OF ECOLOGY ROUTINES [218](#final-overview-of-ecology-routines)</a></p>
<p><a href="#references">REFERENCES [223](#references)</a></p>
<p><a href="#appendix-1-tips-for-calibrating-biological-model-notes-from-the-2015-atlantis-summit">Appendix 1: TIPS FOR CALIBRATING BIOLOGICAL MODEL: NOTES FROM THE 2015 ATLANTIS SUMMIT [228](#appendix-1-tips-for-calibrating-biological-model-notes-from-the-2015-atlantis-summit)</a></p>
<p><a href="#tips-for-parameterizing-biomass-pools-invertebrates"><strong>Tips for parameterizing biomass pools (invertebrates)</strong> [230](#tips-for-parameterizing-biomass-pools-invertebrates)</a></p>
<p><a href="#tips-for-parameterizing-age-structured-vertebrate-groups"><strong>Tips for parameterizing age-structured (vertebrate) groups</strong> [230](#tips-for-parameterizing-age-structured-vertebrate-groups)</a></p>
<p><a href="#starvation-and-respiration">Starvation and respiration [230](#starvation-and-respiration)</a></p>
<p><a href="#movement">Movement [230](#movement)</a></p>
<p><a href="#reproduction">Reproduction [231](#reproduction)</a></p>
<p><a href="#growth-consumption-and-predator-prey-functional-response">Growth, consumption, and predator-prey functional response [231](#growth-consumption-and-predator-prey-functional-response)</a></p>
<p><a href="#other-mortality">Other mortality [232](#other-mortality)</a></p>
<p><a href="#tips-on-parameterizing-biogeochemistry"><strong>Tips on parameterizing biogeochemistry</strong> [232](#tips-on-parameterizing-biogeochemistry)</a></p>
<p><a href="#tips-on-handling-ocean-acidification"><strong>Tips on handling ocean acidification</strong> [232](#tips-on-handling-ocean-acidification)</a></p>
<p>****</p>
<p><strong>Tables</strong></p>
<p><a href="#_Toc526762753">Table 1. Examples of Atlantis model applications and their main purposes: P = primary use of the model, S = secondary use, E = exploratory use beginning (modified after Fulton et al.&nbsp;2011). Some of the model descriptions are included in the useful reading page on the wiki. An updated list of models under development is also available on Atlantis webpage [17](#_Toc526762753)</a></p>
<p><a href="#_Toc526762754">Table 2. Some example findings from Atlantis applications (modified after Fulton et al.&nbsp;2011) [18](#_Toc526762754)</a></p>
<p><a href="#_Toc445899521">Table 3. Main Atlantis output files. The main output files of interest for beginner users are shown in grey boxes. The table first lists Biology related outputs and then Harvest submodel related outputs. Please note, that the list of output files are continuously updated, so check the wiki for any changes. [41](#_Toc445899521)</a></p>
<p><a href="#_Toc526762756">Table 4. Parameters used to describe characteristics of Atlantis variables in the initial_conditions.nc file. These parameters are read in by the readBMTracerInfo() in attracerIO.c [48](#_Toc526762756)</a></p>
<p><a href="#_Toc526762757">Table 5. Hydrodynamic structure of some example Atlantis models. Advect = advection, temp = temperature, salt = salinity. [65](#_Toc526762757)</a></p>
<p><a href="#_Toc526762758">Table 6. Examples of currently used obligatory and optional physics and biogeochemical tracers and variables and their characteristics as defined in the initial_conditions.nc file (collected from different Atlantis applications and used for illustration purposes only). Further description of tracers and physics variables is given in the following chapters on sediment and biological processes. [73](#_Toc526762758)</a></p>
<p><a href="#_Toc526762759">Table 7. List and characteristics of physics variables given in the output.nc files (the same variables are given in the initial_conditions.nc file but without the time dimension). [96](#_Toc526762759)</a></p>
<p><a href="#_Toc526762760">Table 8. State variables tracked in the main Atlantis model (no ice or land included). * indicates optional Si tracking in primary producers that are limited by Si. Primary producer Si limitation is set in functional_groups.csv file parameter IsSiliconDep (see Table 6.2). In models that include C and P, both C and P are tracked like N (so the number of variable tracked is three times larger). [103](#_Toc526762760)</a></p>
<p><a href="#_Toc526762761">Table 9. Parameters in the functional_groups.csv file. [105](#_Toc526762761)</a></p>
<p><a href="#_Toc526762762">Table 10. Routines called for different non-vertebrate GroupTypes in Atlantis code. The Call_Group_Process_Function() routine in atecology.c determines the specific routine used from the atGroupProcess.c file non vertebrate groups. Note, that the greatest diversity ofroutines are called for different benthic invertebrate types (for clarity benthic invertebrates are shown in bold italics). [111](#_Toc526762762)</a></p>
<p><a href="#_Toc526762763">Table 11. List and description of parameters in the run.prm file [113](#_Toc526762763)</a></p>
<p><a href="#_Toc526762764">Table 12. List and description of external forcing options available in Atlantis [122](#_Toc526762764)</a></p>
<p><a href="#_Toc526762765">Table 13. Parameters used for primary producer processes [138](#_Toc526762765)</a></p>
<p><a href="#_Toc526762766">Table 14. General feeding parameters and units [147](#_Toc526762766)</a></p>
<p><a href="#_Toc526762767">Table 15. Parameters defining habitat associations and habitat refuge. [152](#_Toc526762767)</a></p>
<p><a href="#_Toc526762768">Table 16. Parameters defining size refuge for age structured groups [157](#_Toc526762768)</a></p>
<p><a href="#_Toc526762769">Table 17. Parameters defining feeding on fisheries catches [157](#_Toc526762769)</a></p>
<p><a href="#_Toc526762770">Table 18. Equations defining grazing term in the twelve currently available feeding functional responses. Some of these responses are described and compared in Fulton et al.&nbsp;2003a “Mortality and predation in ecosystem models: is it important how these are expressed?” Ecological Modelling 169,157–178 [161](#_Toc526762770)</a></p>
<p><a href="#_Toc526762771">Table 19. Parameters defining respiration rates in age structured groups [167](#_Toc526762771)</a></p>
<p><a href="#_Toc526762772">Table 20. Parameters affecting growth in consumers [173](#_Toc526762772)</a></p>
<p><a href="#_Toc526762773">Table 21. Parameters needed for contaminant uptake model – where XXX is the functional group code and CCC the contaminant name. [175](#_Toc526762773)</a></p>
<p><a href="#_Toc526762774">Table 22. Parameters affecting mortality in consumers [180](#_Toc526762774)</a></p>
<p><a href="#_Toc526762775">Table 23. Parameters determining waste production processes [183](#_Toc526762775)</a></p>
<p><a href="#_Toc526762776">Table 23. Parameters determining maturation and spawn [186](#_Toc526762776)</a></p>
<p><a href="#_Toc526762777">Table 24. Parameters used in the invading species sub-model [210](#_Toc526762777)</a></p>
<p><a href="#_Toc526762778">Table 25. Effects of pH on physiological and ecological processes. See detailed description here [217](#_Toc526762778)</a></p>
<p><a href="#_Toc526762779">Table 26. List of key ecological processes run in the three main distribution types: water colum, sediment, and epibenthic layer. List of the pools tracked in the three types is given in table 8. [221](#_Toc526762779)</a></p>
<p><a href="#_Toc526762780">Table 27. Basic assumptions and formulations in the Atlantis model (modified after Fulton et al.&nbsp;2004b). [222](#_Toc526762780)</a></p>
<p><strong>Figures</strong></p>
<p><a href="#_Toc446445862">Figure 1. Some existing applications of Atlantis modelling framework and specific model domains (see Table 1 for details on numbered models). [16](#_Toc446445862)</a></p>
<p><a href="#_Ref445803305">Figure 2.Schematic representation of the Atlantis model structure and input files. Submodels coloured in green are obligatory. Files marked with * are required, others are optional. [29](#_Ref445803305)</a></p>
<p><a href="#_Toc526762783">Figure 3. Brief overview of the Atlantis main() routine that calls other routines implemented in various submodels. [30](#_Toc526762783)</a></p>
<p><a href="#_Toc526762784">Figure 4. Top: An example cross section of an Atlantis model, showing water column and sediment layers. Oceanic boxes have an open water boundary at the bottom of the deepest water layer. The depths of vertical layers do not have to be the same in all boxes, although most models keep them identical for simplicity (shown with horizontal dashed line). The depth of the bottom water column layer varies depending on the geography of the marine system. Bottom: An example of the horizontal Atlantis model domain (set in the BGM file) [56](#_Toc526762784)</a></p>
<p><a href="file:////Users/ful083/Work/atlantis/manual/final_uploaded_manual_version/AtlantisUserGuide_PartI.docx#_Toc526762785">Figure 5. Example of an Atlantis model geometry (part of a model domain). Boxes are numbered in bold, grey boxes indicate boundary or island (land) boxes. Faces are numbered in italic numbers. Vertices are shown as dots; vertices P1 and P2 describe face 5. [58](#_Toc446445864)</a></p>
<p><a href="file:////Users/ful083/Work/atlantis/manual/final_uploaded_manual_version/AtlantisUserGuide_PartI.docx#_Toc526762786">Figure 6. An example of oceanographic sigma depth layers (grey lines) with an overlay of Atlantis spatial structure. Figure by Javier Porobic Garate [66](#_Toc526762786)</a></p>
<p><a href="#_Toc526762787">Figure 7. Main steps in converting oceanographic files into Atlantis input files, as done using two CSIRO contributed scripts Hydro and Hydroconstruct. Figure by Javier Porobic Garate [67](#_Toc526762787)</a></p>
<p><a href="#_Toc526762788">Figure 8. Illustration of optional physics routines described in detail above. Some boxes have two sediment layers. Vertical mixing can also be set as downwelling, but only one direction is possible in a given simulation. [83](#_Toc526762788)</a></p>
<p><a href="#_Toc526762789">Figure 9. Total final biomass ratios of all Baltic Sea model functional groups in scenarios exploring the effects of optional physics routines. The simulations were run for 20 years. [99](#_Toc526762789)</a></p>
<p><a href="#_Toc526762790">Figure 10. The value of grazing term (Gr) as a function of prey biomass (ranging from 0 to 1x of the predator biomass) and different assumptions on the ratio between C and mum. Predators biomass is 1000mgN, C = 100 m<sup>3</sup> ind<sup>-1</sup> day<sup>-1</sup>. The calculations assume only one prey, no refuge (δ<sub>refuge</sub>=1), maximum availability p<sub>prey,CX</sub> of 1, and assimilation efficiencies E<sub>prey,i</sub> of 1. Black: mum = C, orange: mum = C*3, blue: mum =C/5 [145](#_Toc526762790)</a></p>
<p><a href="#_Toc526762791">Figure 11. Habitat refuge scalar on prey availability as a function of cell area covered with suitable habitats (see box above for details on calculations of suitable cover area). [151](#_Toc526762791)</a></p>
<p><a href="#_Toc526762792">Figure 12. Prey availability scalar Size_Scalar<sub>prey</sub> of different sized prey for different Kmax_coefft values for humped selectivity curve. [154](#_Toc526762792)</a></p>
<p><a href="#_Toc526762793">Figure 13. Prey availability scalar δ<sub>size</sub> of different sized prey for bilogistic selectivity curve. [155](#_Toc526762793)</a></p>
<p><a href="#_Toc526762794">Figure 14. Oxygen limitation scalar δ<sub>O2</sub> for different ambient oxygen concentrations, assuming KO2=50 mgO2 m<sup>-3</sup> and KO2lim = 100 mgO2 m<sup>-3</sup>. [158](#_Toc526762794)</a></p>
<p><a href="#_Toc526762795">Figure 15. Proportion of available energy allocated to SN, depending on the individual’s condition (RN/SN, on x axis), and pR and X_RS parameters. [168](#_Toc526762795)</a></p>
<p><a href="#_Toc526762796">Figure 16. Starvation mortality (per day) for different Kthreshm and mStarve values. [178](#_Toc526762796)</a></p>
<p><a href="#_Toc526762797">Figure 17. Nitrogen flow from consumers to detrital and ammonia pools. Parameters that determine allocation between the pools are shown on the arrows. Optional fluxes are shown in dashed arrows. [181](#_Toc526762797)</a></p>
<p><a href="#_Toc526762798">Figure 18. Tracking of calendar years in Atlantis age groups. Left: three calendar years for each of the four age groups (AG1-AG4) illustrated. Right: dampening of strong year classes through time. [198](#_Toc526762798)</a></p>
<p><a href="#_Toc526762799">Figure 19. The T<sub>scalar</sub> values at different water temperatures. [214](#_Toc526762799)</a></p>
<p><a href="#_Toc526762800">Figure 20. A humped temperature response used when q10_method_XXX=1. [215](#_Toc526762800)</a></p>
<p>****</p>
<p><strong>Acknowledgements</strong></p>
<p>The authors want to express their sincere thanks to everyone who helped to prepare this manual, especially Shane Richards, Ingrid van Putten, Heidi Pethybridge, Cathy Bulman, Javier Porobic Garate, Thibaut Houitte de La Chesnais, Michael Sumner and many others who read the earlier versions of this manual and provided valuable feedback. We would also like to acknowledge that earliest stages of the compilation of the manual was facilitated by the Atlantis Summit (Hawaii 2015), funded by the CSIRO, US National Oceanic and Atmospheric Administration, and the Norwegian Institute of Marine Resources. Lastly, this work would not have been possible without the funding support through the years of the model development of the CSIRO, Australian Fisheries Research and Development Corporation (FRDC), National Oceanic and Atmospheric Administration (NOAA), the Gordon and Betty Moore Foundation, and the David and Lucile Packard Foundation.</p>
<section id="foreward" class="level1">
<h1>0. FOREWARD</h1>
<p>Before delving into Atlantis we would like to provide a little bit of background on the modelling framework and this manual.</p>
<p>Atlantis is just one of many marine ecosystem models, originally known as BM2 (BoxModel 2) it was christened Atlantis by Villy Christensen in South Africa in 2001. Marine ecosystem models have existed for more than 50 years, though they have only grown in popular use since the advent of (fast) modern computing power. They have grown from a biophysical focus to include more and more of the human dimensions. This is reflected in the structure of this manual, which sequentially works through the physical then biological before getting into the human dimensions. Atlantis was originally developed with an eye to temperate marine ecosystems and fisheries, though it has grown through time.</p>
<p>This manual (and the wiki) is intended to help provide guidance on what is contained in each submodel in Atlantis, with explanation of assumptions and some pointers on parameterisation and calibration. While it is not an explicit step-by-step recipe book on how to build a model, it will hopefully help new users to understand the core requirements pf Atlantis, while providing a handy reference for more experienced users on particular variants or processes they may want to add in a specific instance. For those (rare beings) who chose to read the manual end-to-end there is some repetition as we wanted to make sure those who were flitting in and out would contact the important information pertinent to the topic, similarly there is cross referencing between sections where important information is in another section of the manual, That cross referencing is not just for completeness, it is because the processes are interlinked and if you are having a difficulty it may be because of something in those other components.</p>
<p>Atlantis was never intended for anyone else’s use, but was simply the ecological sandbox for me to explore my understanding of the marine world and how to represent it mathematically. It has been a huge shock to see it adopted by others and I remain deeply apologetic for how difficult it is to use (if I had ever known how it would be used I would likely have done things very differently). Atlantis remains a chimera of past decisions (often foist upon us by expediency or technological limitations) and current developments, as we address new questions. We will endeavour to keep the manual and associated wiki up to date as much as possible, but if you feel something is missing or unclear please let us know.</p>
<p>Note that I have shifted from I to we, because Atlantis now represents a group effort and all those who have participated in its development (especially Bec) and the construction of the manual (in particular Asta) should know they have my gratitude and deep appreciation for how outstanding a group of intellectual power they represent. Similarly, the community of Atlantis users represents a welcoming and stimulating group of collaborators!</p>
<p>For anyone new to Atlantis coming to grips with it is like climbing Olympus Mons. It is not for everyone (particularly as it is very data and time intensive!!). Much like when cooking a good stew, go lightly with the spice and concentrate on the hearty base constituents. Most of all try to relax and have fun.</p>
<p>Beth (Somewhere over the Indian Ocean, June 2016).</p>
</section>
<section id="introduction" class="level1">
<h1>1. INTRODUCTION</h1>
<section id="goals-of-this-manual-and-where-to-go-from-here" class="level2">
<h2 class="anchored" data-anchor-id="goals-of-this-manual-and-where-to-go-from-here"><strong>1.1. Goals of this manual and where to go from here</strong></h2>
<p>Before launching into what Atlantis is and what it assumes it is important to explain what the intent of the manual is, how it can be used and the meaning of key terms that will be used throughout.</p>
<p>This manual is intended for Atlantis beginners who have some experience with modelling, running programs in command line (yes, this is how to run Atlantis!), and writing or using C or R scripts for analysing the outputs. While we have provided some information useful during model construction, this is not a comprehensive step-by-step recipe book on how to create a new model from scratch. We assume in many sections that you already have a somewhat operational Atlantis model that you want to use. The manual will also be useful for experienced users, because even experienced users often focus on the area of their interest and may not be thoroughly familiar with other aspects of the modelling tool. Finally, the manual will aim to provide starting points for developers of new Atlantis applications, but you will have to venture out and learn a lot more than what is covered in the manual.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p>As at the time of writing this manual (2016), two versions of Atlantis code are available for to the community<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a>. The main version is called <em>trunk</em>, which is the usual way programmers refer to the main line of the code development. It was released in January 2015 and it includes all the latest developments and updates, and should be used for all new model applications. <strong>This manual, with some rare exceptions, will apply to the <em>trunk</em> version only.</strong> The old version <em>bec_dev</em> is no longer being actively extended and is intended for users who have been using Atlantis for a while and are unable to switch for whatever reason.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>Atlantis is a rapidly developing tool. There are three main resources to keep up with updates:</p>
<p>1. The <a href="https://confluence.csiro.au/display/Atlantis/Atlantis+Ecosystem+Model+Home+Page"><strong>Atlantis Wiki</strong></a> is the first place to go for further information. Also remember to look at the page on <a href="https://confluence.csiro.au/display/Atlantis/Atlantis+FAQ"><em>Frequently Asked Questions</em></a> for troubleshooting issues.</p>
<p>2. The <a href="https://groups.google.com/forum/#!forum/atlantis-ecosystem-model"><strong>Atlantis Google group</strong></a> is a good place to ask questions from the Atlantis user community and search topics of previous discussions.</p>
<p>3. <a href="https://confluence.csiro.au/display/Atlantis/Getting+the+source+code">The code itself!</a> The code is commented and is the best way to understand how different processes are implemented. (We know that may sound confronting and challenging to those less used to code, but doing a word search on the code and looking at what it does is really how Beth and Bec go about answering any questions you ask them… even they don’t keep it all in their heads!)</p>
<table class="caption-top">
<thead>
<tr class="header">
<th><p><strong>Some important references for further information on model details and equations (see Atlantis wiki page on <a href="https://confluence.csiro.au/display/Atlantis/Useful+Reading">useful reading</a>)</strong></p>
<p><strong>Models used as a basis for the physical and biology processes in Atlantis</strong></p>
<p>1) Murray AG, Parslow JS (1999). The analysis of alternative formulations in a simple model of a coastal ecosystem. Ecological Modelling, 119, 149-166.</p>
<p>2) Murray AG, Parslow JS (1997). Port Phillip Bay Integrated Model: Final Report. Technical Report No.44, CSIRO Environmental Projects Office Canberra, ACT, Australia</p>
<p>3) Ebenhöh W, Kohlmeier C, Radford P J (1995). The benthic biological submodel in the European regional seas ecosystem model. Netherlands Journal of Sea Research, 33, 423-452.</p>
<p><strong>Publications explaining which processes from the models above are used in Atlantis (initially called IGBEM and BM2).</strong></p>
<p><strong>READING THESE TWO PUBLICATIONS IS ESSENTIAL FOR ANY ATLANTIS USER!</strong></p>
<p>4) Fulton EA, Smith AD, Johnson CR (2004a). Biogeochemical marine ecosystem models I: IGBEM—a model of marine bay ecosystems. Ecological Modelling, 174, 267-307.</p>
<p>5) Fulton EA, Parslow JS, Smith AD, Johnson CR (2004b) Biogeochemical marine ecosystem models II: the effect of physiological detail on model performance. Ecological Modelling, 173, 371–406</p>
<p><strong>Other Atlantis (or Atlantis related) reports</strong> <strong>and publications</strong></p>
<p>6) Fulton EA (2001) The Effects of Model Structure and Complexity on the Behavior and Performance of Marine Ecosystem Models. PhD Dissertation University of Tasmania</p>
<p>7) Fulton EA, Smith AD, Johnson CR (2003). Mortality and predation in ecosystem models: is it important how these are expressed? Ecological Modelling, 169, 157-178</p>
<p>8) Fulton EA, Smith ADM, Smith DC (2007) Alternative management strategies for Southeast Australian Commonwealth fisheries. AFMA, FRDC</p>
<p>9) Link JS, Gamble RJ, Fulton EA (2011) NEUS—Atlantis: construction, calibration, and application of an ecosystem model with ecological interactions, physiographic conditions and fleet behavior. US Dept. Comm. NOAA Technical Memorandum NMFS-NEFSC-218. 249pp.</p></th>
</tr>
</thead>

</table>
</section>
<section id="terms-and-conventions-used-in-the-manual" class="level2">
<h2 class="anchored" data-anchor-id="terms-and-conventions-used-in-the-manual"><strong>1.2. Terms and conventions used in the manual</strong></h2>
<p>This manual is referenced throughout, and you can click on <u>hyperlinks</u> to take you to appropriate web pages or relevant parts of the manual.</p>
<p>Below are a few terms and standards.</p>
<p><strong>Parameters</strong> and <strong>variables</strong>, used in various input and parameter files, are marked in orange.</p>
<p><strong>Routine</strong> names are shown in <em>italics().</em> In computer programming, routines and subroutines are nearly synonymous terms describing a sequence of code that is intended of to be called repeatedly during the execution of a program. A routine performs a specific task and is relatively independent of the remaining code (although it might use variables defined or calculated in other routines).</p>
<p>Routines are grouped into <strong>files</strong>, used to perform a larger function. For example, the <strong>atphysics.c</strong> file includes a number of routines performing physical processes (gas exchange, deposition, diffusion). The code files will be shown as <strong>bold</strong> . The files are then grouped into larger <strong>libraries</strong>, also shown in <strong>bold</strong>. Currently, Atlantis has 12 libraries, each containing dozens of files that, in turn, each contain several to dozens of routines.</p>
<p>Atlantis parameters can be grouped into <strong>flags</strong> and <strong>parameters</strong> (in narrow sense). <strong>Flags</strong> activate various optional routines; they are used in the if statements in the code (if flagA=1 do routine aaa). For further information about what flags do, check this link to <a href="https://confluence.csiro.au/display/Atlantis/Finding+all+the+options+for+the+flags">flags in the biology parameter file</a> on the Atlantis wiki. There are no clear rules on naming of parameters and flags, so the name alone does not indicate whether it is a parameter or flag.</p>
<p>Atlantis requires a lot of input files. Some of them are used for external forcing of the model (see Figure 2) and will be referred to as <strong>inputs</strong>. They are all listed in the <em>force.prm</em> file (note names of files used to run the model will be given simply in italics, to avoid confusing them with code files). Other input files define parameters or groups used in the submodels or simulation conditions and are referred to as <strong>parameter files</strong>. They are all listed in the BAT file used to initialise the runs.</p>
<p>Throughout the manual important messages and rules are placed in boxes that start with <strong>NOTE!</strong> Suggestions for good practice rules are given as <strong>Good practice tips</strong>. These tips give ideas on how to ensure reproducibility of your work and also highlight some common mistakes.</p>
<p>Most terms will be explained in the relevant sections of the manual. Here we will introduce only absolutely key ones.</p>
<p><strong>Age groups</strong> defines age classes used in the model to represent fully age-structure groups (and can be set to anything from a few months to any number of calendar years). Many parameters for these groups are age-group-specific. All individuals in an age group are considered identical, unless the option of <a href="file:///C:\HY-Data\ASAUDZIJ\documents\manual\distinct">multiple genotypes is used</a>, in which case all individuals within one genotype of one age group are identical.</p>
<p><strong>Fisheries</strong> are fishing units that can have unique and very customisable characteristics, distinguishing them from other fishing units. The number and names of fisheries are given in the fisheries.csv parameter file. If fishing is simulated dynamically (see below) then fisheries behave more like separate fleets with dynamic characteristics.</p>
</section>
<section id="what-is-atlantis" class="level2">
<h2 class="anchored" data-anchor-id="what-is-atlantis"><strong>1.3. What is Atlantis?</strong></h2>
<p>Atlantis is a whole-of-system 3D spatially-explicit marine ecosystem modelling framework based on complementary submodels used to simulate physical and biogeochemical processes, ecology, human uses of marine and coastal systems (primarily fisheries), and management. It is written in the C programming language and was originally based on two other models – the Port Phillip Bay Integrated Model (PPBIM) and the European Regional Seas Ecosystem Model (ERSEM). Atlantis combines different aspects of PPBIM and ERSEM (see Fulton et al.&nbsp;2004a reference in the box above) and adds a lot of new specific features (Fulton et al.&nbsp;2004b). Developed by CSIRO scientists in Australia, it has now been applied to more than 30 locations around the world (Fulton 2004; Fulton et al.&nbsp;2011).</p>
<p>The model explicitly tracks the flow of nutrients through the trophic levels, typically using nitrogen as the “common currency” (carbon and phosphorous may also be used). At the core of Atlantis is a deterministic biophysical submodel that is spatially-resolved in three dimensions using a map made up of user defined boxes (polygons) and slab-like depth layers. The model geometry is flexible and should be matched to the major geographical and bioregional features of the simulated marine system. The oceanographic processes, such as water fluxes, salinity and temperature conditions are forced, typically based on outputs from oceanographic models (although they can be read in from observation data or calculated endogenously within Atlantis). Together with the optional forcing of nutrient inputs, these oceanographic processes drive deterministic primary production. The main ecological processes considered are <strong><em>production, consumption and predation, waste production and cycling, migration, reproduction and recruitment, habitat dependency and mortality.</em></strong> Biological functional groups include habitat-forming species like kelp, corals and sponges, as well as other benthic invertebrates, vertebrates, phytoplankton, zooplankton, refractory and labile detritus, and carrion. Predation is based on a prey availability matrix (similar in concept to the vulnerabilities used in the feeding arena theory, Walters and Kitchell 2001) with the final realised diet modified by the predators’ gape limitations and feeding rates, spatiotemporal co-occurrences, the state of habitat refugia (for habitat dependent species) and in some instances nutritional quality and spawning status. Lower trophic levels are treated as biomass pools, whereas some key invertebrates and all vertebrates are represented as age groups. All individuals within an age group are considered identical and their conditions are tracked through energy allocation to structural and reserve nitrogen. The biological components are replicated in each layer of each of the polygons. Movement of organisms between the polygons is simulated by advective transfer for plankton and optional active migration for macrofauna.</p>
<table class="caption-top table">
<tbody>
<tr class="odd">
<td style="text-align: left;">Atlantis is a strategic, exploratory tool for investigating ecological processes, management policies, and broad consequences of alternative scenarios. It is not intended to replace traditional tactical models, such as annually updated stock assessments used to set fisheries quotas.</td>
</tr>
</tbody>
</table>
<hr>
<p>Atlantis has very flexible Harvest and Economics submodels that allow the user to set a large number of alternative harvest scenarios. A limited set of simple representations of other marine sectors (shipping, aquaculture, tourism, mining, energy generation and coastal land uses) can also be used. Pollution, coastal development, run-off and broad-scale environmental change can also be represented using forcing time series. The human use and impacts submodels are optional, which means that simulations can be run without any explicit exploitation or human impact.</p>
<p>The Harvest submodel is mostly focused on the dynamics of fishing fleets. Either stand-alone, or in combination with the Economics submodel, the human dimensions can be customised to simulate exploitation regimes and potential management schemes. These can range from a simple constant fishing rate on a single functional group to complex, dynamic fisheries comprised of interacting fleets, each with its own gear, bycatch and management characteristics, exploratory fishing, and all affected by compliance decisions and economic issues, such as prices of fish quota trading. Alternatively, Atlantis can be coupled with specific stand-alone fleet dynamics models, such as random utility or agent-based models.</p>
<p>Exploration of management outcomes in Atlantis can be done in two ways. The first is a simple scenario projection, where constant management rules are retained unmodified through time. The second is the full management strategy evaluation mode, where the monitoring, assessment and decision making steps (and their feedback onto fishery regulations and behaviours) are represented. In the latter case, the Harvest submodel interacts with the ecosystem, but also supplies ‘simulated data’ to the Assessment submodel. The Assessment submodel is designed to simulate the assessment process with realistic levels of measurement uncertainty, in both measurement bias and variance. These ‘simulated data’ are derived from the outputs from the Biology and Harvest submodels, given a user-specified monitoring scheme that allows mimicking of a wide range of fisheries and survey information. The ‘data’ are then fed into an appropriate assessment model chosen by the user, such as Surplus Production, Adapt VPA or Bayesian integrated stock assessments. Additionally a range of ecological indicators can be calculated. The output of the Assessment submodel is fed to the Management submodel, designed as a set of decision rules and management levers that will affect fisheries. These include gear restrictions, days at sea, quotas, spatial and temporal zoning, discarding restrictions, size limits, bycatch mitigation, or dynamic reference points. It is also possible to track the approximate management costs, as monitoring and enforcement costs.</p>
<p>One of Atlantis’ defining characteristics is its modular construction. A wide range of alternative assumptions and model implementations are provided in each of the submodels. This allows the user to choose the desired complexity– from a relatively simple model of a few groups with simple trophic interactions and catch equations through to complex models, with complicated stock structure, multiple fleets, detailed economics and multiple management levers. However, this flexibility requires that users fully understand the processes and do not resort to ‘default settings’. <strong>Atlantis is not a model where you can simply sit and pull multiple handles to discern what it is doing. It requires digging deeply into its underlying processes, particularly when first building and calibrating a new application.</strong> It also means that building and parameterising a new and complex model will require a lot of time and data, and can also lead to large uncertainties around model outcomes.</p>
</section>
<section id="existing-applications-and-questions-studied" class="level2">
<h2 class="anchored" data-anchor-id="existing-applications-and-questions-studied"><strong>1.4. Existing applications and questions studied</strong></h2>
<p>Most Atlantis development and applications have centred on evaluating alternative management strategies for multispecies fisheries, or exploratory analyses of multiple ecosystem pressures. To date, the Atlantis framework has been applied to over 30 ecosystems around the world (Fig. 1) primarily to study questions of:</p>
<blockquote class="blockquote">
<p>1) overall assessment of alternative management strategies (Kaplan et al.&nbsp;2012, Savina et al.&nbsp;2013, Fulton et al.&nbsp;2014)</p>
<p>2) historical impacts of harvesting (Link et al.&nbsp;2010),</p>
<p>3) compliance with fishery regulations (Ainsworth et al.&nbsp;2012a, 2012b),</p>
<p>4) robustness of ecological indicators (Fulton et al.&nbsp;2005),</p>
<p>5) exploration of fisher behaviour (Kaplan et al.&nbsp;2014),</p>
<p>6) impacts of global change, including acidification ( Kaplan et al.&nbsp;2010; Griffith et al.&nbsp;2011, 2012, Fulton and Gorton 2014; Weijerman et al.&nbsp;2015),</p>
<p>7) effects of body size changes (Audzijonyte et al.&nbsp;2013, 2014, 2015).</p>
<p>8) the implications of model complexity (Fulton et al.&nbsp;2003, Fulton 2004)</p>
<p>More recently Atlantis has been used to consider the impacts of marine and coastal industries other than fishing.</p>
</blockquote>
<p><span id="_Toc446445862" class="anchor"></span>Figure 1. Some existing applications of Atlantis modelling framework and specific model domains (see Table 1 for details on numbered models).</p>
<p><span id="_Toc526762753" class="anchor"></span>Table 1. Examples of Atlantis model applications and their main purposes: P = primary use of the model, S = secondary use, E = exploratory use beginning (modified after Fulton et al.&nbsp;2011). Some of the model descriptions are included in the <a href="https://confluence.csiro.au/display/Atlantis/Useful+Reading">useful reading</a> page on the wiki. An updated list of models under development is also available on Atlantis <a href="http://atlantis.cmar.csiro.au/">webpage</a></p>
<table>
<colgroup>
<col style="width: 20%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 19%">
</colgroup>
<tbody>
<tr>
<td>
System
</td>
<td style="text-align: center;">
<blockquote class="blockquote">
<p>
Complexity
</p>
</blockquote>
</td>
<td style="text-align: center;">
<blockquote class="blockquote">
<p>
Understanding
</p>
</blockquote>
</td>
<td style="text-align: center;">
<blockquote class="blockquote">
<p>
System-level MSE
</p>
</blockquote>
</td>
<td style="text-align: center;">
<blockquote class="blockquote">
<p>
Fisheries
</p>
</blockquote>
</td>
<td style="text-align: center;">
<blockquote class="blockquote">
<p>
Nutrients
</p>
</blockquote>
</td>
<td style="text-align: center;">
<blockquote class="blockquote">
<p>
Mining
</p>
</blockquote>
</td>
<td style="text-align: center;">
<blockquote class="blockquote">
<p>
Climate impacts
</p>
</blockquote>
</td>
<td style="text-align: center;">
<blockquote class="blockquote">
<p>
Catchments
</p>
</blockquote>
</td>
<td style="text-align: center;">
<blockquote class="blockquote">
<p>
Indicators
</p>
</blockquote>
</td>
<td style="text-align: center;">
<blockquote class="blockquote">
<p>
Management
</p>
</blockquote>
</td>
<td>
Reference
</td>
</tr>
<tr>
<td>
<em>Australia</em>
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td>
</td>
</tr>
<tr>
<td>
<blockquote class="blockquote">
<p>
</p><ol type="1">
<li>SE Australia
<p></p>
</li></ol></blockquote>
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
S
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
P
</td>
<td>
Fulton et al 2007
</td>
</tr>
<tr>
<td>
<blockquote class="blockquote">
<p>
</p><ol start="2" type="1">
<li>Victoria/Tasmania
<p></p>
</li></ol></blockquote>
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
S
</td>
<td style="text-align: center;">
S
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
S
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
P
</td>
<td>
Smith et al.&nbsp;2010
</td>
</tr>
<tr>
<td>
<blockquote class="blockquote">
<p>
</p><ol start="3" type="1">
<li>Port Phillip Bay
<p></p>
</li></ol></blockquote>
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
S
</td>
<td style="text-align: center;">
S
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
</td>
<td>
Fulton et al 2004b
</td>
</tr>
<tr>
<td>
<blockquote class="blockquote">
<p>
</p><ol start="4" type="1">
<li>Westernport
<p></p>
</li></ol></blockquote>
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td>
Savina et al 2005
</td>
</tr>
<tr>
<td>
<blockquote class="blockquote">
<p>
</p><ol start="5" type="1">
<li>E Tasmania
<p></p>
</li></ol></blockquote>
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
S
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
E
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
E
</td>
<td>
Johnson et al 2008
</td>
</tr>
<tr>
<td>
<blockquote class="blockquote">
<p>
</p><ol start="6" type="1">
<li>Tasmania
<p></p>
</li></ol></blockquote>
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
S
</td>
<td>
-
</td>
</tr>
<tr>
<td>
<blockquote class="blockquote">
<p>
</p><ol start="7" type="1">
<li>Derwent-Huon
<p></p>
</li></ol></blockquote>
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
E
</td>
<td style="text-align: center;">
E
</td>
<td style="text-align: center;">
E
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
E
</td>
<td style="text-align: center;">
E
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
E
</td>
<td>
Savina 2009
</td>
</tr>
<tr>
<td>
<blockquote class="blockquote">
<p>
</p><ol start="8" type="1">
<li>New South Wales
<p></p>
</li></ol></blockquote>
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
S
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
S
</td>
<td>
Savina et al 2008
</td>
</tr>
<tr>
<td>
<blockquote class="blockquote">
<p>
</p><ol start="9" type="1">
<li>Clarence River
<p></p>
</li></ol></blockquote>
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
E
</td>
<td style="text-align: center;">
E
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
E
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
E
</td>
<td>
Hayes et al 2007
</td>
</tr>
<tr>
<td>
<em>North America</em>
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td>
</td>
</tr>
<tr>
<td>
<blockquote class="blockquote">
<p>
</p><ol start="10" type="1">
<li>NE USA
<p></p>
</li></ol></blockquote>
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
S
</td>
<td style="text-align: center;">
P
</td>
<td>
Link et al 2010
</td>
</tr>
<tr>
<td>
<blockquote class="blockquote">
<p>
</p><ol start="11" type="1">
<li>California Current
<p></p>
</li></ol></blockquote>
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
S
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
S
</td>
<td style="text-align: center;">
P
</td>
<td>
Brand et al 2007
</td>
</tr>
<tr>
<td>
<blockquote class="blockquote">
<p>
</p><ol start="12" type="1">
<li>Central California
<p></p>
</li></ol></blockquote>
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
S
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
P
</td>
<td>
Horne et al.&nbsp;2010
</td>
</tr>
<tr>
<td>
<blockquote class="blockquote">
<p>
</p><ol start="13" type="1">
<li>Chesapeake Bay
<p></p>
</li></ol></blockquote>
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
S
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
P
</td>
<td>
Ihde et al.&nbsp;2016
</td>
</tr>
<tr>
<td>
<blockquote class="blockquote">
<p>
</p><ol start="14" type="1">
<li>Gulf of California
<p></p>
</li></ol></blockquote>
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
S
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
E
</td>
<td style="text-align: center;">
E
</td>
<td style="text-align: center;">
E
</td>
<td style="text-align: center;">
P
</td>
<td>
Ainsworth et al.&nbsp;2012b
</td>
</tr>
<tr>
<td>
<blockquote class="blockquote">
<p>
</p><ol start="15" type="1">
<li>Gulf of Mexico
<p></p>
<p>
</p><ol start="16" type="1">
<li>Campeche Bay
<p></p>
<p>
</p><ol start="17" type="1">
<li>Desoto Bay
<p></p>
</li></ol></li></ol></li></ol></blockquote>
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
<p>
P
</p>
<p>
P
</p>
<p>
P
</p>
</td>
<td style="text-align: center;">
<p>
P
</p>
<p>
P
</p>
</td>
<td style="text-align: center;">
<p>
P
</p>
<p>
P
</p>
<p>
P
</p>
</td>
<td style="text-align: center;">
S
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
S
</td>
<td style="text-align: center;">
<p>
P
</p>
<p>
P
</p>
</td>
<td>
<p>
Ainsworth et al.&nbsp;2015
</p>
<p>
Ortega-Ortiz and Ainsworth in prep.
</p>
<p>
Gosnell S., unpubl.
</p>
</td>
</tr>
<tr>
<td>
<p>
<em>Other areas</em>
</p>
<blockquote class="blockquote">
<p>
</p><ol start="18" type="1">
<li>Guam
<p></p>
</li></ol></blockquote>
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
P
</td>
<td style="text-align: center;">
S
</td>
<td style="text-align: center;">
P
</td>
<td>
Weijerman et al 2015
</td>
</tr>
</tbody>
</table>


















<p>The greatest challenge in implementing an Atlantis model is evaluating uncertainty of parameters and model forecasts. Traditional sensitivity analyses are not possible in models as complex as Atlantis, and therefore parameter uncertainty must be assessed in other ways. New means of achieving this are under development, but historically it has been done using ‘bounded parameterisation’, where extreme sets of parameters still giving realistic biomasses are used to explore sensitivity. For further discussion on evaluating uncertainty in Atlantis, see chapter 3.2 and 6.1 in Fulton et al.&nbsp;2007. To read about uncertainty in ecosystem models in general see Link et al.&nbsp;2012.</p>
<p>For this reason Atlantis is well suited for broader scale exploratory analyses, “what if” scenarios, general evaluation of management strategies and other questions that draw on its ability to simulate multiple interactions between environmental factors, different marine species and humans. It is important to note that predicted biomasses of some functional groups are not necessarily expected to match biomasses predicted from other kinds of models, such as stock assessments (although ideally the trends would match if the same fishing pressure is included in both models). In general Atlantis produces realistic ecosystem structure and function, with biomasses of the correct order of magnitude and plausible dynamics. However, the details can be uncertain and hard to predict with a great deal of precision. For this reason, Atlantis should not be used for tactical short term advice such as setting quotas. However, it could inform fisheries models on factors such as variable natural mortality, as long as the inherent uncertainties are well recognised. The strategic nature of Atlantis is described in detail in Fulton et al.&nbsp;(2011).</p>
<p>Atlantis is not a simple model to implement and even the simplest parameterisation will take many months to a year or more to implement and calibrate. Most models developed from scratch should be considered a close to full time effort for 1-2 years. Including an elaboration of the human dimensions can increase this by a further 6-12 months. Without this level of effort model development can drag on for many years. Once developed however individual simulations can be done quite quickly (on the order of hours to days) so that many questions can be addressed rapidly. The investment in individual models means that re-use of a model for multiple questions make sense, as the models often represent a synthesis of broad system understanding, but care must be taken to make sure this re-use is appropriate. Put thought into whether the model structure captures the correct features, whether the parametrisation decisions still hold. In short go through the model design steps as if you were starting a new and check that the existing model holds up against these needs – that it is a suitable model. For those who would like more guidance on the re-use of ecosystem models please refer to Essington and Plagányi (2013).</p>
<p><span id="_Toc526762754" class="anchor"></span>Table 2. Some example findings from Atlantis applications (modified after Fulton et al.&nbsp;2011)</p>
<table class="caption-top table">
<colgroup>
<col style="width: 12%">
<col style="width: 87%">
</colgroup>
<thead>
<tr class="header">
<th><strong>Model</strong></th>
<th><strong>Findings</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><p>SE Australia</p>
<p>(SE model)</p></td>
<td><p>- Human behavioural uncertainty produces unanticipated outcomes of management decisions and are important to future adaptive capacity</p>
<p>- If companion TACs are implemented this leads to the ecological decline of weak stocks or significant economic costs (e.g.&nbsp;75% of the catch foregone)</p>
<p>- Relative biomass indicators, ratios of biomass (e.g.&nbsp;pelagic:demersal), size structure and size at maturity are consistently the most robust indicators of the effects of fishing (at fine or very large scales)</p>
<p>- Decline in fish body size can greatly increase predation mortality through positive feedback loops in predator-prey age structure. This can prevent complete recovery from fishing, leading to a new and often lower equilibrium biomass</p>
<p>- Small pelagic fish (like sardines) play a much less important role in temperate Australian ecosystems than in classical upwelling systems</p></td>
</tr>
<tr class="even">
<td><p>Victoria/</p>
<p>Tasmania</p>
<p>(SM model)</p></td>
<td><p>- Relative biomass indicators are consistently the most robust indicators of the performance of spatial management (though their performance can be weak at regional scales due to noisy data resulting from overlapping process scales)</p>
<p>- Universal fisheries reference points are unlikely to work</p>
<p>- Some indicators are only effective at small or very large scales due to how species are aggregated and distributed</p>
<p>- Monitoring design is critical for long-term interpretation of time series</p></td>
</tr>
<tr class="odd">
<td><p>Eastern Tasmania</p>
<p>(SETas model)</p></td>
<td>- off Tasmania there appears to be minor effects of fishing squids, but large impacts of targeting myctophids, suggesting a wasp waist system</td>
</tr>
<tr class="even">
<td><p>New South Wales</p>
<p>(NSW model)</p></td>
<td><p>- The size of no-take marine protected areas can be important for their ultimate success, but the redistribution of the catch and effort outside the closed areas is even more important</p>
<p>- Not all species benefit from spatial management (e.g.&nbsp;prey species do not)</p>
<p>- Moderate levels of fishing pressure can lead to an increase in biodiversity</p>
<p>- The explicit representation of ontogeny can be important for correctly representing the dynamics of marine species that show large spatial or habitat shifts with age</p></td>
</tr>
<tr class="odd">
<td>NE United States (NEUS model)</td>
<td><p>- Despite their complexity, clear patterns are detectable in large complicated biophysical systems (making calibrated predictions feasible)</p>
<p>- Modelling squid and shrimp is extremely difficult (a finding recurring in many ecosystem models), but should be attempted as it can lead to new system understanding</p>
<p>- Modelling ecology is much simpler than modelling a coupled ecological-anthropogenic system</p>
<p>- Well calibrated models have high model skill, but typically only for those system components informed by data</p></td>
</tr>
<tr class="even">
<td>California Current (EMOCC model)</td>
<td><p>- Catch shares are more economically efficient, even if catch levels remain unchanged</p>
<p>- Monitoring and enforcement is critically important for fisheries management and guidance of fleet behaviour</p>
<p>- Ocean acidification is likely to lead to the reduction in biomass of predators dependent on shelled invertebrate prey; in contrast other species (e.g.&nbsp;jellyfish) are likely to increase in biomass</p>
<p>- Ratios of functional group biomass (e.g.&nbsp;piscivore : planktivore) are strong indications of community restructuring under fishing</p></td>
</tr>
<tr class="odd">
<td>Central California (ECCAL model)</td>
<td><p>- Fleets are differentially effected by changes in management rules or environmental conditions (some suffer significant economic declines while others see substantial increases)</p>
<p>- Shifting gears to minimise bycatch and habitat destruction does not restore age structures, but spatial closures can</p></td>
</tr>
<tr class="even">
<td>Gulf of California (GOC model)</td>
<td>- Reported catch values are insufficient to cause observed declines in species biomasses (by 3-5 times for most species)</td>
</tr>
</tbody>
</table>
</section>
<section id="how-does-atlantis-compare-to-other-marine-ecosystem-models" class="level2">
<h2 class="anchored" data-anchor-id="how-does-atlantis-compare-to-other-marine-ecosystem-models"><strong>1.5. How does Atlantis compare to other marine ecosystem models?</strong></h2>
<p>The key characteristics of Atlantis is that it is 1) whole-of-system, 2) spatially explicit, 3) age structured, 4) deterministic, and 5) comprised of submodels that simulate human impacts and management.</p>
<p>1) ‘Whole-of-system’ means that Atlantis aims to represent the entire ecosystem (even if greatly simplified) rather than focus on a few species of particular interest. It spans environmental drivers, habitats, the foodweb, and major human uses (fisheries in detail, but also simple representations of tourism, mining, energy generation, ports, shipping and the other coastal uses such as urban developments). Many other ‘whole-of-system’ or ‘end-to-end’ models are being developed but at present Atlantis tends to include more explicit oceanography and nutrient cycling than foodweb models such as Ecosim with Ecopath (EwE; Christensen and Walters 2004) and a more extensive foodweb than targeted system models like OSMOSE (Shin and Cury 2004).</p>
<p>2) Atlantis is spatially explicit, which makes it a useful tool to explore spatial dynamics of various ecological and socio-economic processes (assuming you can parameterise them). In this way the model is similar to Ecospace (Walters et al.&nbsp;1999), OSMOSE , SEAPODYM (Lehodey et al.&nbsp;2008), and other models that operate on an oceanographic domain (Fiechter et al.&nbsp;2014).</p>
<p>3) Atlantis includes two means of representing species – biomass pools (similar to those in EwE, these pools can include a small number of life-stage stanzas) and fully age-structured, where the average size-at-age of a typical individual is tracked along with the membership of that age group. Typically, biomass pools are used for invertebrate groups and age-structured for vertebrates, but that is not a strict requirement. The representation used for a species is dictated by the modeller in the .csv file used to define the configuration of the model groups. The realised growth and size of species represented through age-structured groups is dynamically dependent on food intake, whereas feeding interactions and reproductive output depends on the realised size and condition. This flexible age-size relationship adds temporal variation in size-at-age, essential for capturing important ecosystem dynamics and useful for exploring the ecological and fisheries consequences of changes in individual size and life-histories (Audzijonyte et al.&nbsp;2013, 2014). The combination of spatial structure and temporal age-size dynamics means that Atlantis can produce realistic ecosystem dynamics both temporally and spatially and a lot of attention has been given to developing the models ability to capture age-size dependencies, temporal, habitat and spatial variation (Fulton et al., 2004c). The Atlantis users are therefore encouraged to make full use of these aspects. Alternatively, if the spatial and temporal age dynamics is not important, it might be better to consider using simpler and most likely computationally faster models.</p>
<p>4) In its standard form, Atlantis is not an agent-based (individual-based) model, like OSMOSE or InVitro (Gray et al 2006)<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a>. It simulates temporally dynamic size-age relationship, but at a given time all individuals in an age group in one cell are identical (variation between cells is possible if a species is sedentary). Thus, Atlantis models processes based on representative “average” individuals for an age group, not true (separate and unique) individuals. This makes Atlantis computationally faster, but it does not simulate inter-individual variation and any potential emergent system properties.</p>
<p>5) Atlantis does not simulate stochastic environmental and ecological processes. Although Atlantis can include a small amount of recruitment timing variation, it is generally a deterministic model. This makes Atlantis different from models such as OSMOSE where stochastic processes are an integral part of the model formulation.</p>
<p>6) Atlantis was the first ecosystem model to include management strategy evaluation options - i.e.&nbsp;the inclusion of each part of the adaptive management cycle – the biological world, the human uses (especially fisheries), monitoring and reporting, assessment and management. EwE and OSMOSE now also include an MSE capacity. New features of human impact are constantly being added to Atlantis and the existing set-up is very flexible to customisation. Fisheries are represented in the most detail, but human impacts can also include nutrient and pollutant inputs, tourism, ports, shipping, urban areas, energy generation, heavy industry, mining and other marine and coastal industries.</p>
</section>
<section id="main-assumptions" class="level2">
<h2 class="anchored" data-anchor-id="main-assumptions"><strong>1.6. Main assumptions</strong></h2>
<p>Like any model, Atlantis has a lot of assumptions and it is important to keep them in mind when using the model and interpreting its outputs. The key ecological assumptions are listed here (as all models, regardless of the level of human representation, contain these assumptions) and these and any other assumptions will also be listed in the description of different submodels and processes – in particular refer to section 14 for an extended list of ecological assumptions in tabular form.</p>
<p>1. While not an ecological assumption the first and foremost assumption of Atlantis is that box 0 is a boundary box and box 1 is a dynamic box. If this assumption is violated things will go <strong>terribly</strong> wrong (most likely you model will not run at all).</p>
<p>2. Atlantis is a deterministic model, which means that a specific combination of parameters and initial conditions will produce the same results (except for possible very minor differences due to the way computers round small numbers).</p>
<p>3. State variables, such as salinity, oxygen, nutrients, sediments, organisms – are tracked as passively or actively advected or moving tracers. Passive tracers are distributed by water flows, whereas actively moving tracers (mostly benthic invertebrates and vertebrates) can have different types of movement (density-dependent, seasonal migrations, sedentary).</p>
<p>4. Each cell of the model (one depth layer of one spatial box) is considered uniform in its environmental variables and hydrodynamic processes. The conditions of these variables can change at each time step (usually every 12h) depending on the water fluxes between cells. Epibenthic cells can include different habitats, modelled as a proportion of the whole cell. This proportion is treated abstractly (via statistical means) and is not given a specific geographic location within the cell. The availability of habitat refugia can scale ecological and fisheries processes, such as predation intensity or vulnerability to capture.</p>
<p>5. Predation is determined by the user defined predator-prey interaction matrix that sets a maximum availability of each prey biomass available to a specific predator. The realised consumption is then modified by the actual spatial overlap, biomass of the prey, habitat refuge for the prey, gape limitation of the predator, spawning time of the predator, possible effect of acidification or contaminants on the predator, and availability of fisheries discards as food. The realised consumption therefore can be very different from what is set in the predator-prey matrix. However, if the predator-prey matrix does not allow a specific interaction (the value is set to 0), predation will never occur.</p>
<p>6. Within a layer of a single box, all individuals within an age group or within a single genotype of an age group (if several genotypes per age group are used) of a fully age-structured group are identical. Condition of these cohorts may vary among cells depending on the local food availability and parameterisation used. Fully age-structured groups are typically used to represent vertebrates.</p>
<p>7. Biomass pool groups are represented as a single pool per model cell (one layer of one box). Age structured biomass pool representations can be used (where a separate biomass pool use used per life history stage for the group). Invertebrates are often represented using biomass pools.</p>
<p>8. Energy flow through the system is primarily tracked using nitrogen as a “currency”. The initial source of nitrogen in the system and each trophic group is set in the initial conditions file. Throughout a simulation nitrogen can be added into the system through optional forced inputs (e.g.&nbsp;river inputs) and atmospheric deposition. Nitrogen is lost through optional burial in the sediments (“decay”), denitrification, loss to the atmosphere (outgassing) and any optional forced sinks. The energy flow through diatoms or other user defined Si limited functional groups is tracked through both N and Si. It is possible to setup the energy flows to be tracked through nitrogen, carbon and phosphorus, but that requires significant additional input file preparation and computing resources, and is recommended only in systems where P limitation is essential.</p>
<p>8. Energy allocation in vertebrates is tracked as structural and reserve nitrogen. Structural nitrogen represents bones, scales, and other hard parts that cannot be reabsorbed if feeding conditions are poor. Reserve nitrogen represents muscle, gonad, and other soft structures that may be used as an energy reserve if conditions are poor. Structural nitrogen cannot decrease throughout an individual’s life-time (an individual cannot get smaller). Reserve nitrogen is used during reproduction or starvation and its ratio to structural nitrogen can change through time.</p>
<p>9. Maintenance energy costs are assumed to be covered through assimilation efficiency and optional total-weight specific respiration. However, most currently developed Atlantis applications do not include the optional respiration. Energy loss from reproduction is subtracted from spawn production as a functional group specific constant. There is no explicit inclusion of movement costs, although the time taken to move between cells is explicitly accounted for.</p>
<p>10. The earliest, larval life stages are not modelled explicitly. The recruitment type is defined by the user and occurs at a set time after spawning.</p>
</section>
</section>
<section id="installation-and-running" class="level1">
<h1>2. INSTALLATION AND RUNNING</h1>
<section id="how-to-check-out-and-install-the-code" class="level2">
<h2 class="anchored" data-anchor-id="how-to-check-out-and-install-the-code"><strong>2.1 How to check out and install the code</strong></h2>
<p>Atlantis is dynamic and new modifications are introduced almost every month. In addition, Atlantis can run on Windows, Mac and Linux (Unix) operating systems. Therefore, you cannot download one executable, but have to check out and compile the code yourself. This ensures that you have the latest version of the code (and please remember to update it regularly!) with all associated libraries required to run Atlantis. As new functionality is added to Atlantis, sometimes code updates also means that new parameters must be added to the parameter files. While this can be frustrating, it is still highly recommended to update the code at least once a month, as updates also include bug fixes. To keep informed on new parameters the users are advised to subscribe to Atlantis wiki blog posts. In addition, if you need help from the model developers you are much more likely to get help quickly if they don’t have to update your model first!</p>
<p>Atlantis source code is hosted in a subversion (abbreviated as SVN) repository. SVN is a commonly used version control system that is used to manage the collections of files that are changed over time. It keeps a history of the code base and the changes that have been made to it, so developers can recall past code versions for reproducibility. It also allows several people to work on the code simultaneously and merge it back together. <strong>As a normal user you will have read-only access to the Atlantis code repository.</strong></p>
<p>If you want <strong>to modify the code</strong>, please check out wiki for instructions on how to create a <a href="https://confluence.csiro.au/display/Atlantis/Switch+working+copy+to+another+branch">new branch</a> from your working copy or <a href="http://svnbook.red-bean.com/en/1.5/svn.branchmerge.basicmerging.html">merge</a> your modifications with new updates. The Atlantis wiki also has more information on modifying the code, such as creating and adding <a href="https://confluence.csiro.au/display/Atlantis/Add+a+input+parameter">a new input parameter</a> or adding <a href="https://confluence.csiro.au/display/Atlantis/Adding+a+new+GroupType">a new type of functional groups</a></p>
<p>To check out the code and install the code on your computer follow these steps:</p>
<p><strong>1. Get access to Atlantis SVN repository</strong> by joining the <a href="https://confluence.csiro.au/display/Atlantis/Atlantis+Ecosystem+Model+Home+Page">Atlantis Wiki</a>, signing the user licence agreement and requesting <a href="https://confluence.csiro.au/display/Atlantis/CSIRO+licence+and+repository+request">access to the SVN</a> repository. The response time on this should not take more than a couple of days (if it does poke Beth again as she is likely on the road and has not seen the email).</p>
<p><strong>2. Check out the code from SVN to your computer.</strong> Experienced users can check out the code by using svn commands through a command line terminal (dos or terminal window). Alternatively, you can use special software to make things easier.</p>
<p><strong>Windows</strong> users can use the free package <a href="http://tortoisesvn.tigris.org/">TortoiseSVN</a>, which works like a plugin for Windows Explorer. To check out the code with TortoiseSVN first install it on your computer. Next, create a new folder where you will keep the code, e.g.&nbsp;C:/AtlantisCode. Rick-click the mouse on this folder and chose “SVN Checkout…” option. This will bring up the TortoiseSVN menu:</p>
<p><img src="markdown_output/media/image3.png" style="width:4.15625in;height:3.02917in"></p>
<p>Enter the URL of the code depository, provided on the Atlantis Wiki source code <a href="https://confluence.csiro.au/display/Atlantis/Getting+the+source+%20code">page</a>, making sure the directory to install the code on your computer is correct, and chose “Fully recursive” Checkout Depth and “HEAD revision”. Detailed instructions on how to check out the code with TortoiseSVN are available on the wiki (<a href="file:///C:\HY-Data\ASAUDZIJ\documents\manual\HYPERLINK%20%22https:\confluence.csiro.au\display\Atlantis\TortoiseSVN">here).</a></p>
<p><strong>Mac</strong> users can check out the code using the app <a href="http://versionsapp.com/">Versions</a>, while Linux users can use <a href="http://svnbook.red-bean.com/en/1.6/svn.ref.svn.c.checkout.html">RedBean</a>.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p>When updating the code, make sure you always update to your main directory where all your code is located, such as C:/AtlantisCode, rather than any subfolders within it. This will ensure that libraries are updated properly.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p><strong>3. Install software for compiling the code.</strong> Compiling Atlantis code in Windows and Linux operating systems is easy. It is <strong>a lot harder to set up on Mac operating</strong> systems (although once installed everything runs smoothly enough).</p>
<p>Windows users can build the code using freely available Microsoft Visual Studio software. Currently supported versions are 2005/2008/2010/2012 and, as of March 2016, you can download 2010 and 2012 Visual Studio installers from the Atlantis wiki by clicking on <a href="https://confluence.csiro.au/pages/viewpageattachments.action?pageId=43155524&amp;metadataLink=true">the paperclip sign</a> on the <a href="https://confluence.csiro.au/display/Atlantis/Building+under+windows">wiki</a>.</p>
<p>Instructions for software and building Atlantis under Linux and other platforms are available <a href="https://confluence.csiro.au/display/Atlantis/Building+Atlantis">on the wiki</a>. If you are a Mac user you may find the use of comfort food at this stage a good idea, each Mac installation appears to face its own challenges, sometimes it goes very easily other times you can spend a day or more wrestling to get one of the supporting libraries to work with your architecture. There does not seem to be any rhyme or reason to this! Macs are just special that way.</p>
<p><strong>4. Install the NetCDF 4 libraries.</strong> These libraries are required for running Atlantis. Detailed instructions on installing NetCDF and updating the path are available <a href="https://confluence.csiro.au/display/Atlantis/NetCDF+4+in+Windows">on the wiki</a>. It is very important that you do this properly, as Atlantis will not run otherwise. Windows users can download pre-compiled libraries, Mac and Linux users will need to down load the source and compile locally.</p>
<p><strong>5. Compile the code.</strong> In Windows compiling the code does not require any programming skills. All you need to do is double-click the Visual Studio SLN file (chose the version suitable for your Visual Studio, e.g.&nbsp;Atlantis_VS2010.sln) located in the main directory of where you checked out the code (e.g.&nbsp;C:\programs\Atlantis). This will load the project into Visual Studio. Then go to the Visual Studio Build menu and chose “Build Solution”. Detailed instructions are given on Atlantis Wiki <a href="https://confluence.csiro.au/%20display/Atlantis/Building+Atlantis">building</a> page.</p>
<p><img src="markdown_output/media/image4.jpg" style="width:5.14151in;height:3.63653in"></p>
<p>Once the build has successfully completed you will find the latest executable <em>atlantismain.exe</em> in C:\...\Atlantis\atlantismain\Debug folder (where … indicates your designated directory for the code)</p>
<table class="caption-top">
<colgroup>
<col style="width: 101%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p>You will need to recompile or build the code anew every time you update it from the SVN repository</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>On Linux (and Mac) platforms GCC is recommended for compiling the code. Other options for compiling are also available for users who for any reason do not want to use Microsoft Visual Studio or GCC. These options include the <a href="https://confluence.csiro.au/display/Atlantis/Using+the+Intel+Compiler">Intel compiler</a> and <a href="https://confluence.csiro.au/display/Atlantis/Building+under+cygwin">Cygwin</a>. However, these are not recommended compilers and users will potentially have to work out solutions to issues themselves.</p>
<p><strong>6. Create a directory for your simulations.</strong></p>
<p>This is where you will keep all your parameters and run outputs. This could be called C:\AtlantisRuns\... This is NOT the same directory as the code itself, so please make sure you understand the difference. If you use several models, such as California, SE Australia and Antarctica, might want to keep parameter files of each model in separate directories, e.g.&nbsp;C:\AtlantisRuns\California\, C:\AtlantisRuns\SE_Australia\ and so on.</p>
<p>Please make sure not to use spaces in the name of the directory (e.g.&nbsp;use C:\AtlantisRuns or C:\Atlantis_Runs not C:\Atlantis Runs as the core code of Atlantis was written before spaces were allowed in directory names and it may not correctly read file names if the directory structure includes spaces<a href="#fn3" class="footnote-ref" id="fnref3" role="doc-noteref"><sup>3</sup></a>).</p>
<table class="caption-top">
<thead>
<tr class="header">
<th><p>The <a href="https://confluence.csiro.au/display/Atlantis/Sample+input+files"><strong>South East Tasmanian (SETas) model</strong></a> is freely available from the Atlantis SVN repository as a learning tool. It is a simple model with only a few boxes and few functional groups. All required input and parameter files for SETas model can be downloaded from the wiki and used as templates for new model applications.</p>
<p>Make sure you use SVN checkout to download these files. Don’t use the web browser to copy the files. Also make sure you download files for the correct version of the Atlantis code (<em>trunk</em> or <em>bec_dev</em>)</p></th>
</tr>
</thead>

</table>
<p><strong>7. Check out or copy files of your specific model and latest Atlantis executable into the run directory.</strong></p>
<p>Also copy the latest executable <em>atlantismain.exe</em> from the C:\...\Atlantis\atlantismain\Debug folder into the run directory you created before (e.g.&nbsp;C:\AtlantisRuns).</p>
<p>The executable <strong>must</strong> be in the same directory as the main BAT run file, required to initialise your runs</p>
<table class="caption-top">
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p>Make sure you copy the new executable file from …Atlantis\atlantismain\Debug into your run directory every time you update and recompile the code. Forgetting to do this is one of the more common mistakes.</p>
<p>Alternatively, you can insert a line into the start of your BAT file to copy it before the start of each run if you want to make sure you always have the latest version.</p>
<p>This would look like:</p>
<p>copy C:\...your Code directory \atlantismain\Debug\atlantismain.exe .</p>
<p>The dot at the end of the line means that the file atlantismain.exe from Debug directory will be copied to the same directory as the BAT file.</p>
<p>See the <a href="https://confluence.csiro.au/display/Atlantis/Windows+Command+Prompt">wiki</a> for details.</p></th>
</tr>
</thead>

</table>
<p><strong><u>Good practice tip 1</u></strong></p>
<p>A good practice Atlantis run directory is uncluttered and contains separate folders for:</p>
<p>1) external forcing files or <strong>inputs</strong>. These files are referred to in <em>force.prm</em> file and will rarely need to be modified after the parameterisation of the model. They are usually placed in the ‘inputs’ directory</p>
<p>2) other <strong>parameter</strong> files. These parameters will be listed in the <em>.bat</em> file used to initialise simulations. Depending on the research question, you will probably modify at least some of the biological, harvest or socio-economic parameters in each of the simulations. Although it is not obligatory we suggest that you keep all your parameters in a separate ‘parameters’ directory.</p>
<p>It is a good idea to give parameter files meaningful names. For example, it would be hard to remember what <em>A.prm</em> refers to, whereas <em>Biol_noclimate_change.prm</em> is a lot more informative. However, there is a limit to the length of a filename Atlantis will read-in so if your model won’t run because it can’t find a file and that file does exist but has a long name try shortening the name and see if that helps.</p>
<p><img src="markdown_output/media/image50.tmp" style="width:4.76in;height:2.73311in"></p>
</section>
<section id="input-files" class="level2">
<h2 class="anchored" data-anchor-id="input-files"><strong>2.2. Input files</strong></h2>
<p>Atlantis input files come in six file formats:</p>
<p>1) BGM file defining model geometry (see chapter 3)</p>
<p>2) NC initial conditions and forcing files (see chapter 2.8)</p>
<p>3) CSV files defining biological and fisheries groups (see chapter 6)</p>
<p>4) PRM files listing flags to activate optional calculations and parameters values (described in detail in subsequent chapters)</p>
<p>5) TS files defining time series of external forcing drivers (see chapter 8)</p>
<p>6) BAT file used to initiate the run (see chapter 2.5)</p>
<p>All files except for NC files can be opened and edited using any basic text editing software (such as TextPad or NotePad). Until recently this was the usual way to setup and modify model parameters. However, some PRM files are thousands of lines long, which means that hand editing can be tedious and a source of error. New R-based tools are currently under development to help users visualise and modify parameter files in a more streamlined way (see below)</p>
<p><strong>NOTE!</strong></p>
<p><strong>PC, Mac and Unix computers use different syntax to indicate line end.</strong> This means that should you receive files from someone using a different operating system, the PRM files will have to be converted to match your local operating system. In Linux this can be done using the <a href="https://confluence.csiro.au/display/Atlantis/File+Formats">‘flip’ software</a>. If you only have a few files this can be done by simply opening the parameter files in a text editor (e.g.&nbsp;TextPad in Windows or TextWrangler in Mac) and saving them in a format suitable for your operating system (see menu “File Format”). More info on line terminators can be found on the wiki <a href="http://en.wikipedia.org/wiki/Newline">here</a> or by searching in your favourite search engine.</p>
<p><img src="markdown_output/media/image60.tmp" style="width:5.888in;height:3.82256in"></p>
</section>
<section id="overview-of-atlantis-submodels-and-the-main-routine" class="level2">
<h2 class="anchored" data-anchor-id="overview-of-atlantis-submodels-and-the-main-routine"><strong>2.3. Overview of Atlantis submodels and the main() routine</strong></h2>
<p>Detailed descriptions of Atlantis submodels and specific input files are provided in subsequent chapters. For a start you need to understand which of the files are required to set up a simple simulations and what they mean (<strong>Error! Reference source not found.</strong>). The figure below gives a schematic of the Atlantis submodels overlaid one over another. At the very basis there is the model geometry, Physics and Biogeochemical (Biology and Ecology) submodels. The other, optional, submodels use their outputs to simulate the human domain.</p>
<p><img src="markdown_output/media/image7.jpg" style="width:6.69306in;height:4.50833in"></p>
<p><span id="_Ref445803305" class="anchor"></span>Figure 2.Schematic representation of the Atlantis model structure and input files. Submodels coloured in green are obligatory. Files marked with * are required, others are optional.</p>
<p>The 3D structure of your model domain, or <strong>Model Geometry</strong>, is described in the BGM file. The model geometry is used to simulate all processes in the five submodels.</p>
<p><strong>Physics</strong> and <strong>Biology</strong> submodels are required (although you can switch some parts off for debugging purposes), whereas <strong>Harvest</strong>, <strong>Assessment</strong> and <strong>Economics</strong> submodels are optional. Each submodel has a specific PRM file describing parameters used in it. Biology and Harvest submodels also require a list of the relevant groups, such as biological functional groups or list of fisheries, in the CSV format.</p>
<p><strong>Initial conditions</strong> are the spatially explicit starting values for all modelled state variables (e.g.&nbsp;spatial distribution of numbers of age 1 cod at the start of your simulations). The initial conditions are specified in the initial condition NC file<strong>.</strong></p>
<p>In addition, all simulations require two supporting files <em>force.prm</em> and <em>run.prm</em> (see below for a brief description of all parameter files).</p>
<p>The <em>main()</em> routine is in the <strong>atlantismain.c</strong> file in the <strong>atlantismain</strong> library. It calls in a set order all other routines that initialise and run the requested submodels and routines and write output files. Each submodel and their main routines will be described in detail below, but a general overview is given in Fig. 3.</p>
<p><img src="markdown_output/media/image8.jpeg" style="width:6.96528in;height:5.37153in"></p>
<p><span id="_Toc526762783" class="anchor"></span>Figure 3. Brief overview of the Atlantis <em>main()</em> routine that calls other routines implemented in various submodels.</p>
</section>
<section id="delving-into-the-code-finding-the-files-and-routines-that-describe-the-processes-you-are-interested-in" class="level2">
<h2 class="anchored" data-anchor-id="delving-into-the-code-finding-the-files-and-routines-that-describe-the-processes-you-are-interested-in"><strong>2.4. Delving into the code: finding the files and routines that describe the processes you are interested in</strong></h2>
<p>This chapter aims to give a very brief introduction into the Atlantis code. You do not need to use the code to run simulations. However, if you can navigate around the code and find routines that use the parameters and execute the processes you are interested in, it will most helpful and informative. In the end, this is the best way to understand what exactly the model does.</p>
<p>Of course, reading the code will require at least some programming skills. Check out a number of C tutorials available online for basic understanding about programming and C. The internet is an ever changing place so the best idea is to use your favourite search engine to find some good c tutorials for someone with your level of coding experience. At time of writing the following were potentially helpful sites <a href="http://www.cprogramming.com">www.cprogramming.com</a> and <a href="http://www.tutorialspoint.com/cprogramming">www.tutorialspoint.com/cprogramming</a> and <a href="http://www.imada.sdu.dk/~svalle/courses/dm14-2005/mirror/c/">www.imada.sdu.dk/~svalle/courses/dm14-2005/mirror/c/</a></p>
<p>However, even without a background in c coding, two approaches might be useful in starting to navigate through the code.</p>
<p>The first is the <strong>Search</strong> tool. In Visual Studio (or GCC, or another package) go to Edit-&gt;Find and Replace-&gt;Quick Find, or simply click Ctrl+F. This will bring up the Search menu where you type in the search string, such as the name of parameter.</p>
<p>In the example below, we search for the include_atmosphere parameter, used in the <em>physics.prm</em> file. Make sure you search in the “Entire Solution” if you want to browse through the whole code. You can either scroll through the entire list of search results and click directly on the line of interest (if you have some familiarity with the code) or you can click “Find Next” (if you go this route you may need to click a few times to finally get to one of the routines that uses the parameter). For example for include_atmosphere the relevant routine is called <em>Amonium_ROC()</em>, which is indicated on the top right drop-down menu in Visual Studio, and it is located in the file <strong>atNutrient.c</strong></p>
<p><img src="markdown_output/media/image9.tmp" style="width:6.69306in;height:3.32083in">Here we can see that if include_atmosphere flag is on (set to 1) the code will transfer NH3 values from the atmosphere into the surface cell.</p>
<p>The second approach is to simply browse the libraries, files and routines you might be interested in. Of course, to do this you will need to know the names of routines. This can be done by searching for relevant parameters as shown above or by looking in the drop down list of routine names (most GUIs for interrogating code have such a feature).</p>
<p>As an example we will look at the formulation of the respiration processes in fully age structured groups.</p>
<p>Once you open your solution file in Visual Studio C++, expand the submodel <strong>atecology</strong> (<strong>1.</strong>) and double click the <strong>atvertprocesses.c</strong> file (<strong>2.</strong>). The file will be opened (<strong>3.</strong>) and on the right side you have a drop-down menu to browse through the routines defined in the file (<strong>4.</strong>). By clicking on the <em>Fish_Respiration()</em>routine you will find the description of parameters and equations used to calculate it Note that the first reference to <em>Fish_Respiration()</em> in the drop-down menu has a small arrow next to it. By clicking on this option you will find where in the code the routine is called.</p>
<p>By exploring the code for the routine (<strong>5.</strong>) you can see that the routine needs variables calculated or defined elsewhere in the code, such as <em>guildcase</em>, <em>SN</em>, <em>RN</em>, <em>Dens</em> and <em>*respire</em>, these variables are described within the parentheses <em>( )</em> right after the routine name. The routine also uses six parameters – <em>KA_id</em>, <em>KB_id</em>, <em>KST_id</em>, <em>Ktmp_id, Kthresh2</em> and <em>X_CN</em>. You can find them in the biological parameters file by replacing the ‘<em>id</em>’ with your functional group abbreviation (e.g.&nbsp;<em>KA_FCD)</em>, as defined in the functional group .csv file. Sometimes the <em>_id</em> represents larger categories, such as _fish or _shark, as is the case for <em>KST_fish</em>, but you should be able to work it out by searching for the parameter name.</p>
<p><img src="markdown_output/media/image10.jpeg" style="width:6.95278in;height:5.21458in"></p>
<p>Now you only have to look through the equations and explore different parameter values in R or Excel, or on a piece of paper, and assess the shape of the response curve to different parameter values.</p>
<p>Another useful tool in exploring the code is the <strong>Call Hierarchy</strong> tool. It allows you to see the routines that call or are called from the routine of your interest. In Visual Studio this can be called by clicking the right mouse button while holding the pointer over the routine name and selecting “View Call Hierarchy” option. In the example below, we can see that routine <em>Vertebrate_Activities()</em> is called from <em>Do_Vertebrate_Living()</em> routine, and that many other routines are called from <em>Vertebrate_Activities()</em></p>
<p><img src="markdown_output/media/image11.tmp" style="width:6.69306in;height:3.87917in"></p>
</section>
<section id="how-to-setup-and-start-simulations" class="level2">
<h2 class="anchored" data-anchor-id="how-to-setup-and-start-simulations"><strong>2.5. How to setup and start simulations</strong></h2>
<section id="crash-introduction-into-command-line-interfaces" class="level3">
<h3 class="anchored" data-anchor-id="crash-introduction-into-command-line-interfaces"><em>2.5.1. Crash introduction into Command line interfaces</em></h3>
<p>Atlantis is an “old style” program that runs from a command line. This is done using "<a href="https://confluence.csiro.au/display/Atlantis/Windows+Command+Prompt">Command Prompt</a>" on Windows or "<a href="http://guides.macrumors.com/Terminal">Terminal</a>" on a Mac or Linux. Please familiarize yourself with the command line interface and how to navigate among folders. For most uses you only need to know a few commands and can pick up the required skills in a few minutes. There are plenty of online tutorials to quickly learn about the Command line interface for your operating system (e.g.&nbsp;tutorial.djangogirls.org/en/intro_to_command_line/). Here are a few commands just to get you going (in fact you do not really need much more than that).</p>
<table class="caption-top table">
<colgroup>
<col style="width: 23%">
<col style="width: 76%">
</colgroup>
<thead>
<tr class="header">
<th colspan="2"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><p><strong>What to type:</strong></p>
<p>cd\</p>
<p>cd..</p>
<p>cd C:\AtlantisRuns</p>
<p>cd C:\A + <strong>TAB</strong> key</p>
<p>(type letter A and keep pressing TAB key)</p>
<p>Press arrows up and down</p>
<p>C:</p></td>
<td><p><strong>What this does:</strong></p>
<p>go to the root directory of the drive you are currently in</p>
<p>go one directory level up</p>
<p>go to the specified directory</p>
<p>scroll through all folders within <strong>C:</strong> that start with A (case insensitive). This way you don’t need to type long folder names (and make spelling mistakes)</p>
<p>go to the previously typed commands</p>
<p>Go to drive C: from any other drive you are in</p></td>
</tr>
<tr class="even">
<td><strong>Note:</strong> In Mac and Linux use / instead of \</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="setting-up-the-bat-file" class="level3">
<h3 class="anchored" data-anchor-id="setting-up-the-bat-file"><em>2.5.2. Setting up the BAT file</em></h3>
<p>At this stage you already have 1) compiled the code on your machine, 2) created a run directory for your simulations (C:\AtlantisRuns\MyModel), 3) copied all input and parameter files into the run directory, 4) copied the Atlantis executable (atlantismain.exe) from C:\...\atlantismain\Debug into your run directory.</p>
<p>The next step is to create a BAT file such as <em>run.bat</em> (or any name you want, e.g.&nbsp;<em>NoFishing.bat</em>) and then call it from the command prompt to initialise the simulations. <strong>The BAT file must be in the same directory as the Atlantis executable.</strong></p>
<p>The easiest way to setup and modify the .bat file is in a text editor.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p>The BAT file initialising Atlantis simulation has one long command line calling the Atlantis executable and giving the names of the parameter files required for the run. For example in windows the command would look like:</p>
<p><em><strong>atlantismain.exe -i MySea_initial.nc 0 -o simulationOutput1.nc -r MySea_run.prm -p MySea_physics.prm -f MySea_force.prm -b MySea_biol.prm</strong> -m MySea_migrations.csv <strong>-s Functional_groups.csv</strong> -h MySea_harvest.prm -q MyFisheries.csv –a MySea_assess.prm –e MySea_econ.prm -d OutputFolder1 -t C:\AtlantisRuns\parameters</em></p>
<p>Required parameters and files are shown <strong>in bold</strong>, optional parameters are in regular font.</p>
<p>You will still have to make sure that correct paths of forcing files (for hydrodynamics, salinity, temperature and others) are given in the <em>force.prm</em> file. In the <em>force.prm</em> file the path of the inputs/ folder name often has to given as ../inputs/ which tells the program that the inputs/ folder is in the same higher level directory as the <em>force.prm</em> file.</p>
<p>While the <em>migrations.csv</em> file is marked as optional here, it is usually used as most models include some form of migration into/out of the model.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>If the <em>–t &lt;name of folder containing parameter files&gt;</em> is omitted then Atlantis will expect to find the parameter files in the same directory as the batch file used to run the model. If the Atlantis executable is formally installed after it is compiled (e.g.&nbsp;using sudo make install in <a href="https://confluence.csiro.au/display/Atlantis/Building+under+linux">Linux</a> or adding it to the Windows recognised path names, see the <a href="https://confluence.csiro.au/display/Atlantis/Building+under+windows">wiki</a>) then you do not need to copy the executable file to the run directory and you only need to type the name of your Atlantis executable (e.g.&nbsp;<em>atlantismain</em> in the example) rather than name.exe in the command line.</p>
<p>Names of the parameter files must be given after an appropriate one letter argument which indicates what the following filename represents. For example, <strong><em>-i</em></strong> tells the executable that the following filename is the input file that details initial conditions. The actual names of the parameter files are flexible, as long as the file extensions are correct. For example, the initial conditions file could be named as initial_conditions.nc, start_of_the_run.nc, start_of_the_fun.nc, a.nc or fred.nc. It does not matter so long as it is an NC file. It is <strong>only</strong> the one letter flag command (e.g.&nbsp;-i, -o, -p and others) preceding the filename that tells Atlantis which kind of file to expect.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p>For example, your BAT file could look like the following and it will run perfectly well:</p>
<p><em><strong>BestModellingTool.exe -i BigBang.nc 0 -o TheLastSimulationEver.nc -r HowToRun.prm -p CleverPhysics.prm -f AllThoseInputs.prm -b EatEachOther.prm -s Critters.csv</strong> -m OurMovements.csv -h HowToCatchFish.prm -q Fishers.csv –a TryToAssess.prm –e MBAdomain.prm -d PutEverythingHere -t C:\AtlantisRuns\parameters</em></p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p><strong>Don’t forget to add number 0 after the name of the initial conditions file.</strong> This requirement is a legacy of the model development, and Atlantis will not run without it.</p>
<p>By default all simulation outputs will be placed in the same folder as the BAT file, but you can create an optional output folder (<em>-d OutputFolderName</em>). Specifying output folders is really important when running batch simulations (see below).</p>
</section>
<section id="parameter-files-called-in-the-bat-command-line" class="level3">
<h3 class="anchored" data-anchor-id="parameter-files-called-in-the-bat-command-line"><em>2.5.3. Parameter files called in the BAT command line</em></h3>
<p>This section gives a brief description of the parameter files. They will be described in further details in subsequent chapters:</p>
<table class="caption-top table">
<colgroup>
<col style="width: 2%">
<col style="width: 97%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"><strong><em>-i</em></strong></th>
<th style="text-align: left;">initial conditions of your model run, including physical water column properties, seabed properties and biomasses and size per group per box</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><strong><em>-o</em></strong></td>
<td style="text-align: left;">provides a name for the output files. Please make this an nc file (e.g.&nbsp;<em>simulationOutput1.nc</em>). Atlantis will create a lot of different outputs and will append titles of different output files to your specified name using the nc filename as the core of those names (e.g.&nbsp;<em>simulationOutput1BiomInd.txt</em>, <em>simulationOuput1Catches.txt</em>).</td>
</tr>
<tr class="even">
<td style="text-align: left;"><strong><em>-r</em></strong></td>
<td style="text-align: left;">parameters defining the run setup, such as which modules to turn on (fishery, socio-economics), time step (12 or 24hr), run and stop times, frequency of outputs, flags for debugging and verbosity, details of notes to store in log.txt</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><strong><em>-p</em></strong></td>
<td style="text-align: left;">physics parameters, such as resuspension, point-source scaling, eddy strengths and others. These parameters are not typically changed, although you may want to modify eddy strengths or turn off resuspension.</td>
</tr>
<tr class="even">
<td style="text-align: left;"><strong><em>-f</em></strong></td>
<td style="text-align: left;">forcing file provides pathways to all external (i.e.&nbsp;not simulated within the model) input files, such as hydrodynamics, salinity, temperature, nutrient input sources from river flows or upwelling, climate time series (precipitation, irradiance), historical catch, fuel prices, GDP and complex spatial zoning through time (the list of allowed external forcing files is given the chapter on <em>force.prm</em> below). These files are usually stored in your “inputs” folder.</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><strong><em>-b</em></strong></td>
<td style="text-align: left;">ecological parameters, including feeding preferences, distributions, mortality, reproduction and others</td>
</tr>
<tr class="even">
<td style="text-align: left;"><strong><em>-m</em></strong></td>
<td style="text-align: left;">migration parameters (used to be inside the <em>biology.prm</em> file)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><strong><em>-s</em></strong></td>
<td style="text-align: left;">list of functional biological groups and their key characteristics, such as number of age groups</td>
</tr>
<tr class="even">
<td style="text-align: left;"><strong><em>-h</em></strong></td>
<td style="text-align: left;">parameters on fishing fleets, target species, catch history, discards, changes in selectivity over time, etc. Note that if you use external forcing of catch or fishing mortality, those forcing files must be specified in the <em>force.prm</em> file and not in the <em>harvest.prm</em> file</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><strong><em>-a</em></strong></td>
<td style="text-align: left;">sample design, sampling error structures and basic assessment model parameters</td>
</tr>
<tr class="even">
<td style="text-align: left;"><strong><em>-e</em></strong></td>
<td style="text-align: left;">socio-economics parameters for fleet dynamics, market model, trading model, and black-book based effort allocation model</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><strong><em>-d</em></strong></td>
<td style="text-align: left;">an optional destination folder. If provided, a new folder with this name will be created during the run and all outputs placed in this folder. Using this option is highly recommended</td>
</tr>
<tr class="even">
<td style="text-align: left;"><strong><em>- t</em></strong></td>
<td style="text-align: left;">an optional, but recommended folder, for storing input parameters (see Good Practice Tip 1). If this folder is not provided, all parameters (not inputs) must be in the same directory as the Atlantis bat file, if Atlantis has been formally installed to the command line, or in the same directory as the executable if you are simply copying the exe file to the run directory.</td>
</tr>
</tbody>
</table>
<p>Note, the <strong>forcing files</strong> (such as hydrodynamics, temperature, catch, recruitment and other forcing files) are not included in the <em>BAT</em> command line, because they are all listed in the <em>force.prm</em> file</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>Good practice tip 2</strong></p>
<p>It is a good idea to give informative names to the parameter files. This way you can remember changes made in them. For example, <em>Baltic_biol_CodMort05.prm</em> could denote a biological parameter file with cod mortality set to 0.5. Also make sure you rename the parameter file if you modify it. This will help you keep track of which parameter combination was used for each simulation.</p>
<p>It is <strong>strongly recommended that you keep a good record of all parameter modifications</strong> and outcomes they had on model simulations. We cannot stress this strongly enough! Almost every modeller has experienced difficult moments when trying to trace back what exactly was done at different steps of model development or scenario analyses.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</section>
<section id="setting-up-batch-simulations" class="level3">
<h3 class="anchored" data-anchor-id="setting-up-batch-simulations"><em>2.5.4. Setting up batch simulations</em></h3>
<p>To set up <strong>batch simulations</strong>, simply add more command lines in your BAT file:</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><em>atlantismain.exe -i MySea_initial.nc 0 -<strong>o NoFishing.nc</strong> -r MySea_run.prm -p MySea_physics.prm -f MySea_force.prm -b MySea_biol.prm -s Functional_groups.csv -<strong>h MySea_harvestNoFishing.prm</strong> -q MyFisheries.csv -a MySea_assess.prm -e MySea_econ.prm -<strong>d OutputNoFishing</strong> -t C:\AtlantisRuns\parameters</em></p>
<p><em>atlantismain.exe -i MySea_initial.nc 0 <strong>-o MedFishing.nc</strong> -r MySea_run.prm -p MySea_physics.prm -f MySea_force.prm -b MySea_biol.prm -s Functional_groups.csv -<strong>h MySea_harvestMedFishing.prm</strong> -q MyFisheries.csv –a MySea_assess.prm -e MySea_econ.prm -<strong>d OutputMedFishing</strong> -t C:\AtlantisRuns\parameters</em></p>
<p><em>atlantismain.exe -i MySea_initial.nc 0 -<strong>o HighFishing.nc</strong> -r MySea_run.prm -p MySea_physics.prm -f MySea_force.prm -b MySea_biol.prm -s Functional_groups.csv -<strong>h MySea_harvesHighFishing.prm</strong> -q MyFisheries.csv -a MySea_assess.prm -e MySea_econ.prm -<strong>d OutputHighFishing</strong> -t C:\AtlantisRuns\parameters</em></p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>Batch simulations would normally have some differences between runs (<strong>shown in bold</strong> in the example above). Make sure you place each output in a different folder (-d) and don’t overwrite it! Also make sure you carefully check your command lines to ensure you call the correct parameter files. It is really disappointing to wait for a few days for your batch simulation outputs just to realise that the BAT file had wrong combinations of parameter files in it!</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p>If you run a lot of simulations it is good idea to keep your BAT files handy, so you can easily trace the parameter combinations used. The LOG file produced for each simulation includes the command line of the BAT file. However, some users find it convenient to add new commands to the BAT file and comment (deactivate) the old ones out with # sign.</p>
<p><em><strong>#</strong>atlantismain.exe -i MySea_initial.nc 0 -o NoFishing.nc -r MySea_run.prm -p MySea_physics.prm -f MySea_force.prm -b MySea_biol.prm -s Functional_groups.csv -h MySea_harvestNoFishing.prm -q MyFisheries.csv -a MySea_assess.prm -e MySea_econ.prm -d OutputNoFishing -t C:\AtlantisRuns\parameters</em></p>
<p><em>atlantismain.exe -i MySea_initial.nc 0 <strong>-o MedFishing.nc</strong> -r MySea_run.prm -p MySea_physics.prm -f MySea_force.prm -b MySea_biol.prm -s Functional_groups.csv -<strong>h MySea_harvestMedFishing.prm</strong> -q MyFisheries.csv –a MySea_assess.prm -e MySea_econ.prm -<strong>d OutputMedFishing</strong> -t C:\AtlantisRuns\parameters</em></p>
<p>In this example, Atlantis will skip the first line and will only execute the second one (where the fishing parameters, harvest and output files are changed). This way you can easily keep a record of parameter combinations used in earlier simulations. Of course, if you run dozens of simulations at some point this BAT file will get too long and cluttered and you will need a different bookkeeping system (but make sure you have some system to trace each parameter combination in each simulation!)</p>
<p>Note, that the line here refers to the whole command string, separated by line spaces. One command line is wrapped into four lines in the example above, but it does not contain any line breaks.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</section>
<section id="initialising-simulations" class="level3">
<h3 class="anchored" data-anchor-id="initialising-simulations"><em>2.5.5. Initialising simulations</em></h3>
<p>Once you have all the input and parameter files, EXE and BAT files ready, open the Command Prompt (or Terminal) window, navigate to your model run directory that has the BAT files in it (and the EXE if you have not formally installed Atlantis to the command line paths), type the name of your BAT file and press ENTER.</p>
<p><img src="markdown_output/media/image12.tmp" style="width:6.10377in;height:2.94822in"></p>
<p>Atlantis should start running now! If all is going well, you should be getting short daily messages, looking like this (45.5 days are completed below):</p>
<p><img src="markdown_output/media/image13.tmp" style="width:5.6691in;height:2.77083in"></p>
<p>When the run is complete, Atlantis will finalise writing all the outputs, free arrays and close (the simulation below was run for 20 days):</p>
<p><img src="markdown_output/media/image14.tmp" style="width:5.44in;height:3.48958in"></p>
<p>If, for some reason you need to stop your run before it is finished, go to the output folder and delete the file called “<strong><em>delete_to_halt_run</em></strong>”. This way Atlantis will write all the outputs to date and free memory before stopping the simulation. If you stop Atlantis any other way the files will not be completed properly and will most likely be uselessly incomplete.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p>During the simulations Atlantis runs many outputs into its internal memory. This means that even though all the output files are created at the beginning of the simulation, they may not contain much information until the runs are actually finished. So, do not get disturbed if you want to peek into some output file during the simulation and find it empty, even if the simulation has already been running for a long time.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>Don’t despair if the simulation does not start so smoothly. Error messages are an important part of this adventure! Check out the chapter on troubleshooting and the dedicated Atlantis wiki <a href="https://confluence.csiro.au/display/Atlantis/Common+Windows+Build+Errors">page</a></p>
</section>
</section>
<section id="simulation-outputs" class="level2">
<h2 class="anchored" data-anchor-id="simulation-outputs"><strong>2.6. Simulation outputs</strong></h2>
<p>Once the simulation finishes Atlantis will generate <strong>a lot of</strong> <strong>output files</strong> in TXT and NC formats. The simulation output folder will also contain the <em>log.txt</em> file (with various details or messages generated during the run), as well as <em>exports.ts</em> and <em>inputs.ts</em>, which store information on flows into/out of the model domain, and a number of XML files, which are mostly just automatically generated files that are converted input files restructured to a format that Atlantis can more easily read in. Most users never touch the TS and XML output files. The XML files might be a useful means of storing the input parameters in a database but are otherwise of little ongoing use. The TS files however can be quite informative for showing how much material enters and exits the domain due to advection.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p>The <em>log.txt</em> file does not start with the name given in the <strong><em>–o</em></strong> command. This means that if you accidentally start another simulation with the same output folder name (or without giving the output folder), the <em>log.txt</em> file will be overwritten.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p><img src="markdown_output/media/image15.tmp" style="width:6.08472in;height:7.94028in">The list of output files in your output folder will look like this (it indeed has <strong>a lot of</strong> outputs!):</p>
<p>The table below lists the main output files. Some output files will be produced only if relevant flags are active (e.g.&nbsp;evolution, different management or fisheries options) and are not included in the table.</p>
<p><span id="_Toc445899521" class="anchor"></span></p>
<p>Table 3. Main Atlantis output files. The main output files of interest for beginner users are shown in grey boxes. The table first lists Biology related outputs and then Harvest submodel related outputs. Please note, that the list of output files are continuously updated, so check the <a href="https://confluence.csiro.au/display/Atlantis/Atlantis+Output+Files">wiki</a> for any changes.</p>
<table class="caption-top table">
<colgroup>
<col style="width: 4%">
<col style="width: 63%">
<col style="width: 4%">
<col style="width: 27%">
</colgroup>
<thead>
<tr class="header">
<th><strong>Name</strong></th>
<th><strong>Contents</strong></th>
<th><strong>Units</strong></th>
<th><strong>Comments</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>BiomIndx.txt</td>
<td><strong>Snapshots</strong> of biomasses of all active species (set in the <em>functional_groups.csv</em> file) across the entire model domain, excluding boundary boxes. Usually this is the first file to explore, but don’t forget to look at the box-specific biomasses!</td>
<td>tonnes</td>
<td>Frequency of output controlled with tsumout in <em>run.prm</em></td>
</tr>
<tr class="even">
<td>BoxBiomass.txt</td>
<td><strong>Snapshots</strong> of biomasses of all active species for each of the model domain boxes, excluding boundary boxes</td>
<td>tonnes</td>
<td>Frequency of output controlled with tsumout in <em>run.prm</em></td>
</tr>
<tr class="odd">
<td>AgeBiomass.txt</td>
<td><strong>Snapshots</strong> of biomasses of all active species per age group across the entire model domain</td>
<td>tonnes</td>
<td><p>This output is given if</p>
<p>flag_age_output is set to 1 in <em>run.prm</em></p></td>
</tr>
<tr class="even">
<td>DietCheck.txt</td>
<td>Proportional make up of a diet of each age class of each functional group. This is the cumulative proportion since the last output</td>
<td>propor-tion</td>
<td>The total sum for each age group is 1 for all prey items</td>
</tr>
<tr class="odd">
<td>DetailedDietCheck.txt</td>
<td>If flagdietcheck is set to 1 then this returns the total consumed biomass since the last output given for each cell (box and layer) of each age group of each functional group. If flagdietcheck is set to 2 then it is the proportional makeup of the diet instead.</td>
<td><p>tonnes or</p>
<p>propor-tion</p></td>
<td>Very useful for understanding model dynamics. Check out R codes for querying these outputs (chapter 2.9)</td>
</tr>
<tr class="even">
<td>Mort.txt</td>
<td>Annual predation (-M) and fishing (-F) mortality of each functional group. This file is currently only useful for looking at <strong>relative M vs F values</strong> for a species, as it does not give accurate mortalities</td>
<td>per year</td>
<td>Note, that if a species is set as isImpacted in the <em>functional_groups.csv</em> file, it will have some F value even if it is not explicitly targeted by fishing.</td>
</tr>
<tr class="odd">
<td>SpecificMort.txt</td>
<td><p>Three values of instantaneous mortality for each functional group, age group and stock. M2 is predation mortality, M1 other natural mortality (starvation, linear mortality), F is fishing mortality.</p>
<p>At present the M and F values for this output are written <strong>before</strong> Atlantis checks that consumption and fishing are not trying to take more than is available in each cell. This means that <strong>M or F in the file can sometimes exceed 1</strong>. This will be corrected in a future release of Atlantis as the code is restructured.</p></td>
<td><p>per</p>
<p>year</p></td>
<td>At this stage the file is useful only as an <strong>indication of relative morality</strong> for qualitative mortality assessment (high/low)</td>
</tr>
<tr class="even">
<td>SpecificPredMort.txt</td>
<td><p><strong>Cumulative</strong> predation mortality of each stock and age group of each functional group (in rows) as imposed by each predator (in columns) over the course of a year.</p>
<p>As with the SpecificMort.txt above, <strong>values can exceed 1</strong>. This will be fixed in the future</p></td>
<td>per year</td>
<td>At this stage the file is useful only as an <strong>indication of relative morality</strong> for qualitative mortality assessment (high/low)</td>
</tr>
<tr class="odd">
<td>YOY.txt</td>
<td>Biomass of recruits per year (which is the same as per spawning event, because a group only spawns once per year) summed over the total model domain. <strong>Snapshot</strong> at the time of spawning, regardless of when the output is written out.</td>
<td>tonnes</td>
<td></td>
</tr>
<tr class="even">
<td>Migration.txt</td>
<td>Total and migrating biomass (and proportion migrating) of migrating groups per adults and juveniles.</td>
<td>tonnes</td>
<td>Data is given for groups that are tagged as migratory in <em>biology.prm</em> file</td>
</tr>
<tr class="odd">
<td>VertSize.txt</td>
<td>Numbers, average sizes and biomass of each migrating age structured group in the model domain at the time of the output. The output does not include sizes or numbers of individuals outside the model at the time of the output.</td>
<td></td>
<td>Data is given for groups that are tagged as migratory in <em>biology.prm</em> file</td>
</tr>
<tr class="even">
<td>.nc</td>
<td>3D (time, box, layer) <strong>snapshots</strong> of the tracer values at the time of the output</td>
<td>mg/m<sup>3</sup> or numb</td>
<td>Frequency of output controlled with toutinc in <em>run.prm</em> file</td>
</tr>
<tr class="odd">
<td>TOT.nc</td>
<td>2D (time, box) <strong>snapshots</strong> of tracer values per box (summed over layers)</td>
<td>tonnes/km<sup>2</sup> or numb</td>
<td>Frequency of output controlled with toutinc in <em>run.prm</em> file</td>
</tr>
<tr class="even">
<td>PROD.nc</td>
<td>2D (time, box) <strong>snapshots</strong> of diagnostic tracers, such as production and grazing for biomass groups, growth and consumption for each age group for each fully age structured group, and a number of other diagnostic indices</td>
<td><p>mg/m<sup>3</sup>/day</p>
<p>or</p>
<p>mg/day/individ</p></td>
<td>Frequency of output controlled with toutinc in <em>run.prm</em> file</td>
</tr>
<tr class="odd">
<td>ANNAGEBIO.nc</td>
<td>Numbers in each <strong>annual</strong> age class of a species (<strong>NOT</strong> the model age group) per box. This can be useful when tracking annual changes in species that have multiple years per age group. It also records the average size of individuals of the associated model age group per box (i.e.&nbsp;all age classes in the age group are assumed to be of the same individual size)</td>
<td><p>number/box</p>
<p>and g/individ/box</p></td>
<td>The output is written only when flag_age_output is set to 1 in <em>run.prm</em> file</td>
</tr>
<tr class="even">
<td>BoxLight.txt</td>
<td><p>Proportion of sun hours per time step</p>
<p>(how long during that time step there was light)</p></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Catch.txt</td>
<td><strong>Cumulative</strong> total catch per functional group over all commercial fisheries</td>
<td>tonnes</td>
<td></td>
</tr>
<tr class="even">
<td>CatchPerFishery.txt</td>
<td><strong>Cumulative</strong> total catch per functional group per fishery</td>
<td>tonnes</td>
<td></td>
</tr>
<tr class="odd">
<td>Discard.txt</td>
<td><strong>Cumulative</strong> total discards per functional group over all commercial fisheries</td>
<td>tonnes</td>
<td></td>
</tr>
<tr class="even">
<td>DiscardPerFishery.txt</td>
<td><strong>Cumulative</strong> total discards per functional group per fishery</td>
<td>tonnes</td>
<td></td>
</tr>
<tr class="odd">
<td>Effort.txt</td>
<td><strong>Cumulative</strong> number of days each fishery spends fishing</td>
<td>days</td>
<td>Output given only if dynamic effort is turned on in <em>harvest.prm</em> file</td>
</tr>
<tr class="even">
<td>HarvestIndx.txt</td>
<td>Fisheries and management indices, e.g.&nbsp;average size of catch, total landings, total discards, the number of changes in gear, bycatch of threatened or endangered species, habitat impacts, total effort expended, social acceptance, conflict between fisheries sectors, management access, management costs</td>
<td></td>
<td>Many indices only make sense when appropriate fisheries flags are turned on (see chapter on the Harvest submodel)</td>
</tr>
<tr class="odd">
<td>RecCatch.txt</td>
<td><strong>Cumulative</strong> total catch per functional group over all recreational fisheries</td>
<td>tonnes</td>
<td>The fleet must be marked as recreational (isREC) in <em>fisheries.csv</em> file</td>
</tr>
<tr class="even">
<td>SSB.txt</td>
<td><strong>Snapshot</strong> of mature biomass at the time of spawning</td>
<td>tonnes</td>
<td></td>
</tr>
<tr class="odd">
<td>TAC.txt</td>
<td>Total allowable catch per species per year. It can be fixed or dynamic dependent on the fishing scenarios</td>
<td>tonnes</td>
<td>Only written when fisheries use TAC for management, set in flagManage in <em>harvest.prm</em></td>
</tr>
<tr class="even">
<td>BrokenStick.txt</td>
<td>Indices related to broken stick harvest strategy (current F, target F, current biomass and resulting scalar to apply as calculated by the broken stick).</td>
<td></td>
<td>Only written when broken stick harvesting is turned on in <em>harvest.prm</em></td>
</tr>
<tr class="odd">
<td>CATCH.nc</td>
<td><strong>2D</strong> (time, box) file of <strong>cumulative</strong> values of catch and discard per species per fishery in numbers and tonnes per box: catch per species per age group (numbers), discards per species&nbsp;per age group (numbers), catch per species per fishery (tonnes), discards per species per fishery&nbsp;(tonnes)</td>
<td>tonnes or number</td>
<td>Frequency of output controlled with toutfinc in <em>run.prm</em> file</td>
</tr>
<tr class="even">
<td>CATCHTOT.nc</td>
<td><strong>2D</strong> (time, box) file of <strong>cumulative</strong> values of catch, discards and recreational catch per each box</td>
<td>tonnes</td>
<td>Frequency of output controlled with toutfinc parameter in <em>run.prm</em> file</td>
</tr>
<tr class="odd">
<td>ANNAGECATCH.nc</td>
<td>Numbers in each <strong>annual</strong> age class of a species (<strong>NOT</strong> the model age group) per species in the catch and discards. This can be useful when tracking annual changes in number for species that have multiple years per cohort</td>
<td>numb</td>
<td>The output is written only when flag_age_output is set to 1 in <em>run.prm</em> file</td>
</tr>
<tr class="even">
<td>inputs.ts</td>
<td>amount of tracers coming from all boundary boxes and time series into all dynamic boxes</td>
<td></td>
<td>Frequency of output controlled with inputs_tout in <em>run.prm</em> file</td>
</tr>
<tr class="odd">
<td>exports.ts</td>
<td>amount of tracers coming from all dynamic boxes and time series into all boundary boxes</td>
<td></td>
<td>Frequency of output controlled with inputs_tout in <em>run.prm</em> file</td>
</tr>
</tbody>
</table>
</section>
<section id="introduction-to-netcdf-files" class="level2">
<h2 class="anchored" data-anchor-id="introduction-to-netcdf-files"><strong>2.7. Introduction to NetCDF files</strong></h2>
<p><a href="https://en.wikipedia.org/wiki/NetCDF">NetCDF (Network Common Data Form)</a> is a set of software libraries and data formats that allow for the creation, access, and sharing of array-oriented scientific data irrespective of type of computer hardware. They are typically used by oceanographers and atmospheric scientists and modellers to store spatial time series information. The NC file format is the computer-readable version, while the CDF format is its unpacked human-readable version that can be viewed and edited with a text editor (such as Notepad or Textpad). The key characteristics of these files are that they are platform-independent and “self-describing”. This means that the file has a header which describes the layout of the rest of the file and array dimensions, followed by sections of data and variable attributes. There are many matlab, R and other tools for interacting with NC files.</p>
<p>The <em>initial_conditions.nc</em> file in Atlantis must list all variables to be tracked throughout the simulation, their initial values or concentrations in the spatial domain, and some of their key characteristics. It is important to get them right. The <em>initial_conditions.nc</em> file is huge and it is not advisable to edit it by hand (see below for tools available to set and modify it). However, it is useful to open and explore the file by converting the NC file to a CDF file.</p>
<p>Two small programs, <strong>ncdump.exe</strong> and <strong>ncgen.exe,</strong> can be used to convert NC files into CDF files (to create human readable and editable files) and from CDF to NC respectively (in case you modified the CDF file and need to convert it back into the computer-readable format). The two packages are included in the NetCDF libraries required for compiling Atlantis (see step 4 in the Chapter 2.1). To convert the files, open the Command Prompt, navigate into the folder that includes the NC or CDF files you want to convert and call the appropriate command with its parameters, as shown below. For example, if you want to convert NC into CDF, type: ncdump.exe somedata.nc &gt; otherdata.cdf. Note, the name of the NC file created by the ncgen.exe, will be the same as the name given in the first line of the CDF file (so NOT necessarily the same name as typed in the command prompt). If you install NetCDF and update the paths as described on the <a href="https://confluence.csiro.au/display/Atlantis/NetCDF+4+in+Windows">wiki</a> the Command Prompt will automatically locate ncdump.exe or ncgen.exe files, so you don’t need to have the EXE files in the same folder as the file you want to convert. The names of the NC and CDF files don’t have to be the same.</p>
<p>My_Data.nc</p>
<p>My_Data.cdf</p>
<p><strong>&gt; ncdump.exe My_Data.nc &gt; My_Data.cdf</strong></p>
<p><strong>&gt; ncgen.exe -b My_Data.cdf</strong></p>
<p><img src="markdown_output/media/image16.png" style="width:5.68056in;height:2.68056in"></p>
<p>Once you converted the NC file into CDF, you can open it using a text editor. Here is an example of an output.nc file (comments after ## are added here for explanation):</p>
<table class="caption-top">
<thead>
<tr class="header">
<th><p>netcdf example { <em>## name of the file</em></p>
<p><em>## header section of the file – describes file contents and dimensions</em></p>
<p><strong>dimensions:</strong></p>
<p>t = UNLIMITED ; // (111 currently) <em>## number of time steps</em></p>
<p>b = 29 ; <em>## number of boxes</em></p>
<p>z = 9 ; <em>## number of layers</em></p>
<p>icenz=2; <em>## OPTIONAL: number of ice layers</em></p>
<p><em>## the</em> <a href="https://confluence.csiro.au/display/Atlantis/Changed+to+prm+files"><em>icenz</em></a> <em>attribute <a href="https://confluence.csiro.au/display/Atlantis/Ice+in+Atlantis">activates the ice model</a></em></p>
<p><strong>variables:</strong> <em>## start listing variable descriptions</em></p>
<p>double t(t) ; <em>## double precision 1D variable - time</em></p>
<p>t:units = "seconds since 2005-01-01 00:00:00 +10" ; <em>## start time of the simulation: <strong>time stamp</strong></em></p>
<p>double volume(t, b, z) ; <em>## 3-dimens. variable: volume (t, b, z)</em></p>
<p>volume:bmtype = "phys" ; <em>## description of Physical variables</em></p>
<p>volume:units = "m3" ;</p>
<p>volume:long_name = "Volume" ;</p>
<p>…</p>
<p>double Det_Si(t, b, z) ; <em>## description of Tracer variables</em></p>
<p>Det_Si:bmtype = "tracer" ;</p>
<p>Det_Si:units = "mg Si m-3" ;</p>
<p>Det_Si:long_name = "Detrital Silica" ;</p>
<p>Det_Si:dtype = 0 ;</p>
<p>Det_Si:sumtype = 1 ;</p>
<p>Det_Si:inwc = 1 ;</p>
<p>Det_Si:insed = 1 ;</p>
<p>Det_Si:decay = 0. ;</p>
<p>Det_Si:dissol = 0 ;</p>
<p>Det_Si:partic = 1 ;</p>
<p>Det_Si:passive = 1 ;</p>
<p>Det_Si:svel = -2.894e-005 ;</p>
<p>Det_Si:xvel = 0. ;</p>
<p>Det_Si:psize = 1.e-005 ;</p>
<p>Det_Si:b_dens = 1000000000. ;</p>
<p>Det_Si:i_conc = 200000000. ;</p>
<p>Det_Si:f_conc = 200000000. ;</p>
<p>…</p>
<p>double Filter_Soft_N(t, b) ; <em>## description of Epibenthic variables</em></p>
<p>Filter_Soft_N:bmtype = "epibenthos" ;</p>
<p>Filter_Soft_N:units = "mg N m-2" ;</p>
<p>Filter_Soft_N:long_name = "Soft substrate filter feeders Nitrogen" ;</p>
<p>Filter_Soft_N:dtype = 0 ;</p>
<p>Filter_Soft_N:sumtype = 1 ;</p>
<p>…</p>
<p><strong>// global attributes:</strong> <em>##global attributes or variables</em></p>
<p>:title = "Baltic" ;</p>
<p>:geometry = "Baltic_aea_v2BGM" ;</p>
<p>:parameters = "atlantismain2.exe -i init_Baltic.nc 0 -o BalticClimate1.nc -s BalticClimate1TOT.nc -g BalticClimate1PROD.nc -c BalticClimate1TOTCATCH.nc -d BalticClimate1CATCH.nc -r Baltic_run.prm -f Baltic_force2015.prm -b Baltic_biol_Climate.prm -h -p Baltic_physics.prm"; <em>##details on BAT and other outputs</em></p>
<p>:wcnz = 8 ; <em>## number of water column layers</em></p>
<p>:sednz = 1 ; <em>## number of sediment layers</em></p>
<p>## data section of the file contains a data entry for each variable declared in the header section</p>
<p><strong>data:</strong> <em>## start of the data for each variable</em></p>
<p>t = 0, 7862400, 15724800, 23587200, 31449600, 39312000, 47174400, <em>## time steps of the simulation</em></p>
<p>55036800, 62899200, 70761600, 78624000, 86486400, 94348800, …</p>
<p>volume = <em>## variable values per layer, box, time</em></p>
<p>0, 0, 0, 0, 1762842878.80397, 8664747499.65088, 2166186874.91272,</p>
<p>2166186874.91272, 216618687.491272,</p>
<p>0, 0, 0, 0, 0, 15949302157.943, 27451466709.024, 27451466709.024,</p>
<p>2745146670.9024, …</p>
<p>… <em>## continued until all variables are listed</em></p>
<p>}</p></th>
</tr>
</thead>

</table>
</section>
<section id="introduction-to-atlantis-variables" class="level2">
<h2 class="anchored" data-anchor-id="introduction-to-atlantis-variables"><strong>2.8. Introduction to Atlantis variables</strong></h2>
<p>There are three kinds of obligatory and two kinds of optional variables used in Atlantis. Variable types are defined by the bmtype parameter:</p>
<p>1) “phys” - physical properties of the model</p>
<p>2) “tracer” – three-dimensional dynamic variables (either in the water column or the sediments) of the model domain</p>
<p>3) “epibenthos” – two-dimensional dynamic benthic variables</p>
<p>4) <a href="https://confluence.csiro.au/display/Atlantis/Changed+to+Initial+Conditions+NetCDF+file">“icetracer”</a> – a variable that exists in ice layers (if the model explicitly includes ice)</p>
<p>5) “<a href="https://confluence.csiro.au/display/Atlantis/Land+in+Atlantis">terrestrial</a>” – tracers used to identify land variables (if the model explicitly includes land)</p>
<p>Each variable declaration includes the expected numerical precision. A few variables (e.g.&nbsp;numlayers) are marked as short, which means they are expecting integer input variables, but most variables are doubles, which indicates that they are a double precision floating point number format (a computer way saying this is a real number not simply an integer). The precision identifier is followed by the variable name and the expected dimensions. Variables can be one dimensional (time), two dimensional (time, box) or three dimensional (time, box, layer). It is important to make sure the data entered has this many dimensions (with no omissions) as Atlantis assumes the dimensions are fully filled when it loads the data.</p>
<p>Next, the file lists the characteristics of the variables. All variables have four general parameters: units, long_name, dtype, sumtype. Units and long_name are only needed for the user and interpretation of results; they are entered in inverted commas ” ” as they will be loaded as strings of text by Atlantis.</p>
<p>The parameters dtype and sumtype are flags whose value determines where the output of the variable value is to be written.</p>
<p>The dtype describes the information that will be written to the output files. It can have four values:</p>
<blockquote class="blockquote">
<p>0: Standard snapshots, sent to the base <em>out.nc</em> output files and summarised in 2D in <em>outTOT.nc</em></p>
<p>1: Fisheries total values, written to <em>fisheresTOT.nc</em> files when the fisheries submodel is active</p>
<p>2: Ecological diagnostics, sent to diagnostic <em>PROD.nc</em> files (production, consumption etc)</p>
<p>3: Detailed fisheries information, typically used for tracers in the dynamic fisheries, sent to base <em>fisheries_out.nc</em> file</p>
</blockquote>
<p>The sumtype parameter can have two values:</p>
<blockquote class="blockquote">
<p>0: indicates that data is written in the 3D format (per layer per box per time step) in the main <em>out.nc</em> file and given in detail</p>
<p>1: indicates that, in addition to the 3D output given in the main <em>out.nc</em> file, summary (e.g.&nbsp;total value per box integrated over all layers) is also written to the <em>outTOT.nc</em> file</p>
</blockquote>
<p>The term “<strong>tracer</strong>” will be used often throughout the manual. Most of the variables belong to this category. They include temperature, salinity, oxygen, nutrients (NO3, NH3, organic nitrogen), bacteria, pelagic organisms and description of age structured groups. Most tracers, like bacteria_N, cod1_numbers are dynamic, which means they participate in the processes and change through time. The <em>initial_conditions.nc</em> file includes some important characteristics of these tracers (Table 4). Other tracers are in fact processes, such as Nitrification, Denitrification, Light adaptation, Sediment Porosity and others. They are initiated as tracers in order to create arrays for them, so Atlantis has a dedicated place to write the output for them.</p>
<p><span id="_Toc526762756" class="anchor"></span>Table 4. Parameters used to describe characteristics of Atlantis variables in the <em>initial_conditions.nc</em> file. These parameters are read in by the <em>readBMTracerInfo()</em> in <u>attracerIO.c</u></p>
<table class="caption-top table">
<colgroup>
<col style="width: 3%">
<col style="width: 37%">
<col style="width: 59%">
</colgroup>
<thead>
<tr class="header">
<th><strong>Parameter</strong></th>
<th><strong>What it means</strong></th>
<th><strong>Where is it used</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>inwc</td>
<td>Tracer present in water column<sup>1</sup></td>
<td><p>Used in most routines that deal with processes in the water column. If a tracer is not present in water column in will be skipped in these routines.</p>
<p>This is only important for tracers that are used by diffusion, decay, deposition, settling or injection routines (nutrients, sediments, plankton). They are not important for fully age structured groups</p></td>
</tr>
<tr class="even">
<td>insed</td>
<td><p>Tracer present in sediments<sup>1</sup></p>
<p><sup>1</sup>A tracer can be present both in water column <strong>AND</strong> in sediments</p></td>
<td>Used in most routines that deal with sediment processes. For example, if a tracer is not allowed in the sediments the deposition routine will not deposit it in the sediment, but will leave it in the bottom water layer</td>
</tr>
<tr class="odd">
<td>dissol</td>
<td>Tracer dissolved in water <sup>2</sup></td>
<td>Used in routines that deal with diffusion and advection processes</td>
</tr>
<tr class="even">
<td>partic</td>
<td><p>Tracer is particulate<sup>2</sup></p>
<p><sup>2</sup>A tracer can only be “dissolved” OR “particulate”</p></td>
<td><p>Used in routines that deal with particulate tracers.</p>
<p>It is recommended to put this value to 1 for all tracers that are not dissolved (which naturally includes biological group tracers)</p></td>
</tr>
<tr class="odd">
<td>passive</td>
<td>Tracer can be passively advected</td>
<td><p>Important for particulate tracers. Only passive particulate tracers are advected (moved by water between cells).</p>
<p>Anything that is particulate and passive also require – particle size, sedimentation velocity (svel), and extra settling velocity (xvel)</p></td>
</tr>
<tr class="even">
<td>decay</td>
<td><p>exponent in the burial or removal rate per second</p>
<p>Note, this is not an actual decomposition into other nutrients (like decay of organic nitrogen into NO3), but a complete removal of the variable from the system, such as burial in the sediments</p></td>
<td><p>Used in the Physics submodel to calculate decay rates</p>
<p>This parameter is very important and will determine <strong>the rate of tracer removal</strong> from the system. It is typically set to 0 for most tracers)</p></td>
</tr>
<tr class="odd">
<td>svel</td>
<td>Sedimentation velocity, m/second</td>
<td>Used in the Physics submodel to calculate sedimentation rate of particulate tracers</td>
</tr>
<tr class="even">
<td>xvel</td>
<td>Extra settling velocity - due to migration of nutrient limitation. This parameter was inherited from PPIB and ERSEM models (see below) and is typically set to 0</td>
<td>Used in the Physics submodel to calculate sedimentation rate of particulate tracers</td>
</tr>
<tr class="odd">
<td>psize</td>
<td><p>Particle size, in meters</p>
<p>Particle size will determine deposition of particulate matter in sediments</p></td>
<td>Used in the Physics submodel to calculate sedimentation and resuspension rate</td>
</tr>
<tr class="even">
<td>b_dens</td>
<td>Particle bulk density (mg*m<sup>-3</sup>)</td>
<td>Used in the Physics submodel to calculate deposition and porosity. Used in <em>deposition()</em> and in calculating sediment density. These parameters in the real world dictate particle movements in real water, but in Atlantis default values would do, because sediment properties are not modelled in such detail.</td>
</tr>
<tr class="odd">
<td>i_conc</td>
<td>Particle initial concentration in the deposits</td>
<td>Used in the Physics submodel to calculate deposition and porosity. It is used in <em>deposition()</em> routine only</td>
</tr>
<tr class="even">
<td>f_conc</td>
<td>Particle final concentration in the deposits, after compaction</td>
<td><strong>This was only used in old models. Current code does not read this in.</strong></td>
</tr>
<tr class="odd">
<td>FillValue</td>
<td>Default value to use when automatically filling empty cells in the initial_conditions.nc arrays</td>
<td>This is often used when tracer the initial concentration in most cells is identical</td>
</tr>
</tbody>
</table>
<p><span id="chapter29" class="anchor"></span></p>
</section>
<section id="tools-and-packages-to-view-and-analyse-atlantis-input-and-output-files" class="level2">
<h2 class="anchored" data-anchor-id="tools-and-packages-to-view-and-analyse-atlantis-input-and-output-files"><strong>2.9 Tools and packages to view and analyse Atlantis input and output files</strong></h2>
<p>The Atlantis community has developed a number of different tools to view and analyse these outputs. The main tools used currently include:</p>
<p>1) An R package “atlantistools” contributed by Alexander Keth, available from https://github.com/alketh/atlantistools</p>
<p>2) A number of excel template files, matlab and R scripts developed by different users and available on <a href="https://confluence.csiro.au/display/Atlantis/Atlantis+Helper+Tools">Atlantis wiki</a></p>
<p>3) An R package “rlantis”, available on <a href="https://github.com/mareframe/rlantis" class="uri">https://github.com/mareframe/rlantis</a></p>
<p>4) netcdf viewing packages <a href="https://confluence.csiro.au/display/Atlantis/How+to+view">Olive and Dive</a> useful for spatial plotting of variables in the NC files</p>
<p>5) R based Shiny application “shinyrAtlantis” contributed by Shane Richards and available on <a href="https://github.com/shanearichards/shinyrAtlantis" class="uri">https://github.com/shanearichards/shinyrAtlantis</a>. This package is very useful for visualising the <em>initial_conditions.nc</em> and parameter files in a spatial context.</p>
<p>A number of other packages and tools are in different stages of development. So stay tuned with updates by joining the <a href="https://groups.google.com/forum/#!forum/atlantis-ecosystem-model">Atlantis google group</a> and regularly checking for updates on the Atlantis wiki. If you have a tool you think others would benefit from please feel free to share it via the wiki or the Atlantis google group.</p>
</section>
<section id="troubleshooting-on-installation-and-running" class="level2">
<h2 class="anchored" data-anchor-id="troubleshooting-on-installation-and-running"><strong>2.10. Troubleshooting on installation and running</strong></h2>
<p>Don’t be disappointed if your first run attempt fails. This is very common! Below is a list of some common errors and more are available on the <a href="https://confluence.csiro.au/display/Atlantis/Common+Windows+Build+Errors">Atlantis wiki</a>:</p>
<p><strong>1. Missing DLL files.</strong> This is one of the most common problems for Windows users. Make sure you have installed the right version of the NetCDF library and updated the path as described on the wiki. If issues still persist, you might need to remove and reinstall the NetCDF library.</p>
<p><img src="markdown_output/media/image17.tmp" style="width:4.84221in;height:3.152in"></p>
<p><strong>2. “Can’t open input data file”.</strong> Make sure the paths specified in the <em>force.prm</em> file are correct.</p>
<p>In the case below there was a typo in the path, the ‘inputs’ directory folder was misspelled</p>
<p>hd0.name ../inputs<strong><u>s</u></strong>/forcisets/Baltic_2005_hydro_22May2014.nc</p>
<p>Once this is fixed into</p>
<p>hd0.name ../inputs/forcisets/Baltic_2005_hydro_22May2014.nc</p>
<p>the run went smoothly</p>
<p><img src="markdown_output/media/image18.tmp" style="width:5.09247in;height:3.28in"></p>
<p><strong>3. Failed to find a parameter.</strong> If some of the required parameters in the PRM files are missing Atlantis will quit and give an error message. In the case below Atlantis could not find a parameter FCD_mL. This happens when mistakes were made in formatting parameter files (e.g.&nbsp;if you used excel to create your file check to see if the PRM file contains ” ” around a parameter name, if does remove them and just leave the name) or when new obligatory parameters were added during the code update to account for new functionality (and you have updated the code on your computer)</p>
<p>Keep an eye on the Atlantis wiki <a href="https://confluence.csiro.au/pages/viewrecentblogposts.action?key=Atlantis">news blog</a> for information on modifications to the code and any new required parameters. To find the missing parameters search the Atlantis wiki for the parameter name to see what is expected.</p>
<p><img src="markdown_output/media/image19.tmp" style="width:5.184in;height:3.35045in"></p>
<p><strong>4. Atlantis crashes with a specific error message on physics, biology or other processes.</strong></p>
<p>Atlantis code has inbuilt checks that will cause the program to quit if things go wrong. This can happen if biomasses of some lower trophic groups go to very low levels, or volumes of cells change too much. These messages cannot all be included here, but the best way to understand them is to search for keywords in the message in the code itself. This approach is also useful in understanding where different parameters are used in the code.</p>
<p>To search for a specific string in the entire code, select Quick Find (Ctrl+F) from the Edit menu, and enter the string you want to find. Make sure you search in the “Entire Solution” (selected in the “Look in:” menu) rather than the current document or routine. Also make sure you enter the text accurately.</p>
<p>When you search for an error message only enter general text, e.g.&nbsp;“ERROR: Volume of cell box”, as in the example shown above. Do not enter specific numbers from the error message, as they are not present in the code (they are specific to that one simulation only). For example, the crash message might say</p>
<blockquote class="blockquote">
<p>ERROR: Volume of cell box 5: layer 1 is zero…”</p>
</blockquote>
<p>However the message built in the code actually looks like this:</p>
<blockquote class="blockquote">
<p>ERROR: Volume of cell box %d: layer %d is zero</p>
</blockquote>
<p>where %d values will be replaced by specific numbers.</p>
<p><img src="markdown_output/media/image20.png" style="width:6.79097in;height:4.90347in"></p>
<p>Finding the routine that leads to the error message may not immediately answer your questions, but it might give ideas about where things went wrong and which parameters might have to change (look at the parameters used in this routine). It is not necessarily straightforward and easy, but as you get used to searching through the code and understand it better, things will slowly start making sense.</p>
<p>A document compiling error messages is available on Google Doc file (available on March 20, 2016):</p>
<p><a href="https://docs.google.com/document/d/1B3X6_Y29gZxtKzjZY2oyylAJBxVtYXXu72pVbwv6mWw/edit" class="uri">https://docs.google.com/document/d/1B3X6_Y29gZxtKzjZY2oyylAJBxVtYXXu72pVbwv6mWw/edit</a></p>
<p>Please check this document for further information and to contribute to it by posting error messages you get.</p>
<p>****</p>
</section>
</section>
<section id="geometry-of-the-model-domain" class="level1">
<h1>3. GEOMETRY OF THE MODEL DOMAIN</h1>
<section id="boxes-and-layers-defining-the-geometry-of-the-model" class="level2">
<h2 class="anchored" data-anchor-id="boxes-and-layers-defining-the-geometry-of-the-model"><strong>3.1. Boxes and layers defining the geometry of the model</strong></h2>
<p>Atlantis is a spatially-explicit model, which means that it simulates spatial variation in biogeochemical and socio-economic processes. The spatial domain or geometry of the specific model consists of boxes or polygons subdivided into vertical sediment, water and optional ice depth layers. Land is also an optional inclusion, represented via sediment layers only. The number and shapes of boxes, as well as a number and depth of vertical layers is defined by the user. One layer of one box is called a cell. A cell is the main model spatial unit. Most biogeochemical, biological or fisheries processes are replicated in each water column cell and are considered uniform within a cell. Movement between the water column cells is by passive advective transfer (forced by hydrodynamic forcing files) or directed active movement. Only sediment relevant processes are executed in the sediments, similarly for ice or land. Direct movement between cells of these special types is not possible for ice or sediments, though some directed movement is possible on land. Movement between the special cell types and water column cells may happen due to processes like deposition and melting.</p>
<p>The advantage of using a unique model polygon and depth domain, rather than a standard grid, is that the user can match the model geometry to the geographical and bioregional features of the simulated marine system. Smaller, higher-resolution polygons can be defined in areas of particular interest (Johnson et al.&nbsp;2011) while open water areas can be modelled as one or several large polygons saving a significant amount of computational time. This means that you should think carefully about defining the model geometry (and forcing file conversion, see Chapter 4). Models focused on coastal fisheries often use bathymetry to define polygons, picking boundary isobaths that are important predictors of species distribution or fisheries management zones - such as states, provinces or marine protected areas. Oceanographers may want to isolate areas with persistent currents, eddies, or other oceanographic features. Biogeographic breaks (headlands, capes) may also be important to consider. Finally, the spatial distribution of biological and fisheries data, and the spatial scale of interest to stakeholders and other users of your model also need to be considered.</p>
<p>In principal users can setup very complex shaped polygons, however, the number of polygons and faces between them will determine the speed of calculations. Remember that all biological and fisheries process calculations are repeated in each layer of each polygon!</p>
<p><img src="markdown_output/media/image21.jpeg" style="width:5.80208in;height:3.48619in"></p>
<p><img src="markdown_output/media/image22.png" style="width:6.21661in;height:4.38904in"></p>
<p><span id="_Toc526762784" class="anchor"></span>Figure 4. Top: An example cross section of an Atlantis model, showing water column and sediment layers. Oceanic boxes have an open water boundary at the bottom of the deepest water layer. The depths of vertical layers do not have to be the same in all boxes, although most models keep them identical for simplicity (shown with horizontal dashed line). The depth of the bottom water column layer varies depending on the geography of the marine system. Bottom: An example of the horizontal Atlantis model domain (set in the BGM file)</p>
</section>
<section id="how-to-define-a-geometry-file-for-a-new-model" class="level2">
<h2 class="anchored" data-anchor-id="how-to-define-a-geometry-file-for-a-new-model"><strong>3.2. How to define a geometry file for a new model?</strong></h2>
<p>The first step in developing a new model is to define the two-dimensional domain specified in the BGM file. Atlantis was created prior to the development of shapefiles that have now become a standard in GIS applications. Hence Atlantis uses its own BGM file format (see <a href="https://confluence.csiro.au/display/Atlantis/Atlantis+BGM+Files">wiki</a> BGM file page), which was a common format for many early box models. These days the developers of new Atlantis applications typically define model geometry and shapes of the polygons using GIS tools, but any other alternative approaches that can provide geographic representation can be used. The GIS tools can export shapefiles, which are converted into the Atlantis-specific BGM file using specially written packages. A new R tool (rbgm, see below) is also being developed to enable users to convert files from GIS or other spatial editing applications directly to BGM files, skipping the shapefile stage (see below). The BGM file is also used by some software packages, such as Dive or Olive, for spatial visualisation of Atlantis outputs.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p>Details of the model geometry are defined in the <strong>BGM</strong> and <em>intial_conditions.nc</em> files.</p>
<p>The <strong>BGM</strong> file defines the <strong>two-dimensional</strong> geometry of the model domain. It includes information on boxes, their shapes, midpoints, faces. It also specifies the maximum depth of the modelled ecosystem and maximum depth of each box.</p>
<p>The <strong><em>initial_condition.nc</em></strong> file includes further information on the <strong>three-dimensional geometry</strong>, specifying numbers and depth of the vertical layers in each box. The maximum number of water column and sediment layers is set in the global attribute section in the <em>initial_conditions.nc</em> file (wcnz and sednz parameters). The actual number of water column and the ID of the top active sediment layer (typically 0) in each box is then given in the numlayers and topk parameters given for each box.</p>
<p>Nominal or initial <strong>depths of the water column and sediment layers</strong> are set in the <em>initial_ condition.nc</em> nominal_dz parameter. Typically they are set to be identical among boxes (see Fig. 4 top panel), except for the bottom layer, but this is not strictly required. The actual depths of the water column and sediment layers can change dynamically through the simulation and are stored in the dz variable. The maximum change in water column depth is set in wc_dz_tol , whereas minimum and maximum depth of the sediments layer(s) is set in the minseddz and maxseddz parameters in the <em>physics.prm.</em></p>
<p>The optional ice layers are included as a fourth dimension in the <em>initial_conditions.nc</em> file. Modelling ice in Atlantis is still under development, see chapter 5.5.2</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>There are <strong>two main types of boxes: dynamic and boundary</strong>. The corners of each polygon are defined by <strong>vertices</strong> and the lines connecting these corners are called <strong>faces</strong>.</p>
<p><span id="_Toc446445864" class="anchor"></span>Figure 5. Example of an Atlantis model geometry (part of a model domain). Boxes are numbered in bold, grey boxes indicate boundary or island (land) boxes. Faces are numbered in <em>italic</em> numbers. Vertices are shown as dots; vertices P1 and P2 describe face 5.</p>
<p><strong>Dynamic boxes</strong> make up the model domain; all biological and socio-economic processes are modelled in them. Dynamic boxes must satisfy the (approximate) requirement of mass onservation<a href="#fn4" class="footnote-ref" id="fnref4" role="doc-noteref"><sup>4</sup></a>, although their volume can change slightly due to currents and tides (just as seen in the real world). In dynamic boxes numeric fluxes of a tracer (e.g.&nbsp;NO3, phytoplankton) due to ecological processes (such as growth and mortality) cannot exceed a certain proportion of its standing biomass. This proportion is set by the RelTol parameter in the <em>biology.prm</em> file (but see the note below on relax_tol).</p>
<p><strong>Boundary boxes</strong> are used only as sources and sinks for advective transport. They represent the “outer world” or the system beyond the model domain and don’t have the requirement of mass conservation. The areas of boundary boxes play no role (they can be set as tiny slivers or large boxes) and there are no water or tracer movements between boundary boxes. Islands inside the model domain are modelled as boundary boxes (unless land has been activated in the model by setting flagAllowLand to 1 in the <em>run.prm</em> file). In models where land is not active islands and other land boxes have zero depth (box 8 in the Fig. 5). However, when land is active (only in the most recently development versions of Atlantis) explicit modelling of land is allowed (see chapter 5.5.1), these boxes use a positive value for total box depth (botz) rather than the negative values used for aquatic boxes.</p>
<p>In the vertical dimension the polygons can be either <strong>non-oceanic</strong> or <strong>oceanic</strong>. Sometimes the Atlantis model does not capture the full depth of the marine ecosystem. For example, the actual maximum depth in the modelled ocean area can be 3500m (shown in the maxbotz parameter in the BGM file), but the model domain only includes the top 2000m (box-specific botz parameters in the BGM file). The polygons for which botz is greater or equal to maxbotz are classified as oceanic (e.g.&nbsp;a maxbotz of -2000 versus a botz of -3500). This means that the bottom water layer in these polygons has an open water boundary with deeper (not explicitly modelled) waters rather than with a sediment layer. In oceanic boxes the sediment layer is still included in the setup (for simplicity) but sediment calculations are not performed in these boxes when the model is run and the biomass of organisms in the sediments are not included in the total biomass calculations. Further, the deepest layers of the oceanic boxes receive nutrient input from “outside” of the model domain, designed to simulate nutrient mixing from deeper layers not included in the model domain.</p>
<p>Below is an example of a BGM file with explanations of parameters given after #</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p># box model geometry based on VMPA_setas_20051216.bgm</p>
<p><em># conversion from lat/long space to x-y space of the model domain. Required for mapping and light level</em></p>
<p><em># calculations (when lim_sum_hours is turned on in the biology.prm file). Definition is based on GIS standards</em></p>
<p><em># The parameters should be <strong>space separated</strong></em></p>
<p>projection proj=aea lat_1=-18 lat_2=-36 lat_0=0 lon_0=134 x_0=3000000 y_0=6000000 ellps=GRS80 towgs84=0,0,0,0,0,0,0 units=m no_defs</p>
<p><em># Number of boxes in horizontal plane</em></p>
<p>nbox 11</p>
<p><em># Number of faces in horizontal plane</em></p>
<p>nface 22</p>
<p><em># Maximum bottom depth (m) of the model domain (not the real ecosystem)</em></p>
<p>maxwcbotz -5000</p>
<p><em># vertices of the polygon defining the boundary of the dynamic model space throughout the total geographical</em></p>
<p><em># extent of the model, creating one big polygon. They are ordered <strong>sequentially in either clockwise or</strong></em></p>
<p><em><strong># anticlockwise</strong> direction (it doesn’t matter which so long as it remains consistent)= in (x,y) format, where x,y is</em></p>
<p><em># in projected (linear transformation) of longitude and latitude</em></p>
<p>bnd_vert 4227984.241 1449270.8</p>
<p>bnd_vert 4279136.436 1451037.995</p>
<p>bnd_vert 4684173.263 1362498.346</p>
<p><strong><em>… and so on for all vertices</em></strong></p>
<p><em># Data for box number 0 ## user defined box number identifier</em></p>
<p>box0.label Box0 <em>##users reference box name, e.g.&nbsp;Box0 or Sydney Harbour</em></p>
<p>box0.inside 4043667.571 1150676.493 <em>## midpoint of polygon</em></p>
<p>box0.nconn 5 <em>## number of faces shared with other dynamic boxes</em></p>
<p>box0.iface 1 10 11 13 17 <em>## id numbers of shared faces, each face has a unique id</em></p>
<p>box0.ibox 0 2 2 2 2 <em>## box that each shared face corresponds to</em></p>
<p>box0.botz -370 <em>## maximum depth of the polygon in the real ecosystem</em></p>
<p>box0.area 3261982282 <em>## area of the box, m<sup>2</sup></em></p>
<p>box0.vertmix 0.000001 <em>## vertical mixing scalar for the polygon</em></p>
<p>box0.horizmix 1 <em>## horizontal transport scalar for the polygon</em></p>
<p>box0.relax_tol 0 <em>## whether relaxing flux tolerance for this box</em></p>
<p>box0.vert 4005591.597 1198429.099 <em>## List of vertices (x,y) in clockwise or counter clockwise order</em></p>
<p>box0.vert 3998613.455 1174049.433</p>
<p>box0.vert 4065533.389 1102640.49</p>
<p>box0.vert 4095090.658 1128446.461</p>
<p>box0.vert 4095245.855 1128504.676</p>
<p>box0.vert 4005591.597 1198429.099</p>
<p><strong><em>… and so on for all boxes</em></strong></p>
<p><em># This section describes each dynamic face (faces of all non-boundary boxes)</em></p>
<p><em># Data for face number 0</em></p>
<p>face0.p1 4005591.597 1198429.099 <em>## one endpoint of face (x1,y1)</em></p>
<p>face0.p2 4095245.855 1128504.676 <em>## other endpoint of face (x2,y2)</em></p>
<p>face0.length 113698.3335 <em>## length of face</em></p>
<p>face0.cs -0.788527462 0.614999546 <em>## Cosine and sine of the slope of the face (slope=sine/cosine)</em></p>
<p>face0.lr 2 0 <em>## Number of the box to the left and to the right as you look</em></p>
<p><em>## from point 1 to point 2. <strong>Correct orientation is crucial!</strong> (see</em></p>
<p><em>## below)</em></p>
<p><strong><em>… and so on for all faces</em></strong></p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Relaxing the flux tolerance check for an individual box</strong></p>
<p>There are times (e.g.&nbsp;during an extreme event, which might be implemented for a specific scenario, or if including restocking/replanting of seagrass for example) when the flux of a variable can legitimately be at a level that would breach the flux tolerance checks in place to ensure numerical stability (as conditioned by RelTol set in the <em>biology.prm</em>). To allow for this there is a property per box in the BGM file, box#.relax_tol , which is typically set to 0 unless absolutely required.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p><strong>Below are some strict rules and best practice suggestions on setting up the model geometry.</strong> These rules are provided here to give general insights into how the geometry file is setup. If you are developing a BGM file for a new model, check more detailed instructions on Atlantis wiki (links given below).</p>
<p><strong>Rules:</strong></p>
<p>1) Box 0 <strong>must</strong> be a <strong>boundary</strong> box and Box 1 <strong>must</strong> be a <strong>dynamic</strong> box.</p>
<p>This rule is a developmental legacy and determined by the way the Atlantis code and oceanographic file conversion was setup during the model development. Box 0 is special because it is set up to have water and nutrient fluxes to nearly all other model boxes. This is set to simulate supply of advective tracers to the oceanic boxes and accommodate any missing water fluxes due to conversion of oceanographic files into Atlantis input files (see chapter 4). While the model is running it also acts as a store for information on any migrating groups currently outside the model domain.</p>
<p>2) <strong>Doughnut shaped boxes are not allowed</strong>.</p>
<p>For example, you cannot have one box around an island. This is because such boxes have their midpoint on land, which would affect calculations of a number of processes relying on midpoint values.</p>
<p>3) The description of boxes, faces and vertices in the BGM file (iface, ibox, bnd_vert parameters) <strong>must be done in the same orientation</strong> – clockwise or counter clockwise. It doesn’t matter which you chose, just make sure it is consistent throughout.</p>
<p>The orientation can be checked using the <a href="https://confluence.csiro.au/display/Atlantis/Checkwinding">Checkwinding</a> package developed by Cameron Ainsworth and available from the Atlantis wiki.</p>
<p>4) Islands <strong>must</strong> be represented as boxes with a depth of 0 (or &gt;0 if land is active), not as empty regions. By default the land boxes are either treated as boundary boxes, but can also be modelled as land, if the land option is activated (see chapter 5.5.1).</p>
<p>5) It is of utmost importance to <strong>correctly identify the boxes to the left and to the right of a face</strong> in the face#.lr parameter (where # stands for the box number). The points P1 and P2 in the Figure 3 describe face 5 between boxes 2 and 6. If face5.p1 is given as P1 and face5.p2 is P2, then left and right boxes are designated as you look from P1 to P2. This means the right box is 6 and the left box is 2 and face5.lr must be set as 2 6. Setting the direction incorrectly inverts the direction of the flow in the hydrodynamics forcing file! <a href="https://confluence.csiro.au/display/Atlantis/Checkwinding">Checkwinding</a> package will check the file for correct right/left designation.</p>
<p>6) The <strong>midpoint</strong> of a polygon is defined in the box#.inside parameter. It is used to: i) calculate the temperature in a box when temperature forcing data is not provided and we are in the southern hemisphere; ii) calculate the sunrise and sunset values based on lat/long when lim_sun_hours is set to 1; iii) calculate distance between boxes and invasion speed if the invasion option is active; iv) calculate distance from a box to a port in the dynamic fishing code. <strong>If the midpoint of the box is misplaced and is not within the actual box, Atlantis will quit when trying to read time series that deposit something into that box or when running dynamic fishing routines.</strong></p>
<p><strong>Suggestions:</strong></p>
<p>7) Keep the number of boxes as small as possible (fewer than 90 is good), remember the fewer the boxes the faster the model will run (e.g.&nbsp;models with 20-30 boxes will run three to four times faster than one with 90 boxes!)</p>
<p>8) Keep the number of dynamic faces (those corresponding to non-boundary boxes) to less than 400, but 100-200 would be even better! It is not recommended to have boxes with very complex shapes (i.e.&nbsp;do not slavishly follow complex isobaths in the GIS when drawing up the box shapes). This is because complex box shapes are 1) is prone to error and 2) may give a false impression of reality. Simple shaped boxes serve as a good reminder that the spatial results presented are only model simulations and should be treated as such</p>
<p>9) When setting up box shapes in GIS, make sure the topology is perfect, which means no gaps or overlaps between polygons. If the shapefile has overlaps or gaps it won’t convert to the BGM file</p>
<p>10) In some cases conversion of water fluxes from oceanographic files to Atlantis hydrodynamic forcing files leaves small coastal boxes isolated, i.e.&nbsp;no water fluxes enter or leave them. This will be covered in more detail in Chapter 5.4. The BGM file has two box-specific water mixing parameters that are used in some physics routines, including those used to correct for the lack of mixing. These parameters allow box-specific mixing scalars to be setup and help to tune vertical and horizontal water flows in the model domain:</p>
<p>box#.vertmix - vertical mixing scalar used only when vert_diffusion or vert_mix are turned on in the <em>physics.prm</em>.</p>
<p>box#.horizmix – horizontal mixing scalar that can be used to correct for hyperdifusion (if insufficient correction was done during the file conversion stage, see Chapter 5.4). It is also used as a box-specific scalar if fill_zero_exchange or horiz_diffusion flags are turned on (set to 1) in the <em>physics.prm</em> file (see Chapter 5 for further details).</p>
</section>
<section id="practical-steps-for-building-the-bgm-files-from-gis-shapefiles" class="level2">
<h2 class="anchored" data-anchor-id="practical-steps-for-building-the-bgm-files-from-gis-shapefiles"><strong>3.3. Practical steps for building the BGM files from GIS shapefiles</strong></h2>
<p><strong>1.</strong> Define polygons using QGIS or similar GIS package. This will produce a shapefile of the model domain. The shapefile is the master of the model geometry, any changes should be made to the shapefile and the BGM should be recreated from that master. Hand editing of the BGM should be avoided<a href="#fn5" class="footnote-ref" id="fnref5" role="doc-noteref"><sup>5</sup></a>, other than to manually set the box-specific mixing values, or to correct any issues resulting from BGM generation (like misplacement of the inside point). See the <a href="https://confluence.csiro.au/display/Atlantis/Atlantis+BGM+Files">Atlantis wiki</a> BGM file page for more information.</p>
<p><strong>2.</strong> Run the BGMeriser package, to convert shapefiles into BGM files. Details on access and installation are given on the wiki (<a href="https://confluence.csiro.au/display/Atlantis/Technical+Information">here)</a> as are details on how to run it (<a href="https://confluence.csiro.au/display/Atlantis/BGMEditor">here</a>).</p>
<p><strong>3.</strong> If the shapefiles do not pass through the BGMeriser, they probably have issues with gaps or overlapping polygons and should be cleaned. Check the <a href="https://confluence.csiro.au/display/Atlantis/Using+QGIS+to+clean+Atlantis+BGM+Shapefiles">instructions</a> on the wiki for tips on cleaning up the shapefiles using QGIS.</p>
<p><strong>4.</strong> Use <a href="https://confluence.csiro.au/display/Atlantis/Checkwinding">Checkwinding</a> to check the orientation of boxes and faces in the BGM file</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p>Please keep in mind that <strong>shapefiles do not preserve all information about the faces</strong> among the boxes (see chapter 3.5). Shapefiles only enumerate faces, but they do not give information about their orientation. They also do not preserve topological information among boxes, i.e.&nbsp;their relationship to one another. Correctly created shapefiles include user defined extra attributes to contain the additional information required to generate a BGM file. However, when shapefiles are edited in a GIS, all face information may change, especially if some boxes are added or deleted. This is why BGMeriser must be run after each shapefile amendment. This is an important disadvantage of using shapefiles for Atlantis model geometry and a <strong>new R package is being developed</strong> to allow users create BGM files directly from GIS or other applications, without having to use shapefiles (see chapter 3.5).</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</section>
<section id="how-to-view-your-model-geometry" class="level2">
<h2 class="anchored" data-anchor-id="how-to-view-your-model-geometry"><strong>3.4. How to view your model geometry?</strong></h2>
<p>The two-dimensional model geometry is now defined in your shapefiles and in BGM files. <strong>The shapefiles can be viewed using QGIS or a similar GIS package.</strong> If shapefiles are not available, BGM files can be converted into shapefiles using <a href="https://confluence.csiro.au/display/Atlantis/BGM2SHP">BGM2SHP</a> package.</p>
<p>The BGM files are also read by various tools used to visualise Atlantis outputs in a spatial context, listed in chapter 2.9</p>
</section>
<section id="using-r-for-bgm-geometry" class="level2">
<h2 class="anchored" data-anchor-id="using-r-for-bgm-geometry"><strong>3.5 Using R for BGM geometry</strong></h2>
<p>There is an R package <a href="https://github.com/mdsumner/rbgm"><strong><em>rbgm</em></strong></a> for working with BGM files contributed by Michael Sumner. The aim of the package is to provide a common platform for reading and writing BGM files, which then also can be used by other Atlantis R applications that require information on the model geometry. Generally, shapefiles do not store all information required to completely re-create a BGM file. One needs at least a line shapefile and a polygon shapefile to store what is in the BGM and also some notes to go with it, which is why a shapefile is only a starting point for a BGM file. Note that BGMeriser <strong>will</strong> create a valid BGM file from a shapefile, but it has to be configured <a href="https://confluence.csiro.au/display/Atlantis/BGMEditor">in the right way</a>.</p>
<p>Using <strong><em>rgbm</em></strong> you can</p>
<ul>
<li><p>read the BGM format, maintaining all topology and attributes</p></li>
<li><p>convert from BGM to GIS-like objects using the Spatial tools in R (<strong>sp</strong> package)</p></li>
</ul>
<p>The “GIS-like” object is an R equivalent of a “shapefile”, which consists of “layers” of geometric shapes (polygons, lines, or points) that are linked to a table of attributes. The BGM geometry is really a combination of information on lines (faces) and polygons (boxes), where each face has information about its orientation and its neighbouring faces and boxes. R uses Spatial layers for separate description of a SpatialPolygonsDataFrame (a polygon shapefile), SpatialLinesDataFrame (a line shapefile) and SpatialPointsDataFrame (a point shapefile).</p>
<p>Once the BGM is loaded in R, it can be used to visualise faces and boxes as lines and polygons, and use these objects to extract data from other sources, like doing spatial queries with other map data, extracting gridded data summaries, and using these for visualization or further analysis (see rbgm description for further information). A goal for <strong><em>rbgm</em></strong> is to drive coupling of Atlantis with physical data and models via R. <strong><em>rbgm</em></strong> is able to write from Spatial forms to shapefiles and other formats, but since we can read directly from BGM or from a GIS or geodatabase, this also means that we don’t necessarily need to create shapefiles at all (soon <strong>rbgm</strong> will be able to create BGM files after reading the input from QGIS or Manifold GIS projects).</p>
<p>A collection of <a href="https://github.com/AustralianAntarcticDivision/box-geometry-models">publicly available BGM files from existing Atlantis models</a> is also available. This collection can be viewed using an R package <a href="https://github.com/AustralianAntarcticDivision/bgmfiles"><strong><em>bgmfiles</em></strong></a>.</p>
</section>
</section>
<section id="hydrodynamic-salinity-and-temperature-forcing-data" class="level1">
<h1>4. HYDRODYNAMIC, SALINITY AND TEMPERATURE FORCING DATA</h1>
<section id="general-introduction-to-forcing-of-atlantis-models" class="level2">
<h2 class="anchored" data-anchor-id="general-introduction-to-forcing-of-atlantis-models"><strong>4.1. General introduction to forcing of Atlantis models</strong></h2>
<p>Like most other ecosystem models, Atlantis does not calculate water fluxes (current, circulation) itself, but uses outputs from specialised oceanographic models such as <a href="https://www.myroms.org/wiki/Documentation_Portal">ROMS</a> (Regional Ocean Modeling System) (Shchepetkin and McWilliams 2005; Haidvogel et al.&nbsp;2008), <a href="http://wp.csiro.au/bluelink/global/">BLUELINK</a> (Brassington et al.&nbsp;2007) or <a href="https://hycom.org/">HYCOM</a> (Chassignet et al.&nbsp;2006). In other words, Atlantis models are forced with hydrodynamic data from oceanographic models. At the minimum level Atlantis requires only a hydrodynamic forcing input file defining water fluxes across boxes and layers. The model will then use simple routines to calculate temperature and salinity conditions in each cell at each simulation time step. However, most Atlantis models, and other ecosystem models (Fiechter et al.&nbsp;2014; Rose et al.&nbsp;2015), also use oceanographic model output to force temperature and salinity conditions (and in some cases also oxygen, pH or other tracers). This is because oceanographic models are specialised to model temperature and salinity conditions, and their calculation of these parameters are deemed to be more accurate than those used in Atlantis. Moreover, taking the physical properties from the oceanographic model means they are consistent with the exchanges (currents) being read in from the oceanographic model. In rare cases, if oceanographic models are not available, water flux information can be pieced together from observations or literature to generate hydrodynamic forcing files manually. Table 5 lists some of the spatial domains and forcing characteristics of example model applications.</p>
<p>Oceanographic models simulate processes on their native grid, which is typically of much higher spatial resolution than Atlantis. The temporal scale in oceanographic models is also often in the range of seconds rather than the 6h, 12h or 24h used in Atlantis. Further, most oceanographic models use the same number of water column layers regardless of total depth (called sigma layers) (Fig. 6). This means that the depth of a layer in oceanographic models is very different in shallow and deep waters, while in Atlantis, water layer depths remain relatively stable (with small variations due to tides). This means that water fluxes and values of all forced parameters must be interpolated onto Atlantis model polygons and cells.</p>
<table class="caption-top table">
<caption><span id="_Toc526762757" class="anchor"></span>Table 5. Hydrodynamic structure of some example Atlantis models. Advect = advection, temp = temperature, salt = salinity.</caption>
<colgroup>
<col style="width: 13%">
<col style="width: 18%">
<col style="width: 9%">
<col style="width: 10%">
<col style="width: 8%">
<col style="width: 10%">
<col style="width: 9%">
<col style="width: 17%">
</colgroup>
<thead>
<tr class="header">
<th></th>
<th style="text-align: center;">SE Australia</th>
<th style="text-align: center;">Eastern Tasmania</th>
<th style="text-align: center;">Port Phillip Bay</th>
<th style="text-align: center;">Western-port</th>
<th style="text-align: center;">NEUS</th>
<th style="text-align: center;">California current</th>
<th style="text-align: center;">Gulf of California</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Key oceanographic effects</td>
<td style="text-align: center;">Boundary currents</td>
<td style="text-align: center;">Boundary current, upwelling</td>
<td style="text-align: center;">River input and slow flushing</td>
<td style="text-align: center;">River input and flushing</td>
<td style="text-align: center;"><p>Gulf Stream,</p>
<p>Boundary current, Georges Bank</p></td>
<td style="text-align: center;"><p>Boundary current, upwelling,</p>
<p>El Nino</p></td>
<td style="text-align: center;">Colorado River input, tidal mixing / stratification</td>
</tr>
<tr class="even">
<td>Number of boxes</td>
<td style="text-align: center;">71</td>
<td style="text-align: center;">11</td>
<td style="text-align: center;">60</td>
<td style="text-align: center;">27</td>
<td style="text-align: center;">30</td>
<td style="text-align: center;">62</td>
<td style="text-align: center;">66</td>
</tr>
<tr class="odd">
<td>Max number of water column layers</td>
<td style="text-align: center;">5</td>
<td style="text-align: center;">6</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">4</td>
<td style="text-align: center;">7</td>
<td style="text-align: center;">6</td>
</tr>
<tr class="even">
<td>Ocean boundary depth</td>
<td style="text-align: center;">2400m</td>
<td style="text-align: center;">2400m</td>
<td style="text-align: center;">45m</td>
<td style="text-align: center;">30m</td>
<td style="text-align: center;">200m</td>
<td style="text-align: center;">1200m</td>
<td style="text-align: center;">2025m</td>
</tr>
<tr class="odd">
<td>Hydrodynamic model used to supply flows</td>
<td style="text-align: center;">OFAM-Bluelink<sup>1</sup></td>
<td style="text-align: center;">OFAM-Bluelink</td>
<td style="text-align: center;">3D hyrdro model<sup>2</sup></td>
<td style="text-align: center;">Empirically defined</td>
<td style="text-align: center;">HYCOM<sup>3</sup></td>
<td style="text-align: center;">ROMS<sup>4</sup></td>
<td style="text-align: center;">ROMS<sup>5</sup></td>
</tr>
<tr class="even">
<td>Hydrodynamic time series length (years)</td>
<td style="text-align: center;">10</td>
<td style="text-align: center;">10</td>
<td style="text-align: center;">5</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">7</td>
<td style="text-align: center;">23</td>
</tr>
<tr class="odd">
<td>Transport model variables</td>
<td style="text-align: center;">Advect, temp, salt</td>
<td style="text-align: center;">Advect, temp, salt</td>
<td style="text-align: center;">Advect</td>
<td style="text-align: center;">Advect</td>
<td style="text-align: center;">Advect, temp, salt</td>
<td style="text-align: center;">Advect, temp, salt</td>
<td style="text-align: center;">Advect, temp, salt</td>
</tr>
<tr class="even">
<td>Environmental drivers</td>
<td style="text-align: center;">Inshore sediment change, temp, pH, nutrients, river flow</td>
<td style="text-align: center;"></td>
<td style="text-align: center;">Nutrients</td>
<td style="text-align: center;">Nutrients</td>
<td style="text-align: center;">Nutrients</td>
<td style="text-align: center;">Temp, nutrients</td>
<td style="text-align: center;">Temp, nutrients, river flow</td>
</tr>
<tr class="odd">
<td>Climate change</td>
<td style="text-align: center;">Yes</td>
<td style="text-align: center;">No</td>
<td style="text-align: center;">No</td>
<td style="text-align: center;">No</td>
<td style="text-align: center;">No</td>
<td style="text-align: center;">Yes (impl.) <sup>6</sup></td>
<td style="text-align: center;">No</td>
</tr>
<tr class="even">
<td>Explicit larval advection</td>
<td style="text-align: center;">No</td>
<td style="text-align: center;">No</td>
<td style="text-align: center;">No</td>
<td style="text-align: center;">No</td>
<td style="text-align: center;">No</td>
<td style="text-align: center;">No</td>
<td style="text-align: center;">Yes</td>
</tr>
</tbody>
</table>
<p>1. Oke et al 2005,</p>
<p>2. Walker 1997a, 1997b,</p>
<p>3. hycom.rsmas.miami.edu/las/,</p>
<p>4. ROMS Hermann – NOAA PMEL/JISAO, Seattle,</p>
<p>5. ROMS Pares-Sierra - Centro de Investigación Científica y de Educación Superior de Ensenada.</p>
<p>6 Implicit handling of climate change means that extra mortality terms (related to acidification) or forced changes (e.g.&nbsp;for temperature) are used rather than flows from down scaled global circulation models.</p>
<p><span id="_Toc526762786" class="anchor"></span>Figure 6. An example of oceanographic sigma depth layers (grey lines) with an overlay of Atlantis spatial structure. Figure by Javier Porobic Garate</p>
</section>
<section id="transforming-oceanographic-model-output-into-atlantis-forcing-files" class="level2">
<h2 class="anchored" data-anchor-id="transforming-oceanographic-model-output-into-atlantis-forcing-files"><strong>4.2. Transforming oceanographic model output into Atlantis forcing files</strong></h2>
<p>The interpolation of the oceanographic output into Atlantis forcing files requires transformation across different shaped grids and depth layers. Water fluxes must be transformed from the oceanographic grid to Atlantis faces, whereas temperature and salinity values must be averaged onto Atlantis cells and time steps. The transformation must also account for hyperdiffusion, which is caused by the different spatial scales of oceanographic and Atlantis models. Because concentration of tracers within a layer of a box is assumed to be uniform, a tracer that enters a box is instantaneously assumed to be equally distributed throughout the entire box. This means that flow of tracers in large boxes would be overstated by orders of magnitude if this hyperdifussion phenomenon is not taken into account. The (crude) correction simply divides the east-west water flows by the width of the box (in metres) in the east-west orientation and north-south flows by the width of the box in the north-south orientation. This is a simple and a standard approach to correct for hyperdifusion and can be further refined using the horiz_mix scalar in the BGM file.</p>
<p>Fortunately, the transformation of oceanographic files is increasingly streamlined. Until recently most model developers used two codes contributed by CSIRO. We will describe them in a more detail below. However, new codes are currently being developed by the Atlantis community, i.e.&nbsp;an <a href="https://confluence.csiro.au/display/Atlantis/R+code+to+process+ROMS+into+Hydro">R script by Cecilia Hansen</a> to construct hydrodynamic files from ROMS files. Please check the Atlantis wiki <a href="https://confluence.csiro.au/display/Atlantis/Forcing+Files">forcing files page</a> and Atlantis google group for any new developments in this area.</p>
<p>The interpolation of oceanographic data onto the Atlantis model geometry requires three main conversions (Fig. 7):</p>
<p>1. Calculate <strong>horizontal</strong> water fluxes across Atlantis polygon faces, i.e.&nbsp;from one polygon to another</p>
<p>2. Calculate <strong>vertical</strong> water fluxes within Atlantis polygons.</p>
<p>3. Calculate <strong>state variables</strong> used for optional forcing, such as salinity and temperature for each cell and time step. For this step temperature and salinity values from high resolution oceanographic models are mapped onto Atlantis cells and a weighted average for each cell is calculated</p>
<p>These three steps constitute the main conversion and are conducted using the <a href="https://confluence.csiro.au/display/Atlantis/Matlab+-+Hydro"><em>Hydro</em></a> Matlab script (follow the link for further instructions on installation and use of the <em>Hydro</em> script and also check <a href="https://confluence.csiro.au/display/Atlantis/Hydro+FAQ"><em>Hydro</em> FAQ</a> for more information). <em>Hydro</em> produces two NC files which are then used by <a href="https://confluence.csiro.au/display/Atlantis/HydroConstruct"><em>Hydroconstruct</em></a> to produce the final Atlantis forcing files. <em>Hydroconstruct</em> makes sure the order of water columns is in the right direction, transforms the water flux data into per layer per box fluxes and also corrects for hyperdifussion (using either area based scaling or area and box shape based scaling). The <em>Hydroconstruct</em> <a href="https://confluence.csiro.au/display/Atlantis/Running+Hydroconstruct">running</a> page and <a href="https://confluence.csiro.au/display/Atlantis/Running+Hydroconstruct?preview=/43155480/43156182/params_17July2012.prm">parameter</a> file provides further information on the script and how to run it.</p>
<p><img src="markdown_output/media/image27.jpeg" style="width:6.88958in;height:2.22639in"></p>
<p><span id="_Toc526762787" class="anchor"></span>Figure 7. Main steps in converting oceanographic files into Atlantis input files, as done using two CSIRO contributed scripts <em>Hydro</em> and <em>Hydroconstruct</em>. Figure by Javier Porobic Garate</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>Good practice tip 3</strong></p>
<p>Interpolation and interpretation of the oceanographic data is one of the main challenges for Atlantis models developers and users, because usually they are biologists and not oceanographers. It is therefore highly recommended to include an oceanographer in the Atlantis team, at least during the model development stage, but also if and when long term forcing of environmental trends is needed.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</section>
<section id="things-to-keep-in-mind-about-hydrodynamic-forcing-files" class="level2">
<h2 class="anchored" data-anchor-id="things-to-keep-in-mind-about-hydrodynamic-forcing-files"><strong>4.3. Things to keep in mind about hydrodynamic forcing files</strong></h2>
<p>Oceanographic modelling has made rapid progress in recent years and availability of high resolution data is increasing, but there are still a <strong>few issued that should be kept in mind:</strong></p>
<p>1. <strong>Absence of long term oceanographic data to study long term trends.</strong> Oceanographers rarely conduct 50+ years of high resolution simulations, yet such time frames are often needed for Atlantis simulations. The length of oceanographic data time series usually more on the order of 1 to 25 years (see Table 5 for examples) and a common practice among Atlantis users is to ‘loop’ over the available oceanographic input years. One should be aware that if long term trends are present in the oceanographic data, they will be seen as jumps in average variable values every time the file is recycled. Also, such recycling alone is not suitable for studying long term environmental trends. Ideally long term forcing files should be obtained directly from long term oceanographic simulations.</p>
<p>As a crude approximation, <strong>long term trends can be forced</strong> onto hydrodynamic files using an <a href="https://confluence.csiro.au/display/Atlantis/R+code%2C+add+climate+trends+to+Hydro.NC">R script</a> contributed by Hem Nalini Morzaria Luna and used for the Gulf of California model. Alternatively, trends in temperature, salinity and pH can be added in the biology.prm file using several Tchange, Schange and PHchange parameters (see below)</p>
<p>2. <strong>Insufficient resolution in coastal areas</strong>. Most oceanographic global and publicly available models do not include coastal areas at sufficiently high resolution (in many old models they are not represented at all). Yet this is where a lot of important processes of nutrient input, recycling, spawning and larval dispersal take place. It is therefore very important to assess the water fluxes in small or shallow coastal areas and make sure that they are sufficiently accurate (see below).</p>
<p>3. <strong>Loss of resolution to represent eddies and upwelling.</strong> Translating high resolution oceanographic models to the Atlantis polygons means that transformed forcing files may not capture the true magnitude of eddy strength or local upwelling, both of which are important for driving local productivity. To compensate for this loss of resolution it is recommended that local forcing of productivity be included using eddy parameters (see below) and/or forcing additional nutrient loading into coastal waters (to simulate upwelling; Brand et al.&nbsp;2007).</p>
<p>4. <strong>Oceanographic models can include ‘drift’</strong>, temporal artefact trends that bias long-term oceanographic outputs. If ‘drift’ is seen to be a problem, modellers can choose a limited set of years of oceanographic output and loop these years.</p>
<p>5. <strong>Artificially inflated vertical fluxes.</strong> Sometimes oceanographic models can inflate the vertical fluxes (they are used for tuning the horizontal fluxes), which can lead to too much vertical mixing. This is becoming less of a problem as the skill of oceanographic models improves with time, but it is still recommended to check the intensity of vertical fluxes. If vertical fluxes appear to be too strong they can be downscaled using vertical diffusion and vertical mixing routines (see below) or by using an R script contributed by Raphael Girardin (details and a link will be added when available, but also check the wiki). Similat ‘jets’ can appear in horizontal fluxes (particularly in older models and most often close to the boundary of the oceanographic model domain).</p>
<p>6. <strong>Underestimated diffusion of tracers between boxes</strong>. In real world there are lots of bi-directional flows between spatial domains that are modelled as distinct Atlantis boxes. However, when these flows are converted into hydrodynamic forcing files, all flows are averaged for each pair of cells, so that only one-directional flow occurs between two cells at each time step. This might under-represent diffusion of tracers between boxes, especially if boxes are large and if averages all go in the same direction in all layers of the two boxes.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p>The <strong>hydrodynamic forcing files</strong> <strong>are automatically</strong> recycled or <strong>looped</strong> for the entire model simulation run. This is <strong>not the case</strong> with other forcing files, such as temperature, salinity or other optional forcing files. To rewind the temperature and salinity forcing files set the flags temp_rewind and salt_rewind to 1 in the force.prm file. If the flags are set to 0 then the simulations will use the last temperature and salinity value once the forcing file time series ends. To rewind any other optional forced tracers make sure you set TracerName_rewind (e.g.&nbsp;pH_rewind) parameter to 1.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p>The hydrodynamic forcing files <strong>are not necessarily the final water transport</strong> in your model. They provide the horizontal and vertical water fluxes, which are then used in the transport model described in the <em>(transportBM)</em> routine in the attransport.c file.</p>
<p>The transport model uses the water exchanges from the hydrodynamic forcing files, but can also:</p>
<p>1) apply water flow corrections to isolated boxes according to specified parameters</p>
<p>2) add local water inputs or sinks, such as river flows or outflows</p>
<p>3) apply water exchanges through the surface layer due to precipitation and evaporation</p>
<p>The final flows of water are given in the output.nc file listed in hdsource, hdsink, eflux and vflux (see description of the ouput file parameters in chapter 5.6). If some of the optional physics routines are used, then these final exhcanges may be different from the hydrodynamic forcing inputs.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p><img src="markdown_output/media/image29.png" style="width:1.18889in;height:4.32292in"></p>
</section>
<section id="numbering-of-vertical-layers-in-the-input-and-output-nc-files" class="level2">
<h2 class="anchored" data-anchor-id="numbering-of-vertical-layers-in-the-input-and-output-nc-files"><strong>4.4. Numbering of vertical layers in the input and output NC files</strong></h2>
<p>Atlantis allows users to define several water column and sediment layers, although most applications have only one sediment layer.</p>
<p>Layer counting starts from the sediment-water interface.</p>
<p>The water layer 0 is always closest to the sediment, the next layer up is layer 1 and so on. Sediment layer 0 is closest to the water, the next layer down is 1 and so on.</p>
<p>In the.nc files water column layers are listed first, followed by the sediment layers.</p>
<p><em>layer 0, layer 1, layer 2, layer 3, layer 4, layer 5, sediment 0</em></p>
<p>In the model setup shown here, the layer corresponding to the water column’s surface layer can be anything from layer 0 to layer 5, depending on the total depth of a box. In deep boxes the surface layer will be layer 5, while for the shallowest boxes the surface will have layer 0 as their surface layer.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p>NOTE!</p>
<p>The ordering of layers in output files is opposite to that in the input file. This is a development legacy aimed to improve the viewing of output files with NC file viewers. This also means that the position of missing water column layers in arrays describing boxes with fewer than maximum layers is reversed between the input and output files.</p>
<p>In the input.nc file a shallow water box with three water layers will look like:</p>
<p><em>layer 0, layer 1, layer 2, 0, 0, 0, sediment 0</em></p>
<p>In the output.nc file the same box will look like:</p>
<p><em>0, 0, 0, layer 0, layer 1, layer 2, sediment 0</em></p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>A good way to understand this is to look at temperature variables. Note that warmer water is on the surface.</p>
<table class="caption-top">
<colgroup>
<col style="width: 50%">
<col style="width: 49%">
</colgroup>
<thead>
<tr class="header">
<th><p>Input.nc file:</p>
<p>Temp =<br>
15.75, 18.54, 18.55, 0, 0, 0, 11.85,<br>
16.38, 17.64, 17.49, 0, 0, 0, 11.77,<br>
15.66, 17.34, 17.31, 0, 0, 0, 11.57,<br>
16.05, 16.92, 16.87, 0, 0, 0, 11.29,<br>
5.15, 11.85, 13.92, 17.02, 19.01, 19.83, 0.32,<br>
5.29, 11.91, 13.85, 16.39, 18.50, 19.48, 0.93,</p>
<p>…</p></th>
<th><p>Output.nc file:</p>
<p>Temp =</p>
<p>0, 0, 0, 15.75, 18.54, 18.55, 11.85,</p>
<p>0, 0, 0, 16.38, 17.64, 17.49, 11.77,</p>
<p>0, 0, 0, 15.66, 17.34, 17.31, 11.57,</p>
<p>0, 0, 0, 16.05, 16.92, 16.87, 11.29,</p>
<p>5.15, 11.85, 13.92, 17.02, 19.01, 19.83, 0.32,</p>
<p>5.29, 11.91, 13.85, 16.39, 18.50, 19.48, 0.93,</p>
<p>…</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</section>
<section id="viewing-and-checking-forcing-input-files-with-r" class="level2">
<h2 class="anchored" data-anchor-id="viewing-and-checking-forcing-input-files-with-r"><strong>4.5. Viewing and checking forcing input files with R</strong></h2>
<p>An R-package <strong><em>shinyrAtlantis</em></strong> is available on <a href="https://github.com/shanearichards/shinyrAtlantis" class="uri">https://github.com/shanearichards/shinyrAtlantis</a> and provides a number of shiny applications to help with viewing Atlantis forcing and parameter files.</p>
<p>For exploring hydrodynamic forcing files the package provides a function ‘make.exchange.object' that reads in the BGM file containing box geometries and the NC file containing the exchanges. The function ’make.exchange.object’ also needs to be provided with the cumulative layer depths. It then produces a data object that can be fed into the shiny application ‘sh.exchange’. The ‘sh.exchange’ provides a visual description of the exchange data. It visualises connections and exchanges among boxes and layers and helps to identify if any boxes are isolated. A time-series of exchanges for each box can also be generated, as well as a two-dimensional visualisation of the horizontal flow throughout the domain. Development is under way to add visualisation of the salinity and temperature forcing files.</p>
</section>
<section id="light-and-noise-pollution" class="level2">
<h2 class="anchored" data-anchor-id="light-and-noise-pollution"><strong>4.6. Light and noise pollution</strong></h2>
<p>A simple representation of noise and light pollution from human sources can be represented as a forcing file. More advanced code linking this pollution to levels of human activity is present in the development branch but not the trunk as yet. In the main trunk code, for now the levels of noise and light pollution must be entered in a forcing NC file in the same way as salinity or temperature.</p>
<p>To activate and load this pollution you need to</p>
<ul>
<li><p><em>run.prm file</em>: set flag_pollutant_impacts to 1</p></li>
<li><p><em>forcing.prm file</em>: set use_pollutantfiles to 1 and then</p></li>
</ul>
<p>nnoise_pollutionfiles X (where X is the number of noise forcing files)</p>
<p>Noise_Pollution0.name path/noise_filename.nc</p>
<p>noise_pollution_rewind 1 (set to 0 if you do not want the file(s) to be rewound and reused)</p>
<p>nlight_pollutionfiles X (where X is the number of light pollution forcing files)</p>
<p>Light_Pollution0.name path/light_pollution_filename.nc</p>
<p>light_pollution_rewind 1 (set to 0 if you do not want the file(s) to be rewound and reused)</p>
<ul>
<li><em>biology.prm</em> <em>file</em>: Add the parameters noise_coefft_XX and light_coefft_XX for each biological group</li>
</ul>
<!-- -->
<ul>
<li><em>groups.csv:</em> Add the columns isLightEffected and isNoiseEffected and set the value to 1 for these if you want noise/light pollution to affect that species group</li>
</ul>
<p>For any groups that are affected then during movement an additional scalar (<span class="math inline">\(\delta_{\mathbb{N}}\)</span>) is applied to each box (making a site less attractive to the species), with:</p>
<p><span class="math inline">\(\delta_{\mathbb{N}} = \ \frac{\kappa_{\eta,s}}{\mathbb{N}_{i,j}}\)</span></p>
<p>Where <span class="math inline">\(\mathbb{N}_{i,j}\)</span> represents the level of the pollutant (noise or light) in layer <em>j</em> of polygon (box) <em>i</em>; and κ<sub>η,s</sub> is the coefficient for that pollutant type.</p>
<p>In addition, metabolic parameters – consumption (C) and growth rates (mum) – are scaled in the same way; while starvation and background (mL, mQ) mortality rates are scaled by the inverse.</p>
</section>
</section>
<section id="the-physics-submodel" class="level1">
<h1>5. THE PHYSICS SUBMODEL</h1>
<p>The Physics submodel deals with most physical and some biogeochemical processes in Atlantis, such as water and gas exchanges and mixing, nutrient inputs, deposition and resuspension of particulate matter, as well as some biological processes in the sediments, such as bioturbation and bioirrigation.</p>
<p>The central part of the Physics submodel is the <strong>transport model</strong> that reads in water exchanges from the hydrodynamic forcing files, applies a number of optional routines to correct for lack of horizontal and vertical mixing (see below), distributes water flows through the model domain, and calculates new concentrations of dissolved or passively advected tracers, such as salinity, temperature, oxygen, pH and pelagic planktonic organisms. These calculations simultaneously account for diffusion, advection and dispersion processes. The transport model is preceded by a number of optional routines executing sediment mixing, resuspension, gas exchange with the atmosphere, bioirrigation and others (see below).</p>
<p>Most of the details in the Atlantis Physics submodel are based on the Port Phillip Bay Integrated Model (PPBIM) described in Murray and Parslow (1997) and Walker (1997) and on the <a href="http://www.meece.eu/library/ersem.html">ERSEM</a> model described in Ebenhöh et al.&nbsp;(1995) (see important references in Chapter 1.5). Atlantis users interested in understanding the physics processes deeper can learn more by consulting these references.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>The parameters and flags used in the Physics submodel are described in</strong>:</p>
<p>1) <em>initial_conditions.nc</em></p>
<p>2) <em>physics.prm, run.prm</em> and <em>force.prm</em> files</p>
<p>3) <em>hydrodynamics.nc</em> forcing input file</p>
<p>4) optional forcing NC and TS input files, such as temperature, salinity, pH, oxygen, contaminants, point source inputs, solar radiation, evaporation, precipitation and others.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>In this chapter we will:</p>
<p>1) introduce physics and biogeochemical tracers and their properties,</p>
<p>2) describe the routines executed in the main <em>physics()</em> routine and parameters in the <em>physics.prm</em> file</p>
<p>3) explain why in some cases the user might need to correct for the lack of the vertical and horizontal mixing in hydrodynamic forcing files</p>
<p>4) give a brief introduction to the latest Atlantis developments allowing for the modelling of ice, land and contaminants</p>
<p>5) provide a small example demonstrating the importance of the optional physics routines</p>
<section id="physical-and-biogeochemical-variables-and-their-characteristics" class="level2">
<h2 class="anchored" data-anchor-id="physical-and-biogeochemical-variables-and-their-characteristics"><strong>5.1. Physical and biogeochemical variables and their characteristics</strong></h2>
<p>The <em>initial_conditions.nc</em> file lists important characteristics of physical and biogeochemical tracers, used for various calculations in the <em>physics()</em> routine. Here is an example of a tracer description:</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p>…</p>
<p>double Det_Si(t, b, z) ;</p>
<p>Det_Si:bmtype = "tracer" ;</p>
<p>Det_Si:units = "mg Si m-3" ;</p>
<p>Det_Si:long_name = "Detrital Silica" ;</p>
<p>Det_Si:dtype = 0 ;</p>
<p>Det_Si:sumtype = 1 ;</p>
<p>Det_Si:inwc = 1 ;</p>
<p>Det_Si:insed = 1 ;</p>
<p>Det_Si:decay = 0. ;</p>
<p>Det_Si:dissol = 0 ;</p>
<p>Det_Si:passive = 1 ;</p>
<p>Det_Si:partic = 1 ;</p>
<p>Det_Si:svel = -2.894e-005 ;</p>
<p>Det_Si:xvel = 0. ;</p>
<p>Det_Si:psize = 1.e-005 ;</p>
<p>Det_Si:b_dens = 1000000000. ;</p>
<p>Det_Si:i_conc = 200000000. ;</p>
<p>Det_Si:f_conc = 200000000. ;</p>
<p>…</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>From this we can see that Det_Si is a three-dimentional variable (t,b,z), found in the water column (inwc=1) and the sediments (insed=1). It is not removed from the system through burial (decay=0), it is not dissolved in the water (dissol=0), but is present in a particulate form (partic=1). The sedimentation velocity of particles is svel=-2.894e-005, and its negative value shows that the particles sink (are not buoyant). The particle size is psize = 1.e-005 m</p>
<p><span id="_Toc526762758" class="anchor"></span>Table 6. Examples of currently used obligatory and optional physics and biogeochemical tracers and variables and their characteristics as defined in the <em>initial_conditions.nc</em> file (collected from different Atlantis applications and used for illustration purposes only). Further description of tracers and physics variables is given in the following chapters on sediment and biological processes.</p>
<table class="caption-top table">
<colgroup>
<col style="width: 10%">
<col style="width: 10%">
<col style="width: 10%">
<col style="width: 10%">
<col style="width: 10%">
<col style="width: 10%">
<col style="width: 10%">
<col style="width: 10%">
<col style="width: 10%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: center;"><strong>Tracer</strong></th>
<th style="text-align: center;"><strong>Name</strong></th>
<th style="text-align: center;"><strong>In water</strong></th>
<th style="text-align: center;"><strong>In sed</strong></th>
<th style="text-align: center;"><strong>Dissol-ved</strong></th>
<th style="text-align: center;"><strong>Parti-culate</strong></th>
<th style="text-align: center;"><strong>Decay</strong></th>
<th style="text-align: center;"><strong>Svel</strong></th>
<th style="text-align: center;"><strong>Psize</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td colspan="9" style="text-align: center;"><em><strong>Obligatory tracers for dynamic properties</strong>. These tracers participate in ecological processes and change as a result. It is very important to get the characteristics of these tracers correct in the initial_conditions.nc file as Atlantis will use them for dynamic calculations.</em></td>
</tr>
<tr class="even">
<td style="text-align: center;">NH3</td>
<td style="text-align: center;">Ammonia</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;"></td>
<td style="text-align: center;">1.e-007</td>
<td style="text-align: center;">NA</td>
<td style="text-align: center;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: center;">NO3</td>
<td style="text-align: center;">Nitrate</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;"></td>
<td style="text-align: center;">1.e-009</td>
<td style="text-align: center;">NA</td>
<td style="text-align: center;">NA</td>
</tr>
<tr class="even">
<td style="text-align: center;">DON</td>
<td style="text-align: center;">Dissolved organic nitrogen</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;"></td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">NA</td>
<td style="text-align: center;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: center;">MicroNut</td>
<td style="text-align: center;">Micronutrients</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;"></td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">NA</td>
<td style="text-align: center;">NA</td>
</tr>
<tr class="even">
<td style="text-align: center;">Oxygen</td>
<td style="text-align: center;">Oxygen</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;"></td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">NA</td>
<td style="text-align: center;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: center;">Si</td>
<td style="text-align: center;">Dissolved silica</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;"></td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">NA</td>
<td style="text-align: center;">NA</td>
</tr>
<tr class="even">
<td style="text-align: center;">Det_Si</td>
<td style="text-align: center;">Detrital silica</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;"></td>
<td style="text-align: center;">x</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">-2.9e-005</td>
<td style="text-align: center;">1.0e-005</td>
</tr>
<tr class="odd">
<td colspan="9" style="text-align: center;"><p><em><strong>Optional tracers for dynamic properties.</strong> Required only if</em></p>
<p><em><sup>1</sup> <strong>-</strong> trackAtomicRatio flag is set to 1 or <sup>2</sup> -</em> <em>flagIsEstuary flag is set to 1 in the run.prm file</em></p></td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong>Tracer</strong></td>
<td style="text-align: center;"><strong>Name</strong></td>
<td style="text-align: center;"><strong>In water</strong></td>
<td style="text-align: center;"><strong>In sed</strong></td>
<td style="text-align: center;"><strong>Dissol-ved</strong></td>
<td style="text-align: center;"><strong>Parti-culate</strong></td>
<td style="text-align: center;"><strong>Decay</strong></td>
<td style="text-align: center;"><strong>Svel</strong></td>
<td style="text-align: center;"><strong>Psize</strong></td>
</tr>
<tr class="odd">
<td style="text-align: center;">P<sup>1</sup></td>
<td style="text-align: center;">Phosphorus</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: center;">TOP<sup>1</sup></td>
<td style="text-align: center;">Total organic phosphorus</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong>Tracer</strong></td>
<td style="text-align: center;"><strong>Name</strong></td>
<td style="text-align: center;"><strong>In water</strong></td>
<td style="text-align: center;"><strong>In sed</strong></td>
<td style="text-align: center;"><strong>Dissol-ved</strong></td>
<td style="text-align: center;"><strong>Parti-culate</strong></td>
<td style="text-align: center;"><strong>Decay</strong></td>
<td style="text-align: center;"><strong>Svel</strong></td>
<td style="text-align: center;"><strong>Psize</strong></td>
</tr>
<tr class="even">
<td style="text-align: center;">Carbon<sup>1</sup></td>
<td style="text-align: center;">Carbon</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: center;">CO2<sup>1</sup></td>
<td style="text-align: center;">Carbon dioxide</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: center;">SED<sup>2</sup></td>
<td style="text-align: center;">Sediments</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;"></td>
<td style="text-align: center;">x</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">-2.0e-005</td>
<td style="text-align: center;">5.0e-006</td>
</tr>
<tr class="odd">
<td colspan="9" style="text-align: center;"><em><strong>Obligatory diagnostic tracers for physical properties and reporting of biogeochemical processes.</strong> These tracers either describe pure physical properties (temperature, light, stress) or outcomes of other processes, e.g.&nbsp;chl_a concentration is a result of primary production and consumption. The properties of these tracers are hardwired in the Atlantis code and the values in the initial_condition.nc file are overwritten during simulations.</em></td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong>Tracer</strong></td>
<td style="text-align: center;"><strong>Name</strong></td>
<td style="text-align: center;"><strong>In water</strong></td>
<td style="text-align: center;"><strong>In sed</strong></td>
<td style="text-align: center;"><strong>Dissol-ved</strong></td>
<td style="text-align: center;"><strong>Parti-culate</strong></td>
<td style="text-align: center;"><strong>Decay</strong></td>
<td style="text-align: center;"><strong>Svel</strong></td>
<td style="text-align: center;"><strong>Psize</strong></td>
</tr>
<tr class="odd">
<td style="text-align: center;">Salt</td>
<td style="text-align: center;">Salinity</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;">x</td>
<td style="text-align: center;"></td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">NA</td>
<td style="text-align: center;">NA</td>
</tr>
<tr class="even">
<td style="text-align: center;">Temp</td>
<td style="text-align: center;">Temperature</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;">x</td>
<td style="text-align: center;"></td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">NA</td>
<td style="text-align: center;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: center;">Light</td>
<td style="text-align: center;">Light intensity on the surface of a cell</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;">x</td>
<td style="text-align: center;"></td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">NA</td>
<td style="text-align: center;">NA</td>
</tr>
<tr class="even">
<td style="text-align: center;">Stress</td>
<td style="text-align: center;">Surface Stress</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;">x</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: center;">Denitrification</td>
<td style="text-align: center;">Denitrification</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;">x</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: center;">Porosity</td>
<td style="text-align: center;">Sediment porosity</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: center;">Nitrification</td>
<td style="text-align: center;">Nitrification</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;">x</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: center;">Chl_a</td>
<td style="text-align: center;">Chlorophyll</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;">x</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td colspan="9" style="text-align: center;"><em><strong>Diagnostic tracers.</strong> They are required in the initial_conditions.nc file and are used during debugging for reporting fine scale information about the group that is tagged by which_check parameter in the run.prm file</em></td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong>Tracer</strong></td>
<td style="text-align: center;"><strong>Name</strong></td>
<td style="text-align: center;"><strong>In water</strong></td>
<td style="text-align: center;"><strong>In sed</strong></td>
<td style="text-align: center;"><strong>Dissol-ved</strong></td>
<td style="text-align: center;"><strong>Parti-culate</strong></td>
<td style="text-align: center;"><strong>Decay</strong></td>
<td style="text-align: center;"><strong>Svel</strong></td>
<td style="text-align: center;"><strong>Psize</strong></td>
</tr>
<tr class="odd">
<td style="text-align: center;">DiagNGain</td>
<td style="text-align: center;">Gain in checked group</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: center;">DiagNLoss</td>
<td style="text-align: center;">Loss in checked group</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: center;">DiagNFlux</td>
<td style="text-align: center;">Flux in checked group</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
</tbody>
</table>
</section>
<section id="the-main-physics-routine" class="level2">
<h2 class="anchored" data-anchor-id="the-main-physics-routine"><strong>5.2. The main <em>physics()</em> routine</strong></h2>
<p>The routine <em>physics()</em> defined in <strong>atphysics.c</strong> is the main physics routine, run at each timestep of the simulation. It is described below in the order of execution. The first column lists the routines, the flags to activate them, and shows which flags are active in the example SETas model. The second column describes routines and parameters. Remember that (as elsewhere in this manual) routines are listed in <em>italics() and</em> files containing these routines are bold and <strong><u>underlined</u></strong> and libraries are <strong>bold</strong>. Parameter names from <em>physics.prm</em> are given <strong>in orange</strong> and those from <em>force.prm</em> file are also marked with an asterisk *.</p>
<p><strong>I. Execute explicit optional physics processes.</strong></p>
<table class="caption-top table">
<thead>
<tr class="header">
<th><p>Add sources and sinks</p>
<p><em>sourceSink()</em> in <strong><u>atsourcesink.c</u></strong></p>
<p>injection</p>
<p>1</p></th>
<th><p>Applies forced inputs and outflows of water and nutrients from point sources, including evaporation and precipitation</p>
<p>The number of point sources and sinks is given in npointss<strong>*</strong>. The inputs are read in as time series TS or NC files and they give details on inflow or outflow of water (m<sup>3</sup>, temp, salinity) and nutrients (NH3, NO3, Labile Detritus, Si). They are also used to force nutrient inputs from river inflows, upwelling, waste disposal and other sources. Theoretically sinks of nutrients and water flows can also be included (negative values in the forcing files), but this has not typically been applied in practice.</p>
<p>The <em>sourceSink()</em> routine also deals with water exchanges due to evaporation (Evaporation<strong>*</strong>) and precipitation (Precipitation<strong>*</strong>).</p>
<p>If the injection flag is set to 0 then evaporation and precipitation will not be done at all. Note that evaporation and precipitation forcing files are provided as two-dimensional NC files, with the values given as millimetres of water input or lost on a xy coordinate grid. These inputs are automatically converted to the Atlantis box geometry by the model.</p></th>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><p>Add solar radiation</p>
<p><em>do_light()</em> in</p>
<p><strong><u>light.c</u></strong></p>
<p>solar_radiation</p>
<p>1</p></td>
<td><p>Add solar radiation energy to the surface of the model domain (Solar_radiation<strong>*</strong>). If forcing TS files of solar radiation are provided solar radiation intensity at each time step will be identical across the entire surface area of the model domain.</p>
<p>This can be changed by setting lim_sun_hours to 1 in <em>biology.prm</em>, which will apply a set of NASA routines to calculate sun hours for each box given its longitude (using lat/long projections from the BGM file)</p></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td><p>Exchange of gasses between the surface water layer and atmosphere</p>
<p><em>gasExchange()</em> in <strong><u>atgas.c</u></strong></p>
<p>atmospherics</p>
<p>1</p></td>
<td><p>Executes <strong>temperature independent</strong> gas exchange with the atmosphere across the water-air interface at the top of the surface layer. This calculates new concentrations of gases in the surface layer and it assumes an infinite source of gases in the atmosphere.</p>
<p>Gas exchange with the surface water layer is governed by the diffusion coefficient, saturation value and stagnant layer thickness of the gas. These coefficients and gases to be used in the routine are hardwired in Atlantis code and defined in the <strong>gaslist[]</strong> data structure in the <strong><u>atgas.c</u></strong> library. Currently only O2 and CO2 are included.</p>
<p>Note that <em>gasExchange()</em> is not the only routine dealing with input of gases from the atmosphere. There is now an option to include explicit deposition of <a href="https://confluence.csiro.au/display/Atlantis/2014/07/10/Atmosphere-sea+fluxes+of+nutrients+and+gases">nutrients and gases from the atmosphere</a>, including O2 and CO2 (see description of include_atmosphere flag below). The include_atmosphere option deals with deposition of nutrients and gases trapped in particulate matter. It is handled in the specific routines defined in <strong><u>atNutrient.c</u></strong> library <strong>atecology</strong> - e.g.&nbsp;O2 is handed by <em>Oxygen_ROC()</em> routine. The deposition of O2 in <em>Oxygen_ROC()</em> is <strong>temperature-specific</strong>.</p>
<p><strong>If both include_atmosphere and atmospherics flags are turned off there will be no O2 exchange with the atmosphere!</strong></p></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td><p>Decay or removal of tracers from the water column and / or sediments.</p>
<p><em>decayBM()</em> in <strong><u>atdecay.c</u></strong></p>
<p>decay_wc</p>
<p>decay_sed</p>
<p>0 0</p></td>
<td><p>Removes tracers from the system, simulating burial in the sediments. The exponent of the removal rate is given in the <em>initial_conditions.nc</em> file :decay parameter, so that removal rate is calculated as [tracer_concentration*exp(:decay*time)]. The removal rate in the sediments can be scaled to that in the water column with decay_sed_scale parameter to set as a multiplier on the standard :decay rate. If the :decay parameter of a tracer is set to 0 in the <em>initial_conditions.nc</em>, no removal of the tracer will occur even if the routine is activated.</p>
<p>Note, the <em>decayBM()</em> routine implements a complete removal of the tracer from the system, not the actual decomposition decay. Even if the decay is deactivated a tracer can still break down through nitrification and denitrification processes. In this case however its nitrogen content may not be entirely removed from the system, but can be transformed into other tracers as specified by the nutrient cycling steps (e.g.&nbsp;organic N is transformed into NO3).</p></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td><p>Resuspension of <strong>particulate</strong> tracers from the sediments into water column</p>
<p><em>resuspendBM()</em> in <strong><u>atsuspension.c</u></strong></p>
<p>resuspension</p>
<p>0</p></td>
<td><p>Calculates erosion of sediments due to shear (bottom) stress as a function of sediment type. Then calculates new concentrations of particulate tracers in the bottom water column layer after the resuspension.</p>
<p>The rate of resuspension depends on the bottom stress, porosity of the sediment and background erosion rate. These parameters are given in the <em>initial_conditions.nc</em> file. The bottom stress can also be supplied as an external BottomStress* forcing file.</p>
<p>To enable the routine, the user <strong>must</strong> provide non-zero stress values. This can be done either by giving non-zero values for the “Stress” tracer in the <em>initial_conditions.nc</em>; or, if NC Stress values are zero, Atlantis will look for an external forcing file. If no forcing file can be found, Atlantis will quit.</p>
<p>The maximum rate for a 1cm sediment depth is set in the max_erosion (m in a single time step). The routine is particularly important when simulating wetlands (shallow ecosystems with many sediment layers).</p></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td><p>Settling of <strong>particulate</strong> tracers from the water column</p>
<p><em>settleBMwc()</em> in <strong><u>atsettle.c</u></strong></p>
<p>settling</p>
<p>1</p></td>
<td><p>Calculates settling (<strong>downward or upward!</strong>) of <strong>particulate</strong> tracers from the water column depending on their settling velocity :svel parameter in initial_conditions.nc. <strong>Only tracers with non-zero :svel and :psize will be acted upon</strong>. This typically includes particulate biogeochemical tracers (Detrital Si) and some phyto- and zooplankton.</p>
<p>If settling velocity is positive, particles are buoyant and rise up through the water column (done in the <em>advect_up()</em> routine). If settling velocity is negative, particles sink towards the bottom (done in the <em>advect_down()</em> routine).</p>
<p>The flux of tracers from the lowest water column layer is added to the uppermost sediment layer if the tracer is allowed to exist in the sediments (:insed parameter in <em>initial_condition.nc</em>). The deposition to sediments is done in the <em>deposit()</em> routine in <strong><u>atdeposition.c</u></strong>.</p>
<p>The deposition depends on the sediment’s and tracer’s porosity. The porosity of the tracers is calculated in the <em>calc_por()</em> routine in <strong>atsedprops.c</strong> using the :psize (particle size) and :b_dens (bulk density) parameters from the <em>initial_condition.nc</em> file. Sediment porosity is set as a “phys” variable in the <em>initial_conditions.nc</em> file. However, if the settling routine is turned on sediment porosity is calculated dynamically through the simulation in the <em>deposit()</em> routine. The sediment porosity is calculated as a ratio of the new water volume in the sediment, after deposition, and the total sediment volume, after deposition.</p>
<p>If a particulate tracer is not allowed to exist in the sediment (:insed is 0 in <em>initial_conditions.nc</em> file), it is retained in the lowest water column layer.</p>
<p>The <em>deposit()</em> routine uses the maxseddz and minseddz parameters to define the maximum and minimum depth of one sediment layer.</p>
<p>If the maxseddz is exceeded everything extra will stay in the bottom water layer and no more deposition will occur.</p></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td><p><strong>II. Execute optional biological sediment processes – bioirrigation and bioturbation:</strong></p>
<p>The bioirrigation and bioturbation routines simulate exchange of dissolved and particulate tracers respectively between sediment layers and between the bottom water column layer and the sediment. The baseline bioirrigation and bioturbation routines have been adopted from the PPBIM. However, Atlantis now also applies a multiplier to the baseline rate, which depends on the biomass of the living organisms in the sediment layers. The multiplier is calculated in the Biology submodel and will be described in the relevant chapter.</p>
<p><strong>The bioturbation routine is only performed when more than one sediment layer is present! Because most models have only one sediment layer, the only way particulate tracers can be returned to the water column is through resuspension.</strong></p>
<p>The full bioirrigation routine also requires more than one sediment layer, but it models simple exchange of dissolved particles even in models with only one sediment layer.</p>
<p>The processes used in Atlantis are similar to those in the PPBIM model, but are dynamically modified by the sediment fauna via an “enhancement” term (similar to that of <a href="http://www.meece.eu/library/ersem.html">ERSEM</a>, Ebenhöh et al.&nbsp;1995).</p>
<p><strong>The “background density” of sediment organisms is set in the <em>initial_conditions.nc</em> file sedbiodens tracer</strong> and it can vary among boxes. Background density represents basic aerobic and anaerobic diversity. This background amount is “enhanced” depending on the abundance of the dynamically modelled infauna and epifauna groups (identified with _INF and _EP names in the GroupType parameter of the <em>functional_groups.csv</em> file). The abundance of dynamically modelled groups changes through time but the background density stays stable. The background biological activity remains present in sediments even if all dynamically modelled groups are wiped out (e.g.&nbsp;in anoxic sediments). Hence bioturbation, bioirrigation and other sediment processes can still occur (if activated).</p>
<p>Bioturbation and bioirrigation are simulated as exchange in thin water and sediment layers, as shown below. The routines take a small sliver of the water layer and sediment layer, swaps them and calculates new concentrations of dissolved and particulate tracers respectively. The thickness of the sliver to be exchanged reflects the intensity of the bioirrigation and bioturbation processes and depends on the amount of the biological activity in the sediment.</p>
<p><img src="markdown_output/media/image30.jpeg" style="width:4.82in;height:1.59333in"></p></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td><p>Bioirrigation: Inflow of water and <strong>dissolved</strong> tracers from the bottom water layer into sediments due to biological activity</p>
<p><em>bioirrig()</em> in <strong><u>atbioirrig.c</u></strong></p>
<p>bioirrigation</p>
<p>1</p></td>
<td><p><em>bioirrig()</em> includes three subroutines (only the last two are done in models with one sediment layer):</p>
<p><em>dissol_diff()</em> calculates diffusion of dissolved tracers across the sediment layers, <strong>if more than one sediment layer is present</strong>. This is calculated as a product of:</p>
<p><strong>-</strong> bi_dissol_kz diffusion constant at zero depth (m<sup>2</sup>s<sup>-1</sup> per animal per m<sup>2</sup>), where the amount of animals per m<sup>2</sup> is set in the <strong>sedbiodens</strong> tracer in the <strong><em>initial_conditions.nc</em></strong> file.</p>
<p>- density of biological activity of _INF and _EP groups</p>
<p>- scalar of biological activity as a function of depth (which depends on the profile described in the biosedprofile)</p>
<p>- scalar of irrigation enhancement due to oxygen concentration</p>
<p><em>exchange()</em> applies the inflow of water and dissolved tracers between bottom water layer and sediments, depending on the concentration of the tracer. This can be seen as a passive exchange of dissolved tracers depending on their concentrations. It is calculated as a product of</p>
<p><strong>-</strong> bi_exchange constant (ms<sup>-1</sup> per animal per m<sup>2</sup>)</p>
<p>- area of the sediment/water interface</p>
<p>- density of biological activity</p>
<p>- scalar of irrigation enhancement</p>
<p><em>injection()</em> does injection of water and dissolved tracers form bottom water layer into sediments, simulating active pumping of water by animals moving in burrows</p>
<p><strong>-</strong> bi_injection constant at zero depth (m<sup>2</sup>s<sup>-1</sup> per animal per m<sup>2</sup>)</p>
<p>- area of the sediment/water interface</p>
<p>- density of biological activity</p>
<p>- scalar of irrigation enhancement</p>
<p>The profile of biological activity with depth is set by the biosedprofile parameter, which can take four functional forms: constant, linear, parabolic, Gaussian</p></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td><p>Bioturbation: movement of <strong>particulate</strong> tracers throughout the sediment layers and into the water column due to biological activity. <strong>This routine works only if more than one sediment layer is present</strong></p>
<p><em>bioturb()</em> in <strong><u>atbioturb.c</u></strong></p>
<p>bioturbation</p>
<p>1</p></td>
<td><p>Only <strong>particulate</strong> tracers that are not macrobenthos and are allowed in the sediments, i.e.&nbsp;sediment grains, settled phytoplankton, microphytobenthos, meiobenthos, detritus and sediment bacteria, can be acted upon by bioturbation.</p>
<p>The <em>bioturb()</em> routine includes three subroutines (like the <em>bioirrig()</em> routine described above):</p>
<p><em>partic_diff()</em> implements particulate tracer diffusion within the sediments and is implemented in the same way as diffusion of dissolved tracers in the <em>dissol_diff()</em> subroutine described above. The diffusion rate is calculated as a product of:</p>
<p>- the diffusion constant bt_partic_kz (m<sup>2</sup>s<sup>-1</sup> per animal per m<sup>2</sup>)</p>
<p>- density of biological activity</p>
<p>- scalar of biological activity as a function of depth (which depends on the profile described in the biosedprofile)</p>
<p>- scalar of bioturbation enhancement</p>
<p><em>expul()</em> applies expulsion of particulate tracers from sediments into the bottom water column. Like above, its constant is set by the bt_expulsion parameter (ms<sup>-1</sup> per animal per m<sup>2</sup>), which is modified by the biomass density, scaling by depth and the bioturbation enhancement scalar.</p>
<p><em>mix_in()</em> applies exchange of particulate matter across sediment layers. The exchange constant is set in bt_exchange (ms<sup>-1</sup> per animal per m<sup>2</sup>) which is modified by biomass density, scaling by depth, and bioturbation enhancement scalar</p></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td><p><strong>III. Execute optional routines to enhance horizontal and vertical diffusion and mixing</strong></p>
<p><strong>(see chapter 5.4 for why these routines may be needed):</strong></p></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td><p>Horizontal diffusion</p>
<p><em>hdiffBMwc()</em> in <strong><u>athdiff.c</u></strong></p>
<p>horiz_diffusion</p>
<p>0</p></td>
<td><strong>Horizontal diffusion is done before the main transport model is run. This was mostly needed in the past, when oceanographic models and their conversion to Atlantis forcing files did not provide sufficient horizontal mixing.</strong></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td><p><a href="https://confluence.csiro.au/display/Atlantis/2014/05/13/Padding+out+mixing">Filler horizontal diffusion</a> <strong>applied for boxes that have not had any water exchanges for a set period of time</strong></p>
<p><em>filler_hdiffBMwc()</em> in <strong><u>athdiff.c</u></strong></p>
<p>fill_zero_exchange</p>
<p>0</p></td>
<td><p>Hydrodynamic models don't always resolve coastlines well, which can lead to some small coastal boxes being left isolated (no water fluxes into or out of them). This optional routine is used to correct for this. It is activated if there is no flux for a set threshold of days, provided in the flush_threshold (often set to 1 day). The routine will force mixing between the isolated box and matching layers of the adjacent boxes.</p>
<p>The routine will apply a constant water column mixing coefficient wc_kz, which indicates the proportion of volume to be mixed (a reasonable value is within a range of 1e-9) and scale it by the relative box volumes.</p>
<blockquote class="blockquote">
<p>The horizontal mixing can be further tuned using box-specific horizontal mixing parameter boxID:horizmix from the BGM file.</p>
<p>To activate this box specific mixing set use_fill_horizmix to 1. Otherwise the baseline diffusion rate among boxes is used.</p>
</blockquote></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td><p>Vertical diffusion added to correct for the lack of vertical circulation (this is not the same as vertical mixing used to imitate eddies, see below)</p>
<p><em>atvdiffBMwc()</em> in <strong><u>atvdiff.c</u></strong></p>
<p>vert_diffusion</p>
<p>0</p></td>
<td><p>Vertical diffusion executed throughout the water layers before the main transport model.</p>
<p>Setting flag mix_deep to 1 activates the replenishment of nutrients in the deep layers of oceanic boxes (those that have open water boundary with deep waters) and set the depth value for deep ocean layers in mix_deep_depth parameter. Nutrient values in the bottom layer of boxes that are deeper than mix_deep_depth will be reset to the original value given in the <em>initial_conditions.nc</em></p>
<p>The nutrients that will be reset are: NH3, NO3, Si, MicroNut and also Phosphorus and Carbon if they are tracked throughout the simulations (trackAtomicRatio flag is set to 1 in the <em>run.prm</em>)</p></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td><p>Vertical mixing added to simulate upwelling and increased mixing by eddies</p>
<p><em>vertical_mixing()</em> in <strong><u>atvmix.c</u></strong></p>
<p>vert_mix</p>
<p>1</p></td>
<td><p>Because processes within one cell are considered uniform, explicit representation of eddies and upwelling would require a lot of small boxes, untenable for Atlantis models. Instead the supply of nutrients from deeper water levels by eddies and upwelling can be represented with this optional vertical mixing routine. The routine applies season and box specific vertical water column mixing, in addition to the vertical exchange forced through hydrodynamic files.</p>
<p>The main difference between vertical diffusion routine above and vertical mixing is that diffusion is typically a slower process of movement from high to low concentration rates. Mixing in turn simulates advection (movement by water currents) and is usually much faster than diffusion. Also vertical mixing can be customised to vary across boxes and seasons.</p>
<p>The total mixing rate (in m<sup>3</sup>s<sup>-1</sup>) is calculated from lower layer up as a product of</p>
<p>- mix_injection coefficient (1e-3)</p>
<p>- Box specific :vertmix scalar defined in the BGM file</p>
<p>- seasonal scale</p>
<p>- eddy scale</p>
<p>- volume of the box</p>
<p>Note that the sign of the global mix_injection and box-specific :vertmix parameters defines the <strong>direction of vertical mixing</strong>. Usually these coefficients are positive, which means the water will flow upwards, but in theory they can also be negative. Either way, <strong>in vertical mixing routine water flows only in one vertical direction</strong> <strong>(up or down)</strong>.</p>
<p>The seasonal scale is modelled as a sine shaped wave (sin(2.0*3.1415*(day_of_the_year-31.0)/365.0) and it’s fluctuations can be increased by setting the mix_season_kz multiplier (set to 10 in SETas model).</p>
<p>The effect of eddies (eddy scale) is activated with eddy_vertmix flag. It can be tuned with two parameters. First, box-specific eddy strengths per each quarter of the year (the model linearly interpolates between these to get eddy strength on any one day) are set with eddy_S1 to eddy_S4. These eddy parameters are used also in scaling <em>Primary_Production()</em> in <strong><u>atprocess.c</u></strong> in the <strong>atecology</strong> library. Therefore, to allow for different effect on primary production and vertical mixing by eddies, the vertical mixing has specific parameter eddy_vertmix. Vertical mixing can further be further scaled with the global scaling coefficient eddy_mixscale</p>
<p>As in the vertical diffusion routine above, activating the flag mix_deep will reset nutrient concentrations in the deepest layers of oceanic boxes</p>
<p>The vertical mixing code within Atlantis to represent exchange between vertical layers is used to represent changes in upwelling or changes in mixing due to (seasonal or long term) modifications of mixed-layer depth that is not sufficiently captured in the hydrodynamics file.</p>
<p>To activate this code set vert_mix to 1 in the <em>physics.prm</em> file. Then in the <em>forcing.prm</em> file add the pointers to the relevant file of scalars:</p>
<p># Vertical mixing scalar files</p>
<p>use_VertMixFiles 1</p>
<p>vertMixScalar_rewind 0</p>
<p>nvertMixScalarfiles 1</p>
<p>vertMixScalar0.name inputs/Filename_VertMix.nc</p>
<p>The structure of the Filename_VertMix.nc is the same as for the temperature or salinity files – i.e.&nbsp;a value per layer per box per timestep included in the file. The equation used to apply the modified vertical mixing is</p>
<p><span class="math display">\[mix_{rate} = mix_{injection}*vertmix*vertmix_{scale}*eddy_{scale}*seasonal_{scale}*{vol}_{box,layer}\]</span></p></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td><img src="markdown_output/media/image31.jpeg" style="width:7.07107in;height:4.27358in"></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td><span id="_Toc526762788" class="anchor"></span>Figure 8. Illustration of optional physics routines described in detail above. Some boxes have two sediment layers. Vertical mixing can also be set as downwelling, but only one direction is possible in a given simulation.</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td><p><strong>IV. Execute the main transport model</strong></p>
<p>Although there is an option to turn the transport model off (advect_diffusion 0) this should <strong>only be done for debugging purposes</strong> as it <strong>stops all water fluxes</strong> in the model domain. Also note that if you turn off the transport model this will stop the the t.hd timestep from incrementing. The timestep is the timestep from the hydro model. It is also used to read in data from the point source input files. So if you turn off the transport model the values read in from the point source files will always be the value corresponding to the first hydro timestep (the first time value in the first hydro forcing file).</p>
<p>The transport model has an option to scale all transport forced from hydrodynamic input file. This is set by setting scale_transport flag to 1 or 2.</p>
<p>If scale_transport is set to 1, the transport will be scaled by prcnt_exchange parameter. For example, setting prcnt_exchange to 0.5 will half all the horizontal and vertical transport, in some older models this was needed to help correct for hyperdifussion.</p>
<p>If scale_transport flag is set to 2, all transport is scaled by the area of the box and ka_exchange multiplier. The area corrected exchanges are multiplied by ka_exchange before they are used (transport_scalar = ka_exchange / box_area). This helps to ensure the transport is not scaled down too much.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p>NOTE!</p>
<p><strong>Do not scale transport by box area if hyperdifussion correction by box area has already been done during the forcing file construction stage, such as in the Hydroconstruct package (see above).</strong></p></th>
</tr>
</thead>
<tbody>
</tbody>
</table></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td><p>Main (advective) transport model</p>
<p><em>transportBM()</em> in</p>
<p><strong>attransport.c</strong></p>
<p>advect_diffusion</p>
<p>1</p></td>
<td><p>This routine executes all forced water fluxes (moves water across cells) and calculates new concentrations of tracers.</p>
<p>Transport model also has a cascade_flows flag which executes transport among boxes with large differences in vertical layers. If cascade_flows is set to 0 the loop calculating transport will only go through the number of water columns in the existing box, if it is set to 1 it will go through the number of water columns in both of the exchanging boxes.</p>
<p>The <em>transportBM()</em> routine calculates new concentrations of tracers, including temperature, salinity and pH. These calculations are done separately in the <em>Properties_At_Depth()</em> routine in <strong>atbiophysics.c</strong> in the <strong>atecology</strong> library. The calculated values are then overwritten if temperature, salinity, pH or other forcing files are provided (see below). The temperature calculations use baseline_temp and temp_ampltiude parameter and will be described in chapters dealing with Biology submodel.</p></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td><strong>V. Read in temperature, salinity and other tracer forcing values and overwrite values calculated in the transport model</strong></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td><p>Read in forcing files for temperature, salinity, pH or other forced tracers and overwrite calculated values</p>
<p><em>tracerForcingBM()</em> in <strong>attempsalt.c</strong></p>
<p>use_tempfiles* 1</p>
<p>use_saltfiles* 1</p>
<p>use_pHfiles* 0</p>
<p>use_force_tracers* 0</p></td>
<td><p>The calculations of tracer values done in the <em>transportBM()</em> routine are overwritten if forced values are provided and appropriate flags activated in the <em>force.prm</em> file</p>
<p>Typically, Atlantis models include temperature and salinity forcing (Table 5), but also pH, oxygen and other tracers set in the <em>initial_conditions.nc</em> can be forced. Such forcing is done if calculations of tracer concentrations done in Atlantis are deemed to be of insufficient accuracy and better information is available from more specialised models or observations (see <a href="https://confluence.csiro.au/display/Atlantis/2013/07/10/Read+in+forcing+data+for+tracers">tracer forcing</a> on the wiki)</p>
<p>The number and names of all tracer forcing files must be listed in the <em>force.prm</em> parameters nforceTracers<strong>*</strong> and tracerNames<strong>*</strong>, eg:</p>
<p>nforceTracers 2</p>
<p>tracerNames 2</p>
<p>Oxygen Light</p>
<p>This is followed by the actual files and parameter indicating whether the tracer values should be looped (rewound) or not.</p>
<p>Oxygen_nFiles 1</p>
<p>Oxygen_File0.name inputs/forcisets/oxygen.nc</p>
<p>Oxygen_rewind 1</p>
<p>Oxygen_wgt_coefft 0.7</p>
<p>The weighting coefficient is what weight to put on the values being read into Atlantis (using a weighted average of Atlantis value and value from the forcing file – so a 1.0 means the read in value is used in place of the native Atlantis value, 0.0 means Atlantis value is used, a value between is a mix of the two)</p>
<p>(see <a href="https://confluence.csiro.au/display/Atlantis/2013/07/10/Read+in+forcing+data+for+tracers">instructions</a> also on the wiki).</p>
<p>Note, that the names of tracers provided in the tracerNames MUST exactly match the names in the <em>initial_conditions.nc</em> file, so that Atlantis knows which tracers to overwrite.</p></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td><strong>NOTE!</strong> <strong>Forcing of tracers, especially those containing nitrogen, should be done with care, as it can easily affect the conservation of mass in the model</strong></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td><p>Forcing species distributions</p>
<p>use_move_entries 1</p></td>
<td><p>To change the distribution of species (for movement) you need to set the flag and then provide the following information for each group you wish to force (this example has two groups, FXX and FYY, and 2 stages)</p>
<p>nforceMoveGroups 4</p>
<p>MoveGroupCodes 4</p>
<p>FXX_stage_0</p>
<p>FXX_stage_1</p>
<p>FYY_stage_0</p>
<p>FYY_stage_1</p>
<p>Also list the distribution nc files - 1 file per "MoveGroupCode"</p>
<p>FXX_stage_0_File.name DistribFilename1.nc</p>
<p>FXX_stage_1_File.name DistribFilename2.nc</p>
<p>FYY_stage_0_File.name DistribFilename3.nc</p>
<p>FYY_stage_1_File.name DistribFilename4.nc</p>
<p>You must also provide the starting time for using the forcing (so it does not have to start on day 0 of the model simulation)</p>
<p>FXX_stage_0_tstart 0</p>
<p>FXX_stage_1_tstart 0</p>
<p>FYY_stage_0_tstart 0</p>
<p>FYY_stage_1_tstart 0</p>
<p>The format of the nc files and other relevant information is available in the section <em>Forcing the model distributions with species distribution models</em>.</p></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td><strong>VI. Do final corrections and scaling</strong></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td><p>Apply temperature, salinity and pH scalars to the biological processes</p>
<p><em>Ecology_Apply_Environ_ Scalars()</em> from <strong>atbiophyscis.c</strong> in <strong>atecology</strong></p></td>
<td><p>Runs optional scaling of biological processes by environmental conditions, such as temperature-, oxygen-, salinity-, pH- dependent feeding rates, mortality or other biological processes.</p>
<p>This will be described further in the Biology routines</p></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td><p>Update eddy values</p>
<p><em>Eddy_Strength_Update()</em> routine in <strong>atphysics.c</strong></p></td>
<td>Conducts linear interpolation of eddy strengths for the current day of the year from the quarterly box specific values given in eddy_S1, eddy_S2, eddy_S3 and eddy_S4.</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td><p>Checks saturation values</p>
<p><em>Saturation_Check()</em> in <strong>atsaturation.c</strong></p></td>
<td>Checks that gases and nutrients haven't surpassed saturation values. If saturation values are exceeded, then concentrations are reset to the maximum saturation concentrations.</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td><p>Get ice related information from input forcing files</p>
<p><em>Get_Ice()</em> in <strong>aticeIO.c</strong></p></td>
<td><p>Reads in ice properties for the current time step from TS or NC forcing files</p>
<p>Only executed if the <em>initial_conditions.nc</em> file has the global attribute icenz (see below and <a href="https://confluence.csiro.au/display/Atlantis/Ice+in+Atlantis">details</a> on wiki), which activates the ice model.</p></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Update time for writing outputs</td>
<td>Update the next time to write the values into <em>inputs.ts</em> and <em>exports.ts</em> files (see Table 3 on the contents of these files)</td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="other-parameters-in-physics.prm-not-included-in-the-physics-routine" class="level2">
<h2 class="anchored" data-anchor-id="other-parameters-in-physics.prm-not-included-in-the-physics-routine"><strong>5.3. Other parameters in <em>physics.prm</em> not included in the <em>physics()</em> routine</strong></h2>
<p>Below are the remaining parameters and physics processes that were not described in the main <em>physics()</em> routine.</p>
<p><em>I. Resetting water column depths</em></p>
<p>wc_dz_tol: This parameter is used by <em>check_wc_dz()</em> routine in the <strong>atvertgeom.c</strong> file of the <strong>atphysics</strong> library. The routine checks the depth of water layers. Transport of water fluxes can in some cases cause too much flux into certain layers. This can be controlled by setting the maximum change in depth allowed in the wc_dz_tol parameter. The model will only allow this much fractional change before dz is reset to nominal value given in the <em>initial_conditions.nc</em> file by the nominal_dz variable. The wc_dz_tol parameter in many models is set to 0.2.</p>
<p>constrain_wc: This is a flag that is called in the <em>box_bio_process()</em> routine in <strong>atbiology.c</strong> and it resets the minimum water column depth to 1m if it becomes any shallower due to water fluxes. Turning this flag on is recommended in models that have a lot of shallow water areas and do not model sediment processes through multiple sediment layers explicitly. This avoids difficulties in modelling nutrient fluxes in very shallow waters.</p>
<p><em>II. Checks for the depth of sediment layer(s)</em></p>
<p>Same checks are done for the depth of sediment layers, which are controlled by maxseddz and minseddz parameters. The depth of the sediment layer(s) is controlled by the <em>check_sed_dz()</em> routine in <strong>atvertgeom.c</strong>. If the sediment layer depth gets too small the layer is merged with an adjacent sediment layer. If the sediment layer depth becomes too large it is split in two (made up of the original layer and a new inserted sediment layer). <strong>This is only used in models that have &gt;1 sediment layer!</strong> In models with only one sediment layer, there are no constraints on the thickness of the sediment layer.</p>
<p><em>III. Atmospheric deposition of gases and nutrients trapped in particulate matter</em></p>
<p>Atlantis has always allowed for atmospheric deposition of NH and NO, but this new functionality makes this explicit and allows users to modify atmospheric concentrations (mg per m<sup>3</sup>) of nutrients and gases trapped in particulate matter. Note, <strong>this is not the diffusion of gases through the water’s surface</strong>, which is dealt with in the <em>gasExchange()</em> routine. The atmospheric deposition is handled in respective nutrient routines in the <strong>atNutrient.c</strong> file in <strong>atecology</strong> library. To apply the atmospheric deposition, set include_atmosphere flag to 1 in the <em>physics.prm</em>. If the flag is set to 0 there will be no implicit deposition from the atmosphere.</p>
<p>If include_atmosphere is active then atmospheric concentrations (mg per m<sup>3</sup>) must be provided for the following seven nutrients and gases (example values given here are taken from the SETas model)</p>
<p>atmospheric_NH 0.017</p>
<p>atmospheric_NO 0.025</p>
<p>atmospheric_F 0.0</p>
<p>atmospheric_P 0.0</p>
<p>atmospheric_Si 0.0</p>
<p>atmospheric_O2 198.9</p>
<p>atmospheric_CO2 0.0</p>
<p><em>IV. Scaling of point sources</em></p>
<p>It is possible to scale the input of nutrients from point sources given in the TS files. This scaling might be useful if simulating increase or decrease of nutrient input through time. The scaling routines are activated with the flag nutrientchange. If the flag is set to 1, then a number of additional parameters should be provided, telling Atlantis how to scale the nutrients. The parameters required for scaling allow detailed scaling of each nutrient in each point source and are explained further on the <a href="https://confluence.csiro.au/display/Atlantis/Scaling+point+sources">wiki</a>.</p>
<p><em>V. Other parameters from old functionality</em></p>
<p>vdiffwt_wc: sets the scalar of water column diffusion and is used in the optional and currently rarely activated vertical and horizontal diffusion routines.</p>
<p>vdiffwt_sed: sets the scalar of diffusion in the sediments. It is used in the bioirrigation routine.</p>
<p>edge_type: a string of box-specific parameters describing the way box edges deal with water fluxes (0=standard, 1=absorptive, 2=reflective, 3=assymetrical scaling). This was used in the early days of Atlantis, when oceanographic files were not as sophisticated as they currently are and transport of tracers had to be manually adjusted. Setting different edge types for different boxes scaled the transport of tracers into and from the boxes. For current models, it is recommended to set the edge_type to 0 (standard) for all boxes.</p>
<p>biooxprofile: This was used in earlier Atlantis versions when the oxygen profile in the sediments was set and not dynamic. Currently the amount of oxygen in the sediments dynamically depends on the amount of biological activity and the parameter is not used.</p>
</section>
<section id="when-to-use-vertical-and-horizontal-transport-and-mixing" class="level2">
<h2 class="anchored" data-anchor-id="when-to-use-vertical-and-horizontal-transport-and-mixing"><strong>5.4. When to use vertical and horizontal transport and mixing?</strong></h2>
<p>The horizontal and vertical diffusion routines <em>hdiffBMwc()</em> and <em>atvdiffBMwc()</em> were created and used most often at a time when oceanographic models were still quite coarse and did not capture the water fluxes with sufficient accuracy. They are rarely used in current models. However, a number of issues still remain due to the loss of spatial resolution when converting files into the Atlantis model domain. This is where activating optional vertical mixing and filler horizontal diffusion routines might be useful:</p>
<p>1) <strong>Eddies</strong>. The spatial scale of most Atlantis models is too coarse for explicitly resolving oceanic eddies. Yet, eddies have a strong influence on productivity and oxygen content and ideally should be included in some form. The effect of eddies can be captured in two non-exclusive ways:</p>
<p>a) Eddy parameters (eddy_S1 to eddy_S4 and eddy_mixscale) parameters are used in the <em>Primary_Production()</em> routine to scale production. If some boxes are known to have strong eddies that amplify local production, the user can modify box- and season-specific eddy parameters to enhance production in certain boxes and certain seasons. The presence of eddies will scale the growth rate of primary producers, so that for the same nutrient concentration more growth is allowed (but of course, no growth is possible if no nutrients are available).</p>
<p>b) Eddy effect on productivity can be simulated by activating the <em>vertical_mixing()</em> routine, where mixing will be enhanced with box- and season-specific eddy scalars. Increased vertical mixing will bring more nutrients to upper water levels and will increase productivity in boxes identified through seasonal eddy_S1 to eddy_S4 values (which can differ box to box, showing relative eddy strength). Differences in vertical mixing intensity among boxes can be further tuned using the box-specific :vertmix parameter in the BGM file.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>Good practice tip 4</strong></p>
<p>Eddies are not currently included in most models, but exploring their potential effect on the model outcomes is recommended. They provide a useful approach for adding spatial realism in primary productivity dynamics and may be particularly important in deeper waters and in systems where eddy effects are known to be strong.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>2) <strong>Upwelling</strong>. The importance of upwelling for primary productivity is well recognised, but it is often difficult to capture the intensity of vertical mixing and nutrient input through the hydrodynamic forcing inputs alone. The latest oceanographic models are getting better in capturing upwelling through vertical fluxes, but in some cases Atlantis modellers might want to add an additional forced time series of point nutrient source into boxes where nutrient supply due to upwelling is not accurately captured. Upwelling would normally only be added into oceanic boxes, i.e.&nbsp;those that do not end with the sediment but with the open deep water boundary. The mixing of the forced nutrient input into the water column might have to be increased using the <em>vertical_mixing()</em> routine and by modifying box specific :vertmix parameter in the BGM file (or by using eddy_S1 to eddy_S4 parameters), to ensure the additional nutrients get properly mixed into the water columns.</p>
<p>Note, that the replenishment of nutrients in the deepest water column layers of the oceanic boxes may not require forced nutrient inputs, as it can be set with the mix_deep flag (see the description of the vertical diffusion routine above) and by normal vertical mixing. Yet, activating the mix_deep will only reset the nutrient concentrations in the deepest water layer to the nominal (initial) values given in the <em>initial_conditions.nc</em>. Oxygen can be added to the list of variables that are included in this representation of the replenishment from deep waters using mix_deep_O2 (by setting it to 1; 0 indicates Oxygen will not be included)</p>
<p>Including a separate input forcing file allows the user to specify time variable nutrient input throughout the year.</p>
<p>3) <strong>Absence of fluxes in small coastal boxes</strong>. Many oceanographic models, unless they were specifically built, still do not capture coastal processes with sufficient detail. This, combined with the loss of spatial resolution, can lead to isolated coastal boxes that receive no water fluxes (especially if the boxes are small or particularly shallow). This would naturally affect the model outcomes in such boxes, leading to high accumulation or depletion of nutrients and biomass.</p>
<p>Ideally, such issues should be resolved in cooperation with oceanographers who developed the models. If this is not possible, the filler horizontal diffusion routine will provide a very crude approximation of mixing for such boxes. Mixing can be further tuned with box specific mixing scalars (:vertmix and :hozirmix) in the BGM file.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>Good practice tip 5</strong></p>
<p>The exploration of vertical and horizontal water fluxes is strongly recommended, especially during the model development stage. Currently there are two complementary ways to do it:</p>
<p>1. Use a <strong><em>shinyrAtlantis</em></strong> R package (see chapter 4.4) to plot fluxes through time and connections among boxes. The tool helps to identify isolated boxes, visualise intensity and direction of fluxes among boxes and also plot the magnitude of vertical fluxes in each box. The user can then compare these fluxes with the existing oceanographic knowledge and decide whether the intensity of vertical mixing should be increased or decreased using box-specific parameters.</p>
<p>2. Use passive inert tracers, such as sediments (SED in the <em>initial_conditions.nc</em>) to explore dispersal of particles through the model domain. Sediments are not acted upon by any organisms, so they won’t be eaten or decay (make sure :decay is 0). The initial concentrations of the tracer can be set in specific locations in the <em>initial_conditions.nc</em> file and their dispersal through the model domain followed in the model output. Alternatively, sediment inputs can be given in the TS files as forced point sources; and their dispersal from the point input locations can then be followed through time.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</section>
<section id="latest-developments" class="level2">
<h2 class="anchored" data-anchor-id="latest-developments"><strong>5.5. Latest developments</strong></h2>
<p>New functionality is added to Atlantis almost every month. At the time of writing of this manual (April 2016) three significant new features were being tested: land, ice and contaminants. They are described here briefly, but the users are advised to check Atlantis wiki for further developments and instructions.</p>
<section id="land" class="level3">
<h3 class="anchored" data-anchor-id="land"><strong><em>5.5.1 Land</em></strong></h3>
<p>It is now possible to include <a href="https://confluence.csiro.au/display/Atlantis/Land+in+Atlantis">land</a> in Atlantis. The key reason to include this functionality is to allow explicit modelling of the movement of nutrients, organisms and human activity between land and water (groups can only make this transition if they are marked as being able to live on land and in the water in the <em>functional_groups.csv</em> file). It also allows for the future development of Atlantis into an integrated water-land model (so you can model land organisms). Currently this may be useful for shallow water areas, harbours and estuaries that have a lot of islands.</p>
<p>The optional land model is turned on by activating the flag flagAllowLand in the <em>run.prm</em> file. Land boxes are set up in the BGM file as having positive botz value. Land occupies water column layer 0 (bottom layer) and processes on land boxes are modelled in two-dimensional space, just like those for epibenthos. To obtain the total biomass, tracer values on land boxes are not multiplied by depth. It is also possible to include specific terrestrial tracers, by setting their bmtype = "terrestrial" in the <em>initial_conditions.nc</em> file. The values for these tracers would be stored in a different place than those for the water column tracers.</p>
</section>
<section id="ice" class="level3">
<h3 class="anchored" data-anchor-id="ice"><strong><em>5.5.2. Ice</em></strong></h3>
<p>Atlantis applications for the northernmost and southernmost domains (Barents Sea, Baltic Sea, Antarctica and others) have prompted the development of an explicit <a href="https://confluence.csiro.au/display/Atlantis/Ice+in+Atlantis">ice</a> model.</p>
<p>The optional ice model is activated when the <em>initial_conditions.nc</em> file has the global icenz variable included. This sets up the maximum number of ice layers. The ice model requires <a href="https://confluence.csiro.au/display/Atlantis/Changed+to+prm+files">ice forcing files</a> and their paths need to be identified in the <em>force.prm</em> (nInceFiles parameter in <em>force.prm</em>)</p>
<p>The <a href="https://confluence.csiro.au/display/Atlantis/Changed+to+prm+files">key characteristics of ice are described</a> in the wiki along with the additional parameters added to the <em>physics.prm</em> - these include maximum and minimum depth of an ice layer (maxicedz and minicedz), the number of ice classes, presence of slush, and specification of the ice model to use. The biological <a href="https://confluence.csiro.au/display/Atlantis/Ice+groups">groups</a> that are allowed in the ice include bacteria, diatoms, mixotrophs and small zooplankton. Activating the ice model requires a number of <a href="https://confluence.csiro.au/display/Atlantis/Changed+to+prm+files">new parameters in the <em>biology.prm</em></a> file too. These describe how to handle albedo, light attenuation, the depth consumers can dig into ice (the ice penetration depth of predators).</p>
</section>
<section id="contaminants" class="level3">
<h3 class="anchored" data-anchor-id="contaminants"><strong><em>5.5.3. Contaminants</em></strong></h3>
<p>It is now possible to track an unlimited number of <a href="https://confluence.csiro.au/display/Atlantis/Contaminants+in+Atlantis">contaminants</a> through the food chain. However, each contaminant requires specific new tracers tracking contaminant concentrations in each functional group (and each age group of fully age structured functional groups), which means significant changes to the input files and longer simulation run times. Each contaminant in the <em>initial_conditions.nc</em> is setup as a separate tracer (e.g.&nbsp;Arsenic). Because contaminants are tracked through all functional groups, a new contaminant tracer must be added for all tracked functional groups, e.g.&nbsp;Sed_Bact_Arsenic, Zoo_Arsenic, Cod1_Arsenic, Cod2_Arsenic, Cod3_Arsenic and so on.</p>
<p>The functional groups take up contaminants either through contact, general uptake or through consumption. For each functional group the user will have to provide a number of <a href="https://confluence.csiro.au/pages/viewpage.action?spaceKey=Atlantis&amp;title=Contaminants+in+Atlantis">new parameters in the <em>biology.prm</em></a> file: the uptake shape (linear or sigmoidal), uptake rate (mgm-<sup>3</sup> per second), 100% and 50% lethal concentration, and time to reach 50% mortality. It is also possible to include an additional contaminant effect on growth.</p>
<p>To set up contaminant tracking set the track_contaminants flag to 1 in the <em>run.prm</em> file and provide contaminant names and forcing files.</p>
<p>The definitions of contaminant tracers are added to the <em>run.prm</em> file. For example, if you want to add a single Arsenic tracer you would add the following to the <em>run.prm</em> file:</p>
<p># Set track contaminants to 1 to have active contaminants and 0 if no contaminants are desired</p>
<p>track_contaminants 1</p>
<p># The number of contaminants to read in</p>
<p>num_contaminants 1</p>
<p># Whether you need to check the contaminant fluxes (for the debugging or calibration of the contaminant model components)</p>
<p>flag_contam_sanity_check 1</p>
<p>To identify the contaminants, add an array of space separated contaminant names.</p>
<p>Atlantis will search for tracers with these names in the initial conditions file.</p>
<p>The number after contaminant names is the number of contaminant names to read in - same as other array definitions in the prm files.</p>
<p>contaminant_names 1</p>
<p>Arsenic</p>
<p>The units of the contaminants are entered as a space separated array.</p>
<p>contaminant_units 1</p>
<p>As/m^3</p>
<p>Atlantis will then expect the following tracer to be defined in the <em>initial_conditions.nc</em> model :</p>
<p>double Arsenic(t, b, z) ;</p>
<p>Arsenic:bmtype = "tracer" ;</p>
<p>Arsenic:units = "As/m^3" ;</p>
<p>Arsenic:long_name = "Amount of Arsenic" ;</p>
<p>Arsenic:sumtype = 1 ;</p>
<p>Arsenic:dtype = 0 ;</p>
<p>Arsenic:inwc = 0 ;</p>
<p>Arsenic:insed = 0 ;</p>
<p>Arsenic:dissol = 0 ;</p>
<p>Arsenic:decay = 0. ;</p>
<p>Arsenic:partic = 0 ;</p>
<p>Arsenic:_FillValue = 0. ;</p>
<p>For each contaminant tracer Atlantis must be provided with a forcing file in netcdf format (the file name and path is given in the <em>force.prm</em> file along with all other forcing time series). These can be all contained in a single netcdf file or with separate files per tracer. The following is an example forcing file in line with our example – the contaminant is for Arsenic in a model, for a model with 2 boxes and 3 water column layers. This example is for a single spill in a single box/layer of the model on the second time step in the file. This assumes the contaminant will be spread around the model using the existing transport model within Atlantis.</p>
<p>netcdf Arsenic_filename {</p>
<p>dimensions:</p>
<p>t = UNLIMITED ; // (3 currently)</p>
<p>b = 2 ;</p>
<p>z = 4 ;</p>
<p>variables:</p>
<p>double t(t) ;</p>
<p>t:units = "2005-01-01 00:00:00 +10" ;</p>
<p>t:dt = 86400. ;</p>
<p>double Arsenic(t, b, z) ;</p>
<p>Arsenic:_FillValue = -999. ;</p>
<p>Arsenic:missing_value = -999. ;</p>
<p>Arsenic:valid_min = 0. ;</p>
<p>Arsenic:valid_max = 300. ;</p>
<p>Arsenic:units = "As/m^3" ;</p>
<p>// global attributes:</p>
<p>:title = "trivial" ;</p>
<p>:geometry = "VMPA_setas.bgm" ;</p>
<p>:parameters = "" ;</p>
<p>data:</p>
<p>t = 0, 86400, 172800 ;</p>
<p>Arsenic =</p>
<p>_, _, _, _,</p>
<p>_, _, _, _,</p>
<p>0.1, _, _, _,</p>
<p>_, _, _, _,</p>
<p>_, _, _, _,</p>
<p>_, _, _, _ ;</p>
<p>}</p>
<p>To identify the forcing file enter the following into the force.prm file:</p>
<p>use_force_tracers 1</p>
<p># The number of tracers to search for in the files and the names of those tracers.</p>
<p>nforceTracers 1</p>
<p>tracerNames 1</p>
<p>Arsenic</p>
<p># Now the actual files. Can have more than one tracer in a file.</p>
<p>Arsenic_nFiles 1</p>
<p>Arsenic_File0.name Arsenic_filename.nc</p>
<p>Arsenic_rewind 0</p>
<p>Arsenic_File0.use_resets 0 #determines whether the values in Atlantis are reset or added to existing values</p>
<p><strong><em>Impact response functions</em></strong></p>
<p>The groups within Atlantis take up contaminants either through contact, general uptake or through consumption.</p>
<p>Some of the relevant&nbsp;parameters (which need to be added to the <a href="https://confluence.csiro.au/display/Atlantis/Changed+to+prm+files"><em>biology.prm</em></a> file) and processes are:</p>
<table class="caption-top table">
<colgroup>
<col style="width: 13%">
<col style="width: 85%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"><strong>Flag (parameter)</strong></th>
<th style="text-align: left;"><strong>Meaning&nbsp;</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">flag_dissolved_pollutants</td>
<td style="text-align: left;">Flag indicating whether the contaminants are dissolved and transmitted in excretion (1) or not (0)</td>
</tr>
<tr class="even">
<td style="text-align: left;">flag_contamMortModel</td>
<td style="text-align: left;">Flag indicating whether top use the simple mortality model from Laender et al 2008 (1) or compound model from InVitro (0)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">flag_contamInteractModel</td>
<td style="text-align: left;">Flag indicating - No contaminant interactions (0), Additive interactions (1), Multiplicative interactions (2), Most limiting contaminant dominates (3)</td>
</tr>
<tr class="even">
<td style="text-align: left;">flag_contamGrowthModel</td>
<td style="text-align: left;">Flag indicating -&nbsp;No growth effects of contaminants (0), InVitro representation (1), logistic model (2)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">flag_contamMove</td>
<td style="text-align: left;">Flag indicating the avoidance scalar used - 0 = none, 1 = knife-edge, 2 = sigmoidal, 3 = left shoulder flat top</td>
</tr>
<tr class="even">
<td style="text-align: left;">flag_contamOnlyAmplify</td>
<td style="text-align: left;">Flag indicating whether the interacting contaminants can only amplify (1) or can buffer/negate each other too (0)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">flag_contamMinTemp</td>
<td style="text-align: left;">Flag indicating the minimum temperature to consider in contaminants relationships that involve temperature corrections (if the realised temperature is lower this minimum value is used instead)</td>
</tr>
<tr class="even">
<td style="text-align: left;">flag_detritus_contam</td>
<td style="text-align: left;">Flag indicating whether running contaminants through detritus</td>
</tr>
<tr class="odd">
<td style="text-align: left;">flag_contam_halflife_spbased</td>
<td style="text-align: left;">Flag indicating whether half-life of contaminant is species specific</td>
</tr>
<tr class="even">
<td style="text-align: left;">flag_contamMaternalTransfer</td>
<td style="text-align: left;">Flag indicating whether maternal transfer enable</td>
</tr>
</tbody>
</table>
<p><em>Other relevant parameters:</em></p>
<p>Contaminants can also affect movement. There is the potential for mobile species (defined by the&nbsp;<em>VerticallyMigrates</em>&nbsp;and&nbsp;<em>HorizontallyMigrates</em>&nbsp;settings for the species in the f<em>unctional_groups.csv</em>&nbsp;file) to avoid the contaminants. It is not always appropriate biomass pool groups to show this behaviour so to turn it off set&nbsp;biopools_dodge_contam&nbsp;0 in the&nbsp;<em>biology.prm</em>.</p>
<p>For each of the contaminants the following parameetrs is required. For contaminant ZZZZ the coefficient indicating degree of dissolubility of the material (used if flag_dissolved_pollutants is set to 1, and determined for each contaminant - 1 all, 0 none) is set using ZZZZ_dissolv_coefft. Half life in seconds is set using ZZZZ_half_life. The flag indicating whether the contaminant effect is temperature dependent is<br>
ZZZZ_temp_depend.</p>
<p>For each functional group XXX the following is also required:</p>
<p>XXX_ZZZZ_uptake_rate - uptake rate (per day -&nbsp; will be converted to per second inside code).</p>
<p>XXX_ZZZZ_InstantDoseMortality – the instant dose contact mortality rate</p>
<p>XXX_ZZZZ_LD100 - the concentration that will kill 100% of the population.</p>
<p>XXX_ZZZZ_LD50 - the concentration that will kill 50% of the population.</p>
<p>XXX_ZZZZ_TimeToLD50 - the number of seconds that it takes to kill 50% of the population (based on Invitro code)</p>
<p>XXX_ZZZZ_avoid - the concentration where animals will begin avoiding the substance</p>
<p>XXX_ZZZZ_K_avoid - the constant used in the avoidance curve (for when there is a slow Gaussian tail to the&nbsp;response rather than a knife edge cut off)</p>
<p>Growth effects due to contamination can also be represented using the following parameters. For each species and each contaminant add</p>
<p>XXX_ZZZZ_GrowthThresh - the tissue contaminant level where growth effects start</p>
<p>XXX_ZZZZ_GrowthEffect - the size of the growth effect (as a scalar)</p>
<p>The flag setting the uptake model XXX_ZZZZ_uptake_option. This currently has three settings:</p>
<p>If set to 1 use a linear uptake (parameter is in parts/sec)</p>
<p><span class="math display">\[C_{a,t} = C_{a,t - 1} + dt \bullet \tau \bullet C_{c,t}\]</span></p>
<p>where <em>C<sub>a,t</sub></em> is the concentration of the contaminant in an agent at time <em>t</em>; <em>C<sub>c,t</sub></em> is the concentration in the water column (or sediments) and τ is the species specific contaminant uptake rate.</p>
<p>If set to 2 then use a sigmoid function</p>
<p><span class="math display">\[C_{a,t} = \frac{C_{a,t}'}{dt}\]</span></p>
<p><span class="math display">\[C_{a,t}' = \left( C_{c,t - 1}^{(1.0 - \varsigma)} - {\left( C_{c,t - 1}^{(1.0 - \varsigma)} - C_{a,t - 1}^{(1.0 - \varsigma)} \right) \bullet e}^{- \mu \bullet dt \bullet (\varsigma + 1.0)} \right)^{\left( \frac{1.0}{1.0 - \varsigma} \right)}\]</span></p>
<p>with <span class="math inline">\(\varsigma\)</span> the uptake constant (between 0 and 1, set to 0.99 by default) and the uptake rate.</p>
<p>And if set to 3 then an alternative sigmoidal uptake is used – this is ideal for representing interactions with a toxin that has no effect until a critical concentration is reached</p>
<p><span class="math display">\[C_{a,t} = \frac{C_{a,t}}{C_{a,t - dt} + \left( C_{a,t} - C_{a,t - dt} \right)e^{- \tau_{s}dt}}\]</span></p>
<p>If you want to allow for uptake of contaminants back out of detritus you also need to set flag_detritus_contam to 1.</p>
<p>Movement and reproduction can both be scaled due to level of contaminants in the organism using XXX_ZZZZ_MoveEffect and XXX_ZZZZ_ReprodEffect.</p>
<p>Maternal transfer is also possible (if the flag flag_contamMaternalTransfer is set to1) in which case the maternal transfer rates need to be set – both that for transfer during conception (recruitment) XXX_ZZZZ_maternal_transfer and during suckling (if have maternal care) XXX_ZZZZ_suckling_transfer.</p>
</section>
</section>
<section id="outputs-of-physics-tracers-in-the-output.nc-file" class="level2">
<h2 class="anchored" data-anchor-id="outputs-of-physics-tracers-in-the-output.nc-file"><strong>5.6. Outputs of physics tracers in the <em>output.nc</em> file</strong></h2>
<p>Values of physical variables throughout the simulation are reported in the main <em>output.nc</em> file. Although not usually analysed in standard simulations, it might be useful to explore these values during the model development stage.</p>
<p><span id="_Toc526762759" class="anchor"></span>Table 7. List and characteristics of physics variables given in the <em>output.nc</em> files (the same variables are given in the <em>initial_conditions.nc</em> file but without the time dimension).</p>
<table class="caption-top table">
<colgroup>
<col style="width: 3%">
<col style="width: 8%">
<col style="width: 87%">
</colgroup>
<thead>
<tr class="header">
<th><strong>Name</strong></th>
<th><strong>Description, units</strong></th>
<th><strong>Comments</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>volume (t, b, z)</td>
<td>Volume of each cell at each time step, m<sup>3</sup></td>
<td>The volume changes due to water fluxes that simulate currents and tides. The change in the box’s volume is seen only as change in the depth and not surface area of the water column. The maximum fractional change in the depth is set by wc_dz_tol, if the depth exceeds this the depth is reset to the nominal_dz values given in the <em>initial_conditions.nc</em>. At present big tides will cause this reset, a flag to disable this is under development so please check the wiki for any updates.</td>
</tr>
<tr class="even">
<td><p>hdsource</p>
<p>(t, b, z)</p></td>
<td>Hydrodynamic sources, 1</td>
<td>A number of fluxes (number connections) into a given box – from other boxes and from itself</td>
</tr>
<tr class="odd">
<td>hdsink (t, b, z)</td>
<td>Hydrodynamic sinks, 1</td>
<td>A number of fluxes (number connections) from a given box – to other boxes and to itself</td>
</tr>
<tr class="even">
<td>eflux (t, b, z)</td>
<td>Hydrodynamic net horizontal exchanges, 1</td>
<td>Total inflow or outflow (positive/negative value) of water into the given cell in the horizontal direction</td>
</tr>
<tr class="odd">
<td>vflux (t, b, z)</td>
<td>Hydrodynamic net vertical exchanges, 1</td>
<td>Total inflow or outflow (positive/negative value) of water into the given cell in the vertical direction</td>
</tr>
<tr class="even">
<td>porosity (t, b, z)</td>
<td>Porosity of the sediment layer, 1</td>
<td>Fractional porosity of the sediment (from 0 to 1). Initial values are set in the <em>initial_conditions.nc</em>, but the property is dynamically calculated after (optional) settling of particulate tracers</td>
</tr>
<tr class="odd">
<td>nominal_dz, (b, z)</td>
<td>Thickness of a water layer, m</td>
<td>Initial or nominal depth of water column and sediment layers in each box. This is <strong>not a dynamic variable</strong>, as it does not have the time dimension. For all but the bottom water column layers the nominal depth is usually a round number (1 to a few hundred meters). The depth of the bottom water column layer varies depending on the terrain of the actual ecosystem (see Fig. 4).</td>
</tr>
<tr class="even">
<td>numlayers (t, b)</td>
<td>Number of layers</td>
<td>Number of layers in each box as set by the user. The number of layers is set in the <em>initial_conditions.nc</em> and currently remains stable through time. New functionality in the future might allow the number of water column layers to change due to large tides.</td>
</tr>
<tr class="odd">
<td>dz (t, b, z)</td>
<td>Thickness of a water layer, m</td>
<td>Dynamic thickness of the water layer, reflecting changes in its volume due to water fluxes (see volume variable above)</td>
</tr>
<tr class="even">
<td>topk (t, b)</td>
<td>Sediment top index</td>
<td>The sediment layer that has the interface with the water. In current models this is always 0, but the variable was used in the PPBIM when multiple dynamic sediment layers were modelled</td>
</tr>
<tr class="odd">
<td>sedbiodepth (t, b)</td>
<td>Depth of biological activity, m</td>
<td>Maximum depth of biological activity in the sediments. The shape of the profile of biological activity with depth is set in the biosedprofile in <em>physics.prm</em>. The actual depth of the activity will change depending on the amount of activity present and the oxygen levels</td>
</tr>
<tr class="even">
<td>seddetdepth (t, b)</td>
<td>Max depth of detritus, m</td>
<td>Depth of detritus in the sediment layer. It determines the depth of the habitat suitable for the infauna and changes through time due to sediment processes. It is dynamically calculated in <em>Bio_Box_Process()</em> routine in <strong><u>atbiology.c</u></strong>.</td>
</tr>
<tr class="odd">
<td>sedoxdepth (t, b)</td>
<td>Depth of oxygen horizon, m</td>
<td>Depth of the oxygenated sediment layer. The initial values are set in the <em>intial_condition.nc</em>, but the actual oxygen layer depth is dynamic and depends on the amount of oxygen in the water column and sediments, and amount of biological activity.</td>
</tr>
<tr class="even">
<td>sedbiodens (t, b)</td>
<td>Biological activity, Animals m<sup>2</sup></td>
<td>The “background density” of biological activity that is always present in the sediment. It is set by the user and can vary among boxes to reflect different quality of the sediment (see description of bioirrigation and bioturbation in chapter 5.2). Even though this variable has time dimension, it does not currently change through time</td>
</tr>
<tr class="odd">
<td>sedirrigenh (t, b)</td>
<td>Bioirrigation enhancement, 1</td>
<td>Scalar of bioirrigation enhancement due to action of infauna species (marked with _INF in the GroupType of the <em>functional_groups.csv</em> file). It is calculated in the <em>Irrig_and_Turb()</em> routine in <strong><u>atprocess.c</u></strong> in <strong>atecology</strong> library</td>
</tr>
<tr class="even">
<td>sedturbenh (t, b)</td>
<td>Bioturbation enhancement, 1</td>
<td>Scalar of bioturbation enhancement due to action of infauna species (marked with _INF in the GroupType of the <em>functional_groups.csv</em> file). It is calculated in the <em>Irrig_and_Turb()</em> routine in <strong><u>atprocess.c</u></strong> in <strong>atecology</strong> library</td>
</tr>
<tr class="odd">
<td>erosion_rate (t, b)</td>
<td>Erosion rate m/s</td>
<td>Dynamic erosion rate variable, which depends on the supplied bottom stress values and the topk variable</td>
</tr>
<tr class="even">
<td>eddy (t, b)</td>
<td>Eddy strength, 1</td>
<td>Eddy strength in each box at each time step, depending on the eddy seasonal values and scalars (see vertical mixing routine description in chapter 5.2)</td>
</tr>
</tbody>
</table>
</section>
<section id="importance-of-optional-physics-routines" class="level2">
<h2 class="anchored" data-anchor-id="importance-of-optional-physics-routines"><strong>5.7. Importance of optional physics routines</strong></h2>
<p>During the model development stage many people focus their attention on the parameters defining biological routines, while the <em>physics.prm</em> file is rarely touched. However, a number of optional routines included in the Physics submodel may have a large effect on the nutrient dynamics – bioirrigation, bioturbation, settling of particulate matter. This in turn is likely to have a large effect on biological groups and especially lower trophic levels. The example below is a quick analysis on the effect of turning some Physics routines off on total biomasses of functional groups in a Baltic Sea model. The model is still under development and is not presented in detail here.</p>
<p><img src="markdown_output/media/image32.jpg" style="width:6.69306in;height:5.08194in"></p>
<p><span id="_Toc526762789" class="anchor"></span>Figure 9. Total final biomass ratios of all Baltic Sea model functional groups in scenarios exploring the effects of optional physics routines. The simulations were run for 20 years.</p>
<p>Separate analysis was also done where six parameters in the <em>physics.prm</em> file defining bioturbation and bioirrigation intensity (typically in the range of 1e-7 to 1e-8) were increased 1000 times. The result is not shown but the effect was very minor, for most groups it made less than 5% difference in the final biomass compared to the scenario using same the physics routines but with default parameter values. Note, that the bioturbation routine is not included here, because the Baltic Sea model has only one sediment layer and hence the routine cannot be performed (setting bioturbation flag to 1 or 0 did not have any effect on biomasses).</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>Good practice tip 6</strong></p>
<p>It is recommended that while developing the model, users conduct a range of simulations with some of the optional routines turned on or off and explore different ways that physical and biological process might affect nutrient turnover.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</section>
<section id="checking-hydrodynamic-transports" class="level2">
<h2 class="anchored" data-anchor-id="checking-hydrodynamic-transports"><strong>5.8. Checking hydrodynamic transports</strong></h2>
<p>In some instances you may want to track whether the transports are matching current patterns. To use it, update the code and</p>
<ol type="1">
<li>In run.prm add</li>
</ol>
<p>flagpassivetracer &nbsp;&nbsp;&nbsp;&nbsp;1&nbsp;&nbsp;&nbsp; # Flag indicating whether a passive tracer is needed for tracking tranpsorts or not (1 = yes, 0 = no)</p>
<ol start="2" type="1">
<li>In the in.nc file you need to add the tracer. In the header (text section at the start of the file) add the following</li>
</ol>
<p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; double SED(t, b, z) ;</p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; SED:bmtype = "tracer" ;</p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; SED:units = "mg N m-3" ;</p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; SED:long_name = "Passive tracer" ;</p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; SED:sumtype = 1 ;</p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; SED:dtype = 0 ;</p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; SED:inwc = 1 ;</p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; SED:insed = 1 ;</p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; SED:dissol = 0 ;</p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; SED:decay = 0. ;</p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; SED:partic = 1 ;</p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; SED:passive = 1 ;</p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; SED:svel = 0. ;</p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; SED:xvel = 0. ;</p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; SED:psize = 1.e-06 ;</p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; SED:b_dens = 1000000000. ;</p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; SED:i_conc = 200000000. ;</p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; SED:f_conc = 200000000. ;</p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; SED:isabs = 0 ;</p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; SED:iscontam = 0 ;</p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; SED:_FillValue = 0. ;</p>
<p>&nbsp;</p>
<p>Then in the data section of the file you need to add</p>
<blockquote class="blockquote">
<p>SED =</p>
<p>&nbsp; N, _, _,;</p>
</blockquote>
<p>Where N is the tracer doping you want to use if you want it to be there at the start of the model. You could also introduce it later on as a point source. Let me know if you want to do that (its like adding a river inputs so really straight forward).</p>
<p>As to the _ you need as many data entries in total as there are layers (water column + sediment layers) * number of boxes in the model (lets call that value “LxB”. Now the N needs one entry so the total number of _ you need is (LxB -1)</p>
<p>So for a model with 3 layers in 4 boxes where N is entering in box 1 you would have</p>
<blockquote class="blockquote">
<p>SED =</p>
<p>&nbsp; _, _, _,&nbsp;</p>
<p>&nbsp;&nbsp;N, _, _,</p>
<p>&nbsp; _, _, _,</p>
<p>&nbsp; _, _, _ ;</p>
</blockquote>
<p>At present it really only makes sense to dope one location at a time with the N otherwise you won’t know where things have come from as you can’t have more than one SED variable at a time. If multiple tracer types are required please contact the developers.</p>
</section>
</section>
<section id="definitions-of-functional-groups" class="level1">
<h1>6. DEFINITIONS OF FUNCTIONAL GROUPS</h1>
<section id="introduction-to-functional-groups" class="level2">
<h2 class="anchored" data-anchor-id="introduction-to-functional-groups"><strong>6.1. Introduction to functional groups</strong></h2>
<p><strong><em>Types of functional groups</em></strong></p>
<p>Biological components are represented as <strong>biomass pools</strong> (most non-vertebrates), <strong>age-structured biomass pools</strong> (some invertebrates) or <strong>age-structured groups</strong> (vertebrates). These three group types are assigned according to the GroupType and NumCohorts parameters given in the <em>functional_groups.csv</em> file (Table 8). Group types FISH, FISH_INVERT, SHARK, MAMMAL, BIRD, REPTILE are assigned to the fully age-structured groups with the number of age groups given by NumCohorts. Other groups are assigned to biomass pools. If NumCohorts for a biomass pool is &gt;1 they will be assigned to an age-structured biomass pool. The age-structured biomass pools are similar to stanzas used in, for example, Ecosim and are used to allow for different juvenile and adult behaviour and parameters. All non-vertebrates that have NumCohorts=1 are assigned to simple biomass pools.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Can a non-vertebrate be modelled as an age-structured group?</strong></p>
<p>Six GroupTypes (FISH, FISH_INVERT, MAMMAL, SHARK, BIRD and REPTILE) are currently modelled as age-structured groups. If full age-structure is needed for a non-vertebrate, it should be set as one of these groups, most likely as FISH_INVERT or FISH. FISH_INVERT currently uses the same code as FISH, but the intent is to use it to represent a size rather than age-based model, which is a more typical representation for key invertebrates (such as exploited species). Note however that setting a group as FISH_INVERT or FISH would not call typical non-vertebrate routines (see below).</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>Atlantis does not explicitly model sex, but represents an average individual. This means that all individuals in a biomass pool and an age cohort (or a genotype in a cohort, if multiple genotypes are modelled) are considered identical in terms of their reproductive output. This may require some caution in parameterisation, e.g.&nbsp;number of offspring per mammal individual.</p>
<p><strong><em>State variables tracked</em></strong></p>
<p><em>Main model</em></p>
<p>In models that are based on N only (i.e.&nbsp;do not explicitly track P and C) the only state variable tracked for most biomass pools is N, but both N and Si fluxes are tracked for Si limited biomass pools, such as diatoms. The Si limited groups are identified from the <em>functional_groups.csv</em> file parameter IsSiliconDep (see below). In age-structured groups three variables are tracked for each age cohort: reserve N (RN), structural N (SN) and numbers (Num). In addition, fluxes in the water column and sediments are tracked through eight <strong>inanimate pools</strong>: ammonia (NH), nitrate (NO), dissolved silica (Si), detrital silica (DSi), dissolved oxygen (O2), dissolved organic nitrogen (DON), labile detritus (DL), and refractory detritus (DR). An additional inanimate pool, carrion (DC), is also required, though it is only dynamically used if the fisheries submodel is active; DC is then used to represent fisheries discards. State variables tracked for each group defined with the GroupType parameter in the <em>functional_groups.csv</em> file are listed in Table 8.</p>
<p><em>Extended models</em></p>
<p>If the model also includes tracking of P and C (trackAtomicRatio is set to 1 in <em>run.prm</em>), then P and C pools are tracked for each functional group, together with N pools. Additional inanimate pools are for C and P are also required. If the model is of an estuary (FlagIsEstuary=1 in <em>run.prm</em> file) a sediment (SED) inanimate pool is added for the water column. If the model tracks contaminants (track_contaminants=1 in <em>run.prm</em>) a separate pool is added for each contaminant for each nitrogen pool tracked, so that concentration of contaminants can be tracked throughout the entire food web. If the model spans polar regions (flagIsPolar=1 in <em>run.prm</em>) the model allows for 24 hours of night/daylight seasonally.</p>
<p><strong><em>Allocation of pools throughout the vertical dimension: water column, epibenthos and sediment</em></strong></p>
<p>Atlantis models 3-5 explicit distribution types of organisms and inanimate pools through the model domain – <strong>water column</strong>, <strong>epibenthic layer</strong> and <strong>sediments</strong> are tracked in all models and <strong>ice</strong> and <strong>land</strong> may also optionally be included. The water column and <strong>sediment</strong> are modelled as three dimensional spaces (units of variables are in m<sup>3</sup>), whereas the epibenthic layer is two-dimensional (units in m<sup>2</sup>). Most organisms are restricted to one type, such as sediments (infaunal invertebrates identified with INF in GroupType name), epibenthic layer (epibenthic organisms, identified with EP in the GroupType name) or water column (fully age-structured groups). Some of the smaller groups (e.g.&nbsp;microflora primary producers) and the inanimate pools can exist in both the sediments and water column (Table 8).</p>
<p>In the Atlantis code the water column, epibenthic, sediment, ice and land distributionns are referred to as “habitatType”. However, remember that they are not the same as the ecological epibenthic habitats modelled in Atlantis (e.g.&nbsp;seagrass, corals, sponges etc.).</p>
<table class="caption-top table">
<caption><span id="_Toc526762760" class="anchor"></span>Table 8. State variables tracked in the main Atlantis model (no ice or land included). * indicates optional Si tracking in primary producers that are limited by Si. Primary producer Si limitation is set in <em>functional_groups.csv</em> file parameter IsSiliconDep (see Table 9). In models that include C and P, both C and P are tracked like N (so the number of variable tracked is three times larger).</caption>
<thead>
<tr class="header">
<th><strong>Group</strong></th>
<th><strong>State variables tracked</strong></th>
<th><strong>GroupType in CSV file or <em>initial_conditions.nc</em></strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong><em>Biomass pools</em></strong></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>pelagic <strong>bacteria</strong></td>
<td>N in water column</td>
<td>PL_BACT</td>
</tr>
<tr class="odd">
<td>sediment <strong>bacteria</strong></td>
<td>N in sediments</td>
<td>SED_BACT</td>
</tr>
<tr class="even">
<td><p><strong><em>(Age-structured) biomass pools</em></strong></p>
<p>These groups can be age structured if NumCohorts set &gt; 1</p></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>non epibenthic <strong>primary producers</strong></td>
<td><p>N in water column</p>
<p>N in sediments</p>
<p>*Si in water column</p>
<p>*Si in sediments</p>
<p>NOTE: Age-structure not recommended</p></td>
<td><p>SM_PHY</p>
<p>LG_PHY</p>
<p>MICROPHTYBENTHOS</p>
<p>DINOFLAG</p></td>
</tr>
<tr class="even">
<td>epibenthic <strong>primary producers</strong></td>
<td>N in epibenthic layer for each age group defined by NumCohorts parameter. If age-structure is selected for SEAGRASS it is not really age structure, but different parts of the plant – with “age 0” assumed to be the pool representing leaves, “age 1” the roots, and “age 2” the epiphytes on the leaves, see details <a href="https://confluence.csiro.au/display/Atlantis/2012/12/12/new+seagrass+modelling+option">here</a></td>
<td><p>PHYTOBEN</p>
<p>SEAGRASS</p></td>
</tr>
<tr class="odd">
<td>infaunal <strong>invertebrates</strong></td>
<td><p>N in sediments for each age group defined by the NumCohorts parameter.</p>
<p>NOTE: Not typically age-structured</p></td>
<td><p>SM_INF</p>
<p>LG_INF</p></td>
</tr>
<tr class="even">
<td>epibenthic <strong>invertebrates</strong></td>
<td>N in epibenthic layer for each age group defined by the NumCohorts parameter</td>
<td><p>SED_EP_FF</p>
<p>SED_EP_OTHER</p>
<p>MOB_EP_OTHER</p>
<p>CORAL</p>
<p>SPONGE</p></td>
</tr>
<tr class="odd">
<td>water column <strong>invertebrates</strong></td>
<td>N in water column for each age group defined by the NumCohorts parameter</td>
<td><p>SM_ZOO</p>
<p>MED_ZOO</p>
<p>LG_ZOO</p></td>
</tr>
<tr class="even">
<td>land <strong>plants</strong></td>
<td>N on land for each age group defined by the NumCohorts parameter</td>
<td><p>MARSH</p>
<p>MANGROVE</p></td>
</tr>
<tr class="odd">
<td>large invertebrates</td>
<td>N in water column for each age group defined by NumCohorts parameter</td>
<td><p>CEP</p>
<p>PWN</p></td>
</tr>
<tr class="even">
<td><strong><em>Age-structured groups</em></strong></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>vertebrates</td>
<td><p>RN in water column</p>
<p>SN in water column</p>
<p>Num in water column</p>
<p>for each cohort defined in</p>
<p>NumCohorts parameter</p></td>
<td><p>FISH</p>
<p>SHARK</p>
<p>MAMMAL</p>
<p>BIRD</p>
<p>REPTILE</p></td>
</tr>
<tr class="even">
<td>size/age structured invertebrates of particular interest</td>
<td><p>RN in water column</p>
<p>SN in water column</p>
<p>Num in water column</p>
<p>for each cohort defined in</p>
<p>NumCohorts parameter</p></td>
<td>FISH_INVERT</td>
</tr>
<tr class="odd">
<td><strong><em>Inanimate pools</em></strong></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Ammonia</td>
<td><p>NH3 in water column</p>
<p>NH3 in sediments</p></td>
<td>NH</td>
</tr>
<tr class="odd">
<td>Nitrate</td>
<td><p>NO3 in water column</p>
<p>NO3 in sediments</p></td>
<td>NO</td>
</tr>
<tr class="even">
<td>Oxygen</td>
<td><p>O2 in water column</p>
<p>O2 in sediments</p></td>
<td>O2</td>
</tr>
<tr class="odd">
<td>Dissolved organic nitrogen</td>
<td><p>DON in water column</p>
<p>DON in sediments</p></td>
<td>DON</td>
</tr>
<tr class="even">
<td>Dissolved silica</td>
<td>Si in water column</td>
<td>Si</td>
</tr>
<tr class="odd">
<td>Detrital silica</td>
<td>DSi in water column</td>
<td>DSi</td>
</tr>
<tr class="even">
<td>Labile detritus</td>
<td><p>N in water column</p>
<p>N in sediments</p></td>
<td>LAB_DET</td>
</tr>
<tr class="odd">
<td>Refractory detritus</td>
<td><p>N in water column</p>
<p>N in sediments</p></td>
<td>REF_DET</td>
</tr>
<tr class="even">
<td>Carrion (only dynamically tracked if fisheries are active)</td>
<td><p>N in water column</p>
<p>N in sediments</p></td>
<td>CARRION</td>
</tr>
<tr class="odd">
<td>Sediment (only if flagIsEstuary=1)</td>
<td>SED in water column</td>
<td>SED</td>
</tr>
</tbody>
</table>
<p><strong><em>Timestep of the model</em></strong></p>
<p>The timestep at which most processes are repeated is typically an <em>adaptive</em> daily (24h) or diurnal (12h) timestep, though tidal models with shorter timesteps (3h or 6h) also exist. This timestep is specified as dt in the <em>run.prm</em> file. Reproduction, recruitment and migration occur less frequently, typically once per year.</p>
<p>The <em>adaptive timestep</em> means that for most of the groups the processes are repeated once per timestep, but for small organisms, such as bacteria and phytoplankton the timestep is reduced to ensure that fluxes into and out of the component with the fastest turnover rates remain stable and numerical artifacts are not introduced into the mathematical solution scheme. The length of the adaptive timestep for such groups is determined to ensure that the relative change (flux) in any variable does not exceed a pre-specified proportion of the standing stock – this tolerance is set in the <em>biology.prm</em> file. The adaptive timestep changes through time, depending on the standing stock of these groups and the temperature, nutrient availability and other factors that affect the turnover rates in these “fast” variables.</p>
<p><strong><em>Biological processes modelled</em></strong></p>
<p>The modelled biological processes include primary production, bacterial processes, feeding, assimilation, waste production, explicit respiration (optional and not used in most models), growth, movement and migration, non-predation mortality, maturation, reproduction and recruitment. In biomass pools all processes are modelled through biomass (nitrogen) turnover. In age-structured groups, the state variables are numbers-at-age and weight-at-age (tracked as reserve and structural nitrogen) which can be converted to biomass.</p>
</section>
<section id="defining-functional-groups" class="level2">
<h2 class="anchored" data-anchor-id="defining-functional-groups"><strong>6.2. Defining functional groups</strong></h2>
<p>The functional group input file sets all the biological groups in the model. It can be set in CSV or XML format (see examples <a href="https://confluence.csiro.au/display/Atlantis/Functional+Group+Definition+File">here</a>). The number of groups is entirely flexible, but each model must include three obligatory detritus groups: carrion (CARRION), labile detritus (LAB_DET) and refractory detritus (REF_DET).</p>
<table class="caption-top table">
<tbody>
<tr class="odd">
<td style="text-align: left;">The <em>functional_groups.csv</em> file defines the biological groups used in the model and some of their key characteristics. These characteristics are used throughout the code to apply different appropriate routines for the group. The number of groups in the <em>functional_groups.csv</em> file must match the number given in the K_num_tot_sp parameter in the <em>run.prm</em> file. The detritus groups must be defined as the last three groups in the file.</td>
</tr>
</tbody>
</table>
<hr>
<p>Currently each group must have the following parameters (characteristics). For new updates, check the Atlantis wiki <a href="https://confluence.csiro.au/display/Atlantis/Functional+Group+Definition+File">functional group page</a></p>
<table class="caption-top table">
<caption><span id="_Toc526762761" class="anchor"></span>Table 9. Parameters in the <em>functional_groups.csv</em> file.</caption>
<thead>
<tr class="header">
<th><strong>Parameter name</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>GroupCode</td>
<td><u>2 or 3 letter code</u> of upper case letters used to identify each functional group. Atlantis will display an error message and quit if the group codes are not the correct format. This code is very important and applied to all functional group specific parameters</td>
</tr>
<tr class="even">
<td>Index</td>
<td>An index of the group. Atlantis doesn't actually read this in but it is helpful for modellers to see as it defines the order of read-in for species vectors and the group ID number if you are trying to activate some of the debug/calibration fprintf statements using whichcheck in the <em>run.prm</em> file. The index must start form 0, as this is how the C programing language counts</td>
</tr>
<tr class="odd">
<td>IsTurnedOn</td>
<td><p>A boolean value (either 0 (false) or 1 (true)) to indicate if the group is active in the current model run. If the group is not active it will be excluded from the model.</p>
<p>This option allows the user to exclude some groups from the model without having to setup new initial conditions and parameter files.</p></td>
</tr>
<tr class="even">
<td>Name</td>
<td>Name of a group. This must match the tracer names in the <em>initial_conditions.nc</em> file as it used to find the tracers in the <em>initial_conditions.nc</em> input file.</td>
</tr>
<tr class="odd">
<td>LongName</td>
<td>The long name written to the NetCDF output files. This name can be a few words long and ideally should be informative. This is especially important when sharing the model with others.</td>
</tr>
<tr class="even">
<td>NumCohorts</td>
<td><p>The number of cohorts in a functional group. If the value is &gt; 1 the groups is assumed to be age-structured (either an age-structured biomass pool or age-structured group).</p>
<p>In the earlier Atlantis versions age structured groups all had to have 10 cohorts. This is now relaxed and there is no upper limit on the number of age-groups to model (but remember, the more age groups the slower the model will run).</p></td>
</tr>
<tr class="odd">
<td>NumGeneTypes</td>
<td>Number of discrete phenotypes in each age group. This option is used when simulating evolution or multiple size-at-age representations and should only be set to &gt;1 if flag_do_evolution is set to 1 in <em>biology.prm</em> file. If more than one phenotype is present, initial conditions will be have to be set up for each phenotype.</td>
</tr>
<tr class="even">
<td>NumStages</td>
<td>This parameter describes the number of separate stages in a functional group, which are set to 1 or 2 if juvenile and adult stages have separate parameters for distribution, mortality, migration, feeding and others. In principle, more than 2 stages are possible, although this has not been tested. Typically all age-structured groups have 2 stages, but age-structured biomass pools can also have separate parameters for juvenile and adult stages and thus have 2 stages.</td>
</tr>
<tr class="odd">
<td>NumSpawns</td>
<td>Number of spawns in the year. Typically one per year, but more than one is possible (especially for ages structured biomass pool groups). The timing of spawn is given in Time_Spawn_XXX parameter in <em>biology.prm</em> file (if NumSpawns is 1 it expects a single value, otherwise it expects a vector of length NumSpawns).</td>
</tr>
<tr class="even">
<td>NumAgeClassSize</td>
<td>Number of calendar years in one age group. This can be set to be anything between a few months to many years and allows simulation of different longevity organisms without having to setup too many age groups (which reduces simulation speed). Note, all individuals in an age group are considered identical (unless different phenotypes are set in NumGeneTypes, in which case all individuals in one phenotype of one age group are identical). However, the proportion of an age group per annual cohort is tracked so that recruitment variation is represented through time and not homogenized away (see chapter 10.9.6).</td>
</tr>
<tr class="odd">
<td>NumStocks</td>
<td><p>This sets the number of geographically distinct stocks for age-structured groups and age-structured biomass pools. The stocks must occupy different boxes or layers, i.e.&nbsp;they do not spatially overlap.</p>
<p>Some biological parameters, like vulnerability to prey can be set as stock specific, which allows for better tuning. It has been applied to some groups that are known to include ecologically distinct stocks in different geographic areas. If NumStocks &gt;1 then the corresponding number of stock specific parameters will be required in the <em>biology.prm</em> input file</p></td>
</tr>
<tr class="even">
<td>VerticallyMigrates</td>
<td>If true then the group can move vertically through the water column. This should be set to 1 except for debugging purposes. If this is set to 0, no vertical movement will happen even if it is specified in the <em>biology.prm</em> file and the vertical distribution of individuals in the <em>initial_conditions.nc</em> file will be retained throughout the simulation.</td>
</tr>
<tr class="odd">
<td>HorizontallyMigrates</td>
<td>If true then the group can move horizontally throughout the model as well as migrate out of the model. No migration will happen if this is set to 0. Biomass pool groups which are marked as passive in the <em>initial_conditions.nc</em> will still be advected even if HorizontallyMigrates is set to 0.</td>
</tr>
<tr class="even">
<td>NumMigrations</td>
<td>The number of migrations per calendar year</td>
</tr>
<tr class="odd">
<td>MultiYrMigrations</td>
<td>Whether any of the migrations are multi-year (1) or simply annual (0). While this column is always required in the groups definition file the associated code will not function unless flag_multiyr_migs is also set to 1 in the <em>run.prm</em> file</td>
</tr>
<tr class="even">
<td>ExternalReproduction</td>
<td>This is whether the group reproduces outside the model (i.e.&nbsp;the value for Time_Spawn_XXX parameter in <em>biology.prm</em> file is set during a day of the year when the species should be outside the model)</td>
</tr>
<tr class="odd">
<td>RecruitType</td>
<td><p>This is where you set the recruitment typ:</p>
<p>0 = independent distribution (use recruit_hdistrib)</p>
<p>1 = at parental locations</p>
<p>2 = external (into Migration Array)</p>
<p>3 = using a forced larval dispersal matrix</p>
<p>4 = external reproducer who has parental care</p></td>
</tr>
<tr class="even">
<td>IsFished</td>
<td>If set to 1 the group can be fished (targeted) by commercial or recreational fisheries and the catch will be reported in the output files, and is analysed in the assessment model. Note that if set to 0, dynamic fishing will still occur if parameters are given in the <em>harvest.prm</em> file.</td>
</tr>
<tr class="odd">
<td>IsImpacted</td>
<td><p>If true then the group can be impacted by fishing, bycatch or incidental fishing related mortality (e.g.&nbsp;destruction of living habitats). This parameter differentiates between targeted and accidental interactions; it is not reported in the landings. This means there must be values in the harvest files for this group. Note the group can be impacted by bycatch/incidental mortality even if no explicit targeted fishing is setup. This means the bycatch/incidental mortality parameters should be assessed carefully in the <em>harvest.prm</em> file.</p>
<p>If both this value and IsFished are set to 0 then all fisheries related impacts on the group are disabled.</p></td>
</tr>
<tr class="even">
<td>isTAC</td>
<td>If true, the group will be under total allowable catch (TAC) management. It will activate specific TAC related code in the Harvest submodel.</td>
</tr>
<tr class="odd">
<td>GroupType</td>
<td><p>The type of species. This is one of the most important parameters, as it tells Atlantis what kind of organism the functional group is representing and what kind of processes should be applied to it. Six group types are allowed for age-structured groups. A lot more group types are available to define invertebrate biomass pool types and these group types are used to dictate the adaptive timestep for simulating proceses, and all of them have some specialized ecological routines applied (see some details about their characteristics <a href="https://confluence.csiro.au/display/Atlantis/AtlantisGroupTypes">here</a>).</p>
<p>The main Atlantis model groups include:</p>
<p><em>AGE STRUCTURED GROUPS</em></p>
<ol type="1">
<li><p>Standard fin-fish or generic vertebrate FISH</p></li>
<li><p>Size/age-structured invertebrate FISH_INVERT</p></li>
<li><p>Chondrichthyans SHARK</p></li>
<li><p>Birds BIRD</p></li>
<li><p>Marine mammanls (potentially terrestrial mammals in the future) MAMMAL</p></li>
<li><p>Reptiles (can be present in the water and on land) REPTILE.</p></li>
</ol>
<p><em>BIOMASS POOL GROUPS</em></p>
<p><strong>Phytoplankton</strong></p>
<ol start="7" type="1">
<li><p>Small Phytoplankton SM_PHY</p></li>
<li><p>Large Phytoplankton LG_PHY (used in combination with IsSiliconDep to define diatoms)</p></li>
<li><p>Trichodesmium and Cyanobacteria TRICHO (N fixers)</p></li>
</ol>
<p><strong>Other primary producers</strong></p>
<ol start="10" type="1">
<li><p>Microphtybenthos MICROPHTYBENTHOS</p></li>
<li><p>Dinoflagellates DINOFLAG</p></li>
<li><p>Phytoben PHYTOBEN (typically used to represent macroalgae)</p></li>
<li><p>Seagrass SEAGRASS</p></li>
<li><p>Turf TURF</p></li>
</ol>
<p><strong>Zooplankton</strong></p>
<ol start="15" type="1">
<li><p>Small Zooplankton SM_ZOO</p></li>
<li><p>Medium Zooplankton MED_ZOO</p></li>
<li><p>Large Zooplankton LG_ZOO</p></li>
<li><p>Jellyfish JELLIES (in the past represented as LG_ZOO)</p></li>
</ol>
<p><strong>Large pelagic invetebrates</strong></p>
<ol start="19" type="1">
<li><p>Cephalopod CEP</p></li>
<li><p>Prawns PWN</p></li>
</ol>
<p><strong>Infauna</strong></p>
<ol start="21" type="1">
<li><p>Small Infauna SM_INF</p></li>
<li><p>Large Infauna LG_INF</p></li>
</ol>
<p><strong>Epibenthic organisms</strong></p>
<ol start="23" type="1">
<li><p>Sediment epibenthic filter feeders SED_EP_FF (often used for bivalves, sponges)</p></li>
<li><p>Benthic grazers SED_EP_OTHER</p></li>
<li><p>Mobile epibenthos MOB_EP_OTHER (often used for crabs, lobster, octopus)</p></li>
<li><p>Corals CORAL</p></li>
<li><p>Sponges SPONGE</p></li>
</ol>
<p><strong>Bacteria</strong></p>
<ol start="28" type="1">
<li><p>Pelagic Bacteria PL_BACT</p></li>
<li><p>Sediment Bacteria SED_BACT</p></li>
</ol>
<p><strong>Obligatory detritus groups</strong></p>
<ol start="30" type="1">
<li><p>Carrion CARRION</p></li>
<li><p>Labile detritus LAB_DET</p></li>
<li><p>Refractory detritus REF_DET</p></li>
<li><p>Additional ice and land based groups are also available.</p></li>
</ol>
<p><strong>Ice dwelling</strong></p>
<ol start="34" type="1">
<li><p>ICE_BACT</p></li>
<li><p>ICE_MIXOTROPHS</p></li>
<li><p>ICE_DIATOMS</p></li>
<li><p>ICE_ZOOBIOTA</p></li>
</ol>
<p><strong>Land dwelling</strong> (vegetation only for now)</p>
<ol start="38" type="1">
<li><p>MARSH</p></li>
<li><p>MANGROVE</p></li>
</ol></td>
</tr>
<tr class="even">
<td>IsPredator</td>
<td>If true then the group is assumed to eat other groups by the assessment code (for the purposes of defining network indices etc), it does not affect the ecological model.</td>
</tr>
<tr class="odd">
<td>IsCover</td>
<td><p>If true the group is classified as a habitat. This means that the group becomes a unique habitat type, that can be used by other groups in the model for shelter etc, in addition to the general four geological habitat types included by default (<strong>reef, flat, soft, canyon).</strong></p>
<p>For example, if IsCover is set to 1 for three functional groups (e.g.&nbsp;BFS, BFF, and MA), the model has a total of seven habitat types. In this case the habitat dependency parameter habitat_XXX in the <em>biology.prm</em> file must have 7 values for each functional group. The living habitats are always the first entries and the geological classes always the last 4 entries. So in this example, the first three entries will refer to the IsCover groups in the order they are listed in the <em>functional_groups.csv</em> file and the last four entries will refer to reef, flat, soft, canyon habitats.</p></td>
</tr>
<tr class="even">
<td>IsSiliconDep</td>
<td>If true, the group is assumed to be Si limited and its energy flows are tracked both through nitrogen and silicon. All Si limited groups must have separate Si tracers set in the <em>initial_conditions.nc</em>, in addition to the usual Nitrogen tracers. At present most models have this set to true for Diatoms and MicroPhytopbenthos.</td>
</tr>
<tr class="odd">
<td>IsAssessed</td>
<td>If true then this group will be assessed in the Assessment submodel. This submodel will generate a number of assessment indices.</td>
</tr>
<tr class="even">
<td>IsCatchGrazer</td>
<td>If true, the group grazes on catch, for instance pinnipeds that eat fish from nets. If set to 1 for a functional group XXX, then availability of catch of each functional group should be given for this group in the pFCXXX parameter in the <em>biology.prm</em> file. The pFC parameters are not required for groups that are not catch grazers.</td>
</tr>
<tr class="odd">
<td>OverWinters</td>
<td>If true, this group will go into hiatus over winter. The overwintering can be triggered and ended by time of the year or temperature. This is set using overwinterStartTofY_XXX, overwinterEndTofY_XXX, overwinterStartTemp_XXX, overwinterEndTemp_XXX parameters in the <em>biology.prm</em> file. These parameters only have to be set for overwintering groups.</td>
</tr>
<tr class="even">
<td>isCultured</td>
<td>This identifies aquaculture groups, and applies aquaculture processes to them. Parameters required for aquaculture groups are described <a href="https://confluence.csiro.au/display/Atlantis/2014/03/06/Aquaculture">here</a>.</td>
</tr>
<tr class="odd">
<td>isHabDepend</td>
<td>Identifies groups that are habitat dependent. Note, this value takes precedence over habitat preference parameters in the <em>biology.prm</em> file. If isHabDepend is set to 0 then the group is not affected by habitats, even if appropriate parameters are given in the <em>biology.prm</em></td>
</tr>
<tr class="even">
<td>numMoveEntries</td>
<td>The number of movement time periods you would like the group to have. Historically this has been fixed at 4 to allow for seasonal movement, but currently any number of moves is allowed. See further details on parameters in <em>biology.prm</em> <a href="https://confluence.csiro.au/pages/viewpage.action?pageId=346916326">here</a></td>
</tr>
<tr class="odd">
<td>isBioEroder</td>
<td>Whether the group bioerodes corals (i.e.&nbsp;sponges). This allows for boring sponges and bioeroder activities in reef models including corals and sponges</td>
</tr>
<tr class="even">
<td>isSupplemented</td>
<td>Whether the group feeds on material from outside the model. If this is set to 1, the group will be assumed to be eating insects or pellets or something from outside the model domain. The proportion of their diet gained in this way (per box) must be set using the Supp_XXX in the <em>biology.prm</em> file.</td>
</tr>
<tr class="odd">
<td>isExternal</td>
<td>Flag to indicate whether the group uses the external population model option (instead of simply holding individuals in hiatus when they have migrated out of the model).</td>
</tr>
<tr class="even">
<td>isLandActive</td>
<td>Marks the group as able to be on land (for models with land and sea). Only required if flagAllowLand is set to 1 in the <em>run.prm</em> file.</td>
</tr>
<tr class="odd">
<td>isLightEffected</td>
<td>Marks the group as susceptible to light pollution. Only required if flag_pollutant_impacts is set to 1 in the <em>run.prm</em> file.</td>
</tr>
<tr class="even">
<td>isNoiseEffected</td>
<td>Marks the group as susceptible to noise pollution. Only required if flag_pollutant_impacts is set to 1 in the <em>run.prm</em> file.</td>
</tr>
</tbody>
</table>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>What is the role of detritus in an ecosystem model? An excerpt from Murray and Parslow (1997) PPBIM report:</strong></p>
<p>“The detrital pool acts as a store of fixed nutrient, which introduces a delay into nutrient recycling. Particulate detritus sinks and thereby transports nutrients out of the surface layer into deep water or to the sediment. Dissolved organic matter (DOM) does not sink, and very large pools of refractory DOM are found in the open ocean and in coastal waters. Horizontal transport of particulate detritus and DOM from regions of high production to more oligotrophic areas can also be important”</p>
<p>“Three forms of non-living organic matter are included in the model: labile particulate detritus, refractory particulate detritus and dissolved organic matter [NOTE – these same pools, plus Carrion, are included in Atlantis, which is based on the PPBIM]. These are represented by the symbols DL, DR and DON respectively. The particulate detrital pool [in nature] contains a wide variety of particles, with diverse sources, ages, sizes and chemical composition. The model includes two functional components: fresh labile detritus breaking down on time scales of a few days, and semi-refractory detritus turning over on time scales of a year. This allows the model to treat perturbations to the nutrient cycle on time scales of days to years.”</p>
<p>“[In nature] The pool of dissolved organic matter also contains a variety of compounds with diverse turnover rates. In oceanic waters there is a large pool of highly refractory DOM, whose average age has been estimated to be as much as 3400 years (Williams 1975). The bulk of this material is essentially irrelevant for the cycling of nutrients on the time scales of [model] interest. However the DOM pool also includes much more labile components. Highly labile sugars and amino acids released through cell lysis or messy feeding may be consumed by bacteria or break down on time scales of hours (Fuhrman 1987) to days (Holibaugh and Azam 1983). This material is not modelled explicitly, but is treated as though it was remineralised immediately to form dissolved inorganic nutrient. The model includes a single DOM pool with an assumed turnover time of the order of a hundred days. … Because it turns over on time scales, which are long compared with both phytoplankton uptake of inorganic N, and transport within the Bay, it plays a role in reducing N sequestration in coastal waters.”</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</section>
<section id="defining-the-grouptype-correctly" class="level2">
<h2 class="anchored" data-anchor-id="defining-the-grouptype-correctly"><strong>6.3. Defining the GroupType correctly</strong></h2>
<p>The definition of a group is very important as it will determine the ecological routines and the adaptive timestep applied for the group (see timestep description in chapter 6.1). Table 10 shows routines that are called for different group types. The routines are further modified depending on whether consumers live in the water column or sediment.</p>
<table class="caption-top table">
<caption><span id="_Toc526762762" class="anchor"></span>Table 10. Routines called for different non-vertebrate GroupTypes in Atlantis code. The <em>Call_Group_Process_Function()</em> routine in atecology.c determines the specific routine used from the atGroupProcess.c file non vertebrate groups. Note, that the greatest diversity ofroutines are called for different benthic invertebrate types (for clarity benthic invertebrates are shown in <em>bold italics</em>).</caption>
<colgroup>
<col style="width: 42%">
<col style="width: 57%">
</colgroup>
<thead>
<tr class="header">
<th><strong>Routine name in Atlantis code</strong></th>
<th><strong>GroupType for which the routine is applied</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><em>Dinoflag_Process()</em></td>
<td><p>DINOFLAG</p>
<p>ICE_MIXOTROPHS</p></td>
</tr>
<tr class="even">
<td><em>Phytoplankton_Process()</em></td>
<td><p>LG_PHY</p>
<p>SM_PHY</p>
<p>PHYTOBEN</p>
<p>SEAGRASS</p>
<p>MICROPHTYBENTHOS</p>
<p>TURF</p>
<p>ICE_DIATOMS</p></td>
</tr>
<tr class="odd">
<td><em>Invert_Consumers_Process()</em></td>
<td><p><strong><em>SM_INF</em></strong></p>
<p><strong><em>LG_INF</em></strong></p>
<p>SM_ZOO</p>
<p>MED_ZOO</p>
<p>LG_ZOO</p>
<p>CEP</p>
<p>PWN</p>
<p>ICE_ZOOBIOTA</p></td>
</tr>
<tr class="even">
<td><em>Epibenthic_Invert_Process()</em></td>
<td><p><strong><em>SED_EP_FF</em></strong></p>
<p><strong><em>MOB_EP_OTHER</em></strong></p></td>
</tr>
<tr class="odd">
<td><em>Sediment_Epi_Other_Process()</em></td>
<td><strong><em>SED_EP_OTHER</em></strong></td>
</tr>
<tr class="even">
<td><em>Coral_Process()</em></td>
<td><p>CORAL</p>
<p>SPONGE</p></td>
</tr>
<tr class="odd">
<td><em>Pelagic_Bacteria_Process()</em></td>
<td>PL_BACT</td>
</tr>
<tr class="even">
<td><em>Ice_Bacteria_Process()</em></td>
<td>ICE_BACT</td>
</tr>
<tr class="odd">
<td><em>Sediment_Bacterica_Process()</em></td>
<td>SED_BACT</td>
</tr>
<tr class="even">
<td><em>Labile_Detritus_Process()</em></td>
<td>LAB_DET</td>
</tr>
<tr class="odd">
<td><em>Refractory_Detritus_Process()</em></td>
<td>REF_DET</td>
</tr>
<tr class="even">
<td><em>Carrion_Process()</em></td>
<td>CARRION</td>
</tr>
</tbody>
</table>
</section>
<section id="modifying-functional-groups-in-parameterised-models" class="level2">
<h2 class="anchored" data-anchor-id="modifying-functional-groups-in-parameterised-models"><strong>6.4. Modifying functional groups in parameterised models</strong></h2>
<p>Details on how to add a new functional group into an existing model are given on the wiki <a href="https://confluence.csiro.au/display/Atlantis/Adding+a+new+group">here</a>. This will require a significant effort and reparameterisation of the model.</p>
<p>Alternatively, if changes to functional groups are needed, it might be easier to use an existing group and change it into something new. You may not want to change the three letter code, as that would involve changing all the parameter names for the group.</p>
</section>
</section>
<section id="run-parameters" class="level1">
<h1>7. RUN PARAMETERS</h1>
<p>The <em>run.prm</em> provides all the key information required to control the model run, load optional submodels and also parameters controlling the debugging and detailed output. The table below lists the main parameters used in the <em>run.prm</em>, but please remember to follow the news on the Atlantis wiki as new parameters are sometimes added.</p>
<p><span id="_Toc526762763" class="anchor"></span>Table 11. List and description of parameters in the <em>run.prm</em> file</p>
<table class="caption-top table">
<thead>
<tr class="header">
<th><strong>Parameter</strong></th>
<th><strong>Meaning</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong><em>General parameters controlling the run</em></strong></td>
<td></td>
</tr>
<tr class="even">
<td>title</td>
<td>Title of the run. Ideally this should be an informative title (e.g.&nbsp;SE Tasmania model with constant forced fishing and base level fishing pressure on FMM increased 2000 fold) as it will written at the start of the <em>log.txt</em> file and into the output.nc file and will be displayed on the top of plots in visualization tools such as OLIVE or DIVE</td>
</tr>
<tr class="odd">
<td>dt</td>
<td>Timestep of the model. The timestep is typically given in hours (e.g.&nbsp;“12 hour”), however Atlantis reads the units together with the numerical value so other units “minute” or “day” may also be used. Typically the timestep is set to 12 or 24 hours, but it can be set to other intervals as well. It is not typically recommended to make it &gt;24h and one should think carefully whether shorter than 12h is really needed, as the model will run slower. Some coastal, very dynamic, areas have 6h or even 3h timesteps, but that will require high resolution hydrodynamic forcing data.</td>
</tr>
<tr class="even">
<td>tstop</td>
<td>Day on which the model run will end. If you want to run the model for 10 years, set this to 3651 (technically 3650, however given most output is done annually it is best to add an extra day to make sure the final output is printed before the model stops).</td>
</tr>
<tr class="odd">
<td>tburnday</td>
<td>Number of days that define the burn-in period. At the end of this period the “virgin” state of the model is stored. This means that every comparison of species biomass to the original or “virgin” state will use values stored at the end of the burn-in period and not those set in the initial conditions. This is useful for models that have unstable dynamics at the start of the model run.</td>
</tr>
<tr class="even">
<td><strong><em>Output files and reporting</em></strong></td>
<td></td>
</tr>
<tr class="odd">
<td>toutstart</td>
<td>Day on which to start writing the output. This is normally set to 0, but can be set to a larger value for long runs if there is a need to save memory and there is no interest in the burn-in period dynamics.</td>
</tr>
<tr class="even">
<td>toutinc</td>
<td><p>Periodicity in days for the Ecology submodel snap shot output to the NC files, in days. In longer runs, this is often set to monthly (30) or yearly (365) so file sizes do not become very large. In poorly behaved runs (that are crashing), or ones with extreme events which condition subsequent behaviour, fine scale output (e.g.&nbsp;at the same level as the timestep) may be required. However, remember that a lot of output is written at each reporting period, so make sure you set reasonable values for long well-behaved runs (quarterly or even yearly).</p>
<p>Note that in Atlantis all years are 365 days long. If monthly output is required and it is set to 30 days then after one year the output would be shifting by 5 days every year.</p></td>
</tr>
<tr class="odd">
<td>toutfinc</td>
<td>Periodicity of the cumulative output from the Harvest submodel (catches, quotas, discards, etc) to the NC files (CATCH.nc, TOTCATCH.nc). These values are the sum of all effort, catch (etc) since the last output dump. Unlike the ecology output, they are not snap shot values.</td>
</tr>
<tr class="even">
<td>tsumout</td>
<td>Periodicity of the output to the TXT files.</td>
</tr>
<tr class="odd">
<td>flagannual_Mest</td>
<td>If 1, estimates of mortality per predator are written out annually, otherwise they are written at the tsumout frequency</td>
</tr>
<tr class="even">
<td>fishout</td>
<td>If 1, it switches on the fisheries output (it is recommended that this is on if you have fisheries in your model, but is unnecessary if it is an ecology model only).</td>
</tr>
<tr class="odd">
<td>flagreusefile</td>
<td><p>Whether to replace or append ouputs to the existing files, when starting a new run. If it is set to 0 Atlantis will quit with a message that a file with such name already exists. If it is set to 1 results will be appended to the existing file, if set to 2 it will overwrite (replace) the existing file without warning.</p>
<p>NOTE: This is old code. If you really want it to append please run a test first to see if it worked. If not, let the developers know, as it likely needs updating. Setting flagreusefile to 0 or 2 is recommended.</p></td>
</tr>
<tr class="even">
<td>flag_age_output</td>
<td>If set to 1, an additional output file <em>agebiomindex.txt</em> will be created and the total biomass output will be given for each cohort. This can bea very useful output file, because without it you will need to run a specially written script over the <em>output.nc</em> file to determine the biomass per age group per functional group.</td>
</tr>
<tr class="odd">
<td><strong><em>Parameters loading different submodel or options</em></strong></td>
<td></td>
</tr>
<tr class="even">
<td>flagecon_on</td>
<td>If 0 the Economics submodel is not loaded</td>
</tr>
<tr class="odd">
<td>flag_fisheries_on</td>
<td>If 0 the Harvest submodel is not loaded</td>
</tr>
<tr class="even">
<td>flag_skip_biol</td>
<td>If 1 the Ecology submodel is not loaded – only for debugging!</td>
</tr>
<tr class="odd">
<td>flag_skip_phys</td>
<td>If 1 the Physics submodel is not loaded. This will actually produce an error message and Atlantis will quit. <strong>So this value should be NEVER set to 1.</strong></td>
</tr>
<tr class="even">
<td>fishmove</td>
<td>If 0 all age structured group movements (vertical and horizontal) are turned off</td>
</tr>
<tr class="odd">
<td>external_populations</td>
<td>Flag indicating whether a age structured population dynamics model is used to track growth and mortality while an age-structured group is outside the model domain (1) or whether a simple growth and mortality correction is applied to migrators on their return to the system (0)</td>
</tr>
<tr class="even">
<td><p><strong><em>Parameters controlling detailed outputs for a specific box and diagnostic reporting</em></strong></p>
<p>The outputs are written into the <em>log.txt</em> file for model tuning and debugging purposes.</p></td>
<td></td>
</tr>
<tr class="odd">
<td>debug_it</td>
<td>If 1 - give detailed outputs of all processes for a selected box and time period into the <em>log.txt</em> file</td>
</tr>
<tr class="even">
<td>checkbox</td>
<td>Box number for which to give detailed outputs into <em>log.txt</em> file</td>
</tr>
<tr class="odd">
<td>checkstart</td>
<td>Day on which the detailed output in the <em>log.txt</em> file for the box above should start</td>
</tr>
<tr class="even">
<td>checkstop</td>
<td>Day on which the detailed output in the <em>log.txt</em> file for the box above should end</td>
</tr>
<tr class="odd">
<td></td>
<td><strong>Note!</strong> Detailed output will likely produce a huge amount of data, so make sure the time period of detailed reporting is not too long or the run is short</td>
</tr>
<tr class="even">
<td>debug</td>
<td><p>This parameter has a lot of options about what topic area or code path to give detailed outputs for (for debugging). These settings trigger fprintf lines in the code and print information to the <em>log.txt</em> file.</p>
<p>Each setting focuses on different processes. Setting it to 0 turns off detailed outputs.</p>
<p>Other parameter values include</p>
<p>debug = 1 (for fisheries catch fprintf)</p>
<p>debug = 2 (for discards)</p>
<p>debug = 3 (for forced fish catches)</p>
<p>debug = 4 (for the assessment model processes)</p>
<p>debug = 5 (for spatial management)</p>
<p>debug = 6 (for basic effort options)</p>
<p>debug = 7 (for economically driven effort allocation model)</p>
<p>debug = 8 (for economic calculations)</p>
<p>debug = 9 (for quota setting and handling)</p>
<p>debug = 10 (for aging in age structured groups)</p>
<p>debug = 11 (for recruitment in age structured groups)</p>
<p>debug = 12 (for spawning in age structured groups)</p>
<p>debug = 13 (for migration in/out of model domain)</p>
<p>debug = 14 (for movement within the model domain)</p>
<p>debug = 15 (for stock-based allocations within a functional group)</p>
<p>debug = 16 (for basic overall biomass calculations)</p>
<p>debug = 17 (for feeding)</p>
<p>debug = 18 (for all ecological processes)</p>
<p>debug = 20 (print biology process function parameters)</p>
<p>debug = 21 (print prey function parameter)</p>
<p>debug = 22 (debug clam - atlantis linkage)</p>
<p>debug = 23 (for deposition in the physics sub-model)</p>
<p>debug = 24 (for the forced mortality scalar)</p>
<p>debug = 25 (for the forced growth scalar)</p>
<p>debug = 26 (for FSPB additions)</p>
<p>debug = 27 (for debugging the C and P code)</p>
<p>debug = 28 (for debugging the SS3 tiered code)</p>
<p>debug = 29 (for external scaling)</p>
<p>This list changes as new debugging issues arise. So if you want to know the latest list, the entire list of currently available debugging options can be seen in the <strong>atlantisboxmodel.h</strong> file (around line 750) in the Header files subfolder of the <strong>atlantismain</strong> library. There is also a list on the Atlantis <a href="https://confluence.csiro.au/display/Atlantis/2010/06/02/New+debug+flag+option+included+-+debug+%3D+24">wiki</a></p></td>
</tr>
<tr class="odd">
<td>fishtest</td>
<td>If 1, count up total population of age structured groups after each main subroutine</td>
</tr>
<tr class="even">
<td>flaggape</td>
<td>Periodically list prey vs gape statistics in <em>log.txt</em></td>
</tr>
<tr class="odd">
<td>flagchecksize</td>
<td>Periodically list relative size in <em>log.txt</em>. This is useful during model calibration for seeing the current size versus size given in the initial conditions.</td>
</tr>
<tr class="even">
<td>flagagecheck</td>
<td>Periodically list age structure per cohort in <em>log.txt</em></td>
</tr>
<tr class="odd">
<td>flagdietcheck</td>
<td><p>Writes detailed diet output per box and layer to <em>DetailedDietCheck.txt</em></p>
<p>If set to 1 then it is total consumption, if set to 2 then it returns proportions</p></td>
</tr>
<tr class="even">
<td>checkNH</td>
<td>Give detailed logged output for NH fluxes in the box selected with checkbox. Useful understanding what groups may be at fault if you get an “unstable variable” crashing the model before completion of the run.</td>
</tr>
<tr class="odd">
<td>checkDL</td>
<td>Give detailed logged output for DL fluxes in checkbox. Useful understanding what groups may be at fault if you get an “unstable variable” crashing the model before completion of the run.</td>
</tr>
<tr class="even">
<td>checkDR</td>
<td>Give detailed logged output for DR fluxes in checkbox. Useful understanding what groups may be at fault if you get an “unstable variable” crashing the model before completion of the run.</td>
</tr>
<tr class="odd">
<td>checkbiom</td>
<td>Give detailed logged output for biomasses of which_check group in the checkbox (see above)</td>
</tr>
<tr class="even">
<td>which_fleet</td>
<td>ID number of fleet to track (if don't want to track anything set this to a value greater than the number of fisheries in your model). The ID is the same as in the <em>fisheries.csv</em> file</td>
</tr>
<tr class="odd">
<td>which_check</td>
<td>ID number of group to track (if don't want to track anything set this to a value greater than the number of functional groups in your model). The ID is the same as in the <em>functional_groups.csv</em> file</td>
</tr>
<tr class="even">
<td>habitat_check</td>
<td>Which distribution type (0-water column, 1- sediment, 2- epibenthic, 3- land, 4-ice) to report for the group marked with which_check. Some groups are ony tracked in one distribution type, others are tracked in both water column and sediments.</td>
</tr>
<tr class="odd">
<td>move_check</td>
<td>ID number of group for which movements should be tracked</td>
</tr>
<tr class="even">
<td><strong><em>Key parameters determining the model runs and inputs</em></strong></td>
<td></td>
</tr>
<tr class="odd">
<td>flaghemisphere</td>
<td>0=southern, 1=northern hemisphere. This flag is important in calculating winds and circulation (affects coriolos and seasons).</td>
</tr>
<tr class="even">
<td>flagAllowLand</td>
<td>If set to 1 if the model includes dynamic land components (this requires a large number of additional inputs). If set to 0 any land boxes are treated as boundary boxes and are not executed.</td>
</tr>
<tr class="odd">
<td>flagIsEstuary</td>
<td>Flag to indicate if the model is estuary. If true a SED sediment tracer is required in the initial conditions input file and tidal emptying of shallow boxes is allowed (with fish moved to the nearest water channel).</td>
</tr>
<tr class="even">
<td>flagIsPolar</td>
<td>Flag to indicate whether the model is in polar regions and so will have 24 hours of nigh/daylight seasonally</td>
</tr>
<tr class="odd">
<td>flag_use_deltaH</td>
<td>Flag to indicate that when using pH tracking and parameter corrections use ∆ [H+] rather than pH <em>per se</em>.</td>
</tr>
<tr class="even">
<td>trackAtomicRatio</td>
<td>If set to 1, then not only N, but also P and C are tracked. This will require many additional tracers in the input files.</td>
</tr>
<tr class="odd">
<td>track_rugosity_arag</td>
<td>If set to 1, rugosity and aragonite saturation will be tracked and additional parameters required. Important for modelling corals and sponges.</td>
</tr>
<tr class="even">
<td>track_pH</td>
<td>If set to 1, pH values will be tracked – important for ocean acidification simulations.</td>
</tr>
<tr class="odd">
<td>track_contaminants</td>
<td>If set to 1, contaminants will be tracked. This will require information on the number and name of the contaminants to be tracked, as well as additional tracers and initial conditions (so as to allow for the tracking of contaminants through each of the functional groups in the model).</td>
</tr>
<tr class="even">
<td><strong><em>Other parameters</em></strong></td>
<td></td>
</tr>
<tr class="odd">
<td>mirror_invalid</td>
<td>If this flag set to 1 and the hydrodynamics model attempts to send water from one cell into an invalid cell (e.g.&nbsp;a layer in a box that does not exist) the water flow will instead will be sent into the layer closest to the intended destination layer. This option was needed because historically some hydrodynamic models tried to send water into layers below what Atlantis was modelling, so water had to be redirected to the extant bottom layer instead. Set this flag to 0 if your hydrodynamic forcing is working well or you want Atlantis to quit with a warning that something is wrong with the hydrodynamic forcing files.</td>
</tr>
<tr class="even">
<td>flag_replicated_old</td>
<td>This flag is used to maintain legacy code options. While we fix true bugs sometimes we also update the code to better represent a process where we had made a rough first attempt that can now be done in a better way. However, such changes can change model dynamics and so for legacy models we need to keep the legacy options. When set to 1, the new code changes are not applied and the old results can be reproduced. However, for all new models it is recommended that this flag be set to 0, as the new code is more rigorous and defensible.</td>
</tr>
<tr class="odd">
<td>flag_replicated_old_PPmort</td>
<td><p>Flag is used to maintain old plankton mortality assumptions (from very early code versions that persisted as potential bug into newer models). Turning it on zeroes out phytoplankton mortality and only allows for lysis in phytoplankton. However, for all new models it is recommended that this flag be set to 0 so parameter application is transparent and contaminants are transferred appropriately.</p>
<p>If you are transitioning an old model and want to have this flag turned off but maintain your old model as close as possible keep your lysis values as is and zero out mL.</p></td>
</tr>
<tr class="even">
<td>flag_replicate_old_calendar</td>
<td>This flag specifically allows for legacy models and the influence of environment on the distribution of recruits to be replicated in the newer code. It is recommended that all new models have this set to 0.</td>
</tr>
<tr class="odd">
<td>flag_old_embryo_init</td>
<td>This flag is to allow legacy models to have the same starting conditions under the new code as they did in older version. In the past, rather than use the stock-recruit relationship for a species to dictate the larvae already in the water column at the start of the model run a simpler linear relationship with spawning stock was used. This has now been replaced by the use of the stock-recruit relationship (for improved consistency). All new models should set this parameter to 0, but legacy models trying to reproduce past model runs should set this value to 1.</td>
</tr>
<tr class="even">
<td>check_dups</td>
<td>If set to 1, Atlantis will check for duplicated parameters in the parameter files before starting the run. This can be used to ensure that same parameter is not entered twice with different values. However, it takes a LONG time to execute (especially for big models). If it is set to 1 the model run will not start for quite a while until all possible duplicates are checked. If you are concerned about duplicate entries, go through the file once with this until all duplicates are found and then turn it off.</td>
</tr>
<tr class="odd">
<td>flagscaleinit</td>
<td>If set to 1 this will cause the vector init_scalar to be read in. This vector contains one value per group listed in the <em>functional_groups.csv</em> file. These values are used to modify (scale) the initial conditions without requiring that nc file to be modified – for the vertebrates the number per age class is scaled and for biomass pools the biomass per cell and layer is scaled (for those groups in the water column, sediment layers and the epibenthos).</td>
</tr>
<tr class="even">
<td>flag_sanity_check</td>
<td>Set this to 1 to make sure that species with a gestation or recruitment period that is about the same length as the time between reproduction (spawning) events do not get cascading population growth (where recruits from one event accidentally also contribute to the next event). To replicate old code versions set this value to 0.</td>
</tr>
<tr class="odd">
<td>flagpassivetracer</td>
<td>Set to 1 if you want to do particle tracking to validate physical forcing. This will require adding the SED_i variable to the <em>initial_conditions.nc</em> file.</td>
</tr>
<tr class="even">
<td>flagenviro_warn</td>
<td>Set to 1 if you want warnings about when environmental conditions constrain spawning or movement in a cell</td>
</tr>
</tbody>
</table>
</section>
<section id="force-parameters" class="level1">
<h1>8. FORCE PARAMETERS</h1>
<p>Atlantis can take a lot of externally forced tracer and parameter values. Details on such forcing files are given in the <em>force.prm</em> file.</p>
<p>The minimum <em>force.prm</em> file will contain the pathname and load information for a hydrodynamics forcing file and nothing else. However, in most models a lot more of physical, ecological and socio-economic processes are forced and pathnames for all these files will be required in <em>force.prm</em>. See table 12 below for the list and application examples of different forcing options.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Absolute time in Atlantis: model time, hydro time and TS time</strong></p>
<p>The time (t) entry in the <em>initial_conditions.nc</em> file indicates the actual absolute time in calendar years that marks the start data and time of the model simulation. For example:</p>
<p>t:units = "seconds since 2000-01-01 00:00:00 +10" ;</p>
<p>This indicates that this model starts at midnight on the first of January 2002. This is the main Atlantis time (called <strong>model time</strong>). The hydrodynamics forcing file also has its own absolute start time, which is called <strong>hydro time</strong>; this does not have not to match with the model time. For example, the hydrodynamics data might have five calendar years and start from 2002 July. Atlantis will try to match the model time to the hydro time. In the example, above Atlantis will start the run by loading in the hydrodynamics beginning in January not July and run from there, matching the month of the year when rewinding the hydro data.</p>
<p>The time series forcing files also have a parameter indicating the absolute time of the start (<strong>TS time</strong>). For example, the forcing data may start from 1990 July. On read in Atlantis will by default convert these times into model time, skipping that part of the time series that occurs before model time begins. In the example here, the first 10 years of TS data will be skipped, and only the forcing information from year 11 will be used. Alternatively, if TS time starts later than the model time, Atlantis will run the simulation withouth any forcing until the model time matches the TS time.</p>
<p>Users have an option to match the TS time not to the model time but to the hydro time. This is done by setting ts_on_hydro_time=1. If ts_on_hydro_time=0 then the default will be used (matching to model time). However, keep in mind that if TS time is matched to hydro time any rewinding of hydrodynamics data will also mean the rewinding of the forcing data. For example, if 50 years of forcing data is available and the model runs for 50 years all forcing will be used if TS and model times match. However, if TS is matched to hydro time and the hydrodynamics is rewound every 5 years, then only 5 years of TS values will be used before the TS file is rewound along with the hydrodynamics data.</p>
<p>Note that the rewinding of TS filed described above (done when ts_on_hydro_time=1) is not the same as specific rewind options associated with most TS files. The specific TS rewind parameters apply when the time series of the run (either on model or hydro time) exceeds the time series given the TS file. See details on the TS file format below</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>It is recommended that users download the sample <a href="https://confluence.csiro.au/display/Atlantis/Sample+input+files">South East Tasmanian (SETas) model</a> input files that give details on the structure of the forcing and other parameter files. The table below lists currently available forcing options</p>
<table class="caption-top">
<thead>
<tr class="header">
<th><p><strong>Format of the time series (TS) forcing files</strong></p>
<p>The TS forcing files have a standard input format read by Atlantis. At the top of the file the user can provide a comment on what the file represents (marked by a single # at the start of each line). Then the user needs to provide a description of variables in each column of the TS data file (Atlantis needs this to know what the variable name is, its units and missing value marker). Atlantis recognises these entries as column description with ## at the start of each line. If a single # is given Atlantis will assume it is a comment and ignore it, so if you are having read-in issues make sure you have all column decriptions beginning with ##. Underneath this header section is the actual data table in columns. The number of columns in the data should be the same as the number of column descriptions.</p>
<p>The first column always indicates time and is given in the units defined (typically days). The data in the TS file can be given at longer time intervals, such as every 30 or every 365 days. If forcing data is not given for every timestep of the model Atlantis will either use the the last valid value (i.e.&nbsp;the last value before the gap in the data) or interpolates between the last valid value and the next (future) value in the time series file. The handling of patchy data is determined by the typeXXXXts parameter, such as typeCatchts or typeDiscardts. If typeXXXXts is 1 the same value is used throughout the period between two forcing time steps, if typeXXXXts is 0 then Atlantis will interpolate the value. Not all forcing files have the typeXXXXts parameter and if this parameter is not available it means that <strong>Atlantis will interpolate values</strong>. For example, if the value on day 1 was 10 and the value on day 5 is 50, setting typeXXXXts to 1 means that on day 2, 3 and 4 the value returned will be 10, while on day 5 value it will be 50. Setting typeXXXXts to 0 means that the value returned on day 2 is 20, day 3 value is 30 and day 4 value is 40. See further details about the interpolation <a href="https://confluence.csiro.au/pages/viewpage.action?spaceKey=Atlantis&amp;title=External+Mortality%2C+Growth+and+Recruitment+Scaling">here</a></p>
<p>Make sure you chose the correct value for the rewind parameter, as that will determine what happens if forcing file time series is shorter than the model run: rewind=0 means that the last value will be used for the remaining of the run, rewind=1 means that the time series will be reused from the start. More details on the structure of different TS files can be found <a href="https://confluence.csiro.au/pages/viewpage.action?spaceKey=Atlantis&amp;postingDay=2010%2F7%2F12&amp;title=Group+size+scale+time+series+input+-+non-compulsory+change+to+forcing+input+file">here</a>.</p>
<p># <strong>THIS FIRST LINES ARE A COMMENT ON THE SOURCE OF THE DATA</strong></p>
<p># GDP time series for Australia 1972 to 2008 based on treasury projections in 2006 Budget</p>
<p># Fuel cost time series from autoregressive cost model based on Australian TACC diesel cost data</p>
<p>#</p>
<p><strong># - EVERYTHING BELOW IS READ BY ATLANTIS AND MUST BE IN A STANDARD FORM</strong></p>
<p>## COLUMNS 3</p>
<p>##</p>
<p>## COLUMN1.name Time</p>
<p>## COLUMN1.long_name Time</p>
<p>## COLUMN1.units days since 1972-01-01 00:00:00 +10</p>
<p>## COLUMN1.missing_value 0</p>
<p>##</p>
<p>## COLUMN2.name GDP</p>
<p>## COLUMN2.long_name GDP</p>
<p>## COLUMN2.units dollars</p>
<p>## COLUMN2.missing_value 0</p>
<p>##</p>
<p>## COLUMN3.name FuelCost</p>
<p>## COLUMN3.long_name FuelCost</p>
<p>## COLUMN3.units dollars</p>
<p>## COLUMN3.missing_value 0</p>
<p>##</p>
<p>1 0.206 0.64</p>
<p>2 0.207 0.68</p>
<p>3 0.208 0.69</p>
<p>4 0.209 0.65</p>
<p>5 0.204 0.64</p>
<p>6 0.202 0.63</p>
<p>7 0.210 0.62</p>
<p><strong>NOTE:</strong> If you are having issues reading a TS file</p>
<ol type="1">
<li><p>Make sure there are no extra spaces or tabs after the name of the column (this can sometimes happen if you have created your TS file in excel and then pasted on to a text editor). Atlantis should now ignore those spaces, but in the past it was an issue.</p></li>
<li><p>Make sure the model is space not tab delimited. Again Atlantis should now be able to read either option, but just to be careful.</p></li>
<li><p>If a collaborator created the file for you and they use a different operating system, make sure the file is in the correct file type for your machine (in text-pad, text wrangler etc check that it is a windows type if you are on a windows machine and linux if you do not use windows). See chapter 2.2 for file conversion details.</p></li>
</ol></th>
</tr>
</thead>

</table>
<table class="caption-top table">
<caption><span id="_Toc526762764" class="anchor"></span>Table 12. List and description of external forcing options available in Atlantis</caption>
<thead>
<tr class="header">
<th><strong>Forcing</strong></th>
<th><strong>Parameters needed in <em>force.prm</em></strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong><em>Forcing of physical parameters</em></strong></td>
<td></td>
</tr>
<tr class="even">
<td>Hydrodynamics</td>
<td><p>This is the minimum forcing required to run Atlantis. See chapter 4 for details on hydrodynamic forcing files.</p>
<p>nhdfiles gives the number of hydrodynamics files that will be supplied (read in). It is possible to put all the forcing time series into one file, but these files can get very big. Therefore, it is often easier to split the series into a number of smaller files. Atlantis will read these files in the order given, using forcing data from the first file, then from the second one and so on. Once the data from the last file is used, Atlantis will return to the first file and start reading from there again.</p>
<p>The hydrodynamics files are automatically looped. For other forcing input files an additional parameter (_rewind) indicates whether the forcing data should be looped when the time series finishes (=1) or stay on the last value (=0)</p>
<p>Note that the pathway to the file is tagged for Atlantis by the parameter name hdN.name where N indictes the number of the file, remembering that numbering in C (and thus Atlantis) starts at 0. So hd0.name, hd1.name, hd2.name and so on.</p>
<p>For example, the following would indicate there are 2 hydrodynamic files and that they are located in the directory the model is being run from.</p>
<p>nhdfiles 2</p>
<p>hd0.name hydrofile1.nc</p>
<p>hd1.name hydrofile2.nc</p>
<p>The following indicate the files are stored in the inputs sub-directory (a directory one level down from the run directory)</p>
<p>hd0.name inputs/hydrofile1.nc</p>
<p>hd1.name inputs/hydrofile2.nc</p>
<p>Finally if the inputs folder is in the directory that is at the same level than the run directory then the path to the hydrodynamics files should read as</p>
<p>hd0.name ../inputs/hydrofile1.nc</p>
<p>hd1.name ../inputs/hydrofile2.nc</p>
<p>For more information on how to indicate a path see, for example, this <a href="https://en.wikipedia.org/wiki/Path_%28computing%29">Wikipedia site</a></p></td>
</tr>
<tr class="odd">
<td>Temperature</td>
<td><p>Atlantis can calculate the temperature dynamically, but most models use temperature forcing from more accurate hydrodynamic modes. See chapter 4 for details of temperature forcing files. The lay out of the list of files is as for the hydro file – with as many files listed as given by the ntempfiles parameter.</p>
<p>use_tempfiles 1 if temperature forcing (NC files) are being used</p>
<p>ntempfiles N where N is the number of files being provided</p>
<p>Temperature0.name temp1.nc the path and filename of the first file</p>
<p>Temperature1.name temp2.nc the path and filename of the second file</p>
<p>Temperature2.name temp3.nc the path and filename of the third file</p>
<p>…</p>
<p>temp_rewind 0 flag indicating whether to rewind (loop over) the temperature forcing data (1) or not (0)</p></td>
</tr>
<tr class="even">
<td>Salinity</td>
<td><p>Atlantis can also load in salinity forcing from more accurate hydrodynamic modes. See chapter 4 for details of salinity forcing files.</p>
<p>The lay out of the salinity parameters and file name list is as for temperature.</p>
<p>use_saltfiles 1</p>
<p>nsaltfiles N</p>
<p>Salinity0.name salt1.nc</p>
<p>Salinity1.name salt2.nc</p>
<p>…</p>
<p>salt_rewind 0 (or 1)</p></td>
</tr>
<tr class="odd">
<td>pH forcing</td>
<td><p>If pH is tracked, it is possible to calculate it dynamically or provide external forcing values from specific models. The pH forcing is provided through spatial NC files in the same way as for temperature and salinity.</p>
<p>use_phFiles 0</p>
<p>npHfiles N</p>
<p>pH0.name pHforceA.nc</p>
<p>pH1.name pHforceB.nc</p>
<p>pH_rewind 0</p></td>
</tr>
<tr class="even">
<td>forcing of other tracers</td>
<td><p>It is possible to force any other tracers with external data. See details on the wiki <a href="https://confluence.csiro.au/display/Atlantis/2013/07/10/Read+in+forcing+data+for+tracers">here</a>. When new data is read in at the end of the timestep any existing value is overwritten with the value from the NC forcing file. This has been used for contaminants and for oxygen (e.g.&nbsp;when internal Atlantis calculations were not reproducing oxygen dynamics sufficiently well). This is a useful option, but should be used with care for any tracers that contains N (and P or C if tracking those tracers), as it can violate the assumption of conservation of matter.</p>
<p>use_force_tracers 1 (this flag indicates whether additional tracer forcing files should be read it (1) or not (0))</p>
<p>nforceTracers N (where N is the number of tracers that will be forced)</p>
<p>tracerNames</p>
<p>nameA</p>
<p>nameB</p>
<p>…</p>
<p>The list of tracer names (matching the names used in the <em>initial_condition.nc</em> file) that will be forced – with each name on a new line. For example,</p>
<p>nforceTracers 2</p>
<p>tracerNames 2</p>
<p>Oxygen</p>
<p>Arsenic</p>
<p>For each tracer listed you must then give the filenames using the format</p>
<p>TRACERNAME_nFiles N (where TRACERNAME is the name of the tracer and N is the number of files to be loaded)</p>
<p>TRACERNAME_File0.name TRACER_1.nc (path and filename for the first forcing file for the tracer)</p>
<p>TRACERNAME_File0.name TRACER_2.nc (path and filename for second file etc)</p>
<p>TRACERNAME_rewind 0 (whether to rewind the forcing files (1) or not (0))</p>
<p>for example</p>
<p>Arsenic_nFiles 1</p>
<p>Arsenic_File0.name Arsenic.nc</p>
<p>Arsenic_rewind 0</p></td>
</tr>
<tr class="odd">
<td>Point sourcing of nutrients</td>
<td><p>Even though Atlantis calculates internal nutrient dynamics, it is relatively common to force additional nutrient inputs (e.g.&nbsp;from sewage outfalls, river mouth, upwellings etc). For example, the California Current, Guam, Gulf of Mexico and Chesapeake Bay models all use nutrient loading -forcing with nutrient time series (Brand <em>et al.</em> 2007; Ainsworth <em>et al.</em> 2015; Weijerman <em>et al.</em> 2015). While such inputs are mostly used to imitate nutrient inputs from sources clearly outside the model domain (e.g.&nbsp;rivers), they can also be used as a proxy for multi-decadal trends in ocean productivity, even if full multi-decadal oceanographic data is not available and must be ‘looped’.</p>
<p>Externally forced nutrients sources (or sinks if negative values are given) are provided as TS files</p>
<p>npointss N (where N is the number of externally supplied point source/sink files)</p>
<p>For each TS file the following details must be given</p>
<p>pss0.name XXXX (where XXX is a meaningful name for the point, such as a river name, used so the modeller knows what this point represents; this name is NOT used by Atlantis)</p>
<p>pss0.location X Y Z (e.g.&nbsp;3951091.3 1789202.7 <strong>-1)</strong></p>
<p>pss0.data PSSTS1.ts (path and filename of the forcing TS file)</p>
<p>pss0.rewind 1 (whether to rewind the forcing file (1) or not (0))</p></td>
</tr>
<tr class="even">
<td></td>
<td><p><strong>Note!</strong> The location values (X Y Z) has three values – coordinates to input the point source (3951091.3 1789202.7) and the depth <strong>(-1).</strong> To make sure the point source is deposited in the desired box it is a good idea to ensure the X Y values match the “inside” point of the box as given in the BGM file (see chapter 3).</p>
<p>The <strong>last number in the location is very important</strong> as it will determine the correct reading of the forcing data. For point source data the depth must be a negative number for it to be deposited in the water. Other time series forcing files (e.g.&nbsp;for catch) follow a similar format to the PSS files – requiring a name, location, data (filename) and rewind flag. However, <strong>for these forcing files the third number gives the box ID number (from BGM file, see chapter 3) not a depth.</strong></p></td>
</tr>
<tr class="odd">
<td>Bottom stress</td>
<td><p>Bottom stress values are used to calculate resuspension and additional macrophyte mortality due to wave action. Atlantis can calculate bottom stress values dynamically, but if processes affected by bottom stress are a focus of the study it is recommended to force stress externally from more precice hydrodynamic models. At present there is no flag to indicate whether this file will be loaded (future releases are likely to switch to a flag as for temperature and salinity, e.g.&nbsp;use_stressfiles). At present if the BottomStress tag is in the <em>force.prm</em> then the listed file will be read automatically.</p>
<p>BottomStress stress.nc (the path and filename of the stress NC input file).</p>
<p>If you do not wish to use bottom stress simply omit the BottomStress entry from the <em>force.prm</em> file.</p></td>
</tr>
<tr class="even">
<td>Precipiation and evaporation</td>
<td><p>See chapter 5.2 on how the precipitation forcing data is used. As in case of bottom stress, precipitation and evaporation are forced through spatial NC files. They will be loaded if the tag name is present in the <em>fore.prm</em> file (if not provided the model will not look to load the files).</p>
<p>Precipitation rain.nc (path and filename for the precipitation)</p>
<p>Evaporation evap.nc (path and filename for the evaporation)</p>
<p><strong>These files will not rewind</strong>.</p></td>
</tr>
<tr class="odd">
<td>Solar radiation</td>
<td><p>Solar radiation can be calculated dynamically by Atlantis or can be forced through a time series TS file. In the later case (TS file used) the same value is used for the entire model domain. If you do not wish to load a solar TS file do not provide a Solar_radiation entry in <em>force.prm</em> and set lim_sun_hours to 1 in the <em>biology.prm</em> file. As with bottom stress the solar radiation file will be loaded if the tag name is present in the <em>fore.prm</em> file</p>
<p>Solar_radiation solar.ts (path and filename for the solar forcing file)</p>
<p>Solar_radiation_rewind 0 (a flag indicating whether to rewind the data file (1) or not (0))</p></td>
</tr>
<tr class="even">
<td>CO2 concentrations</td>
<td><p>Time series of atmospheric CO2 concentrations. This is used to dynamically calculate pH values and is required if trac_pH=1 in <em>run.prm</em>.</p>
<p>See further details on the wiki <a href="https://confluence.csiro.au/display/Atlantis/2012/10/19/Handling+effects+of+acidification">here</a></p></td>
</tr>
<tr class="odd">
<td>Ice</td>
<td>If ice is modelled dynamically, the user must provide ice forcing data. See details on the wiki <a href="https://confluence.csiro.au/display/Atlantis/Ice+Forcing+files">here</a></td>
</tr>
<tr class="even">
<td><strong><em>Forcing of ecological parameters</em></strong></td>
<td></td>
</tr>
<tr class="odd">
<td>Forcing of recruitment</td>
<td><p>Recruitment can be forced in two ways, see details on the wiki <a href="https://confluence.csiro.au/display/Atlantis/Recruitment+forcing">here</a></p>
<p>The first is to use a single recruitment time series across the entire model domain. To do this the tag Recruitment_time_series must be added to the <em>force.prm</em> file</p>
<p>Recruitment_time_series filename.ts (path and filename for the time series)</p>
<p>This time series provides a TS file of recruits entering the entire model domain. Box specific forcing of recruitment is also allowed as described in the link above.</p>
<p>The second way is to allow for time series of recruitment to be read in for specific boxes. To do this set</p>
<p>nMultRects N (the number of boxes where recruitment time series will be provide)</p>
<p>typeMultRects 1 (the type of time series – whether to use the last value (0) or interpolate between values (1))</p>
<p>The for each of the N boxes you need to provide the following:</p>
<p>MultRects0.name Meaningful_name</p>
<p>MultRects0.location X Y BOXID (e.g.&nbsp;4924432.61 3298510.07 94)</p>
<p>MultRects0.data filename1.ts (path and filename of the TS file for box BOXID)</p>
<p>MultRects0.rewind 0 (whether to rewind the file (1) or not (0))</p>
<p>MultRects1.name Meaningful_nameB</p>
<p>MultRects1.location X Y BOXID2</p>
<p>MultRects0.data filename2.ts</p>
<p>MultRects0.rewind 0</p>
<p>…</p></td>
</tr>
<tr class="even">
<td>Recruitment scaling</td>
<td><p>Recruitment can also be scaled by a time series of species-specific scalars provided in a time series (TS) file. To use this option include the tag</p>
<p>Recruitment_enviro_forcing in the <em>force.prm</em> file</p>
<p>Recruitment_enviro_forcing scalarfile.ts (path and filename for the time series)</p>
<p>More details are given on the <a href="https://confluence.csiro.au/display/Atlantis/Recruitment+forcing">wiki</a></p></td>
</tr>
<tr class="odd">
<td>Recruit size</td>
<td><p>Representing differential survival can be done in two ways, both of which use a time series file. To activate these set flag_modify_KWSR in the <em>biology.prm</em> file to a value greater than 0. In both cases KWRR will be reset so that it is KWSR * X_RS. If flag_modify_KWSR is set to 1 then the model will load a time series of multipliers to apply to the constant KWSR supplied for the group in the <em>biology.prm</em> file. If flag_modify_KWSR is set to 2 then the model will load a time series file of values to use for KWSR.</p>
<p>In both cases the file for KWSR is supplied in force.prm as follows</p>
<p>KWSR_forcing KWSR_filename.ts</p></td>
</tr>
<tr class="even">
<td>Larval connectivity matrix</td>
<td><p>To activate set use_larvalfiles 1 in the <em>force.prm</em> and then provide the filename of the connectivity matrix per species using larval dispersal</p>
<p>Larval0.name connectfile.nc (path and filename for the connectivity matrix NC file)</p>
<p>Larval_rewind 0 (a flag indicating whether to rewind the file (1) or not (0))</p>
<p>The NC file must contain an entry XXX_Connnectivity (where XXX is the group code from the <em>functional_groups.csv</em> file). These entries are annual connectivity matrices. If there are les entries than there are years in the run then the rewind flag will dictate whether the file is rewound or not.</p>
<p>This is recent functionality that has not been widely used. It allows to enter larval dispersal matrix. See further details <a href="https://confluence.csiro.au/display/Atlantis/Recruitment+Options">here</a></p>
<p>If you want this connectivity matrix to be used completely in place of recruit_hdistrib (rather than in combination with it) then set the flag larvae_connect_only to 1.</p></td>
</tr>
<tr class="odd">
<td>Linear mortality scaling</td>
<td><p>External scalar on linear mortality (<em>mortsc</em> in chapters on Ecology routines) – can also be supplied as an additional site specific mortality rate rather than a scalar. This can be provided as a TS file or spatially with box-specific values as an NC file.</p>
<p>See details <a href="https://confluence.csiro.au/pages/viewpage.action?spaceKey=Atlantis&amp;title=External+Mortality%2C+Growth+and+Recruitment+Scaling">here</a> on how the external mortality scaling is set and calculated.</p></td>
</tr>
<tr class="even">
<td>Scaling of size, growth rate or maturation proportion</td>
<td><p>The scaling of growth and maturation has been used to explore ecological consequences of long-term trends in life-history traits (e.g.&nbsp;Audzijonyte et al.&nbsp;2013; Audzijonyte &amp; Kuparinen 2016).</p>
<p>The scaling is forced through external forcing scalars on size (SN and RN values), growth rate (mum parameter) or maturation (FSPB parameter). The scalars are applied by scaling the existing values. See further details on size change forcing <a href="https://confluence.csiro.au/display/Atlantis/2010/07/12/Group+size+scale+time+series+input+-+non-compulsory+change+to+forcing+input+file">here</a> and on mum and FSPB scalar <a href="https://confluence.csiro.au/display/Atlantis/2011/03/03/Changing+growth+rates+and+FSPB+over+time">here</a></p></td>
</tr>
<tr class="odd">
<td><p><strong><em>Forcing of fisheries and socio-economic parameters</em></strong></p>
<p>It is possible to provide forced catches, discards, effort and other socio-economic parameters. Forced catches and discards, in particular, have been used to parameterise the model and emergent biomasses given the known catch history. See, for example, details in <a href="https://confluence.csiro.au/display/Atlantis/Useful+Reading?preview=/43155482/341447378/Link_et_al_2011_NEUS_tech_memo.pdf">Fulton et al.&nbsp;2007 and Link et al.&nbsp;2011</a>. The catch and discard forcing files are provided as box-specific species-specific time series. At each timestep Atlantis will then take the forced catch from the species biomass. See further details in the description of the Harvest submodel.</p></td>
<td></td>
</tr>
<tr class="even">
<td>Catch</td>
<td><p>To force catches use the following parameters</p>
<p>nCatchts N (where N is the number of catch force time series (TS) files)</p>
<p>typeCatchts 1 (whether to use the last value (1) or interpolate (0) between the two forcing time steps)</p>
<p>This is followed by the actual box-specific forcing file information. Atlantis will attempt to take the forced catch from the specified box. If not enough biomass is available Altnaits will first try to take the biomass form younger age classes, then may look to neighbouring boxes (depending on the flagimposecatch setting in <em>harvest.prm</em>) and if it still can’t fill it will hold it over (“carry over”) and add the remaining amount to the biomass to take the following day. If at the end of the year there still remains catch that cannot be taken it is noted in the <em>log.txt</em> file and the “carry over catch” is zeroed to start again in the new year.</p>
<p>Catchts0.name meaningful_name (e.g.&nbsp;box0catch; this is NOT used by Atlantis)</p>
<p>Catchts0.location X Y BOXID (e.g.&nbsp;e.g.&nbsp;370889.3287 -717418.8373 0. This is the x, y and box ID indicating the box where the catch will be extracted. It is safest to use the “inside” point of the box BOXID as given in the BGM file - see chapter 3 on details about model geometry)</p>
<p>Catchts0.data catchfile1.ts (path and filename of the catch TS file)</p>
<p>Catchts0.rewind 0 (whether to rewind the data file (1) or not (0))</p></td>
</tr>
<tr class="odd">
<td></td>
<td><strong>Note!</strong> Remember that the third number of the location is the box ID number (from BGM file, see chapter 3). <strong>It is very important to give this number correctly or else Atlantis will not read or apply the forcing inputs correctly.</strong></td>
</tr>
<tr class="even">
<td>Discard</td>
<td><p>The entry format is as for catches above, but instead of catches provide discard values for the specified boxes</p>
<p>nDiscardts N (where N is the number of time series (TS) files)</p>
<p>typeDiscardts 1 (whether to use the last value (1) or interpolate (0) between the two forcing time steps)</p>
<p>Discardts0.name meaningful_name</p>
<p>Discardts0.location X Y BOXID</p>
<p>Discardts0.data disfile1.ts (path and filename of the TS file)</p>
<p>Discardts0.rewind 0 (whether to rewind the data file (1) or not (0))</p></td>
</tr>
<tr class="odd">
<td>MPA</td>
<td><p>This forcing file “closes” a proportion of the box to fishing (with the columns in the TS file being the closure per fishery), simulating marine protected areas (MPA). This is used in place of the MPA vectors in <em>harvest.prm</em> and is useful where MPAs vary through time. Further details are provided in the Harvest submodel description</p>
<p>nMPAts N (where N is the number of time series (TS) files)</p>
<p>typeMPAts 1 (whether to use the last value (1) or interpolate (0) between the two forcing time steps)</p>
<p>MPAts0.name meaningful_name</p>
<p>MPAts0.location X Y BOXID</p>
<p>MPAts0.data mapfile1.ts (path and filename of the TS file)</p>
<p>MPAts0.rewind 0 (whether to rewind the data file (1) or not (0))</p></td>
</tr>
<tr class="even">
<td>Effort</td>
<td><p>This forcing file provides the effort time series to apply in a box (with the columns of the TS file giving the effort per fishery) Further details are provided in the Harvest submodel description</p>
<p>nEffortts N (where N is the number of time series (TS) files)</p>
<p>typeEffortts 1 (whether to use the last value (1) or interpolate (0) between the two forcing time steps)</p>
<p>Effortts0.name meaningful_name</p>
<p>Effortts0.location X Y BOXID</p>
<p>Effortts0.data effortfile1.ts (path and filename of the TS file)</p>
<p>Effortts0.rewind 0 (whether to rewind the data file (1) or not (0))</p></td>
</tr>
<tr class="odd">
<td>Residuals</td>
<td><p>These time series provide the price residuals per market if using the economics model and forcing historical prices. One time series is required per market, with the columns providing the residual values per species.</p>
<p>nResidualts N (where N is the number of time series (TS) files)</p>
<p>typeResidualts 1 (whether to use the last value (1) or interpolate (0) between the two forcing time steps)</p>
<p>Residualts0.name market_name (NOT used by Atlantis)</p>
<p>Residualts0.location X Y BOXID (x,y location and box ID for the port which supplies the market)</p>
<p>Residualts0.data resid_file1.ts (path and filename of the TS file)</p>
<p>Residualts0.rewind 0 (whether to rewind the data file (1) or not (0))</p></td>
</tr>
<tr class="even">
<td>Economic statistics</td>
<td><p>Economic statistics include GDP and fuel costs. They are used in the economically driven dynamic fishing calculations. They are not box specific, but just time series values that apply across the entire model.</p>
<p>nEconts 1 (typically 1 or 0; only non-zero if using the economic effort model)</p>
<p>typeEconts 0 (whether to use the last value (1) or interpolate (0) between the two forcing time steps)</p>
<p>Econts0.name meaningful_name (NOT used by Atlantis)</p>
<p>Econts0.location X Y BOXID (x, y and box ID of any dynamic box in the model; by convention set to the values for the box with the larget market port)</p>
<p>Econts0.data econ_file1.ts (path and filename of the TS file)</p>
<p>Econts0.rewind 0 (whether to rewind the data file (1) or not (0))</p></td>
</tr>
</tbody>
</table>
</section>
<section id="primary-producer-processes" class="level1">
<h1>9. PRIMARY PRODUCER PROCESSES</h1>
<p>The main Atlantis model has six different types of primary producers, including two sizes of phytoplankton (large phytoplankton LG_PHY and small phytoplankton SM_PHY), mixotrohic dinoflagellates (DINOFLAG), microphytobenthos (MICROPHTYBENTHOS) and two epibenthic groups (seagrasses SEAGRASS and other phytobenthos PHYTOBEN). In models including ice, specific ice primary producers are modelled. Simialrly in models that dynamically represent land, the user can define terrestrial vegetation (primary producer) groups, which are restricted to land (e.g.&nbsp;MANGROVE, MARSH, with crops and agriculture also available as in industry if the land use model option is used). The description below will focus on the main model only.</p>
<section id="fluxes-in-primary-producers" class="level2">
<h2 class="anchored" data-anchor-id="fluxes-in-primary-producers"><strong>9.1. Fluxes in primary producers</strong></h2>
<p>In primary producer pools the flux is determined by growth (<em>G<sub>PP</sub></em>), natural mortality (<em>M<sub>PP</sub></em>) and grazing (predation) (<em>Gr<sub>PP,i</sub></em>). In addition, there is an optional encystment at certain times of the year or when temperatures (<em>Ec<sub>out</sub></em>) reach a certain trigger level. Transfer back from cysts occurs when conditions are suitable (<em>Ec<sub>in</sub></em>).</p>
<p>Four types of primary producer groups are tracked separately in both the water column and sediments (SM_PHY, LG_PHY, DINOFLAG, MICROPHTYBENTHOS). Two groups are restricted to the epibenthic layer (PHYTOBEN, SEAGRASS). The general flux equation is</p>
<p><span class="math display">\[\frac{d(PP)}{dt} = G_{PP} - M_{PP} - \sum_{i = predators}^{}{Gr_{PP,i}} - Ec_{out} + Ec_{in}\]</span></p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong> Not all processes are executed in all layers.</p>
<p><strong>Primary producer GroupType processes in water column, sediments and epibenthos</strong></p>
<p>For SM_PHY, LG_PHY, DINOFLAG:</p>
<p><strong>In the water column</strong>: the processes executed are growth (primary production) and mortality due to lysis and grazing.</p>
<p><strong>In the sediments</strong>: there is no primary production (only mixotrophic production in DINOFLAG), no lysis mortality and no grazing, only linear mortality.</p>
<p>For MICROPHTYBENTHOS:</p>
<p><strong>In the water column and sediments</strong>: growth (primary production) and mortality due to grazing are carried out in both locations, with mortality due to lysis in water column or linear mortality in sediments.</p>
<p>For SEAGRASS and PHYTOBEN:</p>
<p><strong>In epibenthic layer</strong>: growth (primary production), linear and extra mortality and mortality due to grazing are all executed.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Role and origin of microphytobenthos in Atlantis</strong></p>
<p>Most ecosystem models do not explicitly model microphytobenthos, which consists of small epibenthic primary producers. The Port Philip Bay Integrated Model (PPBIM) (Muray and Parslow 1997), which is used as a basis for simulating lower trophic level dynamics in Atlantis, included microphytobenthos, because it was well studied in the Port Phillip Bay and its production was shown to be significant. Other estuarine models also use microphytobenthos given its important role in those kinds of ecosystems. However, for the majority of coastal Atlantis models the group is not used due to the limited amount of data available for parameterisation and the deeper water depths, where it plays a less important role.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</section>
<section id="primary-producer-growth" class="level2">
<h2 class="anchored" data-anchor-id="primary-producer-growth"><strong>9.2. Primary producer growth</strong></h2>
<p>The equations and processes governing primary production in Atlantis have been adopted from the PPBIM model of <a href="https://confluence.csiro.au/display/Atlantis/Useful+Reading?preview=/43155482/341447378/Link_et_al_2011_NEUS_tech_memo.pdf">Murray and Parslow (1997)</a>. Primary production processes are executed by <em>Primary_Production()</em> routine in <strong>atprocess.c</strong></p>
<p>Growth of a primary producer <em>pp</em> is determined by <strong>multiplying</strong> the biomass of <em>pp</em> by the maximum specific growth rate (<em>mum</em>) and by three additional potentially limiting factors:</p>
<p>1) nutrient limitation factor</p>
<p>2) light limitation factor</p>
<p>3) space limitation factor for benthic macrophytes</p>
<p>4) optional eddy scalar, representing enhancement of primary production by eddies</p>
<p>5) optional pH scalar, accounting for possible effects of acidification on primary production</p>
<p><span class="math display">\[G_{pp} = mum \bullet B_{PP} \bullet \delta_{light} \bullet \delta_{nutrient} \bullet \delta_{space} \bullet \delta_{eddy} \bullet pHscalar\]</span></p>
<p>The <em>mum</em> parameter defines the maximum growth rate per day (mgN day<sup>-1</sup>).</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Multiplicative model for nutrient and light co-limitation (from Murray and Parslow, 1997, page 26):</strong></p>
<p>“Multiplicative models for nutrient and light co-limitation have been widely used (Fasham et al.&nbsp;1990). A potential problem with multiplicative interactions involving several factors is that predictive growth rates become very low when each factor is moderately limiting. An alternative, Liebig's law of the minimum (ie. multiply mum by the smallest factor) is more often applied to interactions among multiple nutrients.”</p>
<p>Atlantis has different options to design limitation by different nutrients (N, Si and P if phosphorus is tracked), set by the flagnut parameter, with choices of multiplicative or Leibig (minimum nutrient) limitation. However, this does not apply to combined limitation by nutrient, light and space, where simple multiplication is used. The quote above suggests caution when applying multiple limitations on growth, and careful assessment of whether the multiple impacts do not lead to overly constraining limitation.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<section id="primary-producer-nutrient-limitation" class="level3">
<h3 class="anchored" data-anchor-id="primary-producer-nutrient-limitation"><strong><em>9.2.1. Primary producer nutrient limitation</em></strong></h3>
<p>The <strong>nutrient limitation</strong> <em>δ<sub>nutrient&nbsp;</sub></em>scalar in case of limitation based on N only is calculated as</p>
<p><span class="math display">\[\delta_{nutrient} = \frac{DIN}{KN + DIN}\]</span></p>
<p>where <strong>DIN = NH3+NO3</strong> (ammonia and nitrate concentration) and <em>KN</em> is the half saturation constant of nutrient uptake (KN_XXX) in mgN m<sup>-3</sup></p>
<p>Nutrient limitation is calculated by the <em>Nutrient_Lim()</em> routine in <strong>atprocess.c</strong></p>
<p>When phytoplankton is limited by multiple nutrients, such as N and Si in the case of diatoms, the choice of limitation equation is set by the flagnut parameter. This parameter calls three types of nutrient limitation options – Leibig, multicative, and the ERSEM WQI based option. Only the first two options have been more widely used, whereas the third version has been implemented only for testing purposes. The nutrient limitation scalar for the three options is calculated as:</p>
<p><span class="math display">\[\delta_{nutrient} = \min\left( \frac{DIN}{KN + DIN},\frac{Si}{KS + Si} \right)\ \ \ \ \ \ \ \ \ \ \ \ \ if\ Leibig\ limitation\ is\ used\ (flagnut = 0)\]</span></p>
<p><span class="math display">\[\delta_{nutrient} = \sqrt{\frac{DIN}{KN + DIN} \bullet \frac{Si}{KS + Si}}\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ if\ multicative\ limitation\ is\ used\ (flagnut = 1)\]</span></p>
<p><span class="math display">\[\delta_{nutrient} = \frac{2}{\left( \frac{DIN}{KN + DIN} + \frac{Si}{KS + Si} \right)}\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ if\ WQI\ \ limitation\ is\ used\ (flagnut = 2)\]</span></p>
<p>For illustration, assuming a hypothetical case of DIN and Si concentrations of 100 mg m<sup>-3</sup> and half saturation values for N and Si at 50 mg m<sup>-3</sup> and 20 mg m<sup>-3</sup> the three limitation cases would results in scalars of 0.67, 0.75 and 1.41.</p>
<p>Atlantis has an optional nutrient limitation for one additional tracer of micronutrients (currently used for iron). This option is set when flagmicro=1. If micronutrient limitation is used, half-saturation constants (KF_XXX) should be provided for all primary producer groups. Limitation by additional nutrients will be done according to the flagnut option described above.</p>
<p>In models that track P and C (TrackAtomicRatio =1 in <em>run.prm</em> file) primary production will also be limited by P and C. Further details are available on the <a href="https://confluence.csiro.au/display/Atlantis/2013/04/23/New+Phosphorus+tracking">wiki</a>.</p>
</section>
<section id="primary-producer-light-limitation" class="level3">
<h3 class="anchored" data-anchor-id="primary-producer-light-limitation"><strong><em>9.2.2. Primary producer light limitation</em></strong></h3>
<p>Photosynthesis in the model is potentially light-limited and the available suitable light (W m<sup>-2</sup>) – the photosytenthically active radiation (PAR) - in the water column and at the sediment surface is modelled explicitly. The <strong>light limitation</strong> <em>δ<sub>light&nbsp;</sub></em>factor is calculated in <em>Light_Lim()</em> in <strong>atproces.c</strong> as:</p>
<p><span class="math display">\[\delta_{light} = \min\left( \frac{IRR}{KI},1 \right)\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ if\ light\ adaptation\ is\ not\ used\ (flaglight = 0)\]</span></p>
<p>where <em>IRR</em> is the amount of light available for the photosynthesis and <em>KI</em> is the light saturation coefficient (KI_XXX) in W m<sup>-2</sup>.</p>
<p>Atlantis allows for <strong>light adaptation</strong> of primary producers to be represented, which is intended to capture their ability to rapidly adapt to different light conditions. It is used when flaglight = 1 and requires four additional parameters (KIOP_min, KIOP_shift, KI_avail, K_addepth), used to calculate <em>optIRR</em> in <em>Box_Light_Process()</em> in <strong><u>atbiophysics.c</u></strong></p>
<p>If light adaptation is used the light limitation is calculated as</p>
<p><span class="math display">\[\delta_{light} = \min\left( \frac{IRR}{KI + optIRR},1 \right)\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ if\ light\ adaptation\ is\ used\ (flaglight = 1)\]</span></p>
<p>The amount of light falling on the surface of the top layer is either provided through external time series or calculated in Atlantis depending on the lim_sun_hours setting (described in chapter 5, Physics).</p>
<p>The <strong>light attenuation</strong> <em>Kd</em> through the water column (described in Fulton 2001) defines the availability of light to primary producers at different depths and is calculated as</p>
<p><span class="math display">\[Kd = k_{w} + k_{IS} + k_{PN} \bullet \sum_{i = phytopl}^{}{B_{i} + k_{DON} \bullet B_{DON} + k_{DL} \bullet \left( B_{DL} + B_{DR} \right) + k_{SED} \bullet B_{SED}}\]</span></p>
<p>The equation shows that light attenuation is simply defined by the background blue water light absorption coefficient (<em>k<sub>w</sub></em>), additional light absorption due to tanins and other non-biological particulate matter that is important in coastal areas (set in <em>k<sub>IS</sub></em>), and concentration or biomass of light attenuating organisms (plankton) or substances (DON, detrituts, sediments) multiplied by the corresponding light attenuation coefficient. The attenuation is caused by:</p>
<ol type="1">
<li><p>background light absorption by water itself (<em>k<sub>w</sub></em>) – used in the PPBIM model.</p></li>
<li><p>additional background light absorption by water due to tanins and other non-biological particulate matter that is important in coastal areas (set in <em>k<sub>IS</sub></em>). Different coefficients (k_w_shallow and k_w_deep) are used for coastal and oceanic water. This second term of light absorption was not used in PPBIM but was taken from ERSEM</p></li>
<li><p>phytoplankton, with k_PN defining coefficient of light absorption by particulate matter, such as phytoplankton</p></li>
<li><p>dissolved organic nitrogen and the relevant coefficient k_DON</p></li>
<li><p>labile and refractory detritus and the coefficient k_DL</p></li>
<li><p>light absorption by suspended sediment k_SED, used only if the model is estuary (flagIsEstuary=1)</p></li>
</ol>
</section>
<section id="primary-macrophyte-producer-space-limitation" class="level3">
<h3 class="anchored" data-anchor-id="primary-macrophyte-producer-space-limitation"><strong><em>9.2.3. Primary macrophyte producer space limitation</em></strong></h3>
<p>Space limitation is only applied to epibenthic primary producers, defined by PHYTOBEN or SEAGRASS</p>
<p>The <strong>space limitation</strong> <em>δ<sub>space&nbsp;</sub></em>factor is calculated as</p>
<p><span class="math display">\[\delta_{space} = \frac{B_{PP}}{SPmax \bullet P_{area}}\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \]</span></p>
<p>where <em>SPmax</em> is the coefficient setting maximum biomass of the epibenthic primary producer (XXmax) in mgN m<sup>-2</sup> and <em>P<sub>area&nbsp;</sub></em>is the proportion of the cell covered with suitable habitat scaled by the habitat degradation scalar (if used). Space limitation is activated using flagXXXlim.</p>
</section>
<section id="effect-of-eddies-on-primary-production" class="level3">
<h3 class="anchored" data-anchor-id="effect-of-eddies-on-primary-production"><strong><em>9.2.4 Effect of eddies on primary production</em></strong></h3>
<p>The effect of eddies, or the <strong>eddy scalar</strong> <em>δ<sub>eddy</sub></em>, on primary production is modelled by simple scaling (multiplication) using the user defined eddy_scale parameter. This dimensionless parameters is multiplied by the eddy strength in a given box to obtain <em>δ<sub>eddy</sub></em></p>
<p><span class="math display">\[\delta_{eddy} = {eddy}_{scale} \bullet {eddy}_{strength}\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \]</span></p>
<p>Eddy strengths for each box and each quarter are set in the <em>physics.prm</em> file (see Chapter 5.2, section III).</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Effect of eddies on primary production</strong></p>
<p>There is good evidence to suggest that the presence of eddies significantly enhances primary production through vertical water column mixing creating spatial and seasonal variation. There are two ways to add this variation in Atlantis and users are encouraged to make use of these parameters.</p>
<p>First, the effects of eddies on vertical mixing alone can be included through the parameters in the <em>physics.prm</em> (eddy_vertmix, eddy_S1 to eddy_S2, eddy_mixscale – see chapter 5.2, section III). This should increase vertical mixing in the water column and bring extra nutrients to the surface.</p>
<p>Second, the user can add an extra scalar <em>δ<sub>eddy&nbsp;</sub></em>on primary production through the eddy_scale parameter in the <em>biology.prm</em> file. This scalar will take the eddy values and re-scale the primary production calculated for the available amount of light and nutrients based on eddy strength. The second approach assumes that additional nutrients are implicitly supplied through eddies, increasing primary production (these nutrients do not appear in the tracer values but represent just an assumed increase). This <em>ad hoc</em> approach can be taken if the model fails to accurately reflect spatial and seasonal variation in primary production, but the explicit dynamics of nutrient turnover is not resolved accurately enough to enhance primary production sufficiently through nutrient mixing alone. It is not recommended for models that focus on nutrient dynamics and lower trophic levels, but might be useful for models that focus on the dynamics at higher trophic levels and socio-economic aspects.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</section>
</section>
<section id="primary-producer-mortality" class="level2">
<h2 class="anchored" data-anchor-id="primary-producer-mortality"><strong>9.3. Primary producer mortality</strong></h2>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Primary producers in water column, sediments and epibenthos</strong></p>
<p>Primary producer processes are handled differently depending on whether they are in the water column (WC), sediments (SED) or epibenthic layer (EPI).</p>
<p>Four primary producer groups are tracked both in the WC and SED (SM_PHY, LG_PHY, DINOFLAG, MICROPHTYBENTHOS). Two groups are restricted to the epibenthic layer (PHYTOBEN, SEAGRASS)</p>
<p><strong>In WC:</strong> phytoplankton groups (SM_PHY, LG_PHY) can only grow (photosynthesise) when they are in the WC. MICROPHTYBENTHOS can grow both in the WC and in SED; for photosynthesis in the SED they use light conditions at the top of the sediment. DINOFLAG do not photosynthesis in the SED, but they can have mixotrophic feeding. Mortality expressed depends on whether it is in the WC and SED: primary producers are affected by linear mortality (mL_XXX) and optional acidification mortality in both locations, but phytoplankton (SM_PHY and LG_PHY) only suffer lysis (KLYS_XXX) in the WC. DINOFLAG and MICROPHYTOBENTHOS have all mortality terms in both locations. Both lysis and other mortality terms are separately scaled by the <strong><em>external forced optional mortality scalar</em></strong>. All mortality products are transferred into the WC or SED labile detritus pools respectively.</p>
<p><strong>In EPI</strong>: epibenthic primary producers are affected by linear mortality (mL_XXX), extra mortality due to waves or fouling (mS_XXX) and optional acidification mortality. Mortality products are transferred to the SED labile and refractory detritus pools; the split between labile and refractory detritus is determined by the FDL_benthos parameter.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p><strong>Primary producer</strong> mortality in the <strong>water column and sediments</strong> is modelled as</p>
<p><span class="math display">\[M_{PP} = \ \left( \left( \frac{KLYS \bullet B_{PP}}{\delta_{nutrient} + 0.1} \right) + \left( mL \bullet B_{PP} + mA \bullet B_{PP} \right) \right) \bullet mortsc\]</span></p>
<p><strong>Primary producer</strong> mortality in the <strong>epibenthic layer</strong> is modelled as</p>
<p><span class="math display">\[M_{PP,EPI} = \ \left( mL \bullet B_{PP} + mA \bullet B_{PP} + mS \bullet B_{PP} \right) \bullet mortsc + F_{PP}\]</span></p>
<p>where <em>mL</em> (mL_XXX, day<sup>-1</sup>) is the linear mortality rate, <em>KLYS</em> is the lysis rate (KLYS_XXX, day<sup>-1</sup>), <em>mA</em> is the optional acidification mortality, <em>mS</em> is the extra mortality of large epibenthic primary producers calculated below, <em>B<sub>PP</sub></em> is the biomass of the primary producer, <em>mortsc</em> is the external mortality scalar (from forcing files) and <em>F<sub>PP</sub></em> is an optional fishing mortality that is sometimes applied to macroalgae.</p>
<p>The <strong>extra mortality <em>mS</em></strong> in <strong>large</strong> <strong>primary producers</strong> includes mortality due to waves and fouling:</p>
<p><span class="math display">\[mS_{MA} = \ STRESS \bullet \ mS\_\_ MA\]</span></p>
<p><span class="math display">\[mS_{SG} = \ DIN \bullet \ mS\_\_ SG\]</span></p>
<p>where <em>mS_MA</em> is the algal mortality rate (day<sup>-1</sup>) due to abrasion by bottom stress or waves (<em>STRESS</em>), provided through external forcing or calculated in the model (see Chapter 5, Physics) and <em>mS_SG</em> is the seagrass mortality rate (day<sup>-1</sup>) due to fouling by epiphytes in water of high nutrient concentration, and <em>DIN</em> is the concentration of NH3 and NO3.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>What is the difference between lysis and linear mortality?</strong></p>
<p>Lysis represents cell leakage and is inversely related to nutrient concentration, whereas linear mortality is just a constant mortality term. PPBIM only had the linear mortality, but ERSEM also included lysis, and both options are included in Atlantis. The users have an option to use either or both, depending on the mortality parameter values set.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>Finally, the extra mortality will also include any <strong>optional acidification induced mortality</strong>, described in details <a href="https://confluence.csiro.au/pages/viewpage.action?spaceKey=Atlantis&amp;postingDay=2012%2F10%2F19&amp;title=Handling+effects+of+acidification">here</a> and <a href="https://confluence.csiro.au/pages/viewpage.action?spaceKey=Atlantis&amp;postingDay=2015%2F5%2F14&amp;title=Acidification+induced+mortality">here</a> and in Chapter 13. The acidification mortality is only included if the model explicitly includes pH effects (flagmodelpH=1) and primary producers are identified as pH sensitive (flagpHsensitive_XXX=1).</p>
<p>Primary producer mortality is handled by <em>Primary_Production()</em> in <strong>atprocess.c</strong> and <em>Phytoplankton_ Process()</em> in <strong>atGroupProcess.c</strong> routines.</p>
</section>
<section id="encystment-of-primary-producers" class="level2">
<h2 class="anchored" data-anchor-id="encystment-of-primary-producers"><strong>9.4. Encystment of primary producers</strong></h2>
<p>It is possible to include the encystment of primary producers, triggered by time of the year or temperature. This option is only modelled for groups that have isOverWinter set to 1 in the <em>functional_groups.csv</em> file and is described in further detail on the wiki <a href="https://confluence.csiro.au/pages/viewpage.action?pageId=43155510">here</a>.</p>
</section>
<section id="effect-of-temperature-and-salinity-on-primary-production" class="level2">
<h2 class="anchored" data-anchor-id="effect-of-temperature-and-salinity-on-primary-production"><strong>9.5. Effect of temperature and salinity on primary production</strong></h2>
<p>Both light saturation (KI_XXX_T15) and maximum growth rate (mum_XXX_T15) are dependent on <strong>temperature</strong> and will be scaled by <em>Tcorr</em> calculated based on the current water temperature as described in chapter 13.</p>
<p>The <strong>salinity</strong> effect on light saturation and maximum growth rate are optional, where the <em>Scorr</em> is applied only when:</p>
<p>1) an organism is identified as sensitive to salinity, with flagSaltSensitive_XXX</p>
<p>2) an organism an outside the salinity limits defined with XXX_min_move_salt and XXX_max_ move_salt</p>
<p>If the two conditions are fulfilled, the <em>Scorr</em> scalar provided by the user as salt_correction_XXX parameter, is applied to KI_XXX_T15 and mum_XXX_T15 (along with the <em>Tcorr</em> described above).</p>
</section>
<section id="growth-of-mixotrophic-primary-producers" class="level2">
<h2 class="anchored" data-anchor-id="growth-of-mixotrophic-primary-producers"><strong>9.6. Growth of mixotrophic primary producers</strong></h2>
<p>Atlantis includes a mixotrophic dinaflagelate group DINOFLAG, the growth of which is determined by both primary production and heterotrophic consumption.</p>
<p>Phtosynthetic growth of DINOFLAG is as for other primary producers, although there is some increase in efficiency at low light levels, represented here by increasing effective light available via the following equation (which is only used if light levels drop below 10% of KI_XXX):</p>
<p><span class="math inline">\({IRR}_{sp} = KI_{XXX} \bullet \ (0.01 \bullet IRR + 0.018)\)</span></p>
<p>Mixotrophic growth due to consumption is calculated by first determining potential grazing (using the standard feeding routine applied to consumers). Phagotrophic growth then tops-up photosynthetic growth to the maximum possible rate, as if nutrients were non-limiting, unless the maximum growth rate is higher than what could be physically grazed. In this way grazing leads to additional growth compared to photosynthesis alone. Unlike in other consumers mixotrophic groups have no excretion (production of NH) as the nutrients are used in the photosynthetic growth pathway.</p>
<p>Nutrient stress related lysis of DINOFLAG is also corrected to account for the mitigating effects of phagotrophy), with the corection dictated by the proportion of photosynthetic nutrient made up by the grazing.</p>
<table class="caption-top table">
<caption><span id="_Toc526762765" class="anchor"></span>Table 13. Parameters used for primary producer processes</caption>
<thead>
<tr class="header">
<th><strong>Parameter</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong><em>Key parameters defining primary production and primary producer mortality</em></strong></td>
<td></td>
</tr>
<tr class="even">
<td>KN_XXX</td>
<td>Half-saturation constant for XXX growth on DIN (mgN m<sup>-3</sup>)</td>
</tr>
<tr class="odd">
<td>KS_XXX</td>
<td>Half-saturation constant for XXX growth on Si (mgN m<sup>-3</sup> ) – only important for Si limited groups, as defined in the <em>functional_groups.csv</em> file</td>
</tr>
<tr class="even">
<td>KI_XXX_T15</td>
<td>Light saturation of XXX at 15C (W m<sup>-2</sup>) - this parameter is temperature dependent (and potentially salinity dependent), and a scalar is applied based on current temperature (and salinity) conditions.</td>
</tr>
<tr class="odd">
<td>mum_XXX_T15</td>
<td>Maximum primary producer XXX growth rate at 15C (mgN mgN<sup>-1</sup> day<sup>-1</sup>) – this parameter is temperature dependent (and potentially salinity dependent), and a scalar is applied based on current temperature (and salinity) conditions.</td>
</tr>
<tr class="even">
<td>XXmax</td>
<td>Maximum biomass of large epibenthic primary producers (phytobenthos and macroalgae) (mgN m<sup>-2</sup>)</td>
</tr>
<tr class="odd">
<td>mL_XXX</td>
<td>Linear mortality of primary producers (day<sup>-1</sup>)</td>
</tr>
<tr class="even">
<td>KLYS_XXX</td>
<td>Lysis rate of primary producers (day<sup>-1</sup>)</td>
</tr>
<tr class="odd">
<td>mS_XXX</td>
<td>Extra mortality of macroalgae (PHYTOBEN) due to wave action (day<sup>-1</sup>)</td>
</tr>
<tr class="even">
<td>mS_XXX</td>
<td>Extra mortality of seagrasses (SEAGRASS) due to fouling (day<sup>-1</sup>)</td>
</tr>
<tr class="odd">
<td><strong><em>Further modification of primary production</em></strong></td>
<td></td>
</tr>
<tr class="even">
<td>flaglight</td>
<td>Flag indication whether light adaptation in primary producers is activated (0=no, 1=yes), It set to 1 (i.e.&nbsp;activated) it will increase production at low light levels, but it will also require four additional parameters: KIOP_min, KIOP_shift, KI_avail, K_addepth (see SETas model example of <em>biology.prm</em> file)</td>
</tr>
<tr class="odd">
<td>flag_macro_model</td>
<td>Flag indicated whetehr a more detailed seagrass model is activated, see details <a href="https://confluence.csiro.au/display/Atlantis/2012/12/12/new+seagrass+modelling+option">here</a> (0=no, 1=yes). If the flag is set to 1 then three separate pools will be initiated and tracked for each SEAGRASS group type: main biomass, epiphyte biomass and below ground biomass</td>
</tr>
<tr class="even">
<td>flagnut</td>
<td>Nutrient limitation form used if primary production is limited by several nutrients (0=Leibig’s law where limitation is set by the most limiting nutrient; 1= multiplicative; 2=WQI)</td>
</tr>
<tr class="odd">
<td>flagmicronut</td>
<td><p>Flag indications whetehr there is miconutrient limitation (0=no, 1=yes)</p>
<p>If set to 1, half-saturation constant for gthe micronutrient (typically Fe) is required (KF_XXX)</p></td>
</tr>
<tr class="even">
<td>eddy_scale</td>
<td>Coefficient scaling eddy strength impact on phytoplankton growth</td>
</tr>
<tr class="odd">
<td>swr_scalar</td>
<td>Proportion of shortwave radiation available to photosynthesis – usually set to 0.45</td>
</tr>
<tr class="even">
<td>swr_const</td>
<td>Constant in the equation for the shortwave radiation available to photosynthesis – usually set to 1367.0</td>
</tr>
<tr class="odd">
<td>swr_cos_coefft</td>
<td>Coefficient of the cos term in the equation for the shortwave radiation available to photosynthesis – usually set to 0.034</td>
</tr>
<tr class="even">
<td>swr_cos_offset</td>
<td>Offset term in the cos term in the equation for the shortwave radiation available to photosynthesis – usually set to 0.0</td>
</tr>
<tr class="odd">
<td>MB_wc</td>
<td>Scalar for microphytobenthos growth in the water column – growth should be strongly reduced as it is not on the substrate (which is its primary habitat). Only important when microphytobenthos is being modelled.</td>
</tr>
<tr class="even">
<td><h2 id="section" class="anchored"></h2>
<p><strong>If TrackAtomicRatio is set to 1</strong> in <em>run.prm</em> file and the model tracks N:P:C many additional parameters are required to describe P, and C effect on primary production. See details <a href="https://confluence.csiro.au/display/Atlantis/2013/04/23/New+Phosphorus+tracking">here</a> and sample SETas model <em>biology.prm</em> file</p></td>
<td></td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="consumer-processes" class="level1">
<h1>10. CONSUMER PROCESSES</h1>
<p>Consumers in Atlantis are modelled as biomass pools, age-structured biomass pools or age-structured groups. The age-structured groups are typically used for vertebrates and in the text below the terms age-structured groups and vertebrates may be used interchangeably. Non-vertebrates are largely modelled as biomass pools, which in age-structured biomass pools can be explicitly subdivided into ontogenetic stages (similar to stanzas in Ecosim).</p>
<p>The group types FISH, MAMMAL, SHARK, BIRD and FISH_INVERT are automatically assigned to age-structured groups. Other group types that have NumCohort=1 in the <em>functional_groups.csv</em> file are modelled as biomass pools, and if NumCohort&gt;1 they are modelled as age-structured biomass pools, where NumCohort parameter determines how many biomass pools to model.</p>
<section id="fluxes-in-consumer-biomass-pools" class="level2">
<h2 class="anchored" data-anchor-id="fluxes-in-consumer-biomass-pools"><strong>10.1. Fluxes in consumer biomass pools</strong></h2>
<p>In <strong>biomass pool or age-structured biomass pool consumers</strong> (CP in the following equations) the only variable tracked is N (i.e.&nbsp;not Si or the micronutrient; although when P and C tracking is enabled then N, P and C are all tracked for consumers). The flux through a consumer biomass pool is determined by growth (<em>G<sub>CP</sub></em>), natural mortality (<em>M<sub>CP</sub></em>), grazing (predation) for <em>i</em> predators (<em>Gr<sub>CP,i</sub></em>) and optional fishing mortality (<em>F<sub>CP</sub></em>). In addition, there is an optional encystment out of the system at certain times of the year or temperatures (<em>Ec<sub>out</sub></em>) and transfer back from cysts into the system when conditions are suitable (<em>Ec<sub>in</sub></em>).</p>
<p><strong>Biomass pool</strong> consumers maybe in the sediment, epibenthic layer or water column (depending on group type). A single N (P, C) pool is tracked per biomass pool age group (see Table 14):</p>
<p><span class="math display">\[\frac{d(CP)}{dt} = G_{PP} - M_{PP} - \sum_{i = predators}^{}{Gr_{PP,i}} - F_{CP} - Ec_{out} + Ec_{in}\]</span></p>
<p>In <strong>age-structured consumers</strong> (abbreviated as CX) the nitrogen pool is partitioned into the reserve (RN) and structural (SN) nitrogen of an average individual of each age group. Atlantis also tracks the numbers (Num) of individuals per age group. As a result for each age group three state variables are tracked – RN, SN and Num.</p>
<p><strong>Fluxes through the RN and SN pool</strong> are determined by growth (G<sub>CX</sub>), proportion of G<sub>CX</sub> allocated to SN versus RN (λ) and spawn (Sp<sub>CX</sub>), which is taken out of the RN pool on spawning (with timing defined in the <em>biology.prm</em> file, but typically once per year). Details determining λ are given in chapter 10.5.1. Note, that if respiration is included, then growth (G<sub>CX</sub>) can also be negative, and all energy required to cover respiration (or maintenance) costs will be taken out from the RN pool (see chapter 10.4)</p>
<p><span class="math display">\[\frac{d(RN_{CX})}{dt} = G_{CX} \bullet \lambda - Sp_{CX}\]</span></p>
<p><span class="math display">\[\frac{d(SN_{CX})}{dt} = G_{CX} \bullet (1 - \lambda)\]</span></p>
<p>The equations above show that RN pool can change through the year as it will suddenly decrease after the spawn. The SN pool cannot decrease.</p>
<p><strong>The flux of Num</strong> is tracked through individuals. The number of individuals in an age group depends on recruitment or ageing into the age group (<em>Nm<sub>CX-1,&nbsp;ageup</sub></em>), numbers ageing up into the next group (<em>Nm<sub>CX,&nbsp;ageup</sub></em>), natural mortality (<em>M<sub>CX</sub></em>), predation from <em>i</em> predators (<em>Gr<sub>CX,i</sub></em>), optional migration out (<em>T<sub>CXout</sub></em>) and into (<em>T<sub>CXin</sub></em>) the model domain and optional fishing mortality (<em>F<sub>CX</sub></em>).</p>
<p><span class="math display">\[\frac{d(Nm_{CX})}{dt} = Nm_{CX - 1,ageup} - Nm_{CX,ageup} - M_{CX} - \sum_{i = pred}^{}{Gr_{CX,i}} - F_{CX} - T_{CXout} + T_{CXin}\]</span></p>
<p>The numbers are converted to biomass as (SN + RN) · Nums.</p>
</section>
<section id="consumer-feeding" class="level2">
<h2 class="anchored" data-anchor-id="consumer-feeding"><strong>10.2. Consumer feeding</strong></h2>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>What determines growth in consumers?</strong></p>
<p><strong>Growth in biomass pool consumers</strong> is determined by the intake or grazing (Gr) and assimilation efficiency. The consumed food or grazing term (Gr) is then assimilated according to the assimilation efficiencies for different food types (live, plant, labile detritus and refractory detritus). The unassimilated food is sent to the labile detritus, refractory detritus and ammonia pools according to the parameter determining detritus allocation (chapter 10.7), whereas all assimilated food is sent to growth.</p>
<p><strong>Growth in age-structured consumers</strong> is determined by the intake or grazing (Gr), assimilation efficiency and optional maintenance or respiration costs (Rs). The consumed food or grazing term (Gr) is then assimilated according to the assimilation efficiencies for different food types (live, plant, labile detritus and refractory detritus). The unassimilated food is sent to the labile detritus, refractory detritus and ammonia pools according to the parameter determining detritus allocation (chapter 10.7). The assimilated food represents a temporary energy pool (A). This A pool is used to meet the optional maintenance or respiration costs (Rs) and all remaining energy is allocated to SN and RN, i.e.&nbsp;to growth. The proportion of energy going into SN and RN is determined by λ, which is calculated at each time step dynamically depending on the individual’s condition and parameters determining energy allocation rules. The SN pool cannot decrease, whereas the RN pool can decrease, as it is used for reproduction and for meeting optional maintenance needs if assimilated food is insufficient.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>The following chapters will describe these key processes: grazing (Gr), assimilation (A), maintenance (Rs) and energy allocation to SN and RN.</p>
<section id="grazing" class="level3">
<h3 class="anchored" data-anchor-id="grazing"><em>10.2.1 Grazing</em></h3>
<p>Grazing or predatory interactions are modelled in a similar way for both biomass pools and age-structured groups, except that age-structured and biomass pool groups have different options for refuge from predation. Feeding interactions are modelled through biomass, which in age-structured groups is then converted to numbers to track individual mortality. In Atlantis predatory interactions are determined by:</p>
<ol type="1">
<li><p>Physical overlap – prey and predator must be in the same cell at the same time (determined by vertical and horizontal distribution parameters), and if prey is in the sediment the predator must be able to reach it.</p></li>
<li><p>Diet connection matrix (pPREY matrix or detailed ontogenetic diet preferences) that indicate maximum availability of prey to a predator. The actual realised consumption will depend on refuge factors, but if the value in the pPREY matrix is set to 0, no predation will occur.</p></li>
<li><p>Gape limitation for age-structured prey – prey that is too small or too big for the predator (either age-structured or biomass pool) will not be consumed.</p></li>
<li><p>Habitat refuge.</p></li>
<li><p>Environmental factors (temperature, salinity, pH) that can modify predator’s feeding rates, prey’s availability and nutritional content.</p></li>
</ol>
<p>Throughout this chapter for brevity we will refer to both biomass pool (CP) and age-structured (CX) consumers as CX, as in many cases the processes are modelled in the same way for both groups; processes that are modelled differently will be identified throughout.</p>
<p>At the time of writing, Atlantis has twelve different options for modelling a predator’s feeding functional response to the prey’s biomass. There is a lot of theory and debate on what kind of functional response most accurately depicts predator’s behaviour (e.g.&nbsp;Hunsicker et al.&nbsp;2011) and this topic is not covered here.</p>
<p>Currently most Atlantis models use the modified Holling type II response, described in <a href="https://confluence.csiro.au/display/Atlantis/Useful+Reading?preview=/43155482/341447378/Link_et_al_2011_NEUS_tech_memo.pdf">Murray &amp; Parslow (1997)</a> for the Port Phillip Bay Integrated Model. The grazing term (<em>Gr</em>), or the amount of biomass of a specific prey (<em>prey</em>) consumed by a consumer CX (note, this now includes both CP and CX consumers), in the modified Holling type II response is calculated as</p>
<p><span class="math display">\[Gr_{prey} = \ \frac{B \bullet C \bullet B_{prey}^{*}}{1 + \frac{C \bullet \sum_{i}^{}\left( E_{i} \bullet B_{prey,i}^{*} \right)}{mum}}\]</span></p>
<p>where</p>
<p><span class="math display">\[B_{prey}^{*} = p_{prey,CX} \bullet \delta_{overlap} \bullet \delta_{habitat} \bullet \delta_{size} \bullet B_{prey}\]</span></p>
<p>is the available biomass of <em>prey</em> in the cell after taking into account all refuge (<em>δ</em>) options: <em>δ<sub>overlap</sub></em>, <em>δ<sub>habitat&nbsp;</sub></em>and <em>δ<sub>size</sub></em>. The refuge options are described in detail in the next chapter.</p>
<p><em>B</em> is the feeding biomass of a consumer (predator) CX in a given cell (mgN m<sup>-3</sup>) – a proportion of the consumer biomass may not be feeding if it is not active at that time of day or if it is spawning and has been identified as a group that does not feeding while spawning (feed_while_spawnXXX=0). <em>B<sub>prey</sub></em> is the biomass of prey (or fisheries catch available for consumption) in that cell (mgN m<sup>-3</sup>), <em>p<sub>prey,CX</sub></em> is the maximum availability (ranging from 0 to 1) of the <em>prey</em> to the predator CX defined in the pPREY matrix (or optional ontogenetic diet matrices), <em>C</em> is the clearance rate, similar to a search volume of the consumer CX, and <em>mum</em> is the maximum consumption rate, sometimes referred to as maximum growth rate, of the consumer CX.</p>
<p>The units of <em>C</em> and <em>mum</em> in CP consumers are m<sup>3</sup> mgN<sup>-1</sup> day<sup>-1</sup> and mgN mgN <sup>-1</sup> day<sup>-1</sup>. In CX consumers the units depend on the flagfishrates parameter, which defines whether rates are absolute or per unit of consumer biomass. If rates are absolute (flagfishrates=0) then <em>C</em> units are m<sup>3</sup> ind<sup>-1</sup> day<sup>-1</sup> and <em>mum</em> units are mgN ind<sup>-1</sup> day<sup>-1</sup>. If flagfishrates=1 then <em>C</em> units are m<sup>3</sup> mgN<sup>-1</sup> day<sup>-1</sup> and <em>mum</em> units are mgN mgN <sup>-1</sup> day<sup>-1</sup>.</p>
<p>The <em>E<sub>i</sub></em> is the assimilation efficiency of consumer CX on prey group category <em>i,</em> which represents the four prey group categories defined in Atlantis: living prey and fisheries catches, plant prey, labile detritus and carrion (ie. fisheries discards) and refractory detritus. <strong>Note that carrion is included in the labile detritus food category</strong>, and the EDL assimilation efficiency for species that feed on carrion should be higher than what it would be if the species was feeding on labile detritus alone.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Use of assimilation efficiency <em>E<sub>i</sub></em> in the grazing term</strong></p>
<p>Even though assimilation efficiency is included in the grazing term <em>Gr</em> it <strong>does not</strong> define the final assimilated food, but is only used <strong>in modified Holling feeding</strong> responses (see Chapter 10.3) to scale the maximum consumption rate of a consumer. <strong>Higher assimilation efficiencies lead to lower Gr value</strong>, as the predator need not intake as much food as it can extract more energy from what it does consume.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong><em>Should I use absolute or mass-specific</em> C <em>and</em> mum <em>rates (</em>flagfishrates)<em>?</em></strong></p>
<p>The flagfishrates parameter defines whether <em>mum</em> and <em>C</em> parameters are given per individual or per unit of body weight. It is important to remember that in Atlantis the realised size-at-age is dynamic and will change through time, depending on food availability. Setting flagfishrates to 0 or using absolute <em>C</em> and <em>mum</em> values means that regardless of the realised size-at-age <em>C</em> and <em>mum</em> values of an age group will stay the same throughout the simulation and Atlantis will behave more like an age-structured model. Using mass specific values (flagfishrates=1) means that <em>C</em> and <em>mum</em> will vary from year to year depending on the size of individuals in an age group, and behaviour of Atlantis will be closer to the size-structured models.</p>
<p>Regardless of whether absolute or mass-specific rates are used, separate <em>C</em> and <em>mum</em> parameters are required for each age group, so that C_XXX and mum_XXX strings in the <em>biology.prm</em> file contain as many values as there are groups in an age-structured functional group XXX. This means that Atlantis allows users to assume that the relationship between <em>C</em> and <em>mum</em> and body weight can change with ontogeny.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</section>
<section id="what-are-the-c-and-mum-parameters" class="level3">
<h3 class="anchored" data-anchor-id="what-are-the-c-and-mum-parameters"><em>10.2.2 What are the</em> C <em>and</em> mum <em>parameters?</em></h3>
<p>Some Atlantis users have been confused about the meaning of <em>C</em> and <em>mum</em> parameters in the modified Holling representations. The <em>mum</em> parameter was initially adopted by Murray &amp; Parslow (1997), who replaced the standard Holling Type II maximum ingestion rate parameter (inverse of handling time) with a “maximum growth rate” <em>mum</em> by scaling it by assimilation efficiency:</p>
<table class="caption-top table">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>Origin of <em>mum</em> from Murray and Paslow 1997 (page 29):</strong></p>
<p>“In a classical paper, Holling (1966) discussed the theoretical and empirical basis of the functional response relating ingestion per consumer (G) to food density (P). He identified three functional forms:</p>
<p>I) G = min (C*P, Gmax) (bilinear),</p>
<p>II) G = C*P / (1 + C*P/Gmax) (rectangular hyperbola),</p>
<p>III) G = C*P<sup>2</sup> / (1 + C*P<sup>2</sup>/Gmax) (sigmoid or switching).</p>
<p>The parameter Gmax represents a maximum (food saturated) ingestion rate at high food densities, while C is a measure of grazing efficiency at low food densities. In Type I and II, the parameter C has units of volume cleared per grazer per unit time, and represents a maximum clearance rate.</p>
<p>…</p>
<p>In this model, we have used the rectangular hyperbola without thresholds, Type II, as the standard formulation. However, rather than specify the maximum ingestion rate Gmax, we have specified the maximum growth rate, "mum". This is related to Gmax through growth efficiency E, so that mum = Gmax*E, or Gmax = mum/E.”</p>
<p>The growth efficiency described above is the same as assimilation efficiency in Atlantis.</p></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td></td>
</tr>
<tr class="even">
<td><p><strong>NOTE!</strong></p>
<p><strong>What is <em>C</em>?</strong></p>
<p>­<em>C</em> is often referred to as a clearance rate and it is defined as the volume of water searched, but it is different from an often used gut clearance rate that reflects food assimilation efficiency. The <em>C</em> parameter in Atlantis has units of m<sup>3</sup> ind<sup>-1</sup> day<sup>-1</sup> (or m<sup>3</sup> mgN<sup>-1</sup> day<sup>-1</sup>) and reflects the search or “cleared” volume. Note, however, that <em>C</em> is treated differently from true search volume (vl_XXX, vla_XXX, vlb_XXX) parameters used in predcase=5.</p>
<p>The true search volume parameters vl_XXX (for biomass pools) and vla_XXX and vlb_XXX (for age structured groups) are by default mass-specific and the relationship between body weight and search volume does not change. This means that only one entry is required, rather than a separate entry for each age group.</p>
<p>Further, for demersal species (flagdemXXX =1 in the <em>biology.prm</em> file) the search volume is multiplied by 0.5, but this scaling is not applied to the <em>C</em> parameter.</p>
<p><strong>And what is <em>mum</em>?!</strong></p>
<p>The <em>mum</em> parameter is only used in the modified Holling feeding responses (modified after Muray and Parslow, see above). Even though <em>mum</em> is called a maximum growth rate, it is included in the grazing (feeding) and not the growth equation, so it does not define the final growth. The consumed food will first be assimilated and the actual growth will then depend on other factors, such as respiration, reproduction and energy allocation to SN and RN.</p>
<p>As described in Muray and Parslow (1997) mum relates to the maximum ingestion rate (<em>Gmax</em>) as <em>mum = Gmax*E</em>, where <em>E</em> is the assimilation efficiency. The maximum ingestion rate is in turn the inverse of a more familiar “handling time” parameter (<em>ht</em>). Hence <em>mum</em> can also be seen as an inverse of the handling time as <em>mum=E/ht</em></p></td>
</tr>
</tbody>
</table>
<p>There is no clear consensus among currently used models as to the exact relationship between <em>C</em> and <em>mum</em> parameters. Theoretically the <em>mum</em> value should always be larger than <em>C</em>, but in some models and for some species the opposite is true. This is explored in Figure 10, which shows values of <em>Gr<sub>prey</sub></em> for different prey biomass, and different <em>C</em> and <em>mum</em> rates. In general mum sets the level of the eventual asymptote – the higher the <em>mum</em> the higher the realised feeding regardless of the <em>C</em> used. The ratio of <em>C</em> and <em>mum</em> dictates the steepness of the curve. As the ratio <em>C:mum</em> increases (i.e.&nbsp;the more that <em>C</em> exceeds <em>mum</em>) the gentler the curve, rising gradually in a smooth arc to the asymptote. As the ratio <em>C:mum</em> decreases (i.e.&nbsp;the more that <em>mum</em> exceeds <em>C</em>) the more the curve is converted to a flat rate across all but the lowest of prey biomass levels.</p>
<p><img src="markdown_output/media/image33.emf" style="width:5.21686in;height:3.56604in"></p>
<p><span id="_Toc526762790" class="anchor"></span>Figure 10. The value of grazing term (<em>Gr</em>) as a function of prey biomass (ranging from 0 to 1x of the predator biomass) and different assumptions on the ratio between <em>C</em> and <em>mum</em>. Predators biomass is 1000mgN, <em>C</em> = 100 m<sup>3</sup> ind<sup>-1</sup> day<sup>-1</sup>. The calculations assume only one prey, no refuge (<em>δ<sub>refuge</sub></em>=1), maximum availability p<sub>prey,CX</sub> of 1, and assimilation efficiencies <em>E<sub>prey,i</sub></em> of 1. Black: <em>mum</em> = <em>C,</em> orange: <em>mum</em> = <em>C</em>*3, blue: <em>mum</em> =<em>C</em>/5</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong><em>Ensuring meaningful prey choice and diets</em></strong></p>
<p>Atlantis <strong>does not</strong> implement optimal prey choice. The clearance rate <em>C</em> is specific to each predator but identical for all prey. Once all the available prey is assessed (based on overlap, size and habitat refuge) the consumption is applied <strong>uniformly</strong> <strong>across all available prey groups proportionally to the available prey biomass</strong> (prey biomass · pPREY). This means that if the available prey includes 1000mg of clams and 1mg of fish, and the clearance rate determines that only 10% of all available biomass can be consumed for a given time step, the predator will ingest 10% of available clam biomass and 10% of available fish biomass or 100mg of clams and 0.1mg of fish.</p>
<p>This has important implications for optimising the maximum available biomass p<sub>prey,CX</sub> in the prey availability matrix. The availability of biomass pool and especially detritus prey should be low for predators that prefer to eat fish. Otherwise a predator, such as a seal, might entirely fill up on invertebrate prey and impose no top-down control on fishes. Note, this can happen even if the availability of invertebrates to seals is as low as 0.001, as the final proportion in the diet is determined by availability and the biomass in the box. It is of crucial importance to <strong>carefully inspect realised diets</strong> from <em>DietCheck.txt</em> and <em>DetailedDietCheck.txt</em> output files, which can be analysed using for example atlantistools R package or other tools described in chapter 2.9.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<table class="caption-top table">
<caption><span id="_Toc526762766" class="anchor"></span>Table 14. General feeding parameters and units</caption>
<colgroup>
<col style="width: 13%">
<col style="width: 86%">
</colgroup>
<thead>
<tr class="header">
<th><strong>Parameter</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td colspan="2"><strong><em>Key parameters for modified Holling Type II functional response</em></strong></td>
</tr>
<tr class="even">
<td>predcase_XXX</td>
<td>Flag indicating the feeding functional response to use for a predator XXX</td>
</tr>
<tr class="odd">
<td>flagfishrates</td>
<td>Flag defining whether fully age-structured C_ and mum_ parameters are absolute (given per individual) or mass-specific (per unit of body mass)</td>
</tr>
<tr class="even">
<td>mum_XXX_T15</td>
<td>Maximum ingestion rate of biomass pools at 15C</td>
</tr>
<tr class="odd">
<td>C_XXX_T15</td>
<td>Clearance rate of biomass pools at 15C</td>
</tr>
<tr class="even">
<td>mum_XXX</td>
<td>Age specific maximum ingestion rate for age structured groups: mgN ind<sup>-1</sup> day<sup>-1</sup> or mgN mgN<sup>-1</sup> day<sup>-1</sup></td>
</tr>
<tr class="odd">
<td>C_XXX</td>
<td>Age specific clearance rate for age structured groups: m<sup>3</sup> mgN<sup>-1</sup> day<sup>-1</sup> or m<sup>3</sup> ind<sup>-1</sup> day<sup>-1</sup></td>
</tr>
<tr class="even">
<td><p>pPREY1XXX1 pPREY1XXX2</p>
<p>pPREY2XXX1</p>
<p>pPREY2XXX1</p></td>
<td>Maximum availability of prey to a predator XXX (given as a vector with the column the prey group in the same order as in the <em>functional_groups.csv</em> file). Four values are given for each predator-prey combination: juv-juv, juv-ad, ad-juv, ad-ad.</td>
</tr>
<tr class="odd">
<td>E_XXX</td>
<td>Assimilation efficiency when feeding on animal prey</td>
</tr>
<tr class="even">
<td>EPlant_XXX</td>
<td>Assimilation efficiency when feeding on plant prey</td>
</tr>
<tr class="odd">
<td>EDR_XXX</td>
<td>Assimilation efficiency when feeding on refractory detritus</td>
</tr>
<tr class="even">
<td>EDL_XXX</td>
<td>Assimilation efficiency when feeding on labile detritus</td>
</tr>
<tr class="odd">
<td colspan="2"><strong>Parameters for optional further details</strong></td>
</tr>
<tr class="even">
<td>flag_fine_ontogenetic_diets</td>
<td>Flag indicating whether more refined, age-specific, prey availability should be used (so an availability value is required per predator age class rather than per stage (typically juvenile and adult))</td>
</tr>
<tr class="odd">
<td>p_split_YY</td>
<td>Flag identifying prey groups YY for which age-specific diets should be given</td>
</tr>
<tr class="even">
<td>age_structured_prey_XXX</td>
<td>Flag identifying predators for which age-specific diets should be given</td>
</tr>
<tr class="odd">
<td>p_YYXXX</td>
<td>A vector of maximum prey availability for each combination of prey YY and predator XXX identified in p_split_YY and age_structured_prey_XXX parameters</td>
</tr>
</tbody>
</table>
</section>
<section id="what-determines-prey-biomass-available-for-predation" class="level3">
<h3 class="anchored" data-anchor-id="what-determines-prey-biomass-available-for-predation"><strong><em>10.2.3. What determines prey biomass available for predation?</em></strong></h3>
<p>The proportion of prey biomass available to a predator at any given time is determined dynamically as</p>
<p><span class="math display">\[B_{prey}^{*} = p_{prey,CX} \bullet \delta_{overlap} \bullet \delta_{habitat} \bullet \delta_{size} \bullet B_{prey}\]</span></p>
<p>This means that the available prey biomass depends on:</p>
<ol type="1">
<li><p>Static parameter p<sub>prey,CX</sub> (which can range from 0 to 1) defined in the pPREY matrix (or optional ontogenetic diet matrices)</p></li>
<li><p>The <em>δ<sub>overlap</sub></em> refuge defined by co-occurrence in the same cell at a same time step and the predator is active at that time of day. This is determined by horizontal and vertical distribution as well as activity parameters.</p></li>
<li><p>The <em>δ<sub>overlap</sub></em> refuge also defined by the predator’s ability to access at least one of the habitats inhabited by prey. This is determined by the <strong>static</strong> habitat_XXX parameter and is activated only if a secies is habitat dependent (see below)</p></li>
<li><p>The <em>δ<sub>size</sub></em> refuge for age structured prey. This is determined by gape limitation of the predator relative to the prey size.</p></li>
<li><p>The <em>δ<sub>habitat</sub></em> refuge for infaunal (_INF) invertebrates (CX) defining access to biomass pool prey buried in the sediment. This is determined by the depth of the oxygen layer and depth predators can dig into</p></li>
<li><p>The optional <em>δ<sub>habitat</sub></em> refuge for age structured prey. This is determined by optional habitat dependency and habitat refuge parameters.</p></li>
</ol>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>What is <em>p<sub>prey,CX</sub></em>?</strong></p>
<p>This parameter defines the maximum proportion of the prey biomass available to a consumer at a given time. If &lt; 1 then it means that at any given time the consumer cannot access all the prey biomass even if it can fit the prey into its mouth and the prey is not protected by a refuge.</p>
<p>The parameter is similar to the vulnerability parameter in EwE and is influenced by the feeding arena theory.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>The <em>δ<sub>overlap&nbsp;</sub></em>is dependent on the physical overlap of prey and predator. It is determined by:</p>
<ol type="1">
<li><p>horizontal and vertical distribution parameters that define which cell and which distribution type (water column, sediment, epibenthic) each functional group is found in (see Distribution and Movement section below). Feeding of water column predators on sediment prey is determined by the depth consumers can dig into the sediment, whereas predators living inside the sediment cannot feed on the water column prey</p></li>
<li><p>period of the day that a group is active (a group that is not active during that period of the day does not eat, but is available for consumption, see below),</p></li>
</ol>
<p>Atlantis determines whether a prey and predator (both age structured and biomass pools) can access same habitats. If a predator <strong>cannot access any of the habitats inhabited by the prey</strong> (has 0 for prey habitat types), then it also cannot access any prey biomass associated with that habitat type. If a predator <strong>can access at least one</strong> of the habitats that the prey is associated with, then it is assumed to have <strong>access to all of the prey biomass</strong>.</p>
</section>
<section id="habitat-refuge" class="level3">
<h3 class="anchored" data-anchor-id="habitat-refuge"><strong><em>10.2.4. Habitat refuge</em></strong></h3>
<p>The <em>δ<sub>habitat&nbsp;</sub></em>is the <strong>optional prey habitat refuge</strong> which is activated <strong>only</strong> when <strong>all</strong> of the points below are true:</p>
<ol type="1">
<li><p>The general habitat dependency flag flaghabdepend is set to 1</p></li>
<li><p>habdepend_XXX is &gt;0 for the specific prey age-stuctured prey group</p></li>
<li><p>flag_refuge_model &gt; 0, which will activate the standard (=1) or rugosity related (=2) refuge model</p></li>
<li><p>prey is in the bottom water column, where water is in contact with the epibenthic habitats and sediment</p></li>
</ol>
<p>The habitat refuge is calculated in the <em>Vertebrate_Assess_Enviro()</em> routine in <strong>atvertprocesses.c</strong></p>
<p>The <em>δ<sub>habitat&nbsp;</sub></em>is used to reflect the assumption that for <strong>age-structured prey</strong> all habitats can provide some degree of refuge even when predators can access the habitat the prey inhabits. The <em>δ<sub>habitat&nbsp;</sub></em> = 1 for prey species that are not dependent on habitat (XXX_habdepend set to 0), whereas for habitat dependent species it is defined either as a simple habitat refuge (flag_refuge_model =1) or rugosity related (flag_refuge_model=2).</p>
<p>A <strong>simple habitat refuge</strong> is calculated as:</p>
<p><span class="math display">\[\delta_{habitat} = Acov_{prey} \bullet \left( e^{\left( - K_{prey} \bullet Cover_{habitat} + Bcov_{prey} \right)} + \frac{1}{K_{prey}} \right)\]</span></p>
<p>where <em>Cover<sub>habitat</sub></em> is the weighted relative cover in the cell for the prey, <em>Acov<sub>prey</sub></em> (Acov_ad_XXX and Acov_juv_XXX) is a scalar of the overall habitat refuge effect for prey <em>i</em> , <em>Bcov<sub>prey</sub></em> (Bcov_ad_XXX and Bcov_juv_XXX) is the habitat steepness coefficient, <em>K</em><sub>prey</sub> (Kcov_ad_XXX and Kcov_juv_XXX) is the exponent of refuge provided by the habitat.</p>
<p>The availability of suitable cover is calculated as</p>
<p><span class="math display">\[Cover_{habitat} = \ \left( \delta_{substrate,habdegrad} \bullet p_{substrate} + p_{biogenic} \right) \bullet \left( 1 + p_{canyon} \right)\]</span></p>
<p>where <em>δ <sub>substrate,habdegrad</sub></em> is degradation in the physical habitat due to coastal development (set by the flagdegrade flag, which also controls other habitat degradation parameters), <em>p<sub>substrate</sub></em> is the proportion of the cell covered with suitable substrate types, <em>p<sub>biogenic</sub></em> is the proportion of the cell covered by suitable biogenic habitats (determined by the isCover parameter in the <em>functional_groups.csv</em>), <em>p<sub>canyon</sub></em> is the proportion of the cell covered by canyons. Note that canyons are treated differently to other habitats and, for species that live in canyons (set through the habitat_XXX vector) the canyon acts as an enhancement factor (multiplier), where 10% of canyon area in the cell will enhance the habitat cover to 1.1. This is because canyons are known to concentrate production, but their absence does not prevent the establishment and growth of the groups.</p>
<p><strong>Rugosity related habitat refuge</strong> is <strong>only</strong> available for</p>
<ol type="1">
<li><p>models that include corals or sponges and</p></li>
<li><p>is only used for species that are dependent on (interact with) corals.</p></li>
</ol>
<p>Further description of the rugosity related habitat cover is available on the wiki dealing with “Calculation of rugosity related habitat refuge” - <a href="https://confluence.csiro.au/display/Atlantis/2016/09/16/Calculation+of+rugosity+related+habitat+refuge">here</a> (it cover’s models by Bozec and Blackwood in detail).</p>
<p>If flag_refuge_model=2 but a species does not live on corals, then the simple refuge model described above is applied. Three extra parameters are needed for rugosity refuge model: RugCover_Coefft, RugCover_Const, RugCover_Cap.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Calculating the amount of habitat cover available</strong> <em>Cover<sub>habitat</sub></em></p>
<p>The amount of habitat available for the species can be calculated either as a sum of total accessible habitat or as an average proportion of suitable habitats. This is controlled by flag_rel_cover parameter and will give <strong>very different habitat cover outcomes</strong>! For example, a cell that has three suitable habitats, covering 10%, 20% and 30% of the cell area respectively gives the following:</p>
<p>If flag_rel_cover is set to 0 the amount of available cover will be 0.1+0.2+0.3=0.6</p>
<p>If flag_rel_cover is set to 1 the amount of available cover will be (0.1+0.2+0.3)/3=0.2</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>The habitat refuge can be interpreted as a scalar on the available prey biomass determined by the available suitable habitat cover in the cell.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="markdown_output/media/image34.emf" style="width:4.42569in;height:4.25069in" class="figure-img"></p>
<figcaption><span id="_Toc526762791" class="anchor"></span>Figure 11. Habitat refuge scalar on prey availability as a function of cell area covered with suitable habitats (see box above for details on calculations of suitable cover area).</figcaption>
</figure>
</div>
<p><strong>Black</strong>: Acov=1, Bcov=0.6, Kcov=3 (parameters similar to SETas model and originally derived from a meta-analysis of prey accessibility on coral and temperate reefs)</p>
<p><em>Effect of Kcov:</em></p>
<p><strong>Orange</strong>: Acov=1, Bcov=0.6, Kcov=2</p>
<p><strong>Red</strong>: Acov=1, Bcov=0.6, Kcov=1</p>
<p><em>Effect of Bcov:</em></p>
<p><strong>Light blue</strong>: Acov=1, Bcov=0.4, Kcov=3</p>
<p><strong>Dark blue</strong>: Acov=1, Bcov=0.8, Kcov=3</p>
<p>The Acov is a linear scalar that only affects the height of the curve along the y axis. If needed it could be used to scale prey availability to 1 when cover=0</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p>When habitat dependency and refuge are used then the models are often parameterised so that <strong>prey availability is upscaled (&gt;1) when a cell contains zero available habitat</strong> (as prey are more exposed than normal to predators). It is assumed that normally around 0.2-0.3 of the cell will have suitable habitat, which is why prey availability scalar is ~1 at this proportion of cover available (see Fig. 11). This is based on a (now &gt;10 year old) meta-analysis of prey accessibility on coral and temperate reefs.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong><em>Limitations of current implementation of habitat refuge</em></strong></p>
<ol type="1">
<li><p>The curve of habitat refuge does not have a final inflection that would match the case where even good quality habitat has become saturated and fails to provide any refuge for a larger proportion of the total population.</p></li>
<li><p>The curve is deterministic and time invariant so there is no explicit dynamic trade-off between predation risk and hunger-driven searching in open ground as there is in Ecosim (Walters et al 1997).</p></li>
<li><p>This approach ignores the use of cover by predators, where improved cover can actually lead to higher predation.</p></li>
</ol></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>For <strong>biomass pools</strong> (CP) the <em>δ<sub>habitat&nbsp;</sub></em>refuge is available only for prey living in the <strong>sediments</strong> and is calculated by the depth the predators can dig into (set in KDEP_XXX) and depth of the oxygenated layer sedoxdepth, calculated and reported in the <em>output.nc</em> file, see Chapter 5.6). If sedoxdepth &lt; KDEP then no refuge is provided by the sediment (<em>δ<sub>habitat</sub></em> = 1), if sedoxdepth &gt; KDEP then</p>
<p><em>δ<sub>habitat</sub></em> = (sedoxdepth - KDEP)/ sedoxdepth,</p>
<p>For example, if the oxygenated layer depth is twice as deep as the sediment penetration depth, then only half of the sediment living biomass pool will be available (and this will be further scaled by <em>p<sub>prey,CX</sub></em> value). If sedoxdepth=0 a small number (1e-08) is added to avoid division by zero</p>
<table class="caption-top table">
<caption><span id="_Toc526762767" class="anchor"></span>Table 15. Parameters defining habitat associations and habitat refuge.</caption>
<colgroup>
<col style="width: 3%">
<col style="width: 96%">
</colgroup>
<thead>
<tr class="header">
<th><strong>Parameter</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>flaghabdepend</td>
<td>Flag turning on habitat dependency routines. If it is set to 0, all habitat associations and routines are ignored.</td>
</tr>
<tr class="even">
<td>XXX_habdepend</td>
<td>Flag identifying which age structured groups are habitat dependent</td>
</tr>
<tr class="odd">
<td>habitat_XXX</td>
<td>A vector of values identifying which of the available habitat preferences for biomass pool group XXX. Traditionally these used to be only 0 or 1 values, indicating absence or presence in each habitat. Now more resolved setting is allowed. A value &gt; 0 indicated that the organism prefers the habitat and of &lt; 0 it avoids it. The absolute value indicates the relative weighting, where proportion of suitable habitat is calculated as a multiplier of habitat proportion and habitat preference given in habitat_XXX vector.</td>
</tr>
<tr class="even">
<td><p>juv_habitat_XXX</p>
<p>ad_habitat_XXX</p></td>
<td>Same as above but applied for age structured groups and setting available habitats for juvenile and adult life history stages separately</td>
</tr>
<tr class="odd">
<td>flag_refuge_model</td>
<td>If set to 0, no habitat refuge for age structured groups will be applied. If set to 1 the standard refuge is calculated as described above. If set to 2 the habitat refuge is rugosity related</td>
</tr>
<tr class="even">
<td>flag_rel_cover</td>
<td>A flag indicating whether to use cumulative habitat (0) or average relative cover (1) when interacting with predators</td>
</tr>
<tr class="odd">
<td><p>Kcov_juv_XXX</p>
<p>Kcov_ad_XXX</p></td>
<td>exponent of refuge provided by the habitat for age structured groups (provided for juveniles and adults)</td>
</tr>
<tr class="even">
<td><p>Acov_juv_XXX</p>
<p>Acov_ad_XXX</p></td>
<td>scalar of the overall habitat refuge for age structured groups (provided for juveniles and adults)</td>
</tr>
<tr class="odd">
<td><p>Bcov_juv_XXX</p>
<p>Bcov_ad_XXX</p></td>
<td>habitat steepness coefficient for age structured groups (provided for juveniles and adults)</td>
</tr>
<tr class="even">
<td>KDEP_XXX</td>
<td>depth consumers can dig into the sediment, in meters. Used to determine habitat refuge for prey digging in the sediments</td>
</tr>
<tr class="odd">
<td>isCover in .csv</td>
<td>Which epibenthic groups also serve as a habitat</td>
</tr>
</tbody>
</table>
</section>
<section id="size-refuge-from-predation" class="level3">
<h3 class="anchored" data-anchor-id="size-refuge-from-predation"><strong><em>10.2.5. Size refuge from predation</em></strong></h3>
<p>The <strong>non-optional size refuge of age-structured prey</strong> <em>δ<sub>size&nbsp;</sub></em>is defined by gape limitation, which is used to simulated size selectivity and physical limitation of predators when feeding (i.e.&nbsp;what can fit in their mouth). The gape limitation can be calculated in two ways – <strong><em>hard feeding window</em></strong> or <strong><em>smooth feeding curve</em></strong> (defined by the UseHardFeedingWindow flag)</p>
<p>The hard feeding window defines knife-edge size selectivity, where availability of the prey is either zero (prey not available to the predator) or one (prey available). The prey is available to the predator if its size falls within the lower and upper prey selection size limits (KLP_XXX and KUP_XXX), or gape size, of the predator. In these calculations <strong>size is defined by structural nitrogen (SN) only</strong>!</p>
<p><span class="math display">\[\delta_{size} = \ \left\{ \begin{array}{r}
= 1,\ \ if\ KLP \bullet SN &lt; SN_{prey} &lt; KUP \bullet SN \\
= 0,\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ otherwise\
\end{array} \right.\ \]</span></p>
<p>In many models the lower and upper limits are 0.1% and ca 40% of the predator size respectively; so the predator of 1000mg SN could consume age structured prey with from 1 to 400mg (KUP ranges from 15 to 80%).</p>
<p>The smooth feeding window models a smoother transition of prey availability and is set by UseHardFeedingWindow=0. The original implementation of the smooth feeding window used a humped window calculation; now defined with an additional flag UseHumpedFeedingWindow=1. If humped window is used the availability of prey on the upper predator’s size limit is calculated as</p>
<p><span class="math inline">\(\delta_{size} = relSize\  \bullet e^{\left( Kmax \bullet (1 - relSize) \right)}\)</span>, where</p>
<p><span class="math display">\[relSize = \frac{SN_{prey}}{SN \bullet KUP}\]</span></p>
<p>represents <strong>relative size</strong> of the prey compared to the predator’s upper size limit (<em>KUP</em>), and <em>Kmax</em> (Kmax_coefft_XXX) is the steepness of the smooth selectivity curve.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="markdown_output/media/image35.png" style="width:4.73542in;height:3.76389in" class="figure-img"></p>
<figcaption><span id="_Toc526762792" class="anchor"></span>Figure 12. Prey availability scalar Size_Scalar<sub>prey</sub> of different sized prey for different Kmax_coefft values for humped selectivity curve.</figcaption>
</figure>
</div>
<p><strong>Black</strong>: <em>Kmax</em> = 2</p>
<p><strong>Orange</strong>: <em>Kmax</em> = 1</p>
<p><strong>Red</strong>: <em>Kmax</em> = 3</p>
<p><strong>Blue</strong>: <em>Kmax</em> = 5</p>
<p>Predators SN=10mg and KUP=0.4 (shown by vertical blue line). The horizontal black line shows that the maximum value of δ<sub>size</sub> is set to 1 (values larger than one are capped at 1)</p>
<p>As Fig. 12 shows, the humped window calculation is very sensitive to the Kmax_coefft_XXX parameter, where availability of prey at lower size limit can increase very rapidly. An alternative formulation is now available as a logistic shaped curve to calculate δ<sub>size</sub>. This option is appplied if UseHardFeedingWindow = 0 and UseBiLogisticFeedingWindow= 1 <strong>and it is based on total length rather than SN of predator and prey</strong></p>
<p><span class="math display">\[\delta_{size} = \ \left\{ \begin{array}{r}
= \frac{1}{1 + e^{\left( - Kmax \bullet \left( Len_{prey} - Len_{prey,50\% KLP} \right) \right)}},\ \ if\ Len_{prey} &lt; Avail_{top} \bullet Len_{pred} \\
= \frac{1}{1 + e^{\left( Kmax \bullet \left( Len_{prey} - Len_{prey,50\% KUP} \right) \right)}},\ \ \ \ \ \ \ \ \ \ if\ Len_{prey} &gt; Avail_{top} \bullet Len_{pred}
\end{array} \right.\ \]</span></p>
<p>where <em>Kmax</em> is the steepness of the logistic selectivity curve (Kmax_coefft_XXX), <em>Len<sub>prey,50%KLP</sub></em> and <em>Len<sub>prey,50%KUP</sub></em> are midpoints of the logistic selectivity curve or length of prey at which availability is half of the maximum possible value. The <em>Avail<sub>top</sub></em> is the length of prey for which availability is highest.</p>
<p>The <em>Avail<sub>top</sub></em> is calculated as a midpoint between the KLP and KUP values, hence assuming a symmetrical selectivity curve. For example, if the KLP=0.2 and KUP=0.6, the <em>Avail<sub>top</sub></em> = 0.4. Similarly the <em>Len<sub>prey,50%KLP</sub></em> is the midpoint between the lower (KLP_XXX) predator size limits and the <em>Avail<sub>top</sub></em>, and <em>Len<sub>prey,50%KUP</sub></em> is the midpoint between the upper (KUP_XXX) predator size limits and the <em>Avail<sub>top</sub></em>. In the example above the <em>Len<sub>prey,50%KLP</sub></em> = 0.3 and <em>Len<sub>prey,50%KUP</sub></em> = 0.5.</p>
<p>Note, that if the steepness of the curve (<em>Kmax</em> value) is low, the the δ<sub>size</sub> in the equation above may never reach 1 (see Figure 13). It is NOT recommended to use Kmax values lower than 0.5 because it makes the feeding window very wide and allows feeding on almost all sizes of prey (orange line in Fig. 13).</p>
<p><img src="markdown_output/media/image36.png" style="width:4.71875in;height:3.69028in">The bilogistic feeding window uses length rather than SN of prey and predator, because large differences between SN (many orders of magnitude) makes the size scalar highly sensitive to the Kmax parameter.</p>
<p><span id="_Toc526762793" class="anchor"></span>Figure 13. Prey availability scalar δ<sub>size</sub> of different sized prey for bilogistic selectivity curve.</p>
<p><strong>Black</strong>: Kmax = 1, KLP=0.2, KUP=0.6</p>
<p><strong>Red</strong>: Kmax=0.5 , KLP=0.2, KUP=0.6</p>
<p><strong>Orange</strong>: Kmax = 0.2, KLP=0.2, KUP=0.6</p>
<table class="caption-top">
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Calculating length in age-structured groups and biomass pools</strong></p>
<p>If length-based selectivity is selected for a fishery, it will be applied to both age-structured groups and biomass pools. The application of length-based selectivity to biomass pools is a new feature in Atlantis and it is important to get it right, or else invertebrates may not be caught by the gear (see wiki post <a href="https://confluence.csiro.au/pages/viewpage.action?spaceKey=Atlantis&amp;postingDay=2016%2F6%2F2&amp;title=Invertebrate+size+calculations+for+use+in+size+based+selectivity">here</a>)</p>
<p>Length of an age group in age-structured groups is calculated using length-weight conversion parameters (li_a_XXX and li_b_XXX) given in the <em>biology.prm</em> file. For this, first, the structural and reserve nitrogen (RN+SN, in mg) are added up and converted to wet weight in grams (<em>wgt</em>) as:</p>
<p><span class="math display">\[wgt = (RN + SN) \bullet wetdry \bullet X_{CN}/1000\]</span></p>
<p>where <em>wetdry</em> is the wet weight to ash free dry weight ratio (k_wetdry in <em>biology.prm</em>) typically set to 20 in many Atlantis parameter files, based on using C as a proxy for dry weight (but noting that in many models and literature the wet : dry ratio used is often closer to 6-8) and <em>X_CN</em> is the carbon to nitrogen Redfield ratio (X_CN in <em>biology.prm</em>) typically set to 5.7.</p>
<p>The <em>wgt</em> (in grams) is then converted to length (in cm) using the standard equation</p>
<p><span class="math inline">\(wgt = a \bullet {length}^{b}\)</span> or <span class="math inline">\(length = \left( \frac{wgt}{a} \right)^{1/b}\)</span></p>
<p>where <em>a</em> is li_a_XXX and <em>b</em> is li_b_XXX in the <em>biology.prm</em>. Note that these parameters could be set in such a way to return length in metres instead, but by convention it is in cm (as are any other faunal lengths).</p>
<p>Since for <strong>biomass pools</strong> only one nitrogen pool is tracked (individuals are not tracked separately), the structural nitrogen of an average individual must be given in the <em>biology.prm</em> file. This is given in the XXX_sn parameter for each consumer biomass pool group (not plants, bacteria or detrital pools). In addition to the length-based selectivity, this SN value is used in length-based feeding interactions when biomass pool predators feed on the age-structured prey (see chapter 10.2.5).</p>
<p>The length of an average biomass pool invertebrate is calculated in the same way as for age structured groups above. However, before the calculation the SN is converted to the SN+RN weight using the X_RS ratio parameter in the <em>biology.prm</em> file, so that</p>
<p><span class="math display">\[wgt = \ (1.0 + X\_ RS) \bullet SN\]</span></p>
<p>hence assuming that the ratio of hypothetical RN and SN in an invertebrate is optimal. <strong>The length-weight conversion parameters li_a_invert and li_b_invert given in the <em>biology.prm</em> are identical for all biomass pool species.</strong></p></th>
</tr>
</thead>

</table>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Smooth and hard feeding window</strong></p>
<p>Applying hard feeding window (knife-edge selectivity) means that very small changes in prey’s size can abruptly change their availability from 0 to 1. Although this approach is used in a range of ecological models it can lead to abrupt changes in growth or predation mortality between cohorts.</p>
<p>If smoother transition on size selectivity is desired, a smooth feeding window should be used. Note, however, that the “humped” implementation of the smooth feeding curve extends the availability of prey above the upper gape limit (KUP_XXX) of the predator (Fig. 12). The curve is also very sensitive to the Kmax_coefft parameter. Applying this smooth feeding curve might require decreasing the KUP_XXX values.</p>
<p>When using bilogistic curve the prey availability is generally decreased compared to the hard feeding window – the availability is equal to that of hard feeding window only for a small size range of the prey (Fig. 13). This means that when changes from hard to bilogistic curve are applied, the overall prey availability may have to be increased to ensure enough food for the predator.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<table class="caption-top table">
<caption><span id="_Toc526762768" class="anchor"></span>Table 16. Parameters defining size refuge for age structured groups</caption>
<colgroup>
<col style="width: 19%">
<col style="width: 79%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"><strong>Parameter</strong></th>
<th style="text-align: left;"><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">KLP_XXX</td>
<td style="text-align: left;">Min gape limit of the predator (age structured or biomass pool)</td>
</tr>
<tr class="even">
<td style="text-align: left;">KUP_XXX</td>
<td style="text-align: left;">Max gape limit of the predator (age structured or biomass pool)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">UseHardFeedingWindow</td>
<td style="text-align: left;">Flag setting up hard (0) or smooth (1) feeding window</td>
</tr>
<tr class="even">
<td style="text-align: left;">UseBiLogisticFeedingWindow</td>
<td style="text-align: left;">Flag indicating whether using humped (0) or bi-logistic (1) smooth feeding window</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Kmax_coefft_XXX</td>
<td style="text-align: left;">Steepness of the smooth feeding window curve of the predator</td>
</tr>
<tr class="even">
<td style="text-align: left;">XXX_sn</td>
<td style="text-align: left;">Typical size (amount of SN) of a biomass pool consumer to determine the size of age structured prey available to it</td>
</tr>
</tbody>
</table>
</section>
<section id="availability-of-fisheries-catches-to-opportunistic-catch-eaters" class="level3">
<h3 class="anchored" data-anchor-id="availability-of-fisheries-catches-to-opportunistic-catch-eaters"><strong><em>10.2.6. Availability of fisheries catches to opportunistic catch-eaters</em></strong></h3>
<p>The availability of fisheries catches is an optional set of parameters that allow groups to access additional prey. Proportion of catch available from each fishery (set in PropCatch_XXX parameter) is added to the total available prey. No gape limitation (size refuge) is applied to fisheries catches.</p>
<table class="caption-top table">
<caption><span id="_Toc526762769" class="anchor"></span>Table 17. Parameters defining feeding on fisheries catches</caption>
<colgroup>
<col style="width: 9%">
<col style="width: 90%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"><strong>Parameter</strong></th>
<th style="text-align: left;"><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">isCatchGrazer in .csv file</td>
<td style="text-align: left;">Identifies species that can feed on fisheries catches</td>
</tr>
<tr class="even">
<td style="text-align: left;">pFCXXX</td>
<td style="text-align: left;">A vector of values (0 and 1) identifying which of the fisheries catches are available to each catch eating species identified as isCatchGrazer. The vector should be as long as there are fisheries in the <em>fisheries.csv</em> file (assumes the same order as given in the <em>fisheries.csv</em> file).</td>
</tr>
<tr class="odd">
<td style="text-align: left;">PropCatch_XXX</td>
<td style="text-align: left;">Proportion of the catch in each fishery that can be exploited by the isCatchGrazer groups</td>
</tr>
</tbody>
</table>
</section>
<section id="effect-of-temperature-and-salinity-on-feeding-parameters" class="level3">
<h3 class="anchored" data-anchor-id="effect-of-temperature-and-salinity-on-feeding-parameters"><em>10.2.7. Effect of temperature and salinity on feeding parameters</em></h3>
<p>Feeding rates and assimilation efficiencies will be temperature dependent if a species is temperature sensitive. Salinity dependency is optional. See chapter13 for further details.</p>
</section>
<section id="effect-of-oxygen-limitation-on-epibenthic-invertebrate-feeding-rates" class="level3">
<h3 class="anchored" data-anchor-id="effect-of-oxygen-limitation-on-epibenthic-invertebrate-feeding-rates"><em>10.2.8. Effect of oxygen limitation on epibenthic invertebrate feeding rates</em></h3>
<p>Oxygen limitation affects the feeding rates of SED_EP_FF, MOB_EP_OTHER, SED_EP_OTHER, SM_INF, LG_INF, CORAL and SPONGE functional groups (applied in group routines <em>Epibenthic_Ivert_Process()</em>, <em>Invert_Consumer_Process()</em> and <em>Coral_Process()</em>). This dependency is also an option for age structured groups in newer code releases that are currently in the pipeline.</p>
<p>The oxygen limitation scalar <em>δ<sub>O2</sub></em> is applied to clearance rate <em>C</em>. Atlantis has four options to calculate oxygen limitation scalar, set by O2case parameter. The oxygen limitation scalar is calculated by <em>Oxygen()</em> function in <strong><u>atprocess.c</u></strong></p>
<p><strong>1. Ambient oxygen limitation</strong> (O2case=0) uses ambient oxygen levels O<sub>amb</sub>, lethal oxygen concentration (KO2_XXX, mgO2 m<sup>-3</sup>) and limiting oxygen concentration (KO2lim_XXX, mgO2 m<sup>-3</sup>) parameters. Remember that oxygen solubility in seawater at 5C and 1bar pressure is 10 mg/l or 10000 mg m<sup>-3</sup>.</p>
<p><img src="markdown_output/media/image37.wmf" class="img-fluid"></p>
<p><img src="markdown_output/media/image38.png" style="width:4.33889in;height:3.10347in"></p>
<p><span id="_Toc526762794" class="anchor"></span>Figure 14. Oxygen limitation scalar <em>δ<sub>O2&nbsp;</sub></em>for different ambient oxygen concentrations, assuming KO2=50 mgO2 m<sup>-3</sup> and KO2lim = 100 mgO2 m<sup>-3</sup>.</p>
<p><strong>2. Depth based limitation</strong> (O2case=1) only uses sediment depth of half oxygen mortality mD (mD_XXX, m) parameter and depth of the oxygenated sediment layer sedoxdepth (see chapter 5.6) where</p>
<p>δ<sub>O2</sub>=sedoxdepth/(sedoxdepth+mD)</p>
<p>This limitation only reflects reduced feeding within the sediment due to low oxygen conditions, so it is mostly relevant to epibenthic invertebrate biomass pools that can dig into sediment for food or infauna.</p>
<p><strong>3. IGBEM based limitation</strong> (O2case=2) is modified after the ERSEM model and is calculated as:</p>
<p>δ<sub>O2</sub>=O<sub>amb</sub>/(KO2+O<sub>amb</sub>), where O<sub>amb</sub> is the concentration of oxygen in the cell</p>
<p><strong>4. Quadratic limitation</strong> (O2case=3) calculates both ambient and depth based limitation (points 1 and 2 above) and uses the larger of the two values as the oxygen scalar δ<sub>O2</sub></p>
</section>
<section id="effect-of-space-limitation-on-epibenthic-invertebrate-feeding-rates" class="level3">
<h3 class="anchored" data-anchor-id="effect-of-space-limitation-on-epibenthic-invertebrate-feeding-rates"><em>10.2.9. Effect of space limitation on epibenthic invertebrate feeding rates</em></h3>
<p>Space limitation can affect <em>C</em> or <em>mum</em> rates of benthic invertebrate biomass pools if:</p>
<p>1) A biomass pool group XXX is infauna or a filter feeder or mobile epibenthic group (SED_EP_FF and MOB_EP_OTHER GroupType) <strong>and</strong> is habitat dependent (habdependXXX=1) <strong>and</strong> space limitation is active (flagXXXlim=1 or 2). If these three conditions are satisfied then the Crowding Effect scalar is applied to the clearance rate <em>C</em> of XXX (see chapter 10.5.3).</p>
<p>2) A biomass pool group is habitat dependent (with at least some habitat preference parameters not set to 1), benthic and flag_benthos_sediment_link=1. See chapter 10.5.3 for further details.</p>
</section>
<section id="other-factors-affecting-feeding-parameters" class="level3">
<h3 class="anchored" data-anchor-id="other-factors-affecting-feeding-parameters"><em>10.2.10. Other factors affecting feeding parameters</em></h3>
<p>The feeding interactions described above are further dynamically modified by a range of ecological and environmental factors, such as species activity, starvation, or pH.</p>
<p><strong>Species activity</strong></p>
<p>Atlantis has an option to set whether a species is active during the day or night, or all the time. This is done using flagXXXday parameter. It is important to note that if a diurnal timestep is used (i.e.&nbsp;12h or less) and if a species activity preference is set to day or night, it will not be active in the model during the other half of the day – it will not initiate ecological processes such as eating, but can still be preyed upon.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p>If a species is only active during the day or night (flagXXXday set to 0 or 1) it means that <strong>no ecological routines will be initiated by this species during the period of each day that it is inactive.</strong></p>
<p>This means that an inactive species <strong>will not feed, move, reproduce and will not breathe</strong>, if respiration is included in the model. This also means that even if species physically overlap and have diet connections, but a predator is not active at that time of the day, no feeding interaction will occur.</p>
<p>An inactive group is succeptible to fishing and predation.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p><strong>Spawning</strong></p>
<p>Species that are in the spawning period will not feed if the parameter feed_while_spawn is set to 0. The biomass of the spawning stock will subtracted when calculating the feeding biomass of a predator.</p>
<p><strong>Starvation</strong></p>
<p>Starvation has no effect on <em>C</em> and <em>mum</em> in the modified Holling functional responses (predcase 1-3) and ratio dependent feeding responses (predcase 4).</p>
<p>For other feeding responses handling time (ht_XXX in biomass pools and hta_XXX and htb_XXX in age-structured groups) is scaled by KHTD or KHTI scalars when the condition of an age structured group falls below a limit set in Kthresh1 and Kthresh2 (see box below). See chapter 10.5.2 for further details on how starvation is determined and what it affects.</p>
<p><strong>Acidification</strong></p>
<p>Atlantis includes explicit representation of the possible effects of acidification. The model is activated with the flagmodelPH=1 in the <em>biology.prm</em> file. It allows the user to scale prey availability, assimilation efficiency, <em>C</em> and <em>mum</em> rates, mortality, recruitment and other processes. See chapter 13 for further details.</p>
</section>
<section id="other-feeding-functional-responses" class="level3">
<h3 class="anchored" data-anchor-id="other-feeding-functional-responses"><em>10.2.11 Other feeding functional responses</em></h3>
<p>Twelve different feeding functional responses are currently implemented in Atlantis. They are selected using the predcase_XXX parameter, which can take values from 0 to 12 (option 3, Ecosim based feeding rate, is currently unavailable). The first three cases are for the modified Holling Type I-III responses adopted from the PPBIM. They use the <em>mum</em> parameter, as described above. Two responses – minimum-maximum and Holling size dependent type III - are adopted from the ERSEM model (see Table 18), but have not been used in many models. The remaining responses can be grouped into standard Holling responses (Holling type I to IV) and ratio depdenent responses (see e.g.&nbsp;Arditi and Ginzburg 1989 and critique of the approach by Abrams 1994). The ratio dependent responses take into account not only the prey biomass but also the ratio of prey to predator biomass. These responses don’t use the <em>mum</em> parameter but a mass specific handling time (ht_XXX) parameter.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong><em>Search volume (vl) and handling time (ht) as alternative parameters to C and mum</em></strong></p>
<p>The modified Holling type functional responses use age-specific <em>C</em> and <em>mum</em> parameters. For other functional responses either one or both of these parameters are replaced with search volume (vl_XXX, vla_XXX and vlb_XXX) and handling time (ht_XXX, hta_XXX and htb_XXX).</p>
<p>Unlike <em>C</em> and <em>mum</em>, however, the search volume and handling time parameters are always set <strong>based on mass specific structural ash-free dry weight</strong> (not nitrogen!). This means that only one value is provided in the parameter file, and the age specific values for age-structured groups are calculated based on the individual’s body weight. For biomass pools the units of vl_XXX are in m<sup>3</sup> mgN<sup>-1</sup> day<sup>-1</sup>. For age structured groups vla_XXX and vlb_XXX determine an allometric search volume-mass relationship, and they are converted to individual nitrogen-based values in <em>Determine_Fish_Feeding_Prms()</em> in <strong>atvertprocesses.c</strong> where</p>
<p><img src="markdown_output/media/image39.wmf" class="img-fluid"></p>
<p>where SN is the structural nitrogen of an age class and XCN is the C:N Redfield ratio (X_CN in the biology.prm) typically set to 5.7 used to convert nitrogen mass to total ash-free dry mass.</p>
<p>For demersal species (flagdem=1) the search volume VL is halved (VL=VL/2).</p>
<p>The handling time HT is calculated as</p>
<p><img src="markdown_output/media/image40.wmf" class="img-fluid"></p>
<p>where δ<sub>starve</sub> is a scalar that reduces the handling time when an individual is starving (see chapter 10.5.2 on the effects of starvation).</p>
<p><strong>Note, that only SN and not total nitrogen values are used for search volume and handling time calculations.</strong></p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>The table below gives equations for the 12 functional feeding response options. In all the equations B denotes the predator’s biomass, B*<sub>prey</sub> is the available biomass of a given prey, scaled by the optional habitat refuge, and obligatory size and availability parameters, as explained above (note that prey list also includes the available fisheries catch, if the predator is identified as catch-eater in the <em>functional_groups.csv</em> file)</p>
<p><span class="math display">\[B_{prey}^{*} = p_{prey,CX} \bullet \delta_{overlap} \bullet \delta_{habitat} \bullet \delta_{size} \bullet B_{prey}\]</span></p>
<p>and <em>∑B<sub>prey,i</sub></em> is the sum of all available prey.</p>
<table class="caption-top table">
<caption><span id="_Toc526762770" class="anchor"></span>Table 18. Equations defining grazing term in the twelve currently available feeding functional responses. Some of these responses are described and compared in Fulton et al.&nbsp;2003a “Mortality and predation in ecosystem models: is it important how these are expressed?” <em>Ecological Modelling</em> 169,157–178</caption>
<colgroup>
<col style="width: 3%">
<col style="width: 53%">
<col style="width: 43%">
</colgroup>
<thead>
<tr class="header">
<th>Pred case</th>
<th>Functional response</th>
<th>Equations</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>0</td>
<td>Modified Holling type II, adopted from PPBIM</td>
<td><img src="markdown_output/media/image41.wmf" class="img-fluid"></td>
</tr>
<tr class="even">
<td>1</td>
<td><p>Modified Holling type I, adopted from PPBIM</p>
<p>This response has linear increase with prey density, but is capped at the maximum determined by maximum growth rate (<em>mum</em>) divided by the assimilation efficiency on live prey (which is typically highest)</p></td>
<td><p>if (<em>C·B*<sub>prey</sub></em>) &gt; (<em>mum</em> / E<sub>preyLIVE</sub>)</p>
<p><img src="markdown_output/media/image42.wmf" class="img-fluid"></p>
<p>else</p>
<p><img src="markdown_output/media/image43.wmf" class="img-fluid"></p></td>
</tr>
<tr class="odd">
<td>2</td>
<td><p>Modified Holling type III, adopted from PPBIM</p>
<p>Same as Holling type II, but prey biomasses are squared</p></td>
<td><img src="markdown_output/media/image44.wmf" class="img-fluid"></td>
</tr>
<tr class="even">
<td>4</td>
<td><p>Minimum-maximum</p>
<p>Adopted from ERSEM, where it was used to describe fish feeding. However, it is not used in ERSEM anymore, because higher trophic level predators are not currently included in ERSEM</p></td>
<td><p><img src="markdown_output/media/image45.wmf" class="img-fluid"></p>
<p><em>L</em> = lower prey biomass threshold for feeding by predator XX (KL_XX)</p>
<p><em>U</em>= half saturation coefficient for feeding by predator XX(KU_XX)</p></td>
</tr>
<tr class="odd">
<td>5</td>
<td><p>Holling type III – size dependent</p>
<p>Adopted from ERSEM, where it was used to describe seabird and mammal feeding (it is not used in ERSEM anymore, see above)</p></td>
<td><p><img src="markdown_output/media/image46.wmf" class="img-fluid"></p>
<p>VL and HT are search volume and handling time (see above). ∑B <sub>prey.i</sub> is the sum of all available prey.</p>
<p>Remember that for demersal species (flagdem=1), the search volume is halved</p></td>
</tr>
<tr class="even">
<td>6</td>
<td><p>Ratio dependent</p>
<p>See Arditi and Ginzburg 1989 and text above. This approach accounts for competition among predators through ratio of predator and prey biomasses</p></td>
<td><p><img src="markdown_output/media/image47.wmf" class="img-fluid"></p>
<p><em>Compet</em> = interpredator competition. Please check the wiki for further details as this part of the code is changing based on discussion with experts in the field.</p></td>
</tr>
<tr class="odd">
<td>7</td>
<td><p>Standard Holling Type I</p>
<p>These are the standard Holling type responses. They have been added to Atlantis in 2015</p></td>
<td><img src="markdown_output/media/image48.wmf" class="img-fluid"></td>
</tr>
<tr class="even">
<td>8</td>
<td><p>Standard Holling Type II</p>
<p>These are the standard Holling type responses. They have been added to Atlantis in 2015</p></td>
<td><img src="markdown_output/media/image49.wmf" class="img-fluid"></td>
</tr>
<tr class="odd">
<td>9</td>
<td><p>Standard Holling Type III</p>
<p>These are the standard Holling type responses. They have been added to Atlantis in 2015</p></td>
<td><p><img src="markdown_output/media/image50.wmf" class="img-fluid"></p>
<p>As in Holling Type II but the prey and total prey biomasses are squared</p></td>
</tr>
<tr class="even">
<td>10</td>
<td><p>Standard Holling Type IV</p>
<p>These are the standard Holling type responses. They have been added to Atlantis in 2015</p></td>
<td><p><img src="markdown_output/media/image51.wmf" class="img-fluid"></p>
<p>As in Holling Type II but the total prey biomasses are squared</p></td>
</tr>
<tr class="odd">
<td>11</td>
<td><p>Hassel Varley</p>
<p>This response allows for interference among predators</p></td>
<td><p><img src="markdown_output/media/image52.wmf" class="img-fluid"></p>
<p><em>TP</em> = total predator biomass or abundance</p>
<p><em>HVM</em> = is the coefficient of mutual interference for the top carnivores</p></td>
</tr>
<tr class="even">
<td>12</td>
<td><p>Crowley Martin</p>
<p>Like Standard Holling type II but includes competition among predators as in option 6</p></td>
<td><p><img src="markdown_output/media/image53.wmf" class="img-fluid"></p>
<p><em>Compet</em> = interpredator competition. Please check the wiki for further details as this part of the code is changing based on discussion with experts in the field.</p></td>
</tr>
</tbody>
</table>
<p>Earlier when type I response was used it was important to cap the clearance rate of the predator, to ensure it does not get unrealistically high. This was done by setting flag_satiation=1. This approach caps the clearance rate at the value of <em>mum</em>/E<sub>typeLIVE</sub> , where E<sub>typeLIVE</sub> is the assimilation efficiency on live prey, which is typically the highest assimilation efficiency.</p>
<p><span class="math display">\[Gr_{prey} = \ \frac{B \bullet mum \bullet B_{prey}^{*}}{E_{typeLIVE}},\ if\ \ \ \ \ clearance &gt; \ \frac{B \bullet mum}{E_{typeLIVE}}\]</span></p>
<p>The flag_satiation parameter is a special case and in most cases will not be required.</p>
</section>
</section>
<section id="food-assimilation-in-consumers" class="level2">
<h2 class="anchored" data-anchor-id="food-assimilation-in-consumers"><strong>10.3. Food assimilation in consumers</strong></h2>
<p>The total consumed food ∑ <em>Gr<sub>prey</sub></em> is assimilated according to the consumer assimilation efficiencies of four different types of prey <em>i</em>, defined as animals (including fisheries catch), plant, labile detritus and refractory detritus. The assimilation efficiency can be further externally scaled by a scalar <em>gs</em> supplied through forcing files and used to modify growth due to factors not included in the model.</p>
<p>In consumer biomass pools (CP) all assimilated food is directly allocated to growth.</p>
<p><span class="math inline">\(G_{CP} = A_{CP} = \ \sum_{i}^{}{\left( E_{i} \bullet Gr_{prey} \right) \bullet gs}\)</span></p>
<p>In age-structured consumers (CX) the assimilated food is allocated to a temporary pool <em>A</em> (called <em>*growth</em> in Atlantis code). The energy in the pool A is used to meet optional respiration costs and the remainder then allocated to growth of SN and RN pools. Hence the <em>A</em> pool is not tracked but only used for computational purposes.</p>
<p><span class="math inline">\(A_{CX} = \ \sum_{i}^{}{\left( E_{i} \bullet Gr_{prey} \right) \bullet gs}\)</span></p>
<p>Assimilation rate in age structured groups can be affected by temperature, salinity and pH (see chapter 13).</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Accounting for maintenance costs through assimilation efficiency</strong></p>
<p>The assimilation parameter is required for both biomass pools and age-structured groups. However, depending on whether respiration costs are explicitly included in age-structured groups, the assimilation efficiency coefficient can take a different meaning.</p>
<p>In Atlantis explicit mass-specific respiration is only available in age-structured groups, where size of an average individual in an age group is tracked. For biomass pools maintenance costs are assumed to be represented implicitly in assimilation (in)efficiency. Such representation assumes that temperature responses of assimilation efficiency and maintenance processes are the same. It also assumes that when assimilation (<em>Acx</em>) is zero (no food is available) then also respiration or maintenance is zero.</p>
<p>This has important implications for obtaining assimilation efficiency values and comparing them to other models that include maintenance explicitly. For example, in ERSEM model respiration includes two components, basal respiration that is dependent on temperature and independent of the food intake, and activity respiration that represents a fraction of the assimilated food. In Atlantis basal respiration is optional and is only included for age-structured groups. The activity respiration is not explicitly modelled but is implicitly included in the assimilation efficiency.</p>
<p>If respiration of age structured groups is explicitly included in the model (see below), the assimilation efficiency of age structured groups and biomass are not directly comparable and the values for biomass pools should be much lower than that for age-structured groups.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</section>
<section id="maintenance-respiration-in-consumers" class="level2">
<h2 class="anchored" data-anchor-id="maintenance-respiration-in-consumers"><strong>10.4. Maintenance (respiration) in consumers</strong></h2>
<p>In Atlantis the only explicit maintenance costs are those of respiration. Respiration is only included in fully age structured groups and is optional. The majority of existing Atlantis models do not explicitly model maintenance or respiration. The respiration or maintenance costs are assumed to be implicitly accounted for by assimilation inefficiencies (see NOTE! in chapter 10.3).</p>
<p>Age-structured group respiration is carried out by the <em>Fish_Respiration()</em> routine in <strong>atvertprocesses.c</strong></p>
<p>Explicit respiration in the age-structured groups represents base respiration costs and depends on body mass, temperature and condition. It is activated by setting flagresp to 1. The nitrogen used up in respiration is taken out of the temporary <em>A</em> pool (before that pool is allocated to growth) and added to the NH pool.</p>
<p>The nitrogen consumed for respiration is calculated as</p>
<p><span class="math display">\[Rs = \ e^{\left( Ktmp \bullet T_{H2O} \right)} \bullet KA \bullet {Wgt}^{KB}\]</span></p>
<p>where <em>Ktmp</em> (Ktmp_fish, Ktmp_shark, etc), <em>KA</em> (KA_XXX) and <em>KB</em> (KB_XXX) are parameters defined for each age structured group or larger functional groups (fish, sharks, mammals, birds) and <em>Wgt</em> is individual dry weight calculated as</p>
<p><em>Wgt = (SN+RN) · C_XN</em></p>
<p>The C_XN parameter represents N:C Redfield ratio, typically set to 5.7. This means that <strong><em>KA</em> and <em>KB</em> parameters relate to total dry weight</strong> of an individual and <strong>not to the nitrogen</strong> content tracked in the model.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Should models include explicit respiration?</strong></p>
<p>Of course, all organisms have maintenance costs and they have to breathe. The question is whether including these costs implicitly in assimilation efficiencies provides sufficient representation of the physiological-ecological dynamics.</p>
<p>The key difference in using explicit respiration is that it depends on body mass, temperature and condition and is independent of the food intake rate. It is therefore more accurately reflects scaling of respiration to body size and to temperature and allows for different temperature scaling of assimilation and respiration rates.</p>
<p>If a large proportion of the biomass of the total model is in age-structured groups with strong variation in size or environmental conditions, then explicit use of respiration (flagresp=1) is a sensible option to take. However, if there is little variation in the environment, size at age or the model is dominated by biomass pool groups then setting flagres =0 may provide sufficient representation of the system. Also, when the focus of a model is to explore temperature sensitivities or growth and size dynamics of age-structured groups, then including explicit respiration may be advisable.</p>
<p>One of the reasons that Atlantis was developed in place of the earlier IGBEM and ERSEM models is because the explicit physiological detail included in those models was unwieldy to use and calibrate in each new location (imagine that a model harder than Atlantis to use!!). Therefore a simpler representation that produce approximately the same dynamics was developed and implemented in Atlantis (see for example Fulton et al.&nbsp;2004c). Explicit maintenance costs are excluded from many models, and their best representation remains a topic of debate (e.g.&nbsp;Persson et al.&nbsp;2014).</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<section id="scaling-of-respiration-costs-depending-on-individuals-condition" class="level3">
<h3 class="anchored" data-anchor-id="scaling-of-respiration-costs-depending-on-individuals-condition"><strong><em>10.4.1 Scaling of respiration costs depending on individual’s condition</em></strong></h3>
<p>Respiration costs are scaled by a factor KST in starving individuals (see chapter 10.5.2. for further details on starvation).</p>
<p>The optimum RN to SN ratio is defined by a general X_RS parameter, which is typically set to 2.65. If the RN/SN is below the optimum, the respiration costs are scaled by a factor KST. The Kthresh2 parameter controls how far below the optimum RN/SN the individual can be before the scaling of respiration costs starts.</p>
<p><span class="math display">\[if\ \frac{RN}{X_{RS}} \bullet SN &lt; Kthresh2,\ \ \ \ \ \ \ \ \ \ \ \ \ \ \]</span></p>
<p><span class="math display">\[\ Rs = Rs \bullet KST\ \]</span></p>
<table class="caption-top table">
<caption><span id="_Toc526762771" class="anchor"></span>Table 19. Parameters defining respiration rates in age structured groups</caption>
<colgroup>
<col style="width: 26%">
<col style="width: 72%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"><strong>Parameter</strong></th>
<th style="text-align: left;"><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">flagresp</td>
<td style="text-align: left;">Flag turning-on explicit respiration in age structured groups</td>
</tr>
<tr class="even">
<td style="text-align: left;">KA_XXX</td>
<td style="text-align: left;">Scaling coefficient of allometric respiration- weight rate</td>
</tr>
<tr class="odd">
<td style="text-align: left;">KB_xxx</td>
<td style="text-align: left;">Exponent of allometric respiration-weight rate</td>
</tr>
<tr class="even">
<td style="text-align: left;">Ktmp_fish, Ktmp_shark, Ktmp_bird, Ktmp_mammal</td>
<td style="text-align: left;">Parameter defining how respiration increases with temperature in the four group types FISH (and FISH_INVERT), SHARK, BIRD and MAMMAL</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Kthresh2</td>
<td style="text-align: left;">Threshold relative reserve below which respiration rate is scaled by KST</td>
</tr>
<tr class="even">
<td style="text-align: left;">KST_fish, KST _shark, KST _bird, KST _mammal</td>
<td style="text-align: left;">Scalar of respiration when relative reserve drops below Kthresh2 for the four group types FISH (and FISH_INVERT), SHARK, BIRD and MAMMAL</td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="growth" class="level2">
<h2 class="anchored" data-anchor-id="growth"><strong>10.5. Growth</strong></h2>
<section id="growth-and-energy-allocation-in-age-structured-groups" class="level3">
<h3 class="anchored" data-anchor-id="growth-and-energy-allocation-in-age-structured-groups"><strong><em>10.5.1. Growth and energy allocation in age-structured groups</em></strong></h3>
<p>Once the optional respiration costs (<em>Rs</em>) are subtracted from the temporary pool <em>A</em>, <strong>ALL remaining</strong> energy is allocated to RN and SN. The energy is partitioned between RN and SN depending on:</p>
<ol type="1">
<li><p>the consumer’s condition, defined by the current ratio of the RN and SN pool and the X_RS parameter that sets the optimal ratio of RN to SN in well fed age structured groups</p></li>
<li><p>the consumer’s preference for rebuilding of reserves over structure, defined by the pR_XXX parameter (where higher values show lower preference for allocating to RN).</p></li>
</ol>
<p>The proportion of energy in consumer CX going to RN and SN is allocated as:</p>
<p><span class="math display">\[G_{RN} = \ (A - Rs) \bullet (1 - \lambda)\]</span></p>
<p><span class="math display">\[G_{SN} = \ (A - Rs) \bullet \lambda\]</span></p>
<p>where</p>
<p><span class="math display">\[\lambda = \ \left\{ \begin{array}{r}
\frac{\frac{1}{X\_ RS} + pR \bullet \left( \frac{RN}{pR \bullet SN} - 1 \right)}{\frac{1}{X\_ RS} + \frac{RN}{pR \bullet SN}},\ if\ (A - Rs) &gt; 0 \\
0,\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ otherwise\
\end{array} \right.\ \]</span></p>
<p>The equation above shows that if respiration costs are higher than assimilated energy at a given time step (Rs&gt;A), then λ=0 and no energy will be allocated to or taken out of SN. Moreover, (A-Rs) will be negative and the difference between A and Rs is taken out of RN.</p>
<p>Traditionally in Atlantis negative (A-Rs) was not taken out of the RN and in fact most models do not include maintenance, so (A-Rs) never gets negative. To enable negative (A-Rs) to be taken out from RN pool, set the flag_shrinkfat=1. This is <strong>strongly recommended for model that use respiration</strong>. If flag_shrinkfat=0 (for legacy issues) the negative (A-Rs) is converted to zero and nothing is taken out from either RN or SN. This means that the only way RN pool can decrease is through spawning, as it is not used to cover maintenance costs during starvation.</p>
<p>Figure 15 shows the value of λ (y axis) depending on the RN/SN. Until the RN/SN is close to the X_RS value, all available energy will go to rebuilding reserves (λ=0). Black and blue lines show energy allocation under two different X_RS parameters. Once RN/SN &gt; X_RS, the condition of the animal is considered to be good and some energy is allocated to growth of SN. Once RN is replenished, all available energy will be directed to SN. The steepness of the curve and value of RN/SN above which all energy will be sent to SN depends on the pR_XXX parameter, where <strong>larger pR values lead to steeper curve and earlier allocation of all energy to SN</strong>.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="markdown_output/media/image54.png" style="width:4.84861in;height:3.44306in" class="figure-img"></p>
<figcaption><span id="_Toc526762795" class="anchor"></span>Figure 15. Proportion of available energy allocated to SN, depending on the individual’s condition (RN/SN, on x axis), and pR and X_RS parameters.</figcaption>
</figure>
</div>
<p><strong>Black</strong>: X_RS=2.65, pR=3.5</p>
<p><strong>Red</strong>: X_RS=2.65, pR=5</p>
<p><strong>Orange</strong>: X_RS=2.65, pR=10</p>
<p><strong>Blue</strong>: X_RS=2, pR=3.5</p>
<p>Another option to consider when thinking about the representation of size in Atlantis is flag_dynamicXRS. This flag indicates whether to use a dynamically changing ratio between reserve and structural mass through time in Atlantis or to simply use a fixed value (as in the standard Atlantis code). This has implications for everything to do with size – how the optimal weight of in individual is calculated, which influences reproductive capacity and allocation of material during growth. These additions are used most commonly when dealing with evolution in Atlantis.</p>
</section>
<section id="when-is-an-age-structured-group-starving-and-what-does-it-mean" class="level3">
<h3 class="anchored" data-anchor-id="when-is-an-age-structured-group-starving-and-what-does-it-mean"><em>10.5.2. When is an age-structured group starving and what does it mean?</em></h3>
<p>Condition of an age-structured group is determined by the optimal RN to SN ratio, defined by the X_RS parameter, which is typically set to 2.65. When RN &lt; X_RS*SN the age structured group is considered to be starving (or at least food deprived).</p>
<p>This interpretation is different from some other models (e.g.&nbsp;Dynamic Energy Budget approach), where even a small amount of reserve available means that an individual has reserves and is in good condition. In Atlantis RN is not treated as pure reserve, but also includes gonad weight, fat and other body parts that can be reabsorbed during starvation.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p>If respiration is not included, the only way RN pool can decrease is through spawning. This means that a group will always have lower than optimum RN/SN ratio after a spawning event, until RN pool is replenished (even when abundant food is available). If no food is available, the RN/SN ratio will not decrease further, because no maintenance costs are required.</p>
<p>If respiration is included and flag_shrinkfat = 1 then RN pool will decrease at low food abundance and an age group will be “starving” as it is usually understood.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>There are different levels of starvation and they have increasing effects on ecological processes:</p>
<p><em>1) When RN/X_RS*SN &lt; 1</em></p>
<p>- the amount of spawn produced by an individual is reduced (see Spawn in Chapter 10.8).</p>
<p><em>2) When RN / X_RS*SN &lt; Kthresh1</em></p>
<p>- handling time is scaled by the KHTD parameter value. This is only applied if predcaseXXX &gt; 4</p>
<p><em>3) When RN / X_RS*SN &lt; Kthresh2</em></p>
<p>- handling time is scaled by the KHTI parameter value. This is only applied if predcaseXXX &gt; 4</p>
<p>- respiration rate is scaled by the KST parameter. This is only used if flagresp = 1 (respiration used)</p>
<p>Additionally, starvation will lead to starvation mortality, as specified by the starvation mortality parameter, see Chapter 10.6.2</p>
</section>
<section id="growth-in-consumer-biomass-pools" class="level3">
<h3 class="anchored" data-anchor-id="growth-in-consumer-biomass-pools"><strong><em>10.5.3 Growth in consumer biomass pools</em></strong></h3>
<p>No explicit respiration costs are included in biomass pools, hence all assimilated energy <em>A<sub>CP</sub></em> is allocated to growth. However, in epibenthic organisms space limitation scalar <em>δ<sub>space</sub></em> is applied to limit growth in crowded conditions</p>
<p><span class="math display">\[G_{CP} = \ A_{CP} \bullet \delta_{space}\]</span></p>
<p>The <em>δ<sub>space</sub></em> scalar is calculated differently in different invertebrate GroupTypes, and is determined by habitat dependency and/or the flagXXXlim parameter.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Different meaning of flagXXXlim and space limitation in different invertebrate groups</strong></p>
<p>Space limitation is controlled by flagXXXlim and is applied differently for different types of invertebrates:</p>
<p><strong>For filter feeding and mobile epibenthic</strong> <strong>invertebrates</strong> (<strong>SED_EP_FF</strong> and <strong>MOB_EP_OTHER</strong>) flagXXXlim space limitation will be either simple (flagXXXlim=1) or ERSEM based (flagXXXlim=2) – called through <em>Epibenthic_Invert_Processes()</em>routine</p>
<p>If flagXXXlim=0 there is no space limitation</p>
<p>If flagXXXlim=1 space limitation is calculated as δ<sub>space</sub> scalar and applied <strong>to the biomass</strong>.</p>
<p>If flagXXXlim=2 space limitation is calculated as Crowding effect and applied to <strong>clearance rate</strong> (C)</p>
<p>For <strong>infaunal</strong> and other <strong>sediment invertebrates and primary producers</strong> (<strong>SM_INF, LG_ING</strong>, <strong>SED_EP_OTHER, CORAL and SPONGE</strong>) flagXXXlim means either no limitation (flagXXXlim=0) or simple limitation (flagXXXlim=1) – called through <em>Invertebrate_Consumer_Process()</em>, <em>Sediment_Epi_Other_Process()</em> or <em>Coral_Process()</em>routines</p>
<p>If flagXXXlim=0 no space limitation is applied</p>
<p>If flagXXXlim=1 space limitation is calculated as δ<sub>space</sub> scalar and applied <strong>to the biomass</strong>.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>The space limitation scalar δ<sub>space</sub> is calculated as</p>
<p><span class="math display">\[\delta_{space} = \ \frac{1 - B_{CP}}{Max_{CP} \bullet habitat}\]</span></p>
<p>where <em>Max<sub>CP</sub></em> is the XXXmax parameter defining maximum biomass mgN m<sup>-2</sup> and <em>habitat</em> is the proportion of habitat suitable for the consumer (area of the habitat available for the species while accounting for its optional degradation). The <em>B<sub>CP</sub></em> is the consumer biomass (mgN per m<sup>2</sup>).</p>
<p>The Crowding effect is applied when flagXXXlim=2. It is based on ERSEM II model (Blackford 1997) and requires three extra parameters (XXX_low, XXX_sat, XXXthresh). The Crowing effect is <strong>applied on</strong> <strong>clearance rates and not on biomass</strong>.</p>
<p><img src="markdown_output/media/image55.wmf" class="img-fluid"></p>
<p>where <em>θ</em><sub>CXlow</sub> is the crowding lower threshold (XXX_low, mgN m<sup>-2</sup>), <em>κ</em><sub>CXsat</sub> is the crowding half saturation level (XXX_sat, mgN m<sup>-2</sup>), <em>κ</em><sub>CXthresh</sub> is the crowding threshold (XXXthresh, mgN m<sup>-2</sup>)</p>
<p>By default a consumer can only have intraspecific (intra functional group) competition. However, if flag_competing_epiff=1 then interspecies competition is included and <em>B<sub>CP&nbsp;</sub></em>is calculated as the total biomass of all corals (CORAL), sponges (SPONGE), epibenthic macrophytes (PHYTOBEN, SEAGRASS) and filter feeders (SED_EP_FF), which means that all epifauna will compete for space. When flag_competing_epiff=1 the user can also allow the maximum available habitat area to be higher than 1. This means that the epifaunal groups can grow on top of each other (as happens in complex reef structures). This option is set with the max_available_habitat parameter, which acts as a scalar on the available habitat. Only set this value to &gt;1 when you have good reasons to increase the habitat availability to the epifauna.</p>
<p>The δ<sub>space</sub> space limitation scalar is applied either to the biomass or to the clearance rate parameter <em>C</em> (see NOTE! box above). If more habitat limitation is needed, the <em>mum</em> values of invertebrate <strong>epibenthic and sediment</strong> biomass pool consumers (SM_INF, LG_INF, SED_EP_FF, SED_EP_OTHER, MOB_EP_OTHER) can be scaled by the available habitat. This scaling is applied by setting flag_benthos_sediment_link=1. If the scaling is applied the <em>mum</em> values of the epibenthic and sediment invertebrate biomass pool consumers will be multiplied (scaled) by the proportion of the cell with the suitable habitat (accounting for degradation effects)</p>
<p>mum*<sub>CP</sub> = mum<sub>CP</sub>·<em>habitat</em></p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Habitat availability for biomass pool consumers</strong></p>
<p>Traditionally in Atlantis biomass pool consumers could only use the physically defined habitat (reef, flat, soft, canyon). These physical habitats are typically static throughout the simulation, i.e.&nbsp;their proportion in the cell does not change (unless degraded, see below).</p>
<p>This has now been modified, so that biomass pool groups can also use and be affected by the dynamic biohabitats created by functional groups identified with the IsCover parameter in the <em>functional_groups.csv</em>. Such habitats include corals, sponges, mussel beds and similar. To allow the biomass pool to access the biohabitats set flag_invert_biohab=1</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<table class="caption-top table">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Degrading the geological properties</strong></p>
<p>The physical habitats are typically static throughout an Atlantis simulation, i.e.&nbsp;their proportion in the cell does not change. However, in some circumstances change is desired (e.g.&nbsp;representing a scenario of coastal development or dredging etc). In this case set flagdegrade to 1 and provide values for REEFchange_max_num, FLATchange_max_num and SOFTchange_max_num. Additionally the following parameters will be needed</p>
<p>REEFchange_start N where N is the number of changes given by REEFchange_max_num</p>
<p>d1 d2…. dN where d1, d2, dN etc represent the day of the run where a change begins</p>
<p>REEFchange_period N where N is the number of changes given by REEFchange_max_num</p>
<p>p1 p2…. pN where p1, p2, pN etc represent the period in days that a change lasts</p>
<p>REEFchange_mult N where N is the number of changes given by REEFchange_max_num</p>
<p>m1 m2…. mN where m1, m2, mN etc are the multiplicative level of change</p>
<p>Similarly for SOFT and FLAT.</p>
<p><strong>Note, users must ensure that when multipliers are applied to habitat cover, the total cell cover of geological habitats must still sum to 1.</strong> Otherwise, the suitable area in the cell will shrink or expand. So if a REEF habitat is degraded and, for example, a multiplier of 0.5 is applied, other habitats must have multipliers &gt;1 to compensate for the change.</p>
<p>Lastly the boxes where the change will occur will need to be identified (with the same change applied in all these boxes simultaneously)</p>
<p>Box_degraded N where N is the number of boxes in the model domain</p>
<p>0 0 0 0 1…. 0</p>
<p>where there are N entries, one for each box with 1 indicating changes occur in that box.</p></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="feeding-on-material-outside-the-model" class="level3">
<h3 class="anchored" data-anchor-id="feeding-on-material-outside-the-model"><strong><em>10.5.4 Feeding on material outside the model</em></strong></h3>
<p>During the first few months inside the model domain, salmon will stay close to shore and feed on terrestrial insects. To allow for this external input of food into salmon diet take the following steps</p>
<ol type="1">
<li><p>Set isSupplemented to a value greater than 0 in <em>functional_groups.csv</em>. A value of 1 means all life history stages will consume the supplemental feed, a value of 2 is juveniles only and a value of 3 is adults only.</p></li>
<li><p>Add a vector to the <em>biology.prm</em> file to indicate the degree of supplemental feeding occurring per box. The individual values in this vector must have values between 0 and 1 (as the value indicates for that box how much of the potential supplemental feed is actually accessed). Note the vector does not need to sum to 1.0, just that each individual value in it must be &lt; 1.0. For example, for group XXX in a 5 box model it would look like:</p></li>
</ol>
<p>Supp_XXX 5</p>
<p>0.0 0.1 0.0 0.0 0.8</p>
<p>At present this proportional access is assumed to match across all ages undertaking supplemental feeding. If this is not the case contact the Atlantis model developers for an update to the code. Also note that supplemental feeding only occurs in surface waters.</p>
<table class="caption-top table">
<caption><span id="_Toc526762772" class="anchor"></span>Table 20. Parameters affecting growth in consumers</caption>
<colgroup>
<col style="width: 8%">
<col style="width: 91%">
</colgroup>
<thead>
<tr class="header">
<th><strong>Parameter</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>X_RS</td>
<td>Optimal RN to SN ratio in age-structured groups</td>
</tr>
<tr class="even">
<td>pR_XXX</td>
<td>Parameter controlling preference of energy allocation to RN and SN (Fig. 15)</td>
</tr>
<tr class="odd">
<td>flag_shrinkfat</td>
<td>If set to 1, negative energy budget at a given time step (when respiration costs exceed assimilated food intake) will be taken out of the RN pool. This is important for models that include respiration. If set to 0, negative energy budget is converted to 0.</td>
</tr>
<tr class="even">
<td>flaghomog_sp</td>
<td>If set to 1, body condition of age-structured groups is homogenised across the entire model domain after each timestep. This means that RN and SN ratio will be the same for a functional group throughout the whole model domain. If set to 0, condition can vary among boxes depending on food availability</td>
</tr>
<tr class="odd">
<td>flagXXXlim</td>
<td>Flag defining what kind of space limitation should be applied to biomass pool consumers or benthic primary producers (see NOTE! in chapter 10.5.3)</td>
</tr>
<tr class="even">
<td>XXXmax</td>
<td>maximum biomass of a biomass pool consumer (mgN m<sup>-2</sup>), used in simple space limitation (flagXXXlim=1)</td>
</tr>
<tr class="odd">
<td><p>XXX_low</p>
<p>XXX_sat</p>
<p>XXXthresh</p></td>
<td>Lower crowing threshold, crowding half saturation and crowding threshold (all in mgN m<sup>-2</sup>) used to calculate crowding effect in ERSEM based space limitation (flagXXXlim=2)</td>
</tr>
<tr class="even">
<td>flag_competing_epiff</td>
<td>If set to 1, space competition will be calculated taking into account all epifaunal groups (all epifaunal groups are assumed to compete for space) rather than only intra-functional competition</td>
</tr>
<tr class="odd">
<td>max_available_habitat</td>
<td>Scalar for maximum available habitat for epifaunal groups. If &gt; 1 epifaunal groups can grow on top of each other</td>
</tr>
<tr class="even">
<td>flag_benthos_sediment_link</td>
<td>Flag activating additional space limitation scaling in epibenthic and sediment biomass pool consumers, where <em>mum</em> values are scaled by the available habitat in the cell</td>
</tr>
<tr class="odd">
<td>flag_invert_biohab</td>
<td>Flag allowing biomass pool consumers to access biohabitats, identified through IsCover parameter in the <em>functional_groups.csv</em> (e.g.&nbsp;corals, sponges, mussel beds).</td>
</tr>
<tr class="even">
<td>flagdegrade</td>
<td>Flag initiating geological habitat (REEF, FLAT, SOFT) degradation. If set to 1, additional parameters will be required as explained in chapter 10.5.3</td>
</tr>
<tr class="odd">
<td>Supp_XXX</td>
<td>Proportion of a groups diet coming from outside the model domain (defined as a vector with one entry per box). On needed if isSupplemented is set to 1 in the <em>runs.prm</em> file.</td>
</tr>
</tbody>
</table>
</section>
<section id="effect-of-contaminants-on-the-growth-model" class="level3">
<h3 class="anchored" data-anchor-id="effect-of-contaminants-on-the-growth-model"><strong><em>10.5.5 Effect of contaminants on the growth model</em></strong></h3>
<p>A contaminant tracer(s) is traced through the food web to allow for assessment of the accumulation of contaminants in the food chain. The handling and introduction of these contaminants follows the standard Atlantis contaminant procedures. The level of accumulated contaminants in the individual can impact its growth using the following logistic curve to set the scalar to be applied to the growth rate (mum_XXX), this again follows the same general form as for the turbidity effects:</p>
<p><span class="math inline">\(f(x) = \ 1 - \frac{L}{1 + e^{- a(x - b)}}\)</span></p>
<p>A value must be provided per functional group per contaminant for L, a and b. In addition, the flag flag_contamGrowthModel must be set in <em>run.prm</em> to indicate how to apply the impact of contaminant on growth – 0 = no effect, 1 = InVitro method (using a simple scalar), 2 = logistic effects model. At present the contaminant may be “metabolized” by setting the decay terms in the standard Atlantis represent.</p>
<p>Growth effects due to contamination can also be represented using a simple scalar – set flag_contamGrowthModel to 1 and then for each species and each contaminant provide the following parameters in the <em>biology.prm</em> file.</p>
<p># The tissue contaminant level where growth effects start</p>
<p>XXX_Arsenic_GrowthThresh 10</p>
<p># The size of the growth effect (as a scalar)</p>
<p>XXX_Arsenic_GrowthEffect 0.9</p>
<p><em>Contaminant Uptake</em></p>
<p>The groups within Atlantis take up contaminants either through contact, general uptake or through consumption.</p>
<p>For each group the contaminant option needs to be set. There are currently two options for uptake:</p>
<ol type="1">
<li>Linear uptake (parameter is in parts/sec):</li>
</ol>
<blockquote class="blockquote">
<p><span class="math display">\[C_{uptake} = \tau \bullet C\]</span></p>
</blockquote>
<p>where <em>τ</em> is the uptake rate and C is the ambient contaminant level in the environment.</p>
<ol start="2" type="1">
<li>Sigmoidal uptake – this is ideal for representing interactions with a toxin that has no effect until a critical concentration is reached</li>
</ol>
<p><span class="math inline">\(C_{uptake} = \frac{1}{dt} \bullet \left( \frac{C}{C_{in} + \left( C - C_{in} \right) \bullet e^{- \tau \bullet dt}} - C_{in} \right)\)</span></p>
<p>where <em>C<sub>in</sub></em> is the contaminant level within the functional group (i.e.&nbsp;internal level).</p>
<p>The following parameters need to be added to the <em>biology.prm</em> file for each contaminant (where CCCC is the name of the contaminant:</p>
<p>CCC_half_life - Half life in seconds</p>
<p>CCC_temp_depend - Flag indicating whether the contaminant effect is temperature dependent</p>
<p>If there is more than one contaminant in the model, then interactions between them should be noted. Functional group level parameters are also required per contaminant included (see Table below).</p>
<p>Contaminants can also affect movement. There is the potential for mobile species (defined by the <em>VerticallyMigrates</em> and <em>HorizontallyMigrates</em> settings for the species in the <em>functional_groups.csv</em> file) to avoid the contaminants. It is not always appropriate biomass pool groups to show this behaviour so to turn it off set biopools_dodge_contam 0 in the <em>biology.prm</em>.</p>
<table class="caption-top table">
<caption><span id="_Toc526762773" class="anchor"></span>Table 21. Parameters needed for contaminant uptake model – where XXX is the functional group code and CCC the contaminant name.</caption>
<colgroup>
<col style="width: 23%">
<col style="width: 75%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"><strong>Parameter</strong></th>
<th style="text-align: left;"><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">XXX_CCC_uptake_option</td>
<td style="text-align: left;">Uptake option. 1 = linear, 2 = sigmoidal, 3 = piecewise linear</td>
</tr>
<tr class="even">
<td style="text-align: left;">XXX_CCC_uptake_rate</td>
<td style="text-align: left;">Uptake rate in mg/m^3 per second</td>
</tr>
<tr class="odd">
<td style="text-align: left;">XXX_CCC_ InstantDoseMortality</td>
<td style="text-align: left;">Instantaneous contact mortality rate</td>
</tr>
<tr class="even">
<td style="text-align: left;">XXX_CCC_TimeToLD50</td>
<td style="text-align: left;">The number of seconds that it takes to kill of 50% of the population</td>
</tr>
<tr class="odd">
<td style="text-align: left;">XXX_CCC_LD50</td>
<td style="text-align: left;">The concentration that will kill 50% of the population</td>
</tr>
<tr class="even">
<td style="text-align: left;">XXX_CCC_LD100</td>
<td style="text-align: left;">The concentration that will kill 100% of the population</td>
</tr>
<tr class="odd">
<td style="text-align: left;">XXX_CCC_LDChronic</td>
<td style="text-align: left;">The concentration where chronic effects persist</td>
</tr>
<tr class="even">
<td style="text-align: left;">XXX_CCC_LDslope</td>
<td style="text-align: left;">Slope of the internal LDC relationship</td>
</tr>
<tr class="odd">
<td style="text-align: left;">XXX_CCC_Cx</td>
<td style="text-align: left;">First constant used in piecewise interpolation of mortality effects across multiple contaminants</td>
</tr>
<tr class="even">
<td style="text-align: left;">XXX_CCC_Cy</td>
<td style="text-align: left;">Second constant used in piecewise interpolation of mortality effects across multiple contaminants</td>
</tr>
<tr class="odd">
<td style="text-align: left;">XXX_CCC_spL</td>
<td style="text-align: left;">The minimum of the growth effects curve if the logistic effects curve is used</td>
</tr>
<tr class="even">
<td style="text-align: left;">XXX_CCC_spA</td>
<td style="text-align: left;">The curve of the growth effects curve if the logistic effects curve is used</td>
</tr>
<tr class="odd">
<td style="text-align: left;">XXX_CCC_spB</td>
<td style="text-align: left;">The midpoint of the growth effects curve if the logistic effects curve is used</td>
</tr>
<tr class="even">
<td style="text-align: left;">XXX_CCC_ GrowthThresh</td>
<td style="text-align: left;">The tissue contaminant level at which growth effects manifest if using the InVitro growth effects model</td>
</tr>
<tr class="odd">
<td style="text-align: left;">XXX_CCC_EC50</td>
<td style="text-align: left;">Exponent used in the InVitro growth effects model for primary producers and microfauna</td>
</tr>
<tr class="even">
<td style="text-align: left;">XXX_CCC_ECslope</td>
<td style="text-align: left;">Slope of the InVitro growth effects model for primary producers and microfauna</td>
</tr>
<tr class="odd">
<td style="text-align: left;">XXX_CCC_GrowthEffect</td>
<td style="text-align: left;">The size of the growth effect (as a scalar) on consumers if using the InVitro growth effects model</td>
</tr>
<tr class="even">
<td style="text-align: left;">XXX_CCC_MoveEffect</td>
<td style="text-align: left;">Effect on movement (as a scalar)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">XXX_CCC_ReprodEffect</td>
<td style="text-align: left;">Effect on reproduction (as a scalar)</td>
</tr>
<tr class="even">
<td style="text-align: left;">XXX_CCC_sp_ContamScalar</td>
<td style="text-align: left;">The scalar used to represent the generic effects of contaminants on other physiological parameters</td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="mortality-in-consumers" class="level2">
<h2 class="anchored" data-anchor-id="mortality-in-consumers"><strong>10.6. Mortality in consumers</strong></h2>
<p>Mortality for age-structured groups is modelled through the numbers of individuals lost, whereas for biomass-pools it is expressed as biomass lost. The general non-predation and non-fishing mortality equation is:</p>
<p><span class="math display">\[M_{xx} = \ \left( \left( mL + mQ \bullet xx + mSt + \left( 1 - \delta_{O2} \right) \bullet mO + mA + mE \right) \bullet xx \right) \bullet mortsc\]</span></p>
<p>where <em>mL</em> is linear mortality (day<sup>-1</sup>), <em>mQ</em> is quadratic mortality (day<sup>-1</sup>), <em>mSt</em><strong><sub>,</sub></strong> is <strong>age-structured group only</strong> starvation mortality (day<sup>-1</sup>), <em>mO</em> is <strong>benthic invertebrate only</strong> low oxygen mortality (day<sup>-1</sup>), δ<sub>O2</sub> is the oxygen limitation scalar (see below), <em>mA</em> is an optional acidification mortality (day<sup>-1</sup>), <em>mE</em> is the optional extra mortality of <strong>age-structured groups only</strong> (due to seabirds and fish not included in the model) (day<sup>-1</sup>), <em>xx</em> is the biomass or numbers in the biomass pool or age group and <em>mortsc</em> is the optional external mortality scalar, provided through forcing files (see details <a href="https://confluence.csiro.au/pages/viewpage.action?spaceKey=Atlantis&amp;title=External+Mortality%2C+Growth+and+Recruitment+Scaling">here</a>). Below these mortality terms are discussed for age-structured and biomass pool groups separately.</p>
<p>Note that the oxygen mortality is assumed to only apply to benthic invertebrates as they cannot easily escape the oxygen minima, whereas fish and other vertebrates will move away (ultimately disappearing from the model if the entire domain becomes too oxygen poor and if the group is dependent on the oxygen – flagO2depend).</p>
<section id="mortality-in-age-structured-groups" class="level3">
<h3 class="anchored" data-anchor-id="mortality-in-age-structured-groups"><strong><em>10.6.1. Mortality in age-structured groups</em></strong></h3>
<p>The natural non-predation mortality of an age-structured consumer CX (in numbers not biomass) is calculated for each age group as:</p>
<p><span class="math display">\[M_{CX} = \ \left( \left( mL + mQ \bullet Num_{CX} + mSt + mA + mE \right) \bullet Num_{CX} \right) \bullet mortsc\]</span></p>
<p>where <em>mL</em> is an stage-specific (juvenile and adult) linear mortality for CX (mL_XXX, day<sup>-1</sup>), <em>mQ</em> is an stage-specific quadratic mortality for CX (mQ_XXX, day<sup>-1</sup>), <em>mSt</em> is the starvation mortality (day<sup>-1</sup>, see below), <em>mA</em> is an optional acidification mortality (see below), <em>mS</em> is the optional extra mortality due to seabirds and fish not included in the model (adopted from ERSEM I, Bryant <em>et al</em>. 1995) (mS_XXX, day<sup>-1</sup>), <em>Num</em> is the numbers in an age group and <em>mortsc</em> is the optional external mortality scalar, provided through forcing files. See chapter on the forcing file for details on extra mortality scaling.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Terminal or senescence mortality in age structured groups and age structured biomass pools</strong></p>
<p>At the end of the lifespan the individuals (biomass) can either all die (senesce) or remain in the last cohort (“plus group” in stock assessment jargon) for as long as they survive predation or other mortality factors. The senescence mortality is activated by the parameter flagsenesce.</p>
<p>If flagsenesce=1 all individuals above the maximum age die. The maximum age is set as the number of calendar years in an age group (XXX_AgeClassSize) multiplied by the number of age groups (NumCohorts). If flagsenesce=0 the oldest calendar year group in the oldest age group retains all the individuals (see chapter 10.9.6 on tracking calendar years in age groups).</p>
<p>Terminal mortality (mT) behaves somewhat similarly for biomass pools – indicating what proportion of the oldest age class dies due to old age.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>The mortality calculations are done in <em>Vert_Mortality()</em> routine in <strong>atvertprocesses.c</strong></p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Role of linear and quadratic mortality in multi-species models</strong></p>
<p>Linear and quadratic mortality terms are included to account for mortality factors not explicitly represented in the model – such as disease, behaviour-based density dependence (e.g.&nbsp;male rivalry), predators not explicitly represented in the model and other factors. Ideally, the linear mortality terms should be as low as possible (1-e15) or even zero across the board, while quadratic mortality should be low (or zero) for lower trophic levels and of moderate size for the largest of the top predators (where unrepresented density dependent factors appear to be missing from the model as formulated).</p>
<p>If high linear and quadratic mortality values are used, the model will be insensitive to species interactions. High linear and quadratic mortalities mean that the species will be controlled by these terms alone and predation will have little to no effect on the group’s biomass. As a result, the ecosystem model will end up being a set of parallel simulations of many single species models.</p>
<p>The exception is for top predators, such as mammals or seabirds, where predation mortality alone is typically too low to impose adequate control. In these cases, higher terms of quadratic mortalities are often applied. There is a good argument that for mammals and birds competition for breeding sites, disease and other factors indeed may be the main processes controlling population growth, and they should be represented via the quadratic mortality term.</p>
<p>Rates of linear and quadratic mortality are provided for juveniles and adults separately in age-structured groups. The age groups that belong to the juvenile versus adult stage are determined based on the XXX_age_mat parameter in the <em>biology.prm</em> file, which sets the first mature (adult) age group.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</section>
<section id="starvation-mortality-in-age-structured-groups" class="level3">
<h3 class="anchored" data-anchor-id="starvation-mortality-in-age-structured-groups"><strong><em>10.6.2. Starvation mortality in age-structured groups</em></strong></h3>
<p>Starvation mortality is only available in age-structured groups, as the condition (ration of SN to RN) is tracked for these groups only. Starvation mortality is modelled as</p>
<p><span class="math display">\[mSt = mStarve \bullet \frac{(Kthreshm \bullet optCond - W)}{optCond}\]</span></p>
<p>where <em>optCond</em> =(1.0 + X_RS)·SN, <em>W</em> is weight calculated as SN+RN, <em>mStarve</em> (mStarve_XXX, day<sup>-1</sup>) is starvation mortality rate, Kthreshm is a global (same for all species) parameter defining the threshold undernourishment levels when starvation mortality starts and X_RS is a global parameter that sets the optimum RN to SN ratio (typically set to 2.65).</p>
<p><strong>Note</strong>, that weight (W) used here is the <strong>nitrogen weight</strong>, defined as SN+RN and is different from weight used in respiration (where AFDW is used and calculated as N multiplied by the nitrogen to carbon Redfield ratio X_CN).</p>
<p>The calculations for starvation mortality are done in the <em>Vert_Mortality()</em> routine in <strong><u>atvertprocesses.c</u></strong></p>
<p><img src="markdown_output/media/image56.png" style="width:4.31111in;height:3.46181in"></p>
<p><span id="_Toc526762796" class="anchor"></span>Figure 16. Starvation mortality (per day) for different Kthreshm and mStarve values.</p>
<p><strong>Black</strong>: Kthrem=0.9, mStarve=0.001</p>
<p><strong>Orange</strong>: Kthrem=0.8, mStarve=0.001</p>
<p><strong>Red</strong>: Kthrem=0.7, mStarve=0.001</p>
<p><strong>LightBlue</strong>: Kthrem=0.9, mStarve=0.0005</p>
<p><strong>Blue</strong>: Kthrem=0.9, mStarve=0.01</p>
</section>
<section id="extra-mortality-in-age-structured-groups" class="level3">
<h3 class="anchored" data-anchor-id="extra-mortality-in-age-structured-groups"><strong><em>10.6.3. Extra mortality in age-structured groups</em></strong></h3>
<p>The <strong>optional</strong> extra mortality <em>mS</em> allows users to simulate <strong>quarterly</strong> <strong>age structured group</strong> mortality by seabirds and fish (or other top predators) <strong>not included</strong> in the model. It is given by mS_SBXXX (seabird imposed mortality on age structured group XXX) and mS_FDXXX (fish imposed mortality on age structured group XXX) parameters, each with four values, providing mortality for each quarter of the year.</p>
<p>Additional parameters describe the assimilation efficiency of these top predators not included in the model; the unassimilated food is all added to the ammonia pool (see chapter 10.7 for further details)</p>
<p>Note that the extra mortality is rarely used now, because models typically resolve all important functional groups explicitly in the model. As a result this form of mortality will likely be removed in future code releases.</p>
</section>
<section id="mortality-in-biomass-pool-consumers" class="level3">
<h3 class="anchored" data-anchor-id="mortality-in-biomass-pool-consumers"><strong><em>10.6.4. Mortality in biomass pool consumers</em></strong></h3>
<p>Non-predation mortality in consumers is expressed in term of the biomass lost to the biomass pool, where separate calculations are done for each pool in age-structured biomass pools.</p>
<p><span class="math display">\[M_{CP} = \ \left( \left( mL + mQ \bullet B_{CP} + \left( 1 - \delta_{O2} \right) \bullet mO + mA \right) \bullet B_{CP} \right) \bullet mortsc\]</span></p>
<p>Only one linear (<em>mL</em>) and quadratic (<em>mQ</em>) mortality parameter is provided for biomass pool groups, whereas for age-structured biomass pools separate parameters are needed for each pool. The oxygen mortality depends on the oxygen limitation scalar and the oxygen mortality rate (mO_XXX; day<sup>-1</sup>). The oxygen limitation mortality is determined by the <em>δ<sub>O2</sub></em> oxygen scalar, described in chapter 10.2.8.</p>
<p>As in age-structured groups, the age-structured biomass pools can have senescence mortality (see chapter 10.6.1).</p>
</section>
<section id="temperature-and-acidification-effect-on-mortality" class="level3">
<h3 class="anchored" data-anchor-id="temperature-and-acidification-effect-on-mortality"><strong><em>10.6.5. Temperature and acidification effect on mortality</em></strong></h3>
<p>Linear and quadratic mortality parameters can be temperature dependent, as described in chapter 13. If a model includes acidification effects (flagmodelpH=1) then parameters imposing additional acidification induced mortality can be specified.</p>
</section>
<section id="turbidity-effect-on-mortality" class="level3">
<h3 class="anchored" data-anchor-id="turbidity-effect-on-mortality"><strong><em>10.6.6. Turbidity effect on mortality</em></strong></h3>
<p>A logistic curve is applied to represent the escapement from predation due to turbidity:</p>
<p><span class="math inline">\(f(x) = \ 1 - \frac{L}{1 + e^{- a(x - b)}}\)</span></p>
<p>The value of the escapement is a proportion (i.e.&nbsp;between 0 and 1). At present this escapement only modifies the availability of the prey to its predator during feeding, it does not alter the prey’s habitat dependent movement (i.e.&nbsp;the prey species move to their preferred habitat and ignore any turbidity effects, if you want prey to take turbidity into account as cover so that it modifies their movement please let us know and we will update the code).</p>
<p>To activate this effect (i) flagIsEstuary must be set to 1 in the <em>run.prm</em> file so that the SED_i tracer is active in the model – it will need to be included in the <em>initial_conditions.nc</em> file and can also be included in point source inputs (.ts files) to represent any material coming down the rivers into the model domain. The flag turbid_refuge_XXX for each species – set it to 1 if the prey benefits from turbidity, 0 if not. Then for each predator you need to enter the following three vectors, which include one value for each stage of the functional group (as defined by NumStages in the <em>functional_groups.csv</em> file):</p>
<p>XXX _turbid_L which defines the minimum of the curve (per stage)</p>
<p>XXX _turbid_a which defines the slope of the curve (per stage)</p>
<p>XXX _turbid_b which defines the midpoint of the curve (per stage)</p>
<p>These vectors have the format like:</p>
<p>XXX _turbid_L 2</p>
<p>1.0 0.8</p>
<table class="caption-top table">
<caption><span id="_Toc526762774" class="anchor"></span>Table 22. Parameters affecting mortality in consumers</caption>
<colgroup>
<col style="width: 11%">
<col style="width: 87%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"><strong>Parameter</strong></th>
<th style="text-align: left;"><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">mL_XXX</td>
<td style="text-align: left;">Stage-specific (juvenile and adult) linear mortality, day<sup>-1</sup></td>
</tr>
<tr class="even">
<td style="text-align: left;">mQ_XXX</td>
<td style="text-align: left;">Stage-specific (juvenile and adult) quadratic mortality, day<sup>-1</sup></td>
</tr>
<tr class="odd">
<td style="text-align: left;">mS_XXX</td>
<td style="text-align: left;">Optional extra mortality due to seabirds and fish not included in the model, day<sup>-1</sup></td>
</tr>
<tr class="even">
<td style="text-align: left;">flagsenesce</td>
<td style="text-align: left;">If set to 1, all individuals or biomass in age-structured groups and age-structured biomass pools that are above the maximum age die (see NOTE! in chapter 10.6.1)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mT</td>
<td style="text-align: left;">Terminal senescence mortality in age-structured biomass pools (proportion of biomass that dies from senescence when above maximum age)</td>
</tr>
<tr class="even">
<td style="text-align: left;">mStarve_XXX</td>
<td style="text-align: left;">Starvation mortality, day<sup>-1</sup></td>
</tr>
<tr class="odd">
<td style="text-align: left;">mS_SBXXX</td>
<td style="text-align: left;">Extra mortality of age-structured groups due to seabirds not included in the model, day<sup>-1</sup></td>
</tr>
<tr class="even">
<td style="text-align: left;">mS_FDXXX</td>
<td style="text-align: left;">Extra mortality of age-structured groups due to fish not included in the model, day<sup>-1</sup></td>
</tr>
<tr class="odd">
<td style="text-align: left;">mO_XXX</td>
<td style="text-align: left;">Oxygen mortality rate of biomass pool consumers, day<sup>-1</sup></td>
</tr>
<tr class="even">
<td style="text-align: left;">Kthreshm</td>
<td style="text-align: left;">Threshold starvation level, when starvation mortality starts (chapter 10.6.2)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">turbid_refuge_XXX</td>
<td style="text-align: left;">Flag indicating whether a group has its mortality effected by turbidity</td>
</tr>
<tr class="even">
<td style="text-align: left;">XXX _turbid_L</td>
<td style="text-align: left;">defines the minimum of the turbidity effect on mortality curve (per stage)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">XXX _turbid_a</td>
<td style="text-align: left;">defines the slope of the turbidity effect on mortality curve (per stage)</td>
</tr>
<tr class="even">
<td style="text-align: left;">XXX _turbid_b</td>
<td style="text-align: left;">defines the midpoint of the turbidity effect on mortality curve (per stage)</td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="waste-production" class="level2">
<h2 class="anchored" data-anchor-id="waste-production"><strong>10.7. Waste production</strong></h2>
<p>The unassimilated food (faeces) and non-predation mortality products are sent to labile detritus (DL), refractory detritus (DR) and ammonia (NH) pools. The production of waste products by biomass pool and age structured group consumers are handled in the same way, but in the case of age structured groups the mortality terms are converted from numbers of individuals to biomass before being used in the following equations.</p>
<p>Flow of nitrogen from food intake to detritus and the ammonia pool can be illustrated with the flowchart in Fig. 17. It shows that first, the assimilation efficiency parameters determine how much of the grazed food is assimilated. The rest is sent to waste products. The waste products are then split between detritus and the ammonia fraction, where the amount of detritus is determined by the detritus proportion parameters (FDG, FDGDL, FDGDR).</p>
<p><img src="markdown_output/media/image57.jpg" style="width:5.79281in;height:6.26042in"></p>
<p><span id="_Toc526762797" class="anchor"></span>Figure 17. Nitrogen flow from consumers to detrital and ammonia pools. Parameters that determine allocation between the pools are shown on the arrows. Optional fluxes are shown in dashed arrows.</p>
<p>Production of <strong>labile detritus (DL)</strong> by consumer group XX is given by:</p>
<p><span class="math display">\[W_{DL} = \ \left( \left( 1 - E_{L} \right) \bullet FDG \bullet \sum_{i = type}^{}{Gr_{CX,i} + \left( 1 - E_{DL} \right) \bullet FDGDL \bullet Gr_{CX,DL} + \left( 1 - E_{DR} \right)} \bullet FDGDR \bullet Gr_{CX,DR} + FDM \bullet M_{CX} \right) \bullet FDL\]</span></p>
<p>where E<sub>L</sub> is the assimilation efficiency on live prey (E_XXX) and plants (EPlant_XXX), <em>Gr<sub>CX,i</sub></em> is the amount of different live prey types consumed, E<sub>DL</sub> assimilation efficiency on labile detritus (EDL_XXX) and E<sub>DR</sub> assimilation efficiency on refractory detritus (EDR_XXX), FDG is the proportion of the unassimilated prey (animal, plant and catch grazing) that is sent to detritus (FDG_XXX), FDGDR is proportion of unassimilated refractory detritus that is sent to detritus (FDGDR_XXX), FDGDL is proportion of unassimilated labile detritus that is sent to detritus (FDGDL_XXX), and FDM is the proportion of mortality products assigned to detritus (FDM_XXX). FDL determines the allocation of detritus between labile and refractory pools and is defined not by species but by four types of groups: fish and sharks (FDL_fish), birds and mammals (FDL_top), water column biomass pools except CEP and PWN (FDL_wc), other non-vertebrate biomass pools (FDL_benth – note that this includes CEP and PWN not just the true benthic invertebrate groups).</p>
<p>Production of <strong>refractory detritus (DR)</strong> uses the same equation, except that:</p>
<ol type="1">
<li><p>the final multiplication is done not by the FDL parameter but by (1-FDL)</p></li>
<li><p>detritus produced from <strong>all</strong> unassimilated food consumed by optional seabirds and fish NOT explicitly represented in the model is added to DR (see Fig 17 and below)</p></li>
</ol>
<p><span class="math display">\[W_{DR} = \ \left( \left( 1 - E_{L} \right) \bullet FDG \bullet \sum_{i = type}^{}{Gr_{CX,i} + \left( 1 - E_{DL} \right) \bullet FDGDL \bullet Gr_{CX,DL} + \left( 1 - E_{DR} \right)} \bullet FDGDR \bullet Gr_{CX,DR} + FDM \bullet M_{CX} \right) \bullet (1 - FDL) + FFSB \bullet M_{S}\]</span></p>
<p>Here <em>M<sub>S</sub></em> represents the optional extra age structured group mortality by predators NOT included in the model (modelled through the mS_XXX parameter), and FFSB is the assimilation <strong>inefficiency</strong> of these age structured groups. The assimilation inefficiency of <strong>seabirds NOT</strong> included in the model is specific for each age structured group consumed and is set in parameters FSBDR_XXX where XXX is the age-structured group eaten. The assimilation inefficiency of <strong>fish NOT</strong> included in the model is same for all prey species and is set in parameter FFDDR.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Assimilation (in)efficiency of predators included and not included in the model</strong></p>
<p>For <strong>predators included in the model</strong> (normal grazing term) assimilation efficiencies E (E, EPlant, EDL and EDR parameters) show the <strong>proportion of assimilated food</strong>. Fraction of unassimilated food then equals (1-E).</p>
<p>For optional <strong>predators NOT included in the model</strong> (represented through extra age structured group mortality terms mS) assimilation parameters FSBDR and FFDDR show the <strong>proportion of unassimilated food</strong>. This proportion of the optional extra mortality term is sent to the refractory detritus pool and the remaining fraction is <strong>taken out of the model</strong>.</p>
<p>It only makes sense to include non-zero parameters for FSBDR_XXX and FFDDR when extra mortality due to seabirds and fish NOT included in the model is represented through mS_SBXXX and mS_FDXXX arrays.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>The remaining fraction of the mortality products (1-FDM) and unassimilated food (1-FDG, 1-FDGDL and 1-FDGDR) and nitrogen used up in respiration is sent to the <strong>ammonia pool</strong>.</p>
<table class="caption-top table">
<caption><span id="_Toc526762775" class="anchor"></span>Table 23. Parameters determining waste production processes</caption>
<colgroup>
<col style="width: 7%">
<col style="width: 92%">
</colgroup>
<thead>
<tr class="header">
<th><strong>Parameter</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><p>E_XXX</p>
<p>EPlant_XXX</p>
<p>EDR_XXX</p>
<p>EDL_XXX</p></td>
<td>Assimilation efficiency of consumer XXX when feeding on: animal prey (and catch), plant prey, refractory detritus and labile detritus</td>
</tr>
<tr class="even">
<td>FDG_XXX</td>
<td>Proportion of unassimilated animal and plant food (faeces) by consumer XXX that becomes detritus</td>
</tr>
<tr class="odd">
<td>FDGDL_XXX</td>
<td>Proportion of unassimilated labile detrital food (faeces) by consumer XXX that becomes detritus</td>
</tr>
<tr class="even">
<td>FDGDR_XXX</td>
<td>Proportion of unassimilated refractory detrital food (faeces) by consumer XXX that becomes detritus</td>
</tr>
<tr class="odd">
<td>FDM_XXX</td>
<td>Proportion of mortality products of group XXX that is assigned to detritus</td>
</tr>
<tr class="even">
<td><p>FDL_fish</p>
<p>FDL_top</p>
<p>FDL_wc</p>
<p>FDL_benth</p></td>
<td>Proportion of labile detritus of all detritus produced by each of the four groups: fish and sharks (fish), birds and mammals (top), water column biomass pools except CEP and PWN (wc) and other non-vertebrate biomass pools including CEP and PWN (benth)</td>
</tr>
<tr class="odd">
<td>FSBDR_XXX</td>
<td>Assimilation inefficiency of seabirds not included in the model that feed on a functional group XXX. This proportion is sent to refractory detritus pool</td>
</tr>
<tr class="even">
<td>FFDDR</td>
<td>Assimilation inefficiency of fish not included in the model (same for all functional groups preyed upon). This proportion is sent to refractory detritus pool</td>
</tr>
</tbody>
</table>
</section>
<section id="spawn" class="level2">
<h2 class="anchored" data-anchor-id="spawn"><strong>10.8. Spawn</strong></h2>
<p>Spawning and recruitment is only explicitly modelled in age structured groups and age-structured biomass pools. They are handled by multiple routines in the <strong>atdemography.c</strong> file. In simple biomass pools reproduction is implicitly included in the growth term.</p>
<section id="sexual-maturation-in-age-structured-groups" class="level3">
<h3 class="anchored" data-anchor-id="sexual-maturation-in-age-structured-groups"><em>10.8.1. Sexual maturation in age-structured groups</em></h3>
<p>Maturation in age-structured groups is determined by two parameters. First, a fraction of each age class that is sexually mature is set in the FSPB_XXX parameter (ogive), with an option for gradual external change in this parameter through forcing files. Second, an optional minimum length (where the allometric length/weight conversion parameters li_a_XXX and li_b_XXX are used to get length) required for sexual maturation is set in min_li_mat_XXX. If this parameter is &gt; 0 then the no maturation will occur in an age class if it is below the maturation length even if FSPB parameter for that age class is &gt; 0.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Juveniles, adults (life history stages) and sexual maturation (or spawning contributions)</strong></p>
<p>Atlantis does not assume that the age of sexual maturation aligns with the ontogenetic shift in the use of habitats and prey. Sexual maturation (or the contribution to spawning in any one year by members of each age class) is given by the spawning ogive FSPB_XXX or the min_li_mat_XXX maturation length.</p>
<p>In contrast the age at which individuals switch from juvenile behaviour (in terms of feeding, habitat use, prey) to adult behaviour is given by XXX_age_mat. This is the first age group where adult behaviour is expressed and is modelled as a hard transition point with all individuals switching to adult behaviour at once.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</section>
<section id="spawning-in-age-structured-groups" class="level3">
<h3 class="anchored" data-anchor-id="spawning-in-age-structured-groups"><strong><em>10.8.2. Spawning in age-structured groups</em></strong></h3>
<p>The number of times an age-structured group spawn per year is given by NumSpawns in the <em>functional_groups.csv</em> file, although it is typically 1 (more frequent spawning is new functionality so let us know if you are having issues with it). When spawning occurs all spawn is produced in one time step. The day of the year that a species XXX spawns is set by the parameter XXX_Time_Spawn (if more than one spawning event occurs per year a vector of values is required). Although there is a parameter called XXX_spawn_period in the <em>biology.prm</em> file, it does not control the spawn period, but is just used to calculate recruitment delays (see below).</p>
<p>For each age group the amount of actual spawn produced is determined according to the condition of individuals in the age group at the time of spawning. The optimum spawning weight (<em>Wsp</em><sub>i</sub>) of an age group <em>i</em> is calculated as</p>
<p><span class="math display">\[Wsp_{i} = (1 + XRS) \bullet SN_{i}\]</span></p>
<p>where X_RS is a parameter defining the optimum ratio of RN to SN in well fed age structured groups (typically set to 2.65) and <em>SN<sub>i</sub></em> is SN of an age group <em>i</em></p>
<p>The amount of spawn (<em>Sp</em>, mgN) produced by an age group is calculated as</p>
<p><span class="math display">\[Sp = \left( \left( W_{sp} \bullet FSP - KSPA \right) - \left( W_{sp} - W \right) \right) \bullet Num \bullet FSPB\]</span></p>
<p>where (W<sub>sp</sub> –W) is the weight difference between the optimal weight for spawn and the actual nitrogen based weight of an individual (W=SN+RN), <em>FSP</em> is the proportion of weight-for-spawn used for spawn (FSP_XXX; typically ranging between 0.25 to 0.42), <em>KSPA</em> is a constant used by the spawning formulation (KSPA_XXX), <em>Nm</em> is the number of individuals in an age group, and <em>FSBP</em> is the proportion of mature reproducing individuals in the age group (FSPB_XXX). For those groups where not every adult reproduces ever year this value should be set &lt;1 even for fully mature age classes. Note, that if individual’s weight is smaller than the ideal weight at given age (defined by X_RS ratio), then the amount of spawn produced is decreased by the weight deficit (W<sub>sp</sub> –W) is subtracted from the spawn that would be produced in case of ideal weight.</p>
<p>The <em>KSPA</em> accounts for costs of reproduction or the amount of N taken out of weight but not included in the produced spawn. It is tuned during the parameterisation, and represents the base investment needed to build gonads and other reproductive materials. It is not however cohort specific, so the same cost is applied to each cohort.</p>
<p>Vertebrate spawn is calculated by the <em>Ecology_Age_Structured_Spawn()</em> routine in <strong>atdemography.c</strong>. This routine is called from two other routines: <em>Ecology_Do_Internal_Age_Structured_Spawning()</em> and <em>Ecology_Do_External_Age_Structured_Spawning()</em>, both in <strong>atdemography.c</strong>. The latter two routines convert individual spawn to spawn produced by a species in the model (or individuals that migrated outside the model domain), summing spawn over age groups to get the total spawn produced by the group.</p>
</section>
<section id="maturation-and-spawn-in-age-structured-biomass-pools" class="level3">
<h3 class="anchored" data-anchor-id="maturation-and-spawn-in-age-structured-biomass-pools"><strong><em>10.8.3. Maturation and spawn in age-structured biomass pools</em></strong></h3>
<p>In age-structured biomass pools the proportion of mature biomass in an age pool is set by the FSPB_XXX parameter. The amount of spawn (mgN) produced by an age biomass pool <em>i</em> is then calculated as</p>
<p><span class="math display">\[Sp_{i} = B_{i} \bullet FSP \bullet FSPB_{i}\]</span></p>
<p>where <em>B<sub>i</sub></em> is the biomass of the age pool (mgN), FSPB is the proportion of the mature biomass in the pool <em>i</em> (FSBP_XXX), and FSP is the proportion of spawning biomass in the mature biomass (FSP_XXX)</p>
<p><strong>The amount of nitrogen taken from the biomass pool after spawning is not necessarily equal to the spawn produced (Sp), but is set by the parameter prop_spawn_lost_XXX.</strong> This lost biomass represents semelparous reproduction and is used to represent the mortality of exhausted adults or can be set = 1 for full semelparity (all individuals die after reproduction, as in some squid species). The lost biomass is treated as normal mortality and sent to DL, DR and NH3 pools.</p>
<p>Reproduction of age structured biomass pools is done in the <em>Invertebrate_Reproduction()</em> routine in <strong>atdemography.c</strong></p>
<table class="caption-top table">
<caption><span id="_Toc526762776" class="anchor"></span>Table 24. Parameters determining maturation and spawn</caption>
<colgroup>
<col style="width: 9%">
<col style="width: 89%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"><strong>Parameter</strong></th>
<th style="text-align: left;"><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">FSPB_XXX</td>
<td style="text-align: left;">A fraction of each age class in an age-structured groups and age-structured biomass pools that is sexually mature</td>
</tr>
<tr class="even">
<td style="text-align: left;">min_li_mat_XXX</td>
<td style="text-align: left;">Minimum length required for sexual maturation in age-structured groups (calculated using length-weight conversion parameters li_a_XXX and li_b_XXX)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">XXX_age_mat</td>
<td style="text-align: left;">The first age group where adult behaviour is expressed in distribution, feeding, prey availability. It is modelled as hard transition and can be different from sexual maturation defined by FSPB_XXX parameters</td>
</tr>
<tr class="even">
<td style="text-align: left;">XXX_Time_Spawn</td>
<td style="text-align: left;">Day of the year that a species XXX spawns (all spawn occurs at once)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">FSP_XXX</td>
<td style="text-align: left;">Proportion of ideal weight for spawn that is used for spawn</td>
</tr>
<tr class="even">
<td style="text-align: left;">KSPA_XXX</td>
<td style="text-align: left;">Constant used by the spawning formulation, it represents reproductive costs and does not contribute to spawn</td>
</tr>
<tr class="odd">
<td style="text-align: left;">prop_spawn_lost_XXX</td>
<td style="text-align: left;">Amount of nitrogen taken from age-structured biomass pool after spawning. Used to represent both nitrogen lost to spawn and reproduction related mortality</td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="recruitment-and-ageing" class="level2">
<h2 class="anchored" data-anchor-id="recruitment-and-ageing"><strong>10.9. Recruitment and ageing</strong></h2>
<p>Atlantis does not currently model the larval period explicitly. In fully age-structured groups and age-structured biomass pools nitrogen produced as spawn is temporarily taken out of the model and is returned as recruits after a set larval period. In simple (non-age structured) biomass pools reproduction is not modelled explicitly. The larval period is set as</p>
<p>XXX_larval_period = XXX_Recruit_Time + XXX_Spawn_Period + optional_small_scale_random_variation</p>
<p>After this time recruits start “arriving” into the model throughout the period set in Recruit_Period_XXX. Nineteen different recruitment options are currently available for converting the base spawn amount into the biomass and numbers of recruits. The most common function applied for fish species is the Beverton-Holt (BH) function, dependent on both spawn produced and biomass of the species. For mammals, birds and some sharks a constant recruit production per adult is often used.</p>
<p>Some of the recruitment options ignore the spawn altogether, others ignore both spawn and species biomass, others use random number generation and others allow for scaling of recruitment by plankton production. In most cases the amount of nitrogen in spawn is not equal to the amount of nitrogen in recruits. Recruitment is modelled in the same way in age-structured groups and in age-structured biomass pools, but in age-structured groups the recruited biomass is subsequently converted to numbers, using the SN and RN values of recruits provided by the user (KWSR_XXX and KWRR_XXX respectively).</p>
<p>The amount of recruits produced can be influenced by one or several of these factors, and none of them are obligatory:</p>
<ol type="1">
<li><p>amount of spawn produced</p></li>
<li><p>stock biomass</p></li>
<li><p>phytoplankton</p></li>
<li><p>zooplankton</p></li>
<li><p>external recruitment forcing</p></li>
</ol>
<p>Recruited individuals are attributed to the first year of the first age group. The day before the recruits “arrive”, all other individuals age one year up, and can be allocated into a different age group depending on the number of years in each age group. Ageing is done in one time step, regardless of whether recruits arrive all at once or during a certain period of time.</p>
<section id="routines-used-to-calculate-recruitment-in-age-structured-groups" class="level3">
<h3 class="anchored" data-anchor-id="routines-used-to-calculate-recruitment-in-age-structured-groups"><strong><em>10.9.1 Routines used to calculate recruitment in age-structured groups</em></strong></h3>
<p>Several routines in Atlantis are involved in obtaining the final number of recruits. For age-structured groups (vertebrates), the total spawn produced by a species (calculated in the routines above, in the chapter on spawning) is passed to:</p>
<ol type="1">
<li><p>The <em>Ecology_Find_Embryos()</em> routine, which applies environmental scalars, such as scaling by plankton production. It also applies external recruitment forcing, for cases where recruitment time series are supplied. In the case of Beverton-Holt or Ricker recruitment, which are dependent on the spawn or species biomass, nothing is done in this routine.</p></li>
<li><p>The value above is then passed to the <em>Get_Recruits()</em> routine which calculates the recruitment (<em>Rs</em>) depending on the chosen recruitment option.</p></li>
<li><p>Next, two routines <em><strong>–</strong> Get_Vertical_Recruit_Position()</em> and <em>Find_Final_Recruit_Distribtuion() <strong>–</strong></em>determine the horizontal and vertical distribution of new recruits, depending on the distribution parameters provided by the user, whether the group has maternal care, set as flagmotherXXX or bear live young, set as flagbearliveXXX (in which case the recruits arrive where the adults are), and any optional environmental limitations (salinity, temperature, oxygen). The flagmotherXXX parameter allows users to set semelparous reproduction, where value of &lt;0 kills reproducing (FSPB_XXX proportion) adults after reproduction.</p></li>
<li><p>Next, the routine <em>Get_Settlers()</em> distributes the recruits on the temporal scale, depending on the length recruitment period and shape of the recruitment distribution.</p></li>
<li><p>Finally, immediately before the recruits arrive, all other age groups age one year up. This is done in <em>Update_Ageing_Numbers()</em> and a few other routines in the <strong><u>atdem</u>ography.c</strong> (which track the relative proportions per annual age class per age group)</p></li>
</ol>
<p>Next we will give details on how Atlantis calculates recruitment (<em>Rs</em>) from spawn (<em>Sp</em>) for different recruitment options. Equations below combine calculations done in both <em>Ecology_Find_Embryos()</em> and <em>Get_Recruits()</em> routines.</p>
</section>
<section id="stock-recruitment-options" class="level3">
<h3 class="anchored" data-anchor-id="stock-recruitment-options"><strong><em>10.9.2. Stock-recruitment options</em></strong></h3>
<p>The most common Beverton-Holt recruitment (selected as flagrecruitXXX=3) is implemented as:</p>
<p><span class="math display">\[Rc = \ \frac{Sp \bullet BHa}{Biom + BHb}\]</span></p>
<p>where <em>BHa</em> and <em>BHb</em> are species-specific Beverton-Holt alpha and Beverton-Holt beta parameters (BHalpha_XXX and Bhbeta_XXX, mgN), <em>Sp</em> is the spawn produced (mgN, see Spawn chapter) and <em>Biom</em> is the species total biomass. Note that because this <em>Rc</em> is dealing with nitrogen and not recruits entering a fishery, the obtained values will not match those given in stock assessment models.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Converting recruit biomass <em>Rc</em> to numbers</strong></p>
<p>The <em>Rc</em> is calculated in nitrogen biomass. For age-structured groups it is then converted to numbers using the recruit SN and RN values provided by the user as KWSR_XXX and KWRR_XXX parameters .</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>Other recruitment options are:</p>
<p><strong>1) Constant recruitment flagrecruitXXX=1</strong></p>
<p>Recruitment is constant through time and is independent on the stock size. The KDENR_XXX parameter in <em>biology.prm</em> file gives the number of “arriving” each year.</p>
<p><span class="math display">\[Rc = \ KDENR\]</span></p>
<p>This is a useful representation if recruitment is dictated by a large external stock not represented in the model.</p>
<p><strong>2) Recruitment determined by chlA concentration</strong> <strong>flagrecruitXXX=2</strong></p>
<p><strong>Recruitment is calculated separately in each cell</strong>. It is determined by the amount of chlA in the cell, species specific PP_XXX parameter relating recruitment to primary production and a ref_chl parameter providing the reference ChlA level (from both LG_PHY and SM_PHY)</p>
<p><span class="math display">\[Rc = PP \bullet \ \frac{chlA}{ref_{chl}}\]</span></p>
<p><strong>3) Beverton-Holt spawn and species biomass dependent recruitment flagrecruitXXX=3</strong></p>
<p>This is the most commonly applied recruitment option for fish and is described above.</p>
<p><strong>4) Random lognormal recruitment flagrecruitXXX=4</strong></p>
<p>In this option recruitment is completely random and does not depend either on the spawn produced or on the biomass of the species. The centre and variation of the lognormal distribution curve (δ<sub>lonorm</sub>), from which the recruitment is sampled, is set by the lognorm_mu and lognorm_sigma parameter respectively. Further, each species has a specific recruitment multiplier XXX_log_mult <em>(l_multiplier</em> in the equation below) converting the lognormal value, which is typically in the range from 0 to 1 (depending on lognorm_mu and lognorm_sigma values) to the actual number of recruits.</p>
<p><span class="math display">\[Rc = \ \delta_{lonorm} \bullet l_{multiplier}\]</span></p>
<p>Note, that even though Atlantis is fundamentally a deterministic model (the same results are obtained with same sets of conditions and parameters) the lognormal recruitment option would generate different results for each simulation, adding some stochasticity to Atlantis simulation outcomes.</p>
<p><strong>5)</strong> <strong>Recruitment determined by phyto- and zooplankton abundance flagrecruitXXX=5</strong></p>
<p>Recruitment is calculated similarly to the case 2, but now depends not on ChlA concentration but on the sum of the large phytoplankton and small zooplankton biomass, such that</p>
<p><em>plankton</em>=LG_PHY+LG_ZOO+MED_ZOO+SM_ZOO</p>
<p>The same two parameters PP_XXX and ref_chl used in the primary production based option are also used here to relate plankton levels to recruitment. Note that in ChlA or plankton dependent recruitment, spawn or species biomass is not taken into account. <strong>SM_PHY is not included here</strong> because it is not believed to be an important fish larvae food source.</p>
<p><span class="math display">\[Rc = PP \bullet \ \frac{plankton}{ref_{chl}}\]</span></p>
<p><strong>6)</strong> <strong>Beverton-Holt recruitment with lognormal variation and plankton dependency flagrecruitXXX=6</strong></p>
<p>This recruitment option combines options 3, 4 and 5. Recruitment depends on spawn and species biomass, as in option 3, plankton levels as in option 5, and is also modified with a lognormal scalar δ<sub>lonorm</sub> calculated based on lognorm_mu and lognorm_sigma parameters, as described in option 4.</p>
<p><span class="math display">\[Rc = \delta_{lonorm} \bullet \frac{Sp \bullet BHa}{Biom + BHb} \bullet \ \frac{plankton}{ref_{chl}}\]</span></p>
<p><strong>7)</strong> <strong>Beverton-Holt recruitment with encouraged recovery flagrecruitXXX=7</strong></p>
<p>Recruitment is calculated as in option 3, but a species-specific recruitment encouragement multiplier recover_mult_XXX is applied to <em>Rc</em> if a species biomass is below a threshold set in the recover_trigger parameter. This threshold levels shows a proportion of the initial stock biomass below which the recruitment multiplier is applied. The recruitment multiplier is applied for the number of years given in recover_subseq parameter. Moreover, the recruitment multiplier will only kick-in after a set period of days after the threshold stock proportion has been reached. This lag between low stock levels and encouraged recruitment is set in recover_span parameter. This recruitment option represents adaptive stocking or density dependent processes that only activate at low recruit levels.</p>
<p><strong>8)</strong> <strong>Beverton-Holt recruitment with prescribed recovery flagrecruitXXX=8</strong></p>
<p>Same as above, but the recruitment encouragement multiplier recover_mult_XXX is applied to <em>Rc</em> regardless of the stock levels for the period set in recover_span and recover_subseq parameters (potentially represented a period where an unknown environmental or ecological driver facilitated recruitment).</p>
<p><strong>9)</strong> <strong>Ricker recruitment flagrecruitXXX=9</strong></p>
<p>Ricker recruitment is not dependent on the spawn produced, but is based on biomass and the two parameters Ralpha_XXX and Rbeta_XXX. The <em>Rc</em> is calculated as</p>
<p><span class="math display">\[Rc = Biom \bullet e^{Ralpha \bullet \left( 1 - \frac{Biom}{Rbeta} \right)}\]</span></p>
<p>The Ricker recruitment parameter Ralpha and Rbeta are sometimes interpreted as r or intrinsic population growth and K or carrying capacity (see <a href="https://en.wikipedia.org/wiki/Ricker_model">here</a> for example)</p>
<p><strong>10) Beverton-Holt recruitment based only on species biomass flagrecruitXXX=10</strong></p>
<p>Calculated similarly to option 3, but instead of spawn (Sp), only <strong>total species</strong> biomass is used.</p>
<p><span class="math display">\[Rc = \ \frac{Biom \bullet BHa}{Biom + BHb}\]</span></p>
<p><strong>11) Recruitment is equal to spawn flagrecruitXXX=11</strong></p>
<p>This is the simplest recruitment option, where the nitrogen in recruits is simply equals the nitrogen spawned.</p>
<p><span class="math display">\[Rc = \ Sp\]</span></p>
<p><strong>12) Fixed number of offspring per individual flagrecruitXXX=12</strong></p>
<p>In this option the KDENR_XXX parameter provides the number of offspring per individual. This recruitment option is often applied for mammals, birds and viviparous sharks. When setting this option remember that Atlantis models an average individual, hence the actual number of offspring per female may need to be halved in a population of equal sex ratio or reduced if there is in utero dynamics (such as siblicide in sharks), or if mortality sources outside the model domain (e.g.&nbsp;land-based predation of turtle eggs) needs to be implicitly represented.</p>
<p>In this case, recruitment produced by an age group is calculated as</p>
<p><span class="math display">\[Rc = \ KDENR \bullet FSPB \bullet Num\]</span></p>
<p>where <em>FSPB</em> is the proportion of mature individuals in the age group (FSPB_XXX) and <em>Num</em> is the total number of individuals in the age group. Note, that for this recruitment option spawn will still be calculated and taken out of the model, but recruitment will only be determined by the number of offspring per individual multiplied by the number of spawning individuals.</p>
<p><strong>13) Recruitment forced from time series file flagrecruitXXX=13</strong></p>
<p>In this option the recruitment value Rc is taken from externally forced time series files. The name of the recruitment time series file must be provided in the <em>force.prm</em> file. In this option, spawn will still be calculated and taken out of the model, but recruitment will only be determined by external forcing.</p>
<p><strong>14) Larval recruitment –</strong> This option leaves space for explicit larval recruitment to be tracked, but it is currently not available in Atlantis.</p>
<p><strong>15) Jacknife recruitment flagrecruitXXX=15</strong></p>
<p>This option allows two levels of recruitment depending on the spawning stock biomass (SSB), calculated as the wet weight in tonnes (not nitrogen weight!) of spawning individuals and a set threshold <em>B<sub>jack</sub></em>, calculated as a fraction of total biomass:</p>
<p><span class="math display">\[B_{jack} = jack_{b} \bullet Biom\]</span></p>
<p>If SSB &lt; <em>B<sub>jack</sub></em></p>
<p><span class="math display">\[Rc = jack_{a} \bullet SSB\]</span></p>
<p>If SSB ≥ <em>B<sub>jack</sub></em></p>
<p><span class="math display">\[Rc = jack_{a} \bullet B_{jack}\]</span></p>
<p>The jack_a_XXX is interpreted as slope of recruitment, usually set as biomass of recruits / SSB. The jack_b_XXX is empirical proxy for a relationship with the carrying capacity (in the North Sea they used 0.3 * long term average total biomass). This recruitment relationship is taken from the North Sea stock assessment - see a more detailed description on the <a href="https://confluence.csiro.au/display/Atlantis/2013/04/23/Jackknife+recruitment">wiki</a></p>
<p><strong>16) Coral recruitment flagrecruitXXX=16</strong></p>
<p>Coral (and sponge) recruitment is determined by three specific coral recruitment parameters CrecruitA_XXX, CrecruitB_XXX and CrecruitC_XXX and a KDENR_XXX parameter setting a total number of recruits.</p>
<p>The recruitment is calculated as</p>
<p><span class="math display">\[Rc = KDENR + CrecA \bullet max\left( 0,\left( CrecB - e^{- CrecC \bullet B^{*}} \right) \right)\]</span></p>
<p>where B* is the spawning biomass (if corals are modelled as age-structured biomass pools the spawning biomass will be determined by the FSPB parameter).</p>
<p><strong>17) Baltic Sea version of the Ricker recruitment flagrecruitXXX=17</strong></p>
<p>The Baltic Sea version of the Ricker recruitment uses wet weight (not nitrogen based!) biomass B<sub>WW</sub> in <strong>thousands of tonnes</strong> and calculates Rc as:</p>
<p><span class="math display">\[Rc = 1000 \bullet Biom_{WW} \bullet Ralpha \bullet e^{- 1 \bullet Rbeta \bullet Biom_{WW}}\]</span></p>
<p><strong>18) Multiple time series of recruitment – specific forcing in separate boxes flagrecruitXXX=18</strong></p>
<p>This option allows forcing of recruitment from specific boxes. Numbers of recruits are forced by external forcing files and they only arrive in boxes specified in the forcing files. The horizontal and vertical distribution of these recruits is then determined according to the general rules, described in the next chapter below. See further description on Atlantis <a href="https://confluence.csiro.au/display/Atlantis/2014/05/12/Multiple+recruitment+time+series+turning+up+in+different+boxes">wiki</a>.</p>
<p><strong>19) Beverton-Holt recruitment based only on spawn only flagrecruitXXX=19</strong></p>
<p>Calculated similarly to option 3, but instead of biomass, only species spawn (<em>Sp</em>) is used.</p>
<p><span class="math display">\[Rc = \ \frac{Sp \bullet BHa}{Sp + BHb}\]</span></p>
<p>Finally, in addition to the recruitment options above, it is also possible to supplement recruits with external stocking. This is used to imitate, for example, supplementary stocking of salmon. Recruit stocking for a species XXX is set setting flagstockingXXX=1 and providing details on recruit stocking files. These files, and the code the recruit stocking triggers, are the same used to apply multiple time series of recruitment (flagrecruitXXX=18). See <a href="https://confluence.csiro.au/display/Atlantis/2015/03/24/Recruit+stocking">this</a> post on the wiki for further details.</p>
<p><strong>20)</strong> <strong>Ricker recruitment using SSB flagrecruitXXX=20</strong></p>
<p>As for option 9, but in this case SSB not total population biomass, with <em>Rc</em> calculated as</p>
<p><span class="math display">\[Rc = SBB \bullet e^{Ralpha \bullet \left( 1 - \frac{SSB}{Rbeta} \right)}\]</span></p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Interspecies dependency scalar</strong></p>
<p>It is possible in Atlantis to scale recruitment by the biomass of another species Y. This could be used to simulate, for example, consumption of cod eggs by sprat in the Baltic Sea. The interspecies dependency of species XXX recruitment is set with a flag intersp_depend_recruit_XXX, where 1 means linear scaling and 2 is inverse scaling. Parameter intersp_depend_sp_XXX gives the ID number of species Y (from the functional_groups.csv file). The intersp_depend_scale_XXX is a scalar to apply to recruitment, showing the influence of the biomass of species group Y on the recruitment of XXX.</p>
<p>If recruitment of species XXX is dependent of species Y, and intersp_depend_recruit_XXX=1 recruitment Rc of XXX is calculated as</p>
<p>Rc=Rc*intersp_depend_scale*biomassY</p>
<p>If intersp_depend_recruit_XXX=2</p>
<p>Rc=Rc*intersp_depend_scale/biomassY</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Calculating recruitment for different stocks</strong></p>
<p>If several stocks per species are used (identified with NumStocks parameter in the functional_ group.csv file) the procedures described above calculate recruitment for each stock separately, using the biomass of that stock rather than the total species biomass. The user also has an option to use stock specific recruitment scalars recSTOCK_XXX to upscale or downscale recruitment in different stocks.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</section>
<section id="environmental-habitat-and-external-recruitment-scalars" class="level3">
<h3 class="anchored" data-anchor-id="environmental-habitat-and-external-recruitment-scalars"><strong><em>10.9.3. Environmental, habitat and external recruitment scalars</em></strong></h3>
<p>The recruitment value <em>Rc</em> calculated above can be further modified by habitat availability, biomass of other species, environmental conditions, and additional recruit supplementing (stocking). All of these factors are optional and will be applied as multipliers on the base recruitment. Most of them affect recruitment in a negative way. When several scalars are applied (pH, temperature, contaminants), the user must be careful about the multiplicative nature of all these scalars, as they can drastically reduce recruitment.</p>
<p>The pH, temperature, salinity, contaminant and external time series scalars are calculated in <em>Get_Enviro_Recruit_Forcing()</em> routine. They use pHcorr, Tcorr and Scorr values that are applied to different physiological processes of a species and calculated in relevant pH and temperature routines (see chapter 13).</p>
<p><strong>1. pH scalar</strong></p>
<p>If a species is sensitive to pH (flagfecundsensitive_XXX=1) its recruitment will be scaled by a pH scalar, based on the pHcorr scalar calculated in the pH related routines (see below). The scalar is calculated as</p>
<p>pHscalar=1/pHcorr</p>
<p>as it assumes that pH effect on recruitment will be negative (scalar will be &lt;1)</p>
<p><strong>2. Temperature scalar</strong></p>
<p>To activate temperature dependent recruitment for any species set flagtempdepend_reprod to 1 and then set species parameters to indicate how the dependency is expressed. If a species is sensitive to temperature (flagtempsensitiveXXX) <strong>with the kind of effect dictated by the recruitment temperature dependency</strong> (flagq10receffXXX=1 or 2 – 1 for poorer when cooler, 2 if poorer when warmer, 0 indicates no actual effect regardless of flagtempsensitiveXXX setting) then recruitment is scaled by a multiplicative effect of pH and temperature (pHscalar*Tcorr). The pH effect will be 1 if a species is only sensitive to temperature and not pH. If species is temperature insensitive (flagq10receffXXX=0) only the pHscalar on recruitment is applied. If the pHscalar·Tcorr &gt; 1 then the inverse is applied to scale the recruitment (1/(pHscalar·Tcorr)) as the model assumes that effect of temperature and pH on recruitment will be negative</p>
<p>The <strong>effect of temperature</strong> on recruitment of temperature sensitive species (flagtempsensitiveXXX=1) is also simulated through minimum and maximum spawn temperatures (XXX_min_spawn_temp and XXX_max_spawn_temp). If temperature in the cell is outside these ranges, then the scalar is set to 0 and recruitment <strong>in that cell</strong> will be zero.</p>
<p><strong>3. Salinity scalar</strong></p>
<p>If a species is sensitive to salinity (flagSaltSensitive_XXX=1) the recruitment is first scaled by the Scorr scalar, and second, the models checks whether the salinity conditions in the cell are within the minimum and maximum spawn salinity ranges (XXX_min_spawn_salt and XXX_max_spawn_salt). If salinity in the cell is outside these ranges, then the scalar is set to 0 and recruitment <strong>in that cell</strong> will be zero.</p>
<p><strong>4. Contaminant scalar</strong></p>
<p>If the model tracks contaminants, then a scalar on recruitment due to contaminants is also calculated and applied.</p>
<p><strong>5. Externally forced scalar</strong></p>
<p>Users can supply external recruitment forcing scalar through time series files, listed in the <em>force.prm</em> file (see chapter 8). The external scalar will act as a multiplier on all other scalars calculated above. To indicate that external environmental recruitment forcing will be used set flagtsforcerecruit=1 in the <em>biology.prm</em> file. If this flag is set to 1, but no external forcing is provided then Atlantis will quit. This forcing represents environmental influence on recruitment (such as El Nino etc) and is called from <em>Get_Enviro_Recruit_Forcing()</em>.</p>
<p>On top of these scalars an additional external forcing can be applied to recruitment, in the same way as forcing is applied to growth, mortality or other processes. This recruitment forcing is done separately to allow further representation of external processes, such as for example contaminant spill. This final kind of <em>force.prm</em> recruitment scalar is applied within the <em>Get_Final_Distribution()</em> routine and is called from <em>Ecology_Get_Recruitment_Scalar()</em>.</p>
<p><strong>6. Habitat scalar</strong></p>
<p>Habitat scalar on recruitment acts as an additional multiplier on all environmental scalars above. It is applied if species recruitment is habitat dependent (XXX_rec_HabDepend=1). The habitat scalar is calculated as a proportion of suitable habitat in the cell.</p>
</section>
<section id="spatial-distribution-of-recruits" class="level3">
<h3 class="anchored" data-anchor-id="spatial-distribution-of-recruits"><strong><em>10.9.4. Spatial distribution of recruits</em></strong></h3>
<p>Once the recruitment biomass and/or numbers are calculated Atlantis will first define their vertical and then horizontal distribution. This is done in two routines <em>Get_Vertical_Recruit_Position()</em> and <em>Find_Final_Recruit_Distribtuion()</em>. The first routine uses the XXX_recruit_vdistrib parameters. This parameter provides entries for a maximum number of water layers so the routine rescales the vertical recruit distribution in each box depending on how many layers it has and weighs the vertical distribution towards the bottom or surface depending on whether a species is identified as demersal (flagdemXXX=1) or not.</p>
<p>Next, the routine <em>Find_Final_Recruit_Distribtuion()</em> determines the final spatial distribution of recruits.</p>
<p>If enviro_independ_larvae is set to 0 then all the scalars calculated above (including the environmental scalars) are applied to find the final recruit number for each box and layer given the vertical and horizontal distribution parameters. If enviro_independ_larvae is set to 1 then environmental scalars calculated above are not applied to the recruit_hdistrib and will only apply once the settlers (juveniles) have arrived. Also note the point in the larval period when environmental effects are calculated is set using flag_recruit_effect – set to 1 for the environmental effects to be calculated at the time the recruits settle (i.e.&nbsp;effects are on older larve or very small juveniles) or 0 if the environmental effects are to be calculated around the time of spawning (i.e.&nbsp;very young larvae effected). If any other period of time for the effect is needed contact the developers.</p>
<p>The base horizontal distribution, just like the vertical distribution is given by the user in XXX_recruit_hdistrib parameter, which simply gives proportional number of recruits allocated in each box. This potential distribution is modified based on whether the recruits actually recruit to the same location as the adults (RecruitType is set to 1 in the <em>functional_groups.csv</em> file) and whether local temperature, salinity, habitat conditions are suitable (if enviro_independ_larvae is set to 0).</p>
<p>The final number of recruits in a cell is then given as</p>
<p>Recruits_in_cell = AllRecruits · All_scalars · Vertical_distribution · Horizontal_distribution</p>
<p>It is also possible to apply an externally forced larval dispersal matrix (if RecruitType is set to 3 in the <em>functional_groups.csv</em> file) to modify the horizontal and vertical distribution of the recruits. This is described further <a href="https://confluence.csiro.au/pages/viewpage.action?spaceKey=Atlantis&amp;title=Recruitment+Options">here</a>.</p>
<p>It is possible that the sum of the final Recruits_in_cell may not sum to 1.0. If this final non-normalised value is desired set the flag norm_larval_distrib to 0 (i.e.&nbsp;to have it sum to 1.0 set the flag to 1). It may be desirable to have a non-normalised value if the desires is to capture variation in recruitment due to environmental conditions and dispersal (e.g.&nbsp;if environmental conditions are such that many cells have become unsuitable or if habitat conditions more larvae than would have normally survived under the assumed stock recruit relationship survive).</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Minimum and maximum salinity and temperature can greatly reduce recruitment</strong></p>
<p>It is important to realise that <strong>minimum and maximum spawn temperature and salinity act as scalars on recruitment in a box</strong>. Unlike the way in which settled individuals will contract away from unsuitable conditions during movement, recruit distributions <strong>do not</strong> contract to the boxes with suitable conditions but rather <strong>kill all the recruits</strong> that end up in unsuitable boxes. This means that if many boxes have conditions that are outside the suitable recruitment values the total recruitment in the model can be greatly reduced.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</section>
<section id="factors-defining-temporal-arrival-of-recruits" class="level3">
<h3 class="anchored" data-anchor-id="factors-defining-temporal-arrival-of-recruits"><strong><em>10.9.5. Factors defining temporal arrival of recruits</em></strong></h3>
<p>Once the total number of recruits to arrive in each cell is calculated using the procedures above, their arrival is spread throughout the Recruit_Period_XXX. This is done by the <em>Get_Settlers()</em> routine. Two options are available for this, controlled with a global (applied to all species) flagtrecruitdistrib parameter. The recruit arrival can be flat, so that same number of recruits arrive every day (flagtrecruitdistrib=1) or it can be humped (lognormal) (flagtrecruitdistrib=0).</p>
<p>If a lognormal distribution is used the user should also provide rec_m and rec_sigma parameters, determining the shape of the lognormally shaped curve (the same number of recruits will arrive, but not evenly spread through time).</p>
<p>Also note that there is an additional parameter XXX_cohort_recruit_entry that identifies which cohort recruits enter in. Typically, this is zero, but for multi-year and complicated migration patterns this may be set to a value of one or more.</p>
</section>
<section id="modifying-size-at-recruitment" class="level3">
<h3 class="anchored" data-anchor-id="modifying-size-at-recruitment"><strong><em>10.9.6. Modifying size at recruitment</em></strong></h3>
<p>Representing differential survival can be done in two ways, both of which use a time series file. To activate these set flag_modify_KWSR in the <em>biology.prm</em> file to a value greater than 0.</p>
<p>In both cases KWRR will be reset so that it is KWSR * X_RS.</p>
<p>If flag_modify_KWSR is set to 1 then the model will load a time series of multipliers to apply to the constant KWSR supplied for the group in the <em>biology.prm</em> file. If flag_modify_KWSR is set to 2 then the model will load a time series file of values to use for KWSR. In both cases the file for KWSR is supplied in <em>force.prm</em> as follows</p>
<p>KWSR_forcing <em>KWSR_filename.ts</em></p>
<p>The format of the <em>KWSR_filename.ts</em> is as follows for the scalar (flag_modify_KWSR set to 1) option (shown for 1 species, if multiple species are being forced in this way add an additional column for each species)</p>
<p># Scalar for structural sizes of recruits</p>
<p>#</p>
<p>## COLUMNS 2</p>
<p>##</p>
<p>## COLUMN1.name Time</p>
<p>## COLUMN1.long_name Time</p>
<p>## COLUMN1.units days since 2000-01-01 00:00:00 +10</p>
<p>## COLUMN1.missing_value 0</p>
<p>##</p>
<p>## COLUMN2.name XXX</p>
<p>## COLUMN2.long_name XXX</p>
<p>## COLUMN2.units scalar</p>
<p>## COLUMN2.missing_value 0.0</p>
<p>1 0.8</p>
<p>2 2.0</p>
<p>If the absolute value case (flag_modify_KWSR set to 2) is being used then the file format is as follows</p>
<p># Structural sizes of recruits</p>
<p>#</p>
<p>## COLUMNS 2</p>
<p>##</p>
<p>## COLUMN1.name Time</p>
<p>## COLUMN1.long_name Time</p>
<p>## COLUMN1.units days since 2000-01-01 00:00:00 +10</p>
<p>## COLUMN1.missing_value 0</p>
<p>##</p>
<p>## COLUMN2.name XXX</p>
<p>## COLUMN2.long_name XXX</p>
<p>## COLUMN2.units mg N</p>
<p>## COLUMN2.missing_value 0.0</p>
<p>1 110.8</p>
<p>2 112.1</p>
</section>
<section id="stochastic-variation-around-mean-recruitment" class="level3">
<h3 class="anchored" data-anchor-id="stochastic-variation-around-mean-recruitment"><strong><em>10.9.7. Stochastic variation around mean recruitment</em></strong></h3>
<p>It is possible to use all of the above to set the level of recruitment and then add normally distributed random fluctuations around that value. To do this set XXX_flag_recruit_stochastic to 1 in the <em>biology.prm</em> file and then set the sigma for the normal distribution for that species group using XXX_ norm_sigma. It will use the calculated level of recruitment as the “mean” of the distribution.</p>
</section>
<section id="ageing-of-individuals" class="level3">
<h3 class="anchored" data-anchor-id="ageing-of-individuals"><strong><em>10.9.8. Ageing of individuals</em></strong></h3>
<p>As recruits “arrive” they enter the first age group of a species. For species that have more than one calendar year per age group they are entered into the first year of the first age group.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Tracking calendar years within Atlantis age groups</strong></p>
<p>Although some species have several calendar years in one age group for the purposes of simplifying the ecological processes, for aging Atlantis actually tracks individuals of each calendar year within age group separately. This is illustrated in the left Figure 18 below where three calendar years (bars of the same colour) are tracked in each of the four age groups (AG1-AG4). The ageing routines move the individuals from one calendar year to the next (solid line arrows).</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>All individuals in one age group AG are assumed to be identical for predation, reproduction, fishing and other processes. Mortality can be proportionally allocated across annual age cohorts within an age class. However, it is also possible to dampen strong year classes during the ageing process and equalise their relative abundance. This assumes that the most abundant year classes will be more affected by predation and fishing that less abundant ones and that through time the relative abundance of year classes will even out. The figure on the right shows that the abundant youngest year class in year t becomes less abundant in year t+1, while the relative abundance of the other two year classes in that age group increases (shown with bold outlines). In the year t+1, the newly arrived year class is shown in green, and the oldest year class from age group 1 is now in the age group 2.</p>
<p><img src="markdown_output/media/image58.jpg" style="width:2.58333in;height:2.87639in"></p>
<p><img src="markdown_output/media/image59.jpeg" style="width:2.70694in;height:2.70625in"></p>
<p><span id="_Toc526762798" class="anchor"></span>Figure 18. Tracking of calendar years in Atlantis age groups. Left: three calendar years for each of the four age groups (AG1-AG4) illustrated. Right: dampening of strong year classes through time.</p>
<p>Dampening of strong year classes during ageing routines is controlled by the flagrecpeakXXX parameter used in the <em>Update_Age_Distrib()</em> routine. If flagrecpeakXXX=0 then dampening of strong year classes will use the recruitRangeFlat multiplier, and if the flagrecpeakXXX=1 then dampening of strong year classes will use the recruitRange multiplier. By setting the values for recruitRangeFlat and recruitRange the user can control how much dampening will occur. If no dampening is wanted set the values to 1.</p>
<p>The dampening effect and the related parameters described above are <strong>irrelevant for species that only have one calendar year in an age group</strong>.</p>
</section>
</section>
</section>
<section id="distribution-and-movement" class="level1">
<h1>11. DISTRIBUTION AND MOVEMENT</h1>
<p>The spatial distribution of organisms in the model domain is first set in the <em>initial_conditions.nc</em> file and is then updated at each time step. The main routine handling the distribution and movement is <em>Ecology_Total_Verts_And_Migration()</em> for fully age structured groups and <em>Ecology_Invert_Migration()</em> for biomass pools.</p>
<p>The simplest form of spatial distribution is called “prescribed movement” and is done when species do not have any density dependent movement (XXX_ddepend_move=0) or migrations (NumMigrations =0 in the <em>functional_groups.csv</em> file). In this case the spatial distribution is determined or prescribed by FXXX_SY parameters where the Y is typically 1-4 (representing seasonal shifts), but can now be set to whatever number the user desires (see details <a href="https://confluence.csiro.au/pages/viewpage.action?pageId=346916326">here</a>). These vectors determine the horizontal distribution, whereas VERTnight_XXX and VERTday_XXX parameters define vertical distribution. This is described in chapter 11.1</p>
<p>When XXX_ddepend_move=1 a species is set as sedentary. This means that once an individual (biomass) recruits into a box, there will be no further horizontal movement. So no prescribed movement parameters (FXXX_S1 to FXXX_S4) will be applied! The vertical movement is still allowed.</p>
<p>When density dependent movement is used (XXX_ddepend_move=2, 3, 5, 7) the parameters for “prescribed” horizontal movement are mostly or entirely overwritten by the movement and distribution due to food availability (except for the movement of the spawning and not feeding individuals during the spawning period, see below). This is a very important aspect of the model and makes full use of the spatially explicit modelling approach. However, for many models parameterising the density dependent movement can be challenging and it is often applied only to a few key species. This is because turning the density dependent movement without proper parameterisation can lead to all individuals accumulating in one box and completely depleting the food resource, or switching between boxes too rapidly. This is described in chapter 11.2</p>
<p>When XXX_ddepend_move=4 no explicit movement, vertical or horizontal, is permitted.</p>
<p>It is also possible to use home range movement (XXX_ddepend_move=6)</p>
<p>Finally, when species have annual migrations then the spatial distribution is also corrected for the proportion of individuals (or biomass) moving into and out of the model domain.</p>
<p>Below the options determining spatial distribution are described in further detail</p>
<section id="spatial-distribution-without-migration-or-density-dependent-movement" class="level2">
<h2 class="anchored" data-anchor-id="spatial-distribution-without-migration-or-density-dependent-movement"><strong>11.1 Spatial distribution without migration or density dependent movement</strong></h2>
<p>Atlantis updates horizontal and vertical distribution of species at each time step. When no density dependent or migratory movement is used, this is simply done by interpolating the proportion of biomass provided by the user in each box for each quarter of the year given in the FXXX_SY (typically FXXX_S1 to FXXX_S4) parameters. Note, that for the remainder of this discussion it will be assumed that numMoveEntries is set to 4 (representing traditional quarterly shifts in movement distributions). The same logic applies if numMoveEntries does not equal 4, just adjust the frequency and number of entries accordingly.</p>
<p>The horizontal distributions are given separately for adult and juvenile stages of the species, and the transition from juveniles to adults is given by the XXX_age_mat parameter, indicating the first mature age class (this does not mean that all individuals in that age class are mature or will spawn, see the chapter on Spawning for further details).</p>
<p>The abundance of species CX (juvenile or adult) in a cell j (<em>Abund<sub>j</sub></em>) at a given timestep is calculated as</p>
<p><span class="math display">\[Abund_{j} = Abund_{total} \bullet \left( \varepsilon \bullet \left( F_{Q + 1,j} - F_{Q,j} \right) + F_{Q,j} \right)\]</span></p>
<p>where <em>Abund<sub>total&nbsp;</sub></em>is the abundance of the species in the entire model domain (in biomass for biomass pools and in numbers), <em>ɛ</em> is the proportion of the quarter of the year that has passed, whereas F<sub>Q</sub> and F<sub>Q+1</sub> is the proportion of biomass in a cell j in quarter Q and Q+1 provided by the user (FXXX_S1 to FXXX_S4 parameters). Note that if Q is the last quarter of the year, then Q+1 is the first quarter of the year.</p>
<p>The equation above can be illustrated with a simple example, where <em>Abund<sub>total&nbsp;</sub></em>=100, F<sub>Q</sub> =0.1, F<sub>Q+1</sub> =0.2, and <em>ɛ</em> =0.5 (the current time step is half way between the start of the quarter Q and quarter Q+1. In this case the <em>Abund<sub>j</sub></em> = 100·(0.5· (0.2-0.1)+0.1)=100·(0.05+0.1)=15. This shows that while at the start of the quarter Q the proportion of the total biomass in box j is 0.1 and at the start of the quarter Q+1 it is 0.2, at the timestep half way through the quarter Q the proportion will be 0.15.</p>
<p>Once Atlantis calculates the horizontal distribution for a given time step it will then apply the vertical distribution VERTnight_XXX and VERTday_XXX parameters to distribute the individuals (or biomass) in each box among the vertical layers.</p>
<p>For epibenthic biomass pool groups it is possible to freeze their distribution based on the distribution in the initial conditions using the flag flagconstrain_epiwander. It is preferable for the distribution to emerge based on the settings of other constrains (like temperature tolerances or depth constraints). However, if all else fails and epibenthos has to be constrained but are growing in boxes that aren’t desired then set flagconstrain_epiwander to 1. This will then mean that non-mobile epibenthic groups (HorizontallyMigrates set to 0 in the <em>functional_groups.csv</em> file) will only be allowed in boxes where their biomass was non-zero in the initial conditions file.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Effects of environmental sensitivity on spatial distribution</strong></p>
<p>To activate temperature dependent movement set flagtempdepend_move to 1. Similarly, for salinity to affect any species set flagsaltdepend to 1 and for oxygen dependent movement set flagO2depend to 1. Once these overall flags are set in <em>biology.prm</em> file, then to detail how those effects are expressed set the species level parameters as described immediately below.</p>
<p>If a species is sensitive to temperature (flagtempsensitiveXXX), salinity (flagSaltSensitive_XXX) or oxygen (flagO2depend global parameter that affects all species), then its spatial distribution will be affected by the values of these environmental variables in a cell.</p>
<p>The setting of these sensitivity flags indicates what form of tolerance to include. If set to zero there is no sensitivity. If set to 1 then if the values in the cell are outside the tolerated range of the species – given by the min_move_temp_XXX, max_move_temp_XXX, min_move_salt_XXX, max_move_salt_XXX, min_O2_XXX parameters – then the abundance of the species in that cell will be set to 0 and the species distribution will be contracted to cells that have suitable conditions. If the flags are set to 2 then a more Gaussian curve is used to set tolerated environmental conditions. In that case you use the min and max values a for the hard edge option, but you must also include a parameter to scale the maximum possible suitability of conditions – XXX_K_temp_const, XXX_K_salt_const and XXX_K_o2_const – which is typically set to 2.0 to allow for relative preference under peak conditions being 1.0 (with poorer conditions having a value less than 1.0). This relative preference then scales the movement distribution calculated based on other movement settings. If this scalar has a value of 0 then conditions are considered beyond tolerable conditions and the group will avoid that cell. If flagenviro_displace is set to 0 then the species will be distributed smoothly over all suitable remaining cells. If flagenviro_displace is set to 1 then the individuals from the cell that is now unsuitable will be moved to adjacent cells only.</p>
<p>If flagenviro_kill is set to 0 then the individuals displaced by the environment will not die but simply move. <strong>If none of the cells have suitable conditions the species will disappear from the model domain.</strong> If flagenviro_kill is set to 1 then any individuals that do not find suitable cells die and are added to water column detritus in their source cell.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</section>
<section id="spatial-distribution-with-density-dependent-movement" class="level2">
<h2 class="anchored" data-anchor-id="spatial-distribution-with-density-dependent-movement"><strong>11.2 Spatial distribution with density dependent movement</strong></h2>
<p>Vertebrates and pelagic invertebrates can have density dependent movement, which means that the spatial distribution will be determined by the food availability in different boxes. At each time step Atlantis will evaluate potential feeding rates of each age group of a species (for which density dependent movement is turned on) in each box of the model. This will construct a spatial landscape of boxes determined by the potential feeding possibilities they provide. The distribution at each step will then be determined according to this landscape and two parameters, setting the threshold difference and ranking (see below). Two types of density dependent movement are available in Atlantis – basic (XXX_ddepend_move=2) and “sticky” (XXX_ddepend_move=3). We first describe how the basic density depdent movement works.</p>
<p>First, Atlantis calculates potential consumption (amount of assimilated food) for each age group in each of the model cells j:</p>
<p><span class="math display">\[A_{potential,j} = A_{potential,j} \bullet E_{1}\]</span></p>
<p>where <em>Gr<sub>potential.j&nbsp;</sub></em>is the potential grazing term of an age group if it was in the box j and E<sub>1</sub> is the assimilation efficiency on the live prey.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p>Currently, potential consumption is calculated assuming the modified Holling type II response (predcase=0) and hard feeding window. If other feeding functional responses are used, be aware that the density dependent movement might not be exactly right (though the relative forage field may be sufficient that it makes little difference). If you are concerned about this please contact the model developers as they do intend to allow for more functional responses in the movement code.</p>
<p>Also, currently the roc_wgt is a global parameter, which means that the same weighting of the potential growth rate surface (i.e.&nbsp;how clumpy to make it around hotspots) is applied to all species regardless of their identity, life history stage or size.</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>Next, Atlantis assess whether this food amount is higher than the threshold <em>γ<sub>thresh</sub></em> that would trigger movement. This threshold set with global k_roc_food parameter and SN amount in the age group <em>i</em>:</p>
<p><span class="math display">\[\gamma_{thresh} = SN_{i}\  \bullet kroc_{food}\]</span></p>
<p>The k_roc_food parameter is a proportional increase of SN weight (new weight / old weight) that would trigger movement into a better box, and it should always be &gt;1 (as fish should not move to boxes where food conditions are worse)</p>
<p>The spatial cells in the model are then ranked depending on whether the <em>A<sub>potential</sub></em> is larger or smaller than the threshold <em>γ<sub>thresh</sub></em></p>
<p><span class="math display">\[A_{potential,j}^{*} = A_{potential,j} \bullet roc_{wgt}\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ if\ A_{potential,j\ \ } &gt; \ \gamma_{thresh}\]</span></p>
<p><span class="math display">\[A_{potential,j}^{*} = \frac{A_{potential,j}}{roc_{wgt}}\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ if\ A_{potential,j\ \ } &lt; \ \gamma_{thresh}\]</span></p>
<p>New proportional abundance F<sub>j,t+1</sub> in the cell j is then calculated as</p>
<p><span class="math display">\[F_{j,t + 1} = \frac{A_{potential,j}^{*}}{\sum_{}^{}A_{potential,j}^{*}}\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \]</span></p>
<p>This can be illustrated with a hypothetical 4 box example with different food availability (potential assimilation rate) and an age cohort that has SN=100, k_roc_food=1.5, and roc_wgt=100</p>
<table class="caption-top table">
<colgroup>
<col style="width: 51%">
<col style="width: 10%">
<col style="width: 8%">
<col style="width: 13%">
<col style="width: 14%">
</colgroup>
<thead>
<tr class="header">
<th></th>
<th><strong>Box1</strong></th>
<th><strong>Box2</strong></th>
<th><strong>Box3</strong></th>
<th><strong>Box4</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Hypothetical potential assimilated food amount (A<sub>potential</sub> )</td>
<td>100</td>
<td>50</td>
<td>200</td>
<td>1000</td>
</tr>
<tr class="even">
<td>Current proportional distribution</td>
<td>0.25</td>
<td>0.25</td>
<td>0.25</td>
<td>0.25</td>
</tr>
<tr class="odd">
<td>Threshold values for each box (γ<sub>thresh</sub>) assuming k_roc_food=1.5</td>
<td>150</td>
<td>150</td>
<td>150</td>
<td>150</td>
</tr>
<tr class="even">
<td>The threshold condition A<sub>potential</sub> versus γ<sub>thresh</sub></td>
<td>100&lt;150</td>
<td>50&lt;150</td>
<td>200&gt;150</td>
<td>1000&gt;150</td>
</tr>
<tr class="odd">
<td><p>Modified A*<sub>potential</sub> after evaluating the threshold condition,</p>
<p>where roc_wgt=100 (A<sub>potential</sub>*roc_wgt or A<sub>potential</sub>/roc_wgt)</p></td>
<td>100/100 = 1</td>
<td><p>50/100</p>
<p>=0.5</p></td>
<td>200*100 =20000</td>
<td>1000*100 =10<sup>5</sup></td>
</tr>
<tr class="even">
<td><p>New distribution A*<sub>potential</sub> / ∑A*<sub>potential,j</sub></p>
<p>(where ∑A*<sub>potential,j</sub>=120001.5)</p></td>
<td>0.00</td>
<td>0.00</td>
<td>0.17</td>
<td>0.83</td>
</tr>
</tbody>
</table>
<p>It can be seen that after one time step the distribution can shift quite drastically. Atlantis users are encouraged to explore different values of roc_wgt and k_roc_food parameters on possible changes in the distributions. To prevent species teleporting about the model based on short term forage field shifts, the movement rate is adjusted by the swimming speed given in Speed_XXX parameter. The rate at which individuals (biomass) can move from one box to another is scaled by this swimming speed divided by the width of the box they are currently in.</p>
<p>It is also possible to use a more conservative approach to density dependent movement by using “sticky” movement XXX_ddepend_move=3. In this case the same test calculating the potential food assimilation is applied to all boxes. However, instead of redistributing all individuals across the model domain, the individuals that are in boxes where A<sub>potential,j</sub> &gt; γ<sub>thresh</sub> remain where they are and only the individuals from boxes where A<sub>potential,j</sub> &lt; γ<sub>thresh</sub> move throughout the model domain. So in the example above, the Box 3 has passed the test yet the proportion of individuals in this box still decreased from 0.25 to 0.17, because Box 4 is better. When “sticky” density dependent movement is used, only individuals from Box 1 and Box2 are redistributed.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Spawning will affect density dependence movement</strong></p>
<p>The density dependent movement mostly overrides the prescribed movement parameters given in the FXXX_S1 to FXXX_S4 (etc) parameters. However, for species that do not feed while spawning (feed_while_spawnXXX=0) the proportion of the age group spawning will not migrate according to the density dependence rules but according to the prescribed movement rules.</p>
<p>There are two options to control this for the simple (not “sticky”) density dependence movement by setting XXX_ddepend_move to 5 or 7.</p>
<p>When XXX_ddepend_move=5 only prescribed movement for the entire species will be used during the spawning period and only simple (non “sticky”) density dependent movement will be used at other times. When XXX_ddepend_move=7 the prescribed movement will be completely ignored and only simple density dependent movement will be applied at all times.</p>
<p>Note, that currently there is no way to eliminate prescribed movement during spawning for “sticky” density dependent movement. This can however be set by selecting a flag feed_while_spawn=1</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</section>
<section id="forcing-the-model-distributions-with-species-distribution-models" class="level2">
<h2 class="anchored" data-anchor-id="forcing-the-model-distributions-with-species-distribution-models"><strong>11.3. Forcing the model distributions with species distribution models</strong></h2>
<p>You may want to change the species distributions through time based on the output of another model, such as a species distribution model. This can be done by loading in forcing files for the new distributions. To do this in <em>force.prm</em> set use_move_entries 1 and then provide the information on the groups to influence and files to use as follows (this example has two groups, FXX and FYY, and 2 stages)</p>
<p>nforceMoveGroups 4</p>
<p>MoveGroupCodes 4</p>
<p>FXX_stage_0</p>
<p>FXX_stage_1</p>
<p>FYY_stage_0</p>
<p>FYY_stage_1</p>
<p>Also list the distribution nc files - 1 file per "MoveGroupCode"</p>
<p>FXX_stage_0_File.name DistribFilename1.nc</p>
<p>FXX_stage_1_File.name DistribFilename2.nc</p>
<p>FYY_stage_0_File.name DistribFilename3.nc</p>
<p>FYY_stage_1_File.name DistribFilename4.nc</p>
<p>You must also provide the starting time for using the forcing (so it does not have to start on day 0 of the model simulation)</p>
<p>FXX_stage_0_tstart 0</p>
<p>FXX_stage_1_tstart 0</p>
<p>FYY_stage_0_tstart 0</p>
<p>FYY_stage_1_tstart 0</p>
<p>An example of the format of each forcing nc file is shown below – for an example with 6 boxes and 3 time steps. You need one file per Group_Stage combination being forced.</p>
<p>netcdf filename {</p>
<p>dimensions:</p>
<p>b = 6 ;</p>
<p>t = UNLIMITED ; // (3 currently)</p>
<p>variables:</p>
<p>double t(t) ;</p>
<p>t:long_name = "time" ;</p>
<p>t:units = "seconds since 2005-01-01 00:00:00 +10" ; ## must be the same as the model</p>
<p>t:dt = 21600. ; ## must be entered but not actually used</p>
<p>double RAT_stage_0(t, b) ;</p>
<p>FXX_stage_0:units = " " ;</p>
<p>FXX_stage_0:_FillValue = 0.0 ;</p>
<p>FXX_stage_0:missing_value = 0.0;</p>
<p>FXX_stage_0:valid_min = 0.0;</p>
<p>FXX_stage_0:valid_max = 1.0;</p>
<p>// global attributes:</p>
<p>:geometry = "model.bgm" ;</p>
<p>data:</p>
<p>t =</p>
<p>0, 2592000, 5184000; ## See Note below about the fact the model does not really check alignment of these times and model time so use the forced tstart time sensibly.</p>
<p>FXX_stage_0 =</p>
<p>0, 0.2, 0.5, 0.1, 0.1, 0.1, ## There must be a value for each box for each time slice included in the file</p>
<p>0, 0.3, 0.2, 0.2, 0.2, 0.1, ## The distribution values for any one time slice must sum to 1.0</p>
<p>0, 0.5, 0.1, 0.1, 0.1, 0.2;</p>
<p>}</p>
<p>Note that for biomass pool groups (i.e.&nbsp;groups without age structure), treat it as having one cohort (cohort 0). So the distribution should be listed as for “stage_0” (so use: XXX_stage_0, XXX_stage_0.nc, XXX_stage_0_tstart for group XXX).</p>
<p>Also note that regardless of the t values provided in the forcing file it will start applying the distributions at the tstart value listed in <em>force.prm</em>. This means that if you give it a tstart value that is midway through a year it will use the first entry in the file as the distribution for that time of year (irrespective of what the t value is in the file, sorry this is an artifact of how the forcing works for tracers in the model). The model expects the distributions provided for each time slice to sum to 1.0. No more distributions will be read in once the end of the file is reached - <strong>it will not rewind</strong> it will just use the final years’ worth of distributions repeatedly into the future.</p>
<p>The number of time slices per year to include in the forcing file must match the value for numMoveEntries in <em>functional_groups.csv</em> to ensure that the species distribution model slices are correctly read in Atlantis. The number of time slices per year should also be reflected by the initial seasonal distributions in the <em>biology.prm</em> file seasonal distributions. For instance, in the case of a monthly-resolved species distribution model FXXX_S1 to FXXX_S12 should be provided.</p>
</section>
<section id="migration-into-and-out-of-the-model-domain" class="level2">
<h2 class="anchored" data-anchor-id="migration-into-and-out-of-the-model-domain"><strong>11.4. Migration into and out of the model domain</strong></h2>
<section id="annual-migrations" class="level3">
<h3 class="anchored" data-anchor-id="annual-migrations"><strong><em>11.4.1. Annual migrations</em></strong></h3>
<p>It is possible to allow a part or all of an age cohort or biomass pool to leave the model domain for a set period of time. This is used to imitate seasonal migrations to areas outside the model domain. This option is typically used for large mammals such as whales, which only spend a specific period of time within the model domain, but it can be applied to any group. For age-structured groups separate migration parameters are given for juveniles and adults, and one parameter is needed for biomass pools. Migration is handled by the <em>Ecology_Invert_Migration()</em> and E<em>cology_Total_Verts_And_ Migration()</em> routines in <strong>atmovement.c</strong></p>
<p>The relevant parameters defining the migration are detailed in the <em>functional_groups.csv</em> and <em>migrations.csv</em> files.</p>
<p>In <em>functional_groups.csv</em> set whether a group is migrating by setting: NumMigrations &gt; 1. This is the number of migrations initiated per calendar year. Also set whether those migrations are multiyear or not (MultiYrMigrations = 1 means at least one of the migrations started per year can last for many years).</p>
<p>An example of the format of <em>migrations.csv is shown below</em>, but for each migration you must set each of the following parameters. Note that if multiple stages migrate at the same time please provide an entry for each stage for that migration. If an animal matures into another stage while it has migrated away that is fine without an additional entry as it is covered by the original StartStage. If a biomass pool there is simply one entry for each migration for the group.</p>
<ul>
<li><p>StartStage for each migration.</p></li>
<li><p>MigID is the ID number for each migration per year for that stage and group. The order is based on the StartTofY (i.e.&nbsp;in order of the migrations based on when individuals leave the model domain). This ID starts with 0 and then increments from there (so if there were 3 migrations per year for example these would have IDs 0, 1, 2). The numbering resets for each stage.</p></li>
<li><p>StartTofY for each migration. This is the day of the year (calendar day) when that stage of the group migrates out of the model.</p></li>
<li><p>EndTofY for each migration. This is the day of the year (calendar day) when migration ends and the individuals return to the model domain.</p></li>
<li><p>Leave_Period is the number of days over which individuals leave the system as they migrate away</p></li>
<li><p>Return_Period is the number of days over which individuals return to the system</p></li>
<li><p>IsAnnualMigration is whether this particular migration happens annually (1) or stretches for multiple years (0)</p></li>
<li><p>MinTimeAway while a value is required for all migrations, the value is really only used for multi-year migrations. It is the minimum number of years that migrants stay out of the model.</p></li>
<li><p>MaxTimeAway while a value is required for all migrations, the value is really only used for multi-year migrations. It is the maximum number of years that migrants stay out of the model.</p></li>
<li><p>MigrateStockReturn specifies which stock the migrants return to. In most models there is only one stock so simply set it to 0 here.</p></li>
<li><p>MigPropSizeInc indicates whether the species has grown while outside the model domain. For age structured groups this indicates the proportional increase in both SN and RN while outside the model domain, for biomass groups it is straight biomass change. Growth outside the model domain adds nitrogen into the system.</p></li>
<li><p>MigPropSurvive is the survivorship of migrants (proportion of biomass or numbers returning back into the model domain). The remaining nitrogen (1-FSM) is lost from the model.</p></li>
<li><p>PartialExit is whether for the multi-year migrations only some of the age group leaves and for those that don’t leave they remain resident rather than attempt to leave in subsequent years.</p></li>
<li><p>MinAgeLeave if the model has a mix of leaving and resident in multi-year migrations (i.e.&nbsp;PartialExit = 1) this is the youngest cohort leaving. If only the youngest age group migrating set this to 0.</p></li>
<li><p>MaxAgeLeave if the model has a mix of leaving and resident in multi-year migrations (i.e.&nbsp;PartialExit = 1) this is the oldest cohort leaving. If only the youngest age group migrating set this to 0 (or other juvenile residents will leave in subsequent years).</p></li>
<li><p>ReprodAllowed is whether individuals participating in this migration can spawn when they reach maturity (1 = yes, 0 = no). If set to 0 then in multi year spawners (like salmon), the individuals will not spawn even if they reach adult ages during that migration.</p></li>
</ul>
<p>In the <em>biology.prm</em> file you also need to set k_migslow to indicate if the species migrates all at once or gradually. If k_migslow is set to 1 then there will be gradual migration using the Leave_Period and Return_Period (if it is set to 0 then the migrants leave or return in 1 time step). If k_migslow=1 and Leave_Period and Return_Period must be &gt;0 or else Atlantis will quit with an error message.</p>
<p>Within the model, when it is a migration time, the code checks what proportion of the age class migrates out of each box and take that number (biomass) out of the model. This value set in the vector MigIOBox_XXX (with a value given per box in the model) in the <em>biology.prm</em> file. Separate vectors are required juveniles and adults in age-structured groups. The amount removed from the model is stored in the MIGRATION array.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>NOTE!</strong></p>
<p><strong>Example of the migrations.csv file</strong></p>
<table>
<colgroup>
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 3%">
<col style="width: 5%">
<col style="width: 4%">
<col style="width: 7%">
<col style="width: 7%">
<col style="width: 8%">
<col style="width: 7%">
<col style="width: 7%">
<col style="width: 7%">
<col style="width: 4%">
<col style="width: 8%">
<col style="width: 5%">
<col style="width: 6%">
<col style="width: 6%">
</colgroup>
<thead>
<tr>
<th style="text-align: center;">
GroupCode
</th>
<th style="text-align: center;">
StartStage
</th>
<th style="text-align: center;">
MigID
</th>
<th style="text-align: center;">
StartTofY
</th>
<th style="text-align: center;">
EndTofY
</th>
<th style="text-align: center;">
Leave_Period
</th>
<th style="text-align: center;">
Return_Period
</th>
<th style="text-align: center;">
isAnnualMigration
</th>
<th style="text-align: center;">
MinTimeAway
</th>
<th style="text-align: center;">
MaxTimeAway
</th>
<th style="text-align: center;">
<p>
MigrateStock
</p>
<p>
Return
</p>
</th>
<th style="text-align: center;">
<p>
MigProp
</p>
<p>
SizeInc
</p>
</th>
<th style="text-align: center;">
MigPropSurvive
</th>
<th style="text-align: center;">
PartialExit
</th>
<th style="text-align: center;">
MinAgeLeave
</th>
<th style="text-align: center;">
MaxAgeLeave
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: center;">
FAA
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
182
</td>
<td style="text-align: center;">
258
</td>
<td style="text-align: center;">
15
</td>
<td style="text-align: center;">
15
</td>
<td style="text-align: center;">
1
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
2.7
</td>
<td style="text-align: center;">
0.72
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
</tr>
<tr>
<td style="text-align: center;">
FAA
</td>
<td style="text-align: center;">
1
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
182
</td>
<td style="text-align: center;">
258
</td>
<td style="text-align: center;">
15
</td>
<td style="text-align: center;">
15
</td>
<td style="text-align: center;">
1
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0.67
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
</tr>
<tr>
<td style="text-align: center;">
FBB
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
227
</td>
<td style="text-align: center;">
90
</td>
<td style="text-align: center;">
12
</td>
<td style="text-align: center;">
5
</td>
<td style="text-align: center;">
1
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0.05
</td>
<td style="text-align: center;">
1
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
</tr>
<tr>
<td style="text-align: center;">
FCC
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
213
</td>
<td style="text-align: center;">
67
</td>
<td style="text-align: center;">
24
</td>
<td style="text-align: center;">
18
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
3
</td>
<td style="text-align: center;">
6
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
1.5
</td>
<td style="text-align: center;">
0.9
</td>
<td style="text-align: center;">
1
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
1
</td>
</tr>
<tr>
<td style="text-align: center;">
FCC
</td>
<td style="text-align: center;">
1
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
120
</td>
<td style="text-align: center;">
160
</td>
<td style="text-align: center;">
12
</td>
<td style="text-align: center;">
12
</td>
<td style="text-align: center;">
1
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0.98
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
</tr>
<tr>
<td style="text-align: center;">
FCC
</td>
<td style="text-align: center;">
1
</td>
<td style="text-align: center;">
1
</td>
<td style="text-align: center;">
200
</td>
<td style="text-align: center;">
250
</td>
<td style="text-align: center;">
12
</td>
<td style="text-align: center;">
12
</td>
<td style="text-align: center;">
1
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0.05
</td>
<td style="text-align: center;">
0.95
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
</tr>
<tr>
<td style="text-align: center;">
FCC
</td>
<td style="text-align: center;">
1
</td>
<td style="text-align: center;">
2
</td>
<td style="text-align: center;">
280
</td>
<td style="text-align: center;">
330
</td>
<td style="text-align: center;">
17
</td>
<td style="text-align: center;">
10
</td>
<td style="text-align: center;">
1
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0.9
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
</tr>
<tr>
<td style="text-align: center;">
FXX
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
30
</td>
<td style="text-align: center;">
151
</td>
<td style="text-align: center;">
10
</td>
<td style="text-align: center;">
15
</td>
<td style="text-align: center;">
1
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0.1
</td>
<td style="text-align: center;">
0.75
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
</tr>
<tr>
<td style="text-align: center;">
FXX
</td>
<td style="text-align: center;">
1
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
60
</td>
<td style="text-align: center;">
107
</td>
<td style="text-align: center;">
12
</td>
<td style="text-align: center;">
12
</td>
<td style="text-align: center;">
1
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0.05
</td>
<td style="text-align: center;">
0.93
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
</tr>
<tr>
<td style="text-align: center;">
FXX
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
1
</td>
<td style="text-align: center;">
190
</td>
<td style="text-align: center;">
240
</td>
<td style="text-align: center;">
15
</td>
<td style="text-align: center;">
15
</td>
<td style="text-align: center;">
1
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0.1
</td>
<td style="text-align: center;">
0.755
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
</tr>
<tr>
<td style="text-align: center;">
FXX
</td>
<td style="text-align: center;">
1
</td>
<td style="text-align: center;">
1
</td>
<td style="text-align: center;">
250
</td>
<td style="text-align: center;">
324
</td>
<td style="text-align: center;">
12
</td>
<td style="text-align: center;">
12
</td>
<td style="text-align: center;">
1
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0.05
</td>
<td style="text-align: center;">
1
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
<td style="text-align: center;">
0
</td>
</tr>
</tbody>
</table></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</section>
<section id="multi-year-migrations" class="level3">
<h3 class="anchored" data-anchor-id="multi-year-migrations"><strong><em>11.4.2. Multi-year migrations</em></strong></h3>
<p>To trigger multi-year migrations set MultiYrMigrations to 1 for the group in the <em>functional_groups.csv</em> file. The time of year when the group leaves, returns and the periods of time over which migration occur are set as for annual migrations, as is the MigIOBox_XXX. The number of years spent away – set using MinTimeAway and MaxTimeAway is important for specifying how long these multi-year migrations last and whether there are any staggered returns or the entire set of migrants return together.</p>
</section>
</section>
<section id="external-population-model" class="level2">
<h2 class="anchored" data-anchor-id="external-population-model"><strong>11.5. External population model</strong></h2>
<p>To represent the juvenile (at sea) period, which occurs outside the model domain, a specific population model is triggered, which deals explicitly with growth and mortality during this external period (rather than the simple handling typical of standard Atlantis external migrations). To trigger this population model set external_populations to 1 in the <em>run.prm</em> file and then indicate which species use this option using the isExternal flag in the <em>functional_groups.csv</em> file. This will then execute the external population model on all animals for that species in the MIGRATION array (i.e.&nbsp;any individuals of that functional group that are outside the model domain).</p>
<p>The additional parameters associated with the population dynamics model are from the the von Bertalanffy Growth equation:</p>
<p>Linf_XXX – 'L infinity', the asymptotic length at which growth is zero.</p>
<p>Kbert_XXX – growth rate</p>
<p>tzero_XXX – the age at which the functional group would have had zero size, this is tuned to adjust the equation for the initial size of the functional group.</p>
<p>Additional mortality parameters are also required – these need to be entered as vectors with one entry for each life history stage (i.e.&nbsp;juveniles and adults as for XXX_mL for example):</p>
<p>XXX_mLext – the base back ground mortality applied in the external population model</p>
<p>XXX_mPext – the “P” external mortality rate</p>
<p>These mortalities are applied as to give total external mortality (Next_mort) as:</p>
<p><span class="math display">\[N_{ext\_ mort} = \left( m_{l\_ ext} + m_{p\_ ext} \right) \bullet N_{t}\]</span></p>
<p>where <em>N<sub>t</sub></em> is the number of individuals in an age class at time <em>t</em> and the mortality rates mLext and mPext. This population model also uses the existing Atlantis parameters li_a_XXX, li_b_XXX, pR_XXX and age_mat_XXX.</p>
</section>
<section id="invading-species" class="level2">
<h2 class="anchored" data-anchor-id="invading-species"><strong>11.6. Invading species</strong></h2>
<p>It is possible to allow for an invading species (functional group) to enter the model domain. At present only a couple of simple representations of invasion exist, but further may come later (please check the wiki for the latest options). The parameters required by the current invasion models are given in Table 25.</p>
<p><span id="_Toc526762777" class="anchor"></span></p>
<table class="caption-top table">
<caption>Table 25. Parameters used in the invading species sub-model</caption>
<colgroup>
<col style="width: 4%">
<col style="width: 94%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"><strong>Parameter</strong></th>
<th style="text-align: left;"><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">invading_sp_model</td>
<td style="text-align: left;">0 = no invader; 1 = simple drip-feed entry into one box and spread from there); 2 = points of invasion and spread are determined by habitat and environmental conditions; 3 = complex invasion model (currently not implemented)</td>
</tr>
<tr class="even">
<td style="text-align: left;">InvaderIndex</td>
<td style="text-align: left;">ID number of the guild that is invading (i.e.&nbsp;the ID for the group in the <em>functional_groups.csv</em> file)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">minInvaderAge</td>
<td style="text-align: left;">Minimum age class invading</td>
</tr>
<tr class="even">
<td style="text-align: left;">maxInvaderAge</td>
<td style="text-align: left;">Maximum age class invading</td>
</tr>
<tr class="odd">
<td style="text-align: left;">InvaderEntryBox</td>
<td style="text-align: left;">Box that marks the point of entry for the invading group</td>
</tr>
<tr class="even">
<td style="text-align: left;">InvaderMinDepth</td>
<td style="text-align: left;">Minimum depth invader tolerates</td>
</tr>
<tr class="odd">
<td style="text-align: left;">InvaderMaxDepth</td>
<td style="text-align: left;">Maximum depth invader tolerates</td>
</tr>
<tr class="even">
<td style="text-align: left;">InvaderStartDay</td>
<td style="text-align: left;">Day of the run the invasion begins</td>
</tr>
<tr class="odd">
<td style="text-align: left;">InvaderEndDay</td>
<td style="text-align: left;">Day of run invasion ends</td>
</tr>
<tr class="even">
<td style="text-align: left;">InvaderScalar</td>
<td style="text-align: left;">Proportional scalar to represent drip feed of invaders to existing population – used if invading_sp_model is set to 1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">InvaderSpeed</td>
<td style="text-align: left;">Rate of spread of front of the invading species (m hr-1)</td>
</tr>
<tr class="even">
<td style="text-align: left;">InvadersEntering</td>
<td style="text-align: left;">1 = if invaders begin from 0 baseline (in which case the starting numbers entering the model domain must be specified in InvaderStartNums); 0 = a small number of the invading species (functional group) are already in the model when the run begins and the invasion represents a range shift (or expansion of the population) or a turnover in the species composition of a functional group so the numbers group based on invaders</td>
</tr>
<tr class="odd">
<td style="text-align: left;">InvaderStartNums</td>
<td style="text-align: left;">The number of invaders entering (per age class) on first day of the invasion. If not all age classes are invading then set the numbers in the non-invading age classes to 0.</td>
</tr>
<tr class="even">
<td style="text-align: left;">InvaderEntryLayer</td>
<td style="text-align: left;">The water column layer the invaders enter through in the case where InvadersEntering is set to 1</td>
</tr>
</tbody>
</table>
<p>In all cases invasion is represented via scaling the numbers (or biomass if a biomass pool) in the InvaderEntryBox (between days InvaderStartDay and InvaderEndDay). If invading_sp_model is set to 1 then the dispersal of invaders is given by standard movement for age structured groups and via spilling into adjacent cells for biomass pools (with new cells added once biomasses reach saturation points). If invading_sp_model is set to 2 the dispersal is based on depth, temperature and salinity preferences, habitat and prey fields.</p>
</section>
</section>
<section id="processes-in-bacterial-and-inanimate-pools" class="level1">
<h1>12. PROCESSES IN BACTERIAL AND INANIMATE POOLS</h1>
<section id="bacterial-processes" class="level2">
<h2 class="anchored" data-anchor-id="bacterial-processes"><strong>12.1. Bacterial processes</strong></h2>
<p>sBacteria can be represented explicitly or implicitly in Atlantis. Not all early ecosystem models included bacteria, and PPBIM did not explicitly model bacteria. Below is an excerpt of the PPBIM report justifying the decision not to include bacteria:</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>Reasons for excluding bacteria in PPBIM (from Muray and Parslow 1997)</strong></p>
<p>“Heterotrophic bacteria have been included explicitly in a number of previous models, most notable Fasham et al.&nbsp;(1990). Bacteria in aquatic ecosystems play a number of important roles. The most important role is to attack, ingest and remineralise both particulate and dissolved organic matter, releasing dissolved inorganic nutrients. This role is represented implicitly in the model by assigning a breakdown rate to the detrital pools. Given the other uncertainties associated with the nature of detritus, and the lack of information on bacterial biomass, growth rates and loss rates in Port Phillip Bay, there is little reason to believe that an explicit model of bacterial dynamics would increase the accuracy of predicted detrital breakdown rates.</p>
<p>Bacteria can also serve as a food source for microzooplankton, and thereby divert organic matter which would otherwise be "lost" back to higher trophic levels. This alternative food source may have some impact on microzooplankton dynamics, and their interaction with small phytoplankton. The small phytoplankton – microzooplankton interaction is important under oligotrophic conditions, but much less important under the higher nutrient loadings of concern in Port Phillip Bay. Moreover, owing to the inefficiency of the microbial loop, relatively little bacterial production finds it way to higher trophic levels (Ducklow 1991). Recent studies have suggested that a large fraction of water-column bacterial production is not grazed but is killed by viruses instead (Weinbauer and Peduzzi 1995) leading to a further reduction in utilisation of bacterial production by higher trophic levels (Murray and Eldridge 1994).”</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>Atlantis explicitly models aerobic bacteria in the water column and sediments, identified as pelagic bacteria (PB) and benthic bacteria (BB) respectively. This means that functional groups can feed on bacteria rather than on detritus. The representation of bacteria used in Atlantis does a much better job than old consumption-based representations, which treated bacteria as just another consumer. By making bacterial biomass dependent on the volume of detritus the bacterial processes are dynamically represented in a more rigorous way.</p>
<p>A precursor to Atlantis also included anaerobic bacteria. At present these are only implicitly represented in Atlantis, but a dynamic form will be added in a future release.</p>
<table class="caption-top">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th><p><strong>Key bacterial processes</strong></p>
<p>The growth of PB and BB bacteria is dynamically dependent on the concentration of labile and refractory detritus (DL and DR), where DL is typically considered a primary food source. <strong>Bacteria uptake both DL and DR and produce DR, DON and NH3</strong>. Bacterial wastes do not contribute to DL but convert detritus into nutrients as part of the nitrification-denitrification and remineralisation processes.</p>
<p>Mortality of bacteria is caused by predation (grazing) by different consumers, as well as non-predation mortality due to oxygen limitation, acidification (optional), extra mortality (optional) and linear mortality</p></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
<p>Pelagic bacteria processes are run by the routine <em>Pelagic_Bacteria_Processes()</em> in <strong>atGroupProcess.c</strong></p>
<p>The growth of pelagic bacteria is modelled as</p>
<p><span class="math inline">\(G_{BP} = mum \bullet B \bullet \left( 1 - \frac{B}{XPB_{DL} \bullet DL + XPB_{DR} \bullet DR} \right)^{kBP}\)</span></p>
<p>where mum is the maximum growth rate (mum_PB), XPB<sub>DL</sub> and XPB<sub>DR</sub> is the maximum ratio of the bacteria to DL and DR biomass (XPB_DL and XPB_DR parameters), <em>B</em> is the biomass of the pelagic bacteria, <em>DL</em> and <em>DR</em> is the biomass of labile and refractory detritus and k_PB is the constant determining the exponent of growth (1- linear, 3-drop off before maximum, 5- drop off at maximum). The growth of sediment bacteria (BB) is modelled in the same way, except that mum_BB and k_BB parameters are used.</p>
<p>There is an option to allow for stimulation of bacterial growth by bioirrigation and bioturbation. This option is turned setting flagbactstim=1 and will enhance growth depending on the biomass of infaunal organisms and porosity of the sediment (see chapter 4).</p>
<p>Mortality (non-predation) of pelagic and sediment bacteria is modelled as</p>
<p><span class="math display">\[M_{BP} = \ \left( \left( mL + \left( 1 - \delta_{O2} \right) \bullet mO + mA + mS \right) \bullet B_{BP} \right) \bullet mortsc\]</span></p>
<p>where <em>mL</em> is linear mortality, <em>mA</em> is the acidification mortality, <em>mS</em> is the extra mortality due to acidification effects, <em>mO</em> is the oxygen limitation mortality, where (1-δ<sub>O2</sub>) is the oxygen limitation scalar, <em>B<sub>PB</sub></em> is the biomass of pelagic bacteria (or BB is sediment bacteria are modelled) and <em>mortsc</em> is an forced external mortality scalar.</p>
<p>The uptake of detritus by pelagic bacteria is the modelled as</p>
<p><span class="math display">\[\frac{d(DL)}{dt} = \frac{Gr_{BP} \bullet PB_{DL}}{B_{BP} \bullet E_{DL,BP}}\]</span></p>
<p>where E<sub>DL,&nbsp;PB</sub> is the assimilation efficiency on labile detritus (EDL_BP for pelagic bacteria and EDL_BB for benthic bacteria) and <em>PB_DL</em> is the current (box specific) partitioning of PB to DL (vs DR).</p>
<p>Uptake of refractory detritus DR is modelled in the same way except that PB_DR is used and assimilation efficiencies on refractory detritus is used (EDR_BP and EDR_BB).</p>
</section>
<section id="biogeochemical-processes" class="level2">
<h2 class="anchored" data-anchor-id="biogeochemical-processes"><strong>12.2 Biogeochemical processes</strong></h2>
<p>Details on the fluxes in the inanimate pools have been described in Fulton et al.&nbsp;2004a and more recently in Link et al.&nbsp;2011 <a href="https://confluence.csiro.au/display/Atlantis/Useful+Reading?preview=/43155482/341447378/Link_et_al_2011_NEUS_tech_memo.pdf">Atlantis-NEUS model report</a>. The table of inanimate pools tracked in the water column and sediments is given in chapter 6.1.</p>
</section>
</section>
<section id="influence-of-environmental-factors-on-ecological-processes" class="level1">
<h1>13. INFLUENCE OF ENVIRONMENTAL FACTORS ON ECOLOGICAL PROCESSES</h1>
<section id="temperature-two-methods-to-get-temperature-scalar" class="level2">
<h2 class="anchored" data-anchor-id="temperature-two-methods-to-get-temperature-scalar"><strong>13.1. Temperature: two methods to get temperature scalar</strong></h2>
<p>Temperature can have a large influence on many biogeochemical and ecological processes, but in Atlantis the user has an option to turn the temperature effects off, by setting flagq10=0 (NOT recommended given the fundamental temperature dependence of most physiological processes). Temperature effects are mostly applied through the Tcorr scalar calculated in the <em>Get_Tcorr()</em> routine. Setting flagq10=0 means that Tcorr scalar will be set to 1.</p>
<p>There are two ways to calculate the Tcorr scalar and different methods can be applied for different species, selected with the q10_method_XXX parameter. Some details on the calculations are given <a href="https://confluence.csiro.au/display/Atlantis/Effect+of+temperature+on+functional+groups">here</a>.</p>
<p>If the q10_method_XXX=0 then a simple Q10 based correction scalar is calculated, using the reference temperature of 15C:</p>
<p><span class="math display">\[T_{scalar} = {q10}^{\frac{\left( T_{H2O} - 15 \right)}{10}}\]</span></p>
<p>where q10 is a species-specific parameter provided in the q10_XXX and is typically set to 2. The Figure 19 shows the effect of different q10_XXX values.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="markdown_output/media/image60.emf" style="width:4.34861in;height:3.28333in" class="figure-img"></p>
<figcaption><span id="_Toc526762799" class="anchor"></span>Figure 19. The <em>T<sub>scalar</sub></em> values at different water temperatures.</figcaption>
</figure>
</div>
<p><strong>Black</strong>: q10=2</p>
<p><strong>Orange</strong>: q10=2.5</p>
<p><strong>Red</strong>: q10=3</p>
<p>A more complicated six parameter function is used if q10_method_XXX=1 (based on Gary Griffiths PhD thesis)</p>
<p><span class="math display">\[Tcorr = ln(2) \bullet \phi_{A} \bullet {Cons}_{B}^{Temp} \bullet \text{exp}\left( - \phi_{C} \bullet \frac{\left| Temp - {Temp}_{OPT} \right|^{Cons}}{\phi_{corr}} \right)\]</span></p>
<p>where <em>ø<sub>A</sub></em> is a species-specific coefficient (temp_coefftA_XXX), <em>Cons<sub>B</sub></em> is the global coefficient (temp_coeffB), <em>Temp</em> is ambient water temperature<em>, Temp<sub>OPT</sub></em> is a species-specific optimum temperature (q10_optimal_temp_XXX), <em>ø<sub>C&nbsp;</sub></em>is the global coefficient (temp_coeffC), <em>Cons</em> is a global exponent parameter (temp_exp), and <em>ø<sub>corr</sub></em> is a species q10 correction parameter (q10_correction_XXX).</p>
<p>This function aims to imitate a humped response, where rates are highest at optimum temperature levels and decrease when the temperature is below or above the optimum. The function is very sensitive to changes in parameters and before applying it the users should carefully explore the shape of the function for the chosen parameter values (Figure 20). The black line shows the shape of the response curve for the original parameter values, except for <em>Temp<sub>OPT</sub></em> which was chosen to be 19C</p>
<p>ø<sub>A</sub> (temp_coefftA_XXX) =0.85</p>
<p>Cons<sub>B</sub> (temp_coeffB) = 1.06</p>
<p>Temp<sub>OPT</sub> (q10_optimal_temp_XXX) = 19C</p>
<p>ø<sub>C</sub> (temp_coeffC) = 1</p>
<p>Cons (temp_exp) = 3</p>
<p>ø<sub>corr</sub> (q10_correction_XXX) = 1000</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="markdown_output/media/image61.emf" style="width:5.05486in;height:3.46944in" class="figure-img"></p>
<figcaption><span id="_Toc526762800" class="anchor"></span>Figure 20. A humped temperature response used when q10_method_XXX=1.</figcaption>
</figure>
</div>
<p><strong>Black</strong>: original values as shown in parentheses below</p>
<p><strong>Grey</strong>: Temp<sub>OPT</sub> = 12C (19C)</p>
<p><strong>Green</strong>: ø<sub>A</sub>= 0.95 (0.85)</p>
<p><strong>Orange</strong>: Cons<sub>B</sub> = 1.1 (1.06)</p>
<p><strong>Blue</strong>: ø<sub>C</sub> = 1.1 (1.0)</p>
<p><strong>Pink</strong>: Cons = 2.5 (3.0)</p>
<p><strong>Purple</strong>: ø<sub>corr</sub> = 100 (1000)</p>
</section>
<section id="temperature-effects-on-feeding-parameters-and-assimilation-efficiency" class="level2">
<h2 class="anchored" data-anchor-id="temperature-effects-on-feeding-parameters-and-assimilation-efficiency"><strong>13.2. Temperature: effects on feeding parameters and assimilation efficiency</strong></h2>
<p>Once the Tcorr scalar has been calculated for the species and its ambient temperature in the cell using one of the two methods above, it is applied as a scalar to a range of processes. Typically, parameters that are scaled by Tcorr are indicated with T15, but this is not always the case.</p>
<p><em>A) Primary producers</em></p>
<p>In all primary producers the Tcorr scalar is applied to the light saturation (KI_XXX_T15) and maximum growth rate parameter (mum_XXX_T15)</p>
<p><em>B) Consumer feeding parameters</em></p>
<p>For <strong><em>biomass pools</em></strong> the C_ and mum_ values in the biological parameter file are given as rates at 15C and they are <strong>always</strong> scaled (multiplied) by Tcorr scalar for a given water temperature in a cell.</p>
<p>For <strong>age structured groups</strong> the search volume (vl_a) is always scaled by Tcorr, but the temperature scaling of C_ and mum_ is applied <strong>only</strong> if flagtempsensitiveXXX is set to 1. This is an inherited convention form the models the approaches were taken from and may change in the future.</p>
<p><em>C) Consumer assimilation efficiency – optional</em></p>
<p>Atlantis has an option to set improved or decreased assimilation efficiency depending on the temperature. There is no clear consensus in the ecological community on how temperature affects assimilation efficiency (it appears to be taxa specific); hence the user can decide whether to use this option.</p>
<p>To allow for temperature effect on assimilation efficiency set flagq10effXXX to 1</p>
<p>When <strong>flagq10effXXX = 1</strong> the <strong>efficiency is poorer in cooler water</strong> and the Tcorr scalar (Fig. 20) is applied <strong>only</strong> if water temperature is lower than the optimum or reference level (depending on the q10_method_XXX used).</p>
<p>If <strong>flagq10effXXX= 2</strong> the <strong>efficiency is poorer in warmer water</strong> and the Tcorr scalar (see Fig. 20) is applied <strong>only</strong> if water temperature is higher than the optimum or reference level (depending on the q10_method_XXX used).</p>
<p>In the calculations above if the Tcorr is &gt;1 then the Tcorr=1/T_scalar. This means that Tcorr is always &lt;1, which ensures that efficiency is decreased when correcting for temperature effects.</p>
<p><em>D) Mortality</em></p>
<p>For all species the linear mortality (mL), quadrating mortality (mQ) and extra mortality (mS) values are scaled by Tcorr.</p>
<p><em>E) Physical parameters</em></p>
<p>Parameters that determine the rate of breakdown are all scaled by the Tcorr scalar. They include:</p>
<p>r_DL_T15 – rate of labile detritus breakdown (day<sup>-1</sup>)</p>
<p>r_DC_T15 – rate of carrion breakdown (day<sup>-1</sup>)</p>
<p>r_DR_T15 – rate of refractory detritus breakdown (day<sup>-1</sup>)</p>
<p>r_DON_T15 – rate of dissolved organic nitrogen breakdown (day<sup>-1</sup>)</p>
<p>r_DSi_T15 – rate of detrital silica breakdown (day<sup>-1</sup>)</p>
<p>K_nit_T15 – rate of nitrification by free bacteria (mgN day<sup>-1</sup>)</p>
</section>
<section id="salinity" class="level2">
<h2 class="anchored" data-anchor-id="salinity"><strong>13.3. Salinity</strong></h2>
<p>The <strong>salinity</strong> effects on <strong>biomass pool and age structured group</strong> physiological processes are modelled through an optional Scorr, designed to reflect the sensitivity of physiological processes to salinity conditions. The Scorr scalar is not calculated dynamically, but supplied by the user in salt_correction_XXX parameter. The Scorr scalar is applied only if:</p>
<p>1) an organism is identified as sensitive to salinity, with flagSaltSensitive_XXX</p>
<p>2) an organism an outside the salinity limits defined with XXX_min_salt and XXX_max_salt</p>
<p><strong>The Scorr scalar is applied in the same way as for the Tcorr scalar described above. For age structured groups the Scorr cannot be applied alone without applying the Tcorr scalar.</strong> This means that if a species is identified as sensitive to salinity, but NOT sensitive to temperature, the Scorr scalar will not be applied to the physiological processes that have optional temperature scaling. This is not the case for biomass pool groups, where Scorr is applied regardless.</p>
</section>
<section id="acidification" class="level2">
<h2 class="anchored" data-anchor-id="acidification"><strong>13.4. Acidification</strong></h2>
<p>Atlantis has an option to include effects of acidification on different physiological processes, predatory interactions and non-predation mortality. The pH effects are activated by setting flagmodelpH to 1 and flagpHsensitive_XXX to 1.</p>
<p>The calculation of the pH correction scalar (pHCorr) is described in detail on wiki <a href="https://confluence.csiro.au/display/Atlantis/2012/10/19/Handling+effects+of+acidification">here</a>.</p>
<p>Briefly, depending on the pH_sensitivity_model selected, Atlantis will calculate the pHCorr using monodynamic, non-linear, linear, piecewise or quadratic approaches (see link above for details). The scalar will be &lt;1 at decreasing pH values. It can be based on the pH values of ∆ [H+] (set flag_use_deltaH to 1 to use ∆ [H+]).</p>
<p>As for the Scorr scalar, <strong>for age structured groups</strong> the pHCorr will be applied to the processes affected by temperature (feeding rates, assimilation efficiency, mortality), <strong>only if a species is sensitive to pH</strong> (flagpHsensitive_XXX=1) <strong>and sensitive to temperature.</strong></p>
<p>The pH can also affect other processes, as listed in Table 26.</p>
<p><span id="_Toc526762778" class="anchor"></span>Table 26. Effects of pH on physiological and ecological processes. See detailed description <a href="https://confluence.csiro.au/display/Atlantis/2012/10/19/Handling+effects+of+acidification">here</a></p>
<table class="caption-top table">
<thead>
<tr class="header">
<th><strong>What is affected</strong></th>
<th><strong>How to activate</strong></th>
<th><strong>How is it applied</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><em>Processes for which pHcorr scalar is applied at the same time as the Tcorr scalar, described in chapter 13.2</em></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Growth and non-predation mortality rates</td>
<td><p>flagpHsensitive_XXX=1</p>
<p>For age structured groups it is applied only if:</p>
<p>flagtempsensitiveXXX=1</p></td>
<td>Growth and non-predation mortality are affected in opposite ways by pH. The unmodified C and mum will be multiplied by pHCorr (and decrease as a result) and the unmodified mL and mQ multiplied by 1.0/pHCorr (and increase)</td>
</tr>
<tr class="odd">
<td>Search volume (if predcase=5)</td>
<td>flagpHsensitive_XXX=1</td>
<td><p>The vla_T15 is multiplied by pHCorr (and decreases)</p>
<p>This is only applied for age structured groups</p></td>
</tr>
<tr class="even">
<td>Assimilation efficiency</td>
<td><p>flagpHsensitive_XXX=1</p>
<p>flagtempsensitiveXXX=1</p>
<p>flagq10eff_XXX=1 or 2</p></td>
<td>Four assimilation efficiencies are multiplied by pHCorr (and decreases)</td>
</tr>
<tr class="odd">
<td><em>Processes for which pHcorr scalar is applied differently from Tcorr scalar</em></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Availability of prey to predators</td>
<td><p>flagpHsensitive_XXX=1</p>
<p>flagpredavaileffect_XXX = 1</p></td>
<td><p>For <strong>biomass pool prey</strong> availability to predators (defined in pPREY or ontogenetic diet matrices) is <strong>increased</strong> multiplying by 1.0/pHCorr</p>
<p>For <strong>age structured group prey</strong> availability to predators (defined in pPREY or ontogenetic diet matrices) is <strong>decreased</strong> multiplying by pHCorr</p></td>
</tr>
<tr class="odd">
<td>Nutritional content of a species to its predators; mostly intended to simulate nutritional content of primary producers</td>
<td><p>flagpHsensitive_XXX=1</p>
<p>flagnutvaleffect_XXX=1</p></td>
<td>The amount of prey biomass available to a predator (defined in pPREY or ontogenetic diet matrices) is further multiplied by pHCorr (ot represent that more must be eaten to get the same nutritional content) and in this way decreased, reflecting lower nutritional content</td>
</tr>
<tr class="even">
<td>Reduced larval survival before recruitment</td>
<td><p>flagpHsensitive_XXX=1</p>
<p>flagfecundsensitive_XXX =1</p></td>
<td>The number of recruits is multiplied by pHCorr and therefore decreased</td>
</tr>
<tr class="odd">
<td>Modifying thermal tolerance of a species</td>
<td><p>flagpHsensitive_XXX=1</p>
<p>flagcontract_tol_XXX = 1</p></td>
<td>The thermal tolerance decreases according to contract_tol_XXX parameter which defined the number of degrees to contract the temperature tolerances by as pH drops</td>
</tr>
<tr class="even">
<td>Additional mortality</td>
<td><p>flagpHsensitive_XXX=1</p>
<p>flagPHmortcase &gt;0</p></td>
<td>Extra mortality applied for ALL groups</td>
</tr>
<tr class="odd">
<td>Extra mortality</td>
<td><p>flagpHsensitive_XXX=1</p>
<p>pHmortstart = 1</p></td>
<td>Another logistic extra mortality term is added, see details <a href="https://confluence.csiro.au/display/Atlantis/2015/05/14/Acidification+induced+mortality">here</a></td>
</tr>
</tbody>
</table>
</section>
<section id="oxygen" class="level2">
<h2 class="anchored" data-anchor-id="oxygen"><strong>13.5. Oxygen</strong></h2>
<p>The oxygen dependency is modelled differently from the temperature, salinity and pH effects described above. The oxygen content of the water does not affect the physiological processes, but recruitment and distributions. It also can lead to oxygen stress induced mortality of biomass pool groups.</p>
<p>The effect of oxygen limitation on feeding rates and linear mortality of biomass pool groups is non-optional and is modelled with O2case, mO_XXX, mD_XXX, KO2_XXX and KO2LIM_XXX parameters</p>
<p>The sensitivity of distributions and recruitment on oxygen concentration is optional. It is applied to both age structured groups and biomass pools and is activated by a global flagO2depend parameter and species-specific minimum oxygen concentrations XXX_min_O2 parameter, setting the minimum tolerated oxygen level. If this option is used then species distribution will contract to areas above the minimum oxygen level. Further, the recruits that arrive into cells with oxygen concentrations lower than the minimum will be killed (not contracted!, see chapter 10.9.3.).</p>
</section>
</section>
<section id="final-overview-of-ecology-routines" class="level1">
<h1>14. FINAL OVERVIEW OF ECOLOGY ROUTINES</h1>
<p>The schematic representation below gives a brief overview of the key ecology routines and the order they are called. The Harvest and socio-economic routines are not included in this representation.</p>
<p>The <em>main()</em> routine controls the entire run, and calls the <em>runNextTimeStep()</em> if the run should continue (has not been terminated or the last time step has not been reached yet). The <em>runNextTimeStep()</em> checks the time of the year and initiaties any relevant annual routines, such as migration or spawning (they are executed later, but the initiation is started here). It then calls the <em>Ecology_Box_Biology()</em> routine to run all ecological processes in a given cell. The <em>Ecology_Box_Biology()</em> assesses the environmental conditions (light, oxygen, habitat, eddies) and calls for routines that should be executed in each of the five box types: <strong>water column</strong> (not in contact with the sediment layer), <strong>sediment</strong> (not in contact with the water column), <strong>epibenthic</strong> (bottom water layer and top sediment layer), <strong>ice</strong> and <strong>land</strong>. Within each of these boxes Atlantis first runs the main ecological processes, such as feeding, maintenance (if used), growth, nutrient cycling, waste production. These processes are run on an adaptive time step, which means that for the smallest groups (bacteria, phytoplankton) the processes might have to be dynamically repeated many times within a single model timestep (see introduction in Chapter 6 on adaptive time step). Once these processes are completed, Atlantis then runs the required annual ecology processes, such as reproduction, recruitment and ageing.</p>
<p>The land and ice boxes have only been added recently and are not discussed in details below (the execution of ice is quite similar to the water column, while the land is like the epibenthic layers).</p>
<p>Atlantis runs the water column, sediment or epibenthic processes depending on what kind of interactions are expected (epibenthic processes are run only in the epibenthic boxes). The water column, sediment and epibenthic processes are summarised below.</p>
<p><img src="markdown_output/media/image62.jpg" style="width:6.56814in;height:8.58261in"></p>
<p><span id="_Toc526762779" class="anchor"></span>Table 27. List of key ecological processes run in the three main distribution types: water colum, sediment, and epibenthic layer. List of the pools tracked in the three types is given in table 8.</p>
<p>PP – phytoplankton (including dinoflagelates), ZP – zooplankton, MA – macroalgae and seagrasses, BI – benthic invertebrates, PI – cephalopods and prawns (nektonic pelagic invertebrates), VE – vertebrates, MB - microphytobenthos</p>
<table class="caption-top table">
<colgroup>
<col style="width: 10%">
<col style="width: 6%">
<col style="width: 6%">
<col style="width: 6%">
<col style="width: 6%">
<col style="width: 6%">
<col style="width: 6%">
<col style="width: 6%">
<col style="width: 6%">
<col style="width: 6%">
<col style="width: 6%">
<col style="width: 6%">
<col style="width: 6%">
<col style="width: 6%">
</colgroup>
<thead>
<tr class="header">
<th><strong>Process</strong></th>
<th><strong>N</strong></th>
<th><strong>Si</strong></th>
<th><strong>PB</strong></th>
<th><strong>BB</strong></th>
<th><strong>DL</strong></th>
<th><strong>DR</strong></th>
<th><strong>PP</strong></th>
<th><strong>ZP</strong></th>
<th><strong>MA</strong></th>
<th><strong>BI</strong></th>
<th><strong>PI</strong></th>
<th><strong>VE</strong></th>
<th><strong>MB</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td colspan="14"><em>Water column processes</em></td>
</tr>
<tr class="even">
<td>Used by phytoplankton</td>
<td>+</td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Used by bacteria</td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Input from wastes</td>
<td>+</td>
<td>+</td>
<td></td>
<td></td>
<td>+</td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Mineralisation</td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td>+</td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Nitrification</td>
<td>+</td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Oxygen production</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Nutrient uptake</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Lysis (nutrient stress)</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Feeding</td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td>+</td>
<td>+</td>
<td></td>
</tr>
<tr class="odd">
<td>Growth</td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td>+</td>
<td></td>
<td></td>
<td>+</td>
<td>+</td>
<td></td>
</tr>
<tr class="even">
<td>Respiration</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>(+)</td>
<td></td>
</tr>
<tr class="odd">
<td>Predation mortality/losses</td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td>+</td>
<td>+</td>
<td>+</td>
<td>+</td>
<td></td>
<td></td>
<td>+</td>
<td>+</td>
<td></td>
</tr>
<tr class="even">
<td>Other mortality</td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td>+</td>
<td></td>
<td></td>
<td>+</td>
<td>+</td>
<td></td>
</tr>
<tr class="odd">
<td>Waste production</td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td>+</td>
<td></td>
<td></td>
<td>+</td>
<td>+</td>
<td></td>
</tr>
<tr class="even">
<td colspan="14"><em>Sediment processes</em></td>
</tr>
<tr class="odd">
<td><strong>Process</strong></td>
<td><strong>N</strong></td>
<td><strong>Si</strong></td>
<td><strong>PB</strong></td>
<td><strong>BB</strong></td>
<td><strong>DL</strong></td>
<td><strong>DR</strong></td>
<td><strong>PP</strong></td>
<td><strong>ZP</strong></td>
<td><strong>MA</strong></td>
<td><strong>BI</strong></td>
<td><strong>PI</strong></td>
<td><strong>VE</strong></td>
<td><strong>MB</strong></td>
</tr>
<tr class="even">
<td>Used by microphytobenthos</td>
<td>+</td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Used by bacteria</td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Input from wastes</td>
<td>+</td>
<td>+</td>
<td></td>
<td></td>
<td>+</td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Mineralisation</td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td>+</td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Nitrification</td>
<td>+</td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Denitrification</td>
<td>+</td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Oxygen production</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>+</td>
</tr>
<tr class="odd">
<td>Nutrient uptake</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>+</td>
</tr>
<tr class="even">
<td>Feeding</td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Growth</td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td>+</td>
</tr>
<tr class="even">
<td>Predation mortality/losses</td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td>+</td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td>+</td>
</tr>
<tr class="odd">
<td>Other mortality</td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td>+</td>
</tr>
<tr class="even">
<td>Waste production</td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Bio turbation/irrigation</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td colspan="14"><em>Epibenthic processes</em></td>
</tr>
<tr class="odd">
<td><strong>Process</strong></td>
<td><strong>N</strong></td>
<td><strong>Si</strong></td>
<td><strong>PB</strong></td>
<td><strong>BB</strong></td>
<td><strong>DL</strong></td>
<td><strong>DR</strong></td>
<td><strong>PP</strong></td>
<td><strong>ZP</strong></td>
<td><strong>MA</strong></td>
<td><strong>BI</strong></td>
<td><strong>PI</strong></td>
<td><strong>VE</strong></td>
<td><strong>MB</strong></td>
</tr>
<tr class="even">
<td>Used by macrophytes</td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Input from wastes</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Mineralisation</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Oxygen production</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Nutrient uptake</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Feeding</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Growth</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td>+</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Predation mortality/losses</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td>+</td>
<td></td>
<td></td>
<td>+</td>
<td>+</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Other mortality</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td>+</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Waste production</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Bio turbation/irrigation</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>+</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<table class="caption-top table">
<caption><span id="_Toc526762780" class="anchor"></span>Table 28. Basic assumptions and formulations in the Atlantis model (modified after Fulton et al.&nbsp;2004b).</caption>
<colgroup>
<col style="width: 23%">
<col style="width: 76%">
</colgroup>
<tbody>
<tr class="odd">
<td style="text-align: center;"><strong>Feature</strong></td>
<td style="text-align: center;"><strong>Assumptions and/or formulation notes</strong></td>
</tr>
<tr class="even">
<td style="text-align: center;">General features</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: center;"><blockquote class="blockquote">
<p>biomass units</p>
</blockquote></td>
<td style="text-align: center;">mg N/m3</td>
</tr>
<tr class="even">
<td style="text-align: center;"><blockquote class="blockquote">
<p>input forcing</p>
</blockquote></td>
<td style="text-align: center;">nutrients, temperature and physics on interannual, seasonal, tidal frequencies</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><blockquote class="blockquote">
<p>level of group detail</p>
</blockquote></td>
<td style="text-align: center;">functional group (with a small number of individual species)</td>
</tr>
<tr class="even">
<td style="text-align: center;"><blockquote class="blockquote">
<p>resolution of the formulation used for the biomass pool groups</p>
</blockquote></td>
<td style="text-align: center;">follow the dynamics of the entire biomass pool of the functional group (or species) in the cell</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><blockquote class="blockquote">
<p>resolution of the formulation used for the age structured group groups</p>
</blockquote></td>
<td style="text-align: center;">follow the biomass dynamics (structural and reserve weight) of the ‘average individual’ for the functional group (or species) in the cell and the number of individuals in the cell</td>
</tr>
<tr class="even">
<td style="text-align: center;"><blockquote class="blockquote">
<p>timestep</p>
</blockquote></td>
<td style="text-align: center;">adaptive* daily or diurnal time step</td>
</tr>
<tr class="odd">
<td style="text-align: center;">Process related</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: center;"><blockquote class="blockquote">
<p>bioturbation and bioirrigation</p>
</blockquote></td>
<td style="text-align: center;">yes, simple exchange between layers</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><blockquote class="blockquote">
<p>consumption formulation</p>
</blockquote></td>
<td style="text-align: center;">type II (asymptotic), with an availability parameter which can be habitat dependent</td>
</tr>
<tr class="even">
<td style="text-align: center;"><blockquote class="blockquote">
<p>equations</p>
</blockquote></td>
<td style="text-align: center;">five general sets of rate of change equations used (autrophs, biomass pool and age structured consumer, bacteria, inanimate)</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><blockquote class="blockquote">
<p>formulation detail</p>
</blockquote></td>
<td style="text-align: center;">general: only growth, mortality and excretion explicit</td>
</tr>
<tr class="even">
<td style="text-align: center;"><blockquote class="blockquote">
<p>light limitation</p>
</blockquote></td>
<td style="text-align: center;">optimal irradiance fixed</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><blockquote class="blockquote">
<p>mixotrophy</p>
</blockquote></td>
<td style="text-align: center;">yes, for dinoflagellates (if present)</td>
</tr>
<tr class="even">
<td style="text-align: center;"><blockquote class="blockquote">
<p>nutrient limitation</p>
</blockquote></td>
<td style="text-align: center;">external nutrients determine uptake</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><blockquote class="blockquote">
<p>nutrient ratio</p>
</blockquote></td>
<td style="text-align: center;">Redfield</td>
</tr>
<tr class="even">
<td style="text-align: center;"><blockquote class="blockquote">
<p>oxygen limitation</p>
</blockquote></td>
<td style="text-align: center;">yes</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><blockquote class="blockquote">
<p>sediment burial</p>
</blockquote></td>
<td style="text-align: center;">very low background rate included</td>
</tr>
<tr class="even">
<td style="text-align: center;"><blockquote class="blockquote">
<p>sediment chemistry</p>
</blockquote></td>
<td style="text-align: center;">dynamic, with sediment bacteria</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><blockquote class="blockquote">
<p>shading of primary producers</p>
</blockquote></td>
<td style="text-align: center;">yes</td>
</tr>
<tr class="even">
<td style="text-align: center;"><blockquote class="blockquote">
<p>spatial (or habitat) limitation</p>
</blockquote></td>
<td style="text-align: center;">yes (for benthic or demersal groups (and species))</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><blockquote class="blockquote">
<p>spatial structure</p>
</blockquote></td>
<td style="text-align: center;">flexible with the potential for multiple vertical and horizontal cells</td>
</tr>
<tr class="even">
<td style="text-align: center;"><blockquote class="blockquote">
<p>temperature dependency</p>
</blockquote></td>
<td style="text-align: center;">yes</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><blockquote class="blockquote">
<p>transport model used for hydrodynamics flows</p>
</blockquote></td>
<td style="text-align: center;">yes</td>
</tr>
<tr class="even">
<td style="text-align: center;">Model closure</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: center;"><blockquote class="blockquote">
<p>top predators represented by static loss terms</p>
</blockquote></td>
<td style="text-align: center;">some top predators are included explicitly, but predators not explicitly included in the foodweb are represented using quadratic mortality terms</td>
</tr>
<tr class="even">
<td style="text-align: center;"><blockquote class="blockquote">
<p>mortality terms</p>
</blockquote></td>
<td style="text-align: center;">linear and quadratic</td>
</tr>
<tr class="odd">
<td style="text-align: center;">Vertebrate and fisheries related</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: center;"><blockquote class="blockquote">
<p>age structure</p>
</blockquote></td>
<td style="text-align: center;">multiple age classes (or stages, which equate to life phases), with final age class of each group a “plus group”</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><blockquote class="blockquote">
<p>fishery discards</p>
</blockquote></td>
<td style="text-align: center;">target and bycatch groups (and species)</td>
</tr>
<tr class="even">
<td style="text-align: center;"><blockquote class="blockquote">
<p>incidental mortality due to fishing</p>
</blockquote></td>
<td style="text-align: center;">yes</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><blockquote class="blockquote">
<p>biomass pool fisheries</p>
</blockquote></td>
<td style="text-align: center;">yes</td>
</tr>
<tr class="even">
<td style="text-align: center;"><blockquote class="blockquote">
<p>management</p>
</blockquote></td>
<td style="text-align: center;">variable, may be via effort limitations, gear limitations, minimum legal size, area or temporal closures and may be based on target or endangered stocks</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><blockquote class="blockquote">
<p>stock-recruit relationship</p>
</blockquote></td>
<td style="text-align: center;">Beverton-Holt, productivity-based or constant recruitment</td>
</tr>
<tr class="even">
<td style="text-align: center;"><blockquote class="blockquote">
<p>stock structure</p>
</blockquote></td>
<td style="text-align: center;">depends on recruitment function chosen – may be internal (all the stock within the bay and self-seeds) or external (the reproductive stock outside the bay produces the recruits and the oldest age classes migrate out of the bay to join this stock)</td>
</tr>
</tbody>
</table>
</section>
<section id="references" class="level1">
<h1>REFERENCES</h1>
<p>Abrams, Peter A. "The Fallacies of" Ratio‐Dependent" Predation." Ecology 75, no. 6 (1994): 1842-1850.</p>
<p>Ainsworth, C.H. and Walters, C.J. (2015) Ten common mistakes made in Ecopath with Ecosim modelling. Ecological Modelling 308, 14–17.</p>
<p>Ainsworth, C.H., Morzaria-Luna, H., Kaplan, I.C., Levin, P.S., Fulton, E.A., Cudney-Bueno, R., Turk-Boyer, P., Torre, J., Danemann, G.D., and Pfister, T. 2012a. Effective ecosystem-based management must encourage regulatory compliance: A Gulf of California case study. Mar.&nbsp;Policy 36(6): 1275–1283.</p>
<p>Ainsworth, C.H., Morzaria-Luna, H.N., Kaplan, I.C., Levin, P.S., and Fulton, E.A. 2012b. Full compliance with harvest regulations yields ecological benefits: Northern Gulf of California case study. J. Appl. Ecol. 49(1): 63–72.</p>
<p>Ainsworth, C.H., Schirripa, M.J. and Morzaria Luna, H. (2015) An Atlantis Ecosystem Model for the Gulf of Mexico supporting Integrated Ecosystem Assessment. NOAA Technical Memorandum NMFS-SEFSC-676. Miami, USA.Available from http://dx.doi.org/10.7289/V5X63JVH [accessed 21 December 2015].</p>
<p>Alin, S.R., Feely, R.A., Dickson, A., Hernandez-Ayon, J.M., Juranek, L.W., Ohman, M.D.&nbsp;and Goericke, R. (in revision) Robust empirical relationships for estimating the carbonate system in the southern California Current System using hydrographic data and application to CalCOFI hydrographic cruise data (2005–2011). Journal of Geophysical Research.</p>
<p>Arditi, R., &amp; Ginzburg, L. R. (1989). Coupling in predator-prey dynamics: ratio-dependence. Journal of theoretical biology, 139(3), 311-326.</p>
<p>Audzijonyte, A., Kuparinen A., Gorton R., Fulton E.A. (2013) Ecological consequences of body size decline in harvested fish species: positive feedback loops in trophic interactions amplify human impact. Biology Letters, 9(2):1103</p>
<p>Audzijonyte, A., Kuparinen, A., Fulton, E.A. (2014) Ecosystem effects of contemporary life-history changes are comparable to those of fishing. Marine Ecology Progress Series 495: 219-231</p>
<p>Aydin, K., Gaichas, S., Ortiz, I., Kinzey, D. and Friday, N. (2007) A Comparison of the Bering Sea, Gulf of Alaska, and Aleutian Islands Large Marine Ecosystems Through Food Web Modeling.</p>
<p>Behrenfeld, M.J. and Falkowski, P.G. (1997a) A consumer’s guide to phytoplankton primary productivity models. Limnology and Oceanography 42, 1479–1491.</p>
<p>Behrenfeld, M.J. and Falkowski, P.G. (1997b) Photosynthetic rates derived from satellite-based chlorophyll concentration. Limnology and oceanography 42, 1–20.</p>
<p>Brand, E.J., Kaplan, I.C., Harvey, C.J., Levin, P.S., Fulton, E.A., Hermann, A.J. and Field, J.C. (2007) A Spatially Explicit Ecosystem Model of the California Current’s Food Web and Oceanography. NOAA Technical Memorandum NMFS-NWFSC-84.</p>
<p>Brassington, G.G., Pugh, T., Spillman, C., Schulz, E., Beggs, H., Schiller, A. and Oke, P.R. (2007) BLUElink&gt; Development of operational oceanography and servicing in Australia. Journal of Research and Practice in Information Technology 39, 151–164.</p>
<p>Chassignet, E.P., Hurlburt, H.E., Smedstad, O.M., Halliwell, G.R., Hogan, P.J., Wallcraft, A.J. and Bleck, R. (2006) Ocean prediction with the hybrid coordinate ocean model (HYCOM). Springer.</p>
<p>Christensen, V., and Walters, C.J. 2004. Ecopath with Ecosim: methods, capabilities and limitations. Ecol. Model. 172(2-4): 109–139.</p>
<p>Essington, T.E. and Plagányi, É.E. (2013). "Pitfalls and guidelines for “recycling” models for ecosystem-based fisheries management: evaluating model suitability for forage fish fisheries." ICES Journal of Marine Science 71: 118–127.</p>
<p>Fiechter, J., Rose, K.A., Curchitser, E.N. and Hedstrom, K.S. (2014) The role of environmental controls in determining sardine and anchovy population cycles in the California Current: Analysis of an end-to-end model. Progress in Oceanography.</p>
<p>Fulton, E.A., M. Fuller, A.D.M. Smith, and A.E. Punt. 2004a. Ecological Indicators of the Ecosystem Effects of Fishing: Final Report. Australian Fisheries Management Authority Report, R99/1546.</p>
<p>Fulton, E.A., A.D.M. Smith and C.R. Johnson. 2004b. Effects of spatial resolution on the performance and interpretation of marine ecosystem models. Ecological Modelling 176: 27 – 42.</p>
<p>Fulton E.A,. J.S. Parslow, A.D.M. Smith and C.R. Johnson. 2004c. Biogeochemical Marine Ecosystem Models II: The Effect of Physiological Detail on Model Performance. Ecological Modelling, 173: 371-406.</p>
<p>Fulton, E., Smith, A. and Punt, A. (2005) Which ecological indicators can robustly detect effects of fishing? ICES Journal of Marine Science 62, 540–551.</p>
<p>Fulton, E.A., Smith, A.D.M. and Smith, D.C. (2007) Alternative management strategies for southeast Australian Commonwealth Fisheries: stage 2: quantitative management strategy evaluation. Australian Fisheries Management Authority Report.</p>
<p>Fulton, E.A., Link, J.S., Kaplan, I.C., et al.&nbsp;(2011) Lessons in modelling and management of marine ecosystems: the Atlantis experience. Fish and Fisheries 12, 171–188.</p>
<p>Fulton, E.A., Smith, A.D.M., Smith, D.C. and Johnson, P. (2014) An Integrated Approach Is Needed for Ecosystem Based Fisheries Management: Insights from Ecosystem-Level Management Strategy Evaluation. PLoS ONE 9, e84242.</p>
<p>Gray, R., Fulton, E.A., Little, L.R., Scott, R., 2006. Operating Model Specification Within an Agent Based Framework. North West Shelf Joint Environmental Management Study Technical Report, vol 16. CSIRO, Hobart, Tasmania. 127pp.</p>
<p>Griffith, G.P., Fulton, E.A. and Richardson, A.J. (2011) Effects of fishing and acidification-related benthic mortality on the southeast Australian marine ecosystem. Global Change Biology 17, 3058–307.</p>
<p>Griffith, G.P., Fulton, E.A., Gorton, R., and Richardson, A.J. 2012. Predicting Interactions among Fishing, Ocean Warming, and Ocean Acidification in a Marine System with Whole-Ecosystem Models. Conserv. Biol. 26(6): 1145–1152.</p>
<p>Haidvogel, D., Arango, H., Budgell, W., et al.&nbsp;(2008) Ocean forecasting in terrain-following coordinates: Formulation and skill assessment of the Regional Ocean Modeling System. Journal of Computational Physics 227, 3595–3624.</p>
<p>Horne, P.J., Kaplan, I.C., Marshall, K.N., Levin, P.S., Harvey, C.J., Hermann, A.J., and Fulton, E.A. 2010. Design and parameterization of a spatially explicit ecosystem model of the central California Current.</p>
<p>Hunsicker, Mary E., et al.&nbsp;"Functional responses and scaling in predator–prey interactions of marine fishes: contemporary issues and emerging concepts." Ecology Letters 14.12 (2011): 1288-1299.</p>
<p>Ihde, T. F., I. C. Kaplan, E. A. Fulton, I. A. Gray, M. Hasan, D. Bruce, W. Slacum, and H. M. Townsend. 2016. Design and parameterization of the Chesapeake Bay Atlantis Model: A spatially explicit end-to-end ecosystem model. U.S. Dept. of Commer., NOAA. NOAA Technical Memorandum NMFS-F/SPO-166, 145 p.</p>
<p>Johnson, P., Fulton, E., Smith, D. C., Jenkins, G. P., &amp; Barrett, N. (2011). The use of telescoping spatial scales to capture inshore to slope dynamics in marine ecosystem modeling. Natural Resource Modeling, 24(3), 335-364.</p>
<p>Juranek, L.W., Feely, R.A., Peterson, W.T., et al.&nbsp;(2009) A novel method for determination of aragonite saturation state on the continental shelf of central Oregon using multi-parameter relationships with hydrographic data. Geophysical Research Letters 36.</p>
<p>Kaplan, I.C., Levin, P.S., Burden, M. and Fulton, E.A. (2010) Fishing catch shares in the face of global change: a framework for integrating cumulative impacts and single species management. Canadian Journal of Fisheries and Aquatic Sciences 67, 1968–1982.</p>
<p>Kaplan, I.C., Horne, P.J., and Levin, P.S. 2012. Screening California Current Fishery Management Scenarios using the Atlantis End-to-End Ecosystem Model. Prog. Oceanogr. 102: 5–18.</p>
<p>Kaplan, I.C., Holland, D.S., and Fulton, E.A. 2014. Finding the accelerator and brake in an individual quota fishery: linking ecology, economics, and fleet dynamics of US West Coast trawl fisheries. ICES J. Mar.&nbsp;Sci. J. Cons. 71(2): 308–319.</p>
<p>Kroeker, K.J., Kordas, R.L., Crim, R., et al.&nbsp;(2013) Impacts of ocean acidification on marine organisms: quantifying sensitivities and interaction with warming. Global change biology.</p>
<p>Lehodey, P., Senina, I., and Murtugudde, R. 2008. A spatial ecosystem and populations dynamics model (SEAPODYM) – Modeling of tuna and tuna-like populations. Prog. Oceanogr. 78(4): 304–318. doi:10.1016/j.pocean.2008.06.004.</p>
<p>Link, J.S. (2010) Adding rigor to ecological network models by evaluating a set of pre-balance diagnostics: a plea for PREBAL. Ecological Modelling 221, 1580–1591.</p>
<p>Link, J.S., Fulton, E.A. and Gamble, R.J. (2010) The northeast US application of ATLANTIS: A full system model exploring marine ecosystem dynamics in a living marine resource management context. Progress In Oceanography 87, 214–234.</p>
<p>Methot Jr., R.D., and Wetzel, C.R. 2013. Stock synthesis: A biological and statistical framework for fish stock assessment and fishery management. Fish. Res. 142: 86–99. doi:10.1016/j.fishres.2012.10.012.</p>
<p>Morato, T., E.A. Fulton and T. Pitcher. In press. Approaches to the modelling of seamounts and their fisheries. In, Seamounts, Ecology, Fisheries and Conservation. Fish and Aquatic Resources Series (FAR), Blackwell Publishing, Oxford.</p>
<p>Nihoul, J.C.J. and S. Djenidi, 1998. Chapter 18: Coupled physical, chemical and biological models. In: K.H. Brink and A.R. Robinson (eds), The Sea. John Wiley &amp; Sons Inc: New York, London.</p>
<p>Persson, L., Van Leeuwen, A., &amp; De Roos, A. M. (2014). The ecological foundation for ecosystem-based management of fisheries: mechanistic linkages between the individual-, population-, and community-level dynamics. ICES Journal of Marine Science: Journal du Conseil, fst231.</p>
<p>Rose, K., Allen, J.I., Artioli, Y., et al.&nbsp;(2010) End-To-End Models for the Analysis of Marine Ecosystems: Challenges, Issues, and Next Steps. Marine and Coastal Fisheries: Dynamics, Management, and Ecosystem Science 2, 115–130.</p>
<p>Rose, K.A., Fiechter, J., Curchitser, E.N., et al.&nbsp;(2015) Demonstration of a fully-coupled end-to-end model for small pelagic fish using sardine and anchovy in the California Current. Progress in Oceanography.</p>
<p>Sainsbury, K.J., A.E. Punt A.E. and A.D.M. Smith. 2000. Design of operational management strategies for achieving fishery ecosystem objectives. ICES Journal of Marine Science, 57: 731–741</p>
<p>Savina, M., Condie, S.A., and Fulton, E.A. 2013. The role of pre-existing disturbances in the effect of marine reserves on coastal ecosystems: A modelling approach. PloS One 8(4): e61207.</p>
<p>Shchepetkin, A.F. and McWilliams, J.C. (2005) The regional oceanic modeling system (ROMS): a split-explicit, free-surface, topography-following-coordinate oceanic model. Ocean Modelling 9, 347–404.</p>
<p>Shin, Y.-J., and Cury, P. 2004. Using an individual-based model of fish assemblages to study the response of size spectra to changes in fishing. Can. J. Fish. Aquat. Sci. 61(3): 414–431. doi:10.1139/f03-154.</p>
<p>Smith, D.C., Johnson, P. and Fulton, E.A. (2010) Developing integrated performance measures for spatial management of marine systems. Fisheries Research and Development Corporation Report</p>
<p>Uitz, J., Huot, Y., Bruyant, F., Babin, M. and Claustre, H. (2008) Relating phytoplankton photophysiological properties to community structure on large scales. Limnology and Oceanography 53, 614–630.</p>
<p>Uitz, J., Claustre, H., Gentili, B. and Stramski, D. (2010) Phytoplankton class-specific primary production in the world’s oceans: Seasonal and interannual variability from satellite observations: PHYTOPLANKTON CLASS-SPECIFIC PRODUCTION. Global Biogeochemical Cycles 24, n/a–n/a.</p>
<p>Walters, C., Pauly, D., &amp; Christensen, V. (1999). Ecospace: prediction of mesoscale spatial patterns in trophic relationships of exploited ecosystems, with emphasis on the impacts of marine protected areas. Ecosystems, 2(6), 539-554.</p>
<p>Walters, Carl, and James F. Kitchell. "Cultivation/depensation effects on juvenile survival and recruitment: implications for the theory of fishing." Canadian Journal of Fisheries and Aquatic Sciences 58, no. 1 (2001): 39-50.</p>
<p>Weijerman, M., Fulton, E.A., Kaplan, I.C., Gorton, R., Leemans, R., Mooij, W.M., and Brainard, R.E. 2015. An integrated coral reef ecosystem model to support resource management under a changing climate. PloS One 10(12): e0144165.</p>
<p>Wittmann, A.C. and Pörtner, H.-O. (2013) Sensitivities of extant animal taxa to ocean acidification. Nature Climate Change 3, 995–1001.</p>
</section>
<section id="appendix-1-tips-for-calibrating-biological-model-notes-from-the-2015-atlantis-summit" class="level1">
<h1>Appendix 1: TIPS FOR CALIBRATING BIOLOGICAL MODEL: NOTES FROM THE 2015 ATLANTIS SUMMIT</h1>
<p><em>by Isaac Kaplan and Atlantis community</em></p>
<p>New Atlantis modelers at the Summit requested additional guidance on the steps useful for calibrating models for new systems.</p>
<p>The main software tools used to calibrate the model are OLIVE or DIVE (for visualising spatial output maps in out.nc), a good text editor to look through <em>log.txt</em>, Excel files that use lookup tables to compare model output with observations and assessments (as appropriate) or R (where scripts can be run to do spatial or time series plotting for comparison purposes).</p>
<p>At the Atlantis Summit, two useful R packages were unveiled to help with calibration:</p>
<ul>
<li><p><a href="https://github.com/mareframe/vat" class="uri">https://github.com/mareframe/vat</a></p></li>
<li><p>Alex Keth’s R scripts for Atlantis, which are being incorporated into <a href="https://github.com/r4atlantis/atlantisom" class="uri">https://github.com/r4atlantis/atlantisom</a></p></li>
</ul>
<p>Additional R packages are also being posted on Github:</p>
<ul>
<li>Github <a href="https://github.com/r4atlantis" class="uri">https://github.com/r4atlantis</a> for R packages for Atlantis</li>
</ul>
<p>Conversations about calibration are usually posted to the Atlantis google group (<a href="https://groups.google.com/forum/#!forum/atlantis-ecosystem-model" class="uri">https://groups.google.com/forum/#!forum/atlantis-ecosystem-model</a>), while solutions to problems are posted on the Atlantis wiki (<a href="https://confluence.csiro.au/display/Atlantis/Atlantis+Ecosystem+Model+Home+Page" class="uri">https://confluence.csiro.au/display/Atlantis/Atlantis+Ecosystem+Model+Home+Page</a>).</p>
<p><strong><em>The general goals of calibration are, in this order:</em></strong></p>
<ol type="1">
<li><p>Prevent all species present in the model from going extinct (unless they do so in the time series being fit to)</p></li>
<li><p>Have age structured groups grow such that size-at-age is reasonable (within 20% of initial conditions, typically)</p></li>
<li><p>For species with historical data are available, have the model recreate observations of abundance from surveys or assessments</p></li>
<li><p>For species with no historical data, the model should yield reasonable time series of abundance (especially under perturbation – such as environmental forcing or fishing pressure)</p></li>
<li><p>Capturing observed spatial distributions</p></li>
</ol>
<p>Practical experience suggests that tackling model problems in the order above is best, since this sequence minimizes recalibrating bits too many times.</p>
<p>Once size-at-age is within 20% of the von Bertalanffy curve expected from the literature, then growth is generally acceptable. When calibrating age structured groups, attempt to get relative SN and RN values reasonable before worrying about numbers (den results). Note though that if numbers are too high (by an order of magnitude or more) the group maybe starving itself so you may need to reduce recruitment or increase predation to get numbers under control before you will make any headway with the growth of the group.</p>
<p>During calibration, tuning typically focuses primarily on changing growth rates, consumption rates, linear and quadratic mortality, and recruitment (Beverton Holt) parameters. These variables are in the <em>biology.prm</em> file. Below, XX represents the functional group code (e.g.&nbsp;FVD, FPL, etc.) from the <em>functional_groups.csv</em> file:</p>
<ul>
<li><p><strong>C_XX</strong> consumption rates</p></li>
<li><p><strong>mum_XX</strong> maximum growth rates</p></li>
<li><p><strong>mL_XX</strong> ‘unexplained’ background linear mortality</p></li>
<li><p><strong>mQ_XX</strong> quadratic mortality (‘unexplained’ density dependence)</p></li>
<li><p><strong>pPREYXX</strong> diet or predator-prey interaction terms</p></li>
<li><p><strong>BHalphaXX</strong> Beverton Holt alpha parameters in recruitment</p></li>
</ul>
<p>A thorough exploration of parameter distributions within the Atlantis models would be broadly useful for error checking inputs and understanding reasonable bounds during calibration. As suggested by Ainsworth and Walters (2015), other authors have already built databases and conducted meta-analyses on Ecopath parameters, and comparison of Atlantis values to Ecopath values would be informative. Finally, automated parameter comparison across models lends itself well to basic checks on model assumptions about biomasses and vital rates, akin to the PREBAL diagnostic test of Link (2010).</p>
<p><strong><em>Calibration with fishing</em></strong></p>
<p>A major part of calibrating the model is comparing how it performs when faced with various levels of fishing intensity, both real historical values and hypothetical values. This sort of qualitative exploration can reveal underlying problems, such as parameterizations that involve too little recruitment (and therefore inevitable decline of a stock) or too much recruitment (and therefore unreasonable resilience to fishing). Just as testing Ecopath models with fishing is a crucial step (Ainsworth and Walters 2015), Atlantis models can also use fishing to test basic parameterization of productivity. For heavily fished target species, we roughly expect Atlantis to behave like a single species model (and if not, need to consider why). For lightly fished, nontarget or forage species, we expect ecosystem dynamics to matter a lot more, and Atlantis may (reasonably) diverge from a single species assessment prediction.</p>
<p>Beth’s suggestions on fishing scenarios to run are as follows (these in fact mirror some of the Common Scenarios runs for Atlantis Summit)</p>
<ol type="1">
<li><p>Unfished - Atlantis spinup estimate of the unexploited system (where possible from virgin, but also what system would it head to from current state if unfished)</p></li>
<li><p>Historical removals - can the Atlantis state support historically recorded catches (and suspected misreported catches); does the trajectory match assessment/survey data? (preferably survey data rather than assessment)</p></li>
<li><p>Fish the target(s) at current levels (and do so constantly through time)</p></li>
<li><p>Fish the target(s) at assessment estimate of RBC (Recommended Biological Catch, which may differ from MSY)</p></li>
<li><p>Fish the target(s) at 5xassessment estimate of RBC</p></li>
<li><p>Fish the target(s) at 0.2xassessment estimate of RBC</p></li>
<li><p>Fish the target(s) at 0.2xcurrent levels</p></li>
<li><p>Fish the target(s) at 20xcurrent levels (or more depending on current F)</p></li>
<li><p>Shock the system in some other way (e.g.&nbsp;nutrient pollution, habitat degration, etc.)</p></li>
<li><p>Combinations of shocks (typically high fishing + a non-fishing shock)</p></li>
</ol>
<section id="tips-for-parameterizing-biomass-pools-invertebrates" class="level2">
<h2 class="anchored" data-anchor-id="tips-for-parameterizing-biomass-pools-invertebrates"><strong>Tips for parameterizing biomass pools (invertebrates)</strong></h2>
<p>Benthic invertebrate abundance and distribution data are scarce, especially for non-harvested species, and generalized additive models are one approach to addressing this. Generalized additive models start from patchy observations or field sampling, and use habitat and oceanographic descriptors to predict invertebrate abundance or distribution. This was applied to SEAMAP data in the Gulf of Mexico (Ainsworth <em>et al.</em> 2015), and is being applied in the Bay of Biscay. Expert knowledge was also relied upon to parameterize these species.</p>
<p>Based on responses at the Atlantis Summit, invertebrate rate parameters such as consumption and growth rates are often taken from existing Ecopath models. These models serve as essential steps toward building Atlantis models, yet the assumptions and methods to estimate Ecopath parameters (PB and QB) were not clear to all Atlantis Summit participants. Furthermore, consumption and growth rates for invertebrates were often tuned heavily in Atlantis. A best practice would be to first understand Ecopath parameterization, then document how it informs Atlantis, before beginning the Atlantis calibration.</p>
<p>Invertebrate groups have been particularly hard to calibrate in many Atlantis models. It is not clear if this is due to poor data or noisy population dynamics. Cephalopods, shrimp, jellyfish, and some benthos are problematic. Cephalopods and shrimp often are parameterized with simple juvenile vs adult biomass pools, and users with the setup should view and test the associated code.</p>
<p>Most Atlantis modelers convert from carbon or dry weight to nitrogen using the Redfield ratio. However, greater attention to these conversions is warranted as the same value may not actually be appropriate across vertebrates and invertebrates of different kinds.</p>
</section>
<section id="tips-for-parameterizing-age-structured-vertebrate-groups" class="level2">
<h2 class="anchored" data-anchor-id="tips-for-parameterizing-age-structured-vertebrate-groups"><strong>Tips for parameterizing age-structured (vertebrate) groups</strong></h2>
<section id="starvation-and-respiration" class="level3">
<h3 class="anchored" data-anchor-id="starvation-and-respiration">Starvation and respiration</h3>
<p>Parameterization of starvation is a topic that many Atlantis modelers have not focused upon, but the mortality rate for starving fish can be an influential parameter. Most participants at the Atlantis Summit felt that they should investigate higher values of starvation-induced mortality, as well as testing a model run where all diets are set to 0 (with starvation expected to result).</p>
<p>Very few Atlantis models currently include explicit respiration. In theory, respiration can be included implicitly by reducing assimilation efficiencies (from ~0.8 to 0.3 or less) (but see details on respiration in the manual above). Similarly, most models used a standard temperature effect on metabolism (Q10 = 2), but this should be questioned for new models and applications, particularly those related to climate change.</p>
</section>
<section id="movement" class="level3">
<h3 class="anchored" data-anchor-id="movement">Movement</h3>
<p>A mix of approaches to age structured group movement has been applied within Atlantis, and the exact approach needs to be tailored to the research question and model goals. Density dependent movement has been particularly useful for modeling marine mammals and birds in the Baltic Sea and Northeast USA (Link <em>et al.</em> 2010) and some fish groups in Guam (Weijerman <em>et al.</em> 2015). Temperature-dependent movements have less commonly been applied. Often density dependent movement for some species has been combined with prescribed quarterly spatial distributions for other species; these prescribed movements can be based on expert opinion and rough categorization of species into a few general movement patterns.</p>
</section>
<section id="reproduction" class="level3">
<h3 class="anchored" data-anchor-id="reproduction">Reproduction</h3>
<p>To model recruitment (reproduction) of fish, most Atlantis models to date primarily rely on one of two forms of the Beverton-Holt relationship. This omits the possibility of ecological effects such as cannibalism on larvae or pre-recruits. When using Beverton-Holt, a starting point for parameterization is to use the spreadsheet available on the <a href="https://confluence.csiro.au/pages/viewpage.action?pageId=238256130">wiki</a>, which converts from stock assessment quantities to Atlantis Beverton-Holt parameters. However, subsequent calibration is essential, in large part because ‘recruits’ in Atlantis are much younger than ‘recruits’ in stock assessments, and the intervening mortality means that stock assessment estimates must be scaled up to Atlantis numbers-of-recruits. In general, mammals and birds were often parameterized with fixed offspring per adult.</p>
</section>
<section id="growth-consumption-and-predator-prey-functional-response" class="level3">
<h3 class="anchored" data-anchor-id="growth-consumption-and-predator-prey-functional-response">Growth, consumption, and predator-prey functional response</h3>
<p>There are at least three methods for estimating consumption rates, prior to using these within Atlantis. For mammals and birds, Gompertz growth curves have often been used to specify size-at-age. For fish, size-at-age is often taken from von Bertlanffy growth relationships. Then one option is that consumption may be estimated from allometric relationships<br>
<em>C<sub>i</sub> = a ·</em> (<em>RN<sub>i</sub> + SN<sub>i</sub></em>) <sup>0.7</sup></p>
<p>Where <em>C</em> is consumption <em>, i</em> indexes age<em>, a</em> is a constant, and <em>RN</em> and <em>SN</em> are weight-at-age in terms of reserve and structural nitrogen. This approach is used within the California Current Atlantis model.</p>
<p>A second alternative is that consumption may be estimated from growth:</p>
<p>C<em><sub>i</sub></em> = b <em>·</em> ((<em>RN<sub>i</sub> + SN<sub>i</sub></em>) <em>-</em> (<em>RN<sub>i-1</sub> + SN <sub>i-1</sub></em>))</p>
<p>where <em>b</em> is a conversion efficiency of ~10. This approach is presently used within the Icelandic Atlantis model.</p>
<p>A final alternative in ecosystems with better stomach sampling is that daily ration (consumption per day) can be directly taken from field studies. This approach is possible for the Baltic Sea Atlantis. Finally, it was noted that consumption estimates in Atlantis are maximum consumption, and the estimates above were therefore scaled up by 20%-300% to account for this.</p>
<p>A strong recommendation arose from Atlantis Summit conversations regarding predator-prey functional responses: Atlantis modelers should look at the code, understand the functional response chosen, and test alternative responses. Most models are using modified Holling Type II relationships for all or most of the age structured groups, though some species (such as menhaden and marine mammals in Northeast US) use modified Holling type III. The Holling Type II relationship in Atlantis was originally used for filter feeders, and assumptions need to be understood for age structured groups. An Excel functional response <a href="https://confluence.csiro.au/display/Atlantis/Atlantis+Helper+Tools">demo is available</a> for common functional response.</p>
<p>All users were encouraged to carefully consider and test the functional response options they choose to use in the code. If the traditional modified Holling Type II relationship is chosen, the <em>mum</em> maximum growth rate parameter can usually be inferred from the equations listed above for consumption, and an assumption about assimilation or conversion efficiency. The ‘clearance’ (<em>C_</em>) parameter would be related to water volume filtered by a filter feeding invertebrate; for a fish it is roughly related to search volume. However, modelers in some systems have found that clearance= 0.1 * <em>mum</em> was a good starting point for calibration, while others have had better success with clearance = 10 x <em>mum</em>. Each modeler should test and plot the functional response and the parameters they are using, and consider alternative formulations as well.</p>
<p>During calibration of the model, most users experienced situations with low growth of age structured groups, and in these cases increasing <em>mum</em> was the recommended first step, before adjusting other functional response parameters.</p>
</section>
<section id="other-mortality" class="level3">
<h3 class="anchored" data-anchor-id="other-mortality">Other mortality</h3>
<p>In most Atlantis models, linear and quadratic mortality (<em>mL</em> and <em>mQ</em> in Atlantis) are generally used as calibration parameters of last resort, to represent mortality effects or carrying capacity not explicitly handled the by the model.</p>
</section>
</section>
<section id="tips-on-parameterizing-biogeochemistry" class="level2">
<h2 class="anchored" data-anchor-id="tips-on-parameterizing-biogeochemistry"><strong>Tips on parameterizing biogeochemistry</strong></h2>
<p>Atlantis requires initial conditions for nutrients, primary producers, zooplankton, and detritus. Several modeling groups have used Nutrient-Phytoplantkon-Zooplankton-Detritus (NPZD) model output as the basis for these Atlantis initial conditions. These include the Baltic Sea and Strait of Sicily Atlantis models. This was suggested as a useful approach, as long as the NPZD models have been evaluated for skill against field observations. Some regions have extensive in-situ sampling and accessible databases, which can also be used as initial conditions for Atlantis. An example is <a href="http://gulfatlas.noaa.gov/" class="uri">http://gulfatlas.noaa.gov/</a> in the Gulf of Mexico.</p>
<p>Nutrient loading has been used in Atlantis models (forcing with nutrient time series), including the, California Current, Guam, Gulf of Mexico (Brand <em>et al.</em> 2007; Ainsworth <em>et al.</em> 2015; Weijerman <em>et al.</em> 2015) and Chesapeake Bay and Strait of Sicily model. One advantage of this approach is that it can be used as a proxy for multi-decadal trends in ocean productivity, even if full multi-decadal oceanography (ROMS etc.) is not available and must be ‘looped’. Within the US, the Environmental Protection Agency has provided time series of point-source and non-point-source nutrients and sediments to Atlantis modelers.</p>
</section>
<section id="tips-on-handling-ocean-acidification" class="level2">
<h2 class="anchored" data-anchor-id="tips-on-handling-ocean-acidification"><strong>Tips on handling ocean acidification</strong></h2>
<p>Several distinct approaches to handling ocean acidification impacts have been tested within Atlantis:</p>
<ol type="1">
<li><p>adding linear mortality to calcifying species, across all model polygons (Kaplan <em>et al.</em> 2010; Griffith <em>et al.</em> 2011).</p></li>
<li><p>allowing pH to evolve within Atlantis, but forced by atmospheric pCO2 (Weijerman <em>et al.</em> 2015),</p></li>
<li><p>Importing pH from a ROMS that includes biogeochemistry (California Current work in prep).</p></li>
</ol>
<p>A different approach would be to predict pH or aragonite saturation state from temperature, oxygen, salinity, depth (Alin <em>et al.</em> in revision; Juranek <em>et al.</em> 2009). Based on experience with Common Scenarios at the Atantis Summit, we suggest the first option, before delving into the more detailed approaches.</p>
<p>After pH (or aragonite saturation state) fields are generated within or outside Atlantis, the biological response to pH must be parameterized. This is challenging and involves a high degree of uncertainty; global meta-analyses provide some guidance (Kroeker et al.&nbsp;2013; Wittmann and Pörtner 2013) but local conditions may require additional expertise. One concern was that these biological responses should be updated periodically as new information becomes available.</p>


</section>
</section>


<div id="quarto-appendix" class="default"><section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes"><h2 class="anchored quarto-appendix-heading">Footnotes</h2>

<ol>
<li id="fn1"><p>There is also Beth’s development version, that contains features being actively developed at any instant in time. As those features are tested and verified (and so are less likely to contain a bug that will blow up) they are moved to the trunk version for general access.<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn2"><p>An agent-based extension exists for Atlantis, but it is not currently included in the standard Atlantis package and will not be discussed in the manual<a href="#fnref2" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn3"><p>We do endeavour to update things for new naming conventions, but we also find it is better to err on the side of caution.<a href="#fnref3" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn4"><p>A small amount of mismatch (typically to cover rounding error) is allowed over the total fluxes in a cell in any one time step, but if it becomes too large (i.e.&nbsp;exceeds the flux tolerance set in the <em>biology.prm</em> file, with typical values being on the order of 0.2 mgN*m<sup>-3</sup> per second) the model will quit and provide a warning message, as this indicates that there is a problem with the parameterisation<a href="#fnref4" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn5"><p>Expert users may be able to tinker directly with the BGM file, but extreme care should be taken and it should be documented so any future users can understand the changes made (as the master shapefile file will no longer 100% match the model geometry).<a href="#fnref5" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>